// @kit-ui/admin/lib/AppHeader/AppHeader.stories.js
"use strict";

exports.__esModule = true;
exports.GrowingSearch = exports.SearchExpanded = exports.Search = exports.SubviewExpanded = exports.MenuAndSearch = exports.WithMenu = exports.Default = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _AppHeader = _interopRequireDefault(require("./AppHeader"));

var _AppHeader2 = _interopRequireDefault(require("./AppHeader.example"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _default = {
  title: 'App/AppHeader',
  component: _AppHeader.default,
  args: {
    title: "App",
    subtitle: "Owner"
  },
  argTypes: {}
};
exports.default = _default;

var Template = function Template(args) {
  return /*#__PURE__*/_react.default.createElement(_AppHeader2.default, args);
};

var Default = Template.bind({});
exports.Default = Default;
Default.args = {};
var WithMenu = Template.bind({});
exports.WithMenu = WithMenu;
WithMenu.args = {
  menu: {
    expanded: false
  }
};
var MenuAndSearch = Template.bind({});
exports.MenuAndSearch = MenuAndSearch;
MenuAndSearch.args = {
  menu: {
    expanded: false
  },
  search: {
    expanded: false
  }
};
var SubviewExpanded = Template.bind({});
exports.SubviewExpanded = SubviewExpanded;
SubviewExpanded.args = {
  subview: {
    expanded: true,
    title: "Subview",
    description: "5 actions"
  }
};
var Search = Template.bind({});
exports.Search = Search;
Search.args = {
  search: {
    expanded: false,
    placeholder: "Search app"
  }
};
var SearchExpanded = Template.bind({});
exports.SearchExpanded = SearchExpanded;
SearchExpanded.args = {
  search: {
    expanded: true,
    placeholder: "Search app"
  }
};
var GrowingSearch = Template.bind({});
exports.GrowingSearch = GrowingSearch;
GrowingSearch.args = {
  search: {
    variant: "growing",
    expanded: false,
    placeholder: "Search app"
  }
};
