// @kit-ui/admin/lib/AppHeader/AppNav.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _NavUser = _interopRequireDefault(require("../NavUser/NavUser"));

var _AppSwitcher = _interopRequireDefault(require("../AppSwitcher/AppSwitcher"));

var _styles = require("@material-ui/core/styles");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var useStyles = (0, _styles.makeStyles)(function (theme) {
  var _root;

  return {
    root: (_root = {
      position: "absolute",
      top: 0,
      right: 0,
      zIndex: 2000,
      display: "flex",
      marginTop: theme.spacing(-8),
      "&[aria-expanded=true]": {
        marginTop: 0
      },
      height: theme.spacing(8),
      marginLeft: theme.spacing(1),
      marginRight: theme.spacing(1)
    }, _root["display"] = "flex", _root.alignItems = "center", _root.justifyContent = "flex-end", _root),
    button: {
      width: theme.spacing(6),
      height: theme.spacing(6),
      display: "flex",
      alignItems: "center",
      justifyContent: "center"
    }
  };
});

var AppNav = function AppNav(_ref) {
  var className = _ref.className,
      _ref$expanded = _ref.expanded,
      expanded = _ref$expanded === void 0 ? true : _ref$expanded,
      app = _ref.app,
      user = _ref.user,
      owner = _ref.owner,
      onLogin = _ref.onLogin,
      onLogout = _ref.onLogout,
      onSelect = _ref.onSelect;
  var classes = useStyles();
  return /*#__PURE__*/_react.default.createElement("section", {
    className: className || classes.root,
    "aria-expanded": expanded
  }, user && user.uniqueId && /*#__PURE__*/_react.default.createElement(_AppSwitcher.default, {
    className: classes.button,
    user: user,
    owners: user && user.owners,
    owner: owner,
    onSelect: onSelect
  }), /*#__PURE__*/_react.default.createElement(_NavUser.default, {
    className: classes.button,
    app: app,
    user: user,
    onLogin: onLogin,
    onLogout: onLogout
  }));
};

var _default = AppNav;
exports.default = _default;
module.exports = exports.default;
