// @kit-ui/schema/lib/ColorSettings/ColorSettings.stories.js
"use strict";

exports.__esModule = true;
exports.TextColor = exports.WithSwatches = exports.Default = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _ColorSettings = _interopRequireDefault(require("./ColorSettings"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _default = {
  title: 'Settings/ColorSettings',
  component: _ColorSettings.default,
  args: {
    color: "blue"
  },
  argTypes: {}
};
exports.default = _default;

var Template = function Template(args) {
  return /*#__PURE__*/_react.default.createElement(_ColorSettings.default, args);
};

var Default = Template.bind({});
exports.Default = Default;
Default.args = {};
var WithSwatches = Template.bind({});
exports.WithSwatches = WithSwatches;
WithSwatches.args = {
  colors: ["white", "blue", "red"]
};
var TextColor = Template.bind({});
exports.TextColor = TextColor;
TextColor.args = {
  textColor: true,
  colors: ["white", "blue", "red"]
};
