// @kit-ui/schema/lib/ColorSettings/ColorSettings.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _core = require("@kit-ui/core");

var _tinycolor = _interopRequireDefault(require("tinycolor2"));

var _IconButton = _interopRequireDefault(require("@material-ui/core/IconButton"));

var _ColorSettingsPicker = _interopRequireDefault(require("./ColorSettingsPicker"));

var _ColorSettingsIcon = _interopRequireDefault(require("./ColorSettingsIcon"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

var ColorSettings = function ColorSettings(_ref) {
  var variant = _ref.variant,
      backgroundColor = _ref.backgroundColor,
      colors = _ref.colors,
      value = _ref.value,
      onChange = _ref.onChange;

  var _useState = (0, _react.useState)(false),
      expanded = _useState[0],
      setExpanded = _useState[1];

  var anchorRef = (0, _react.useRef)(null);

  var _onChange = function _onChange(_ref2) {
    var hex = _ref2.hex;
    onChange && onChange(hex);
  };

  var _onToggle = function _onToggle(event) {
    setExpanded(function (prevExpanded) {
      return !prevExpanded;
    });
  };

  var _onCollapse = function _onCollapse(event) {
    setExpanded(false);
  }; // color swatches


  if (!value && colors.length) {
    value = colors[0];
  } // color + contrast


  var primaryColor = (0, _tinycolor.default)(value);
  var color = primaryColor.toHexString();
  var contrastColor = primaryColor.isLight() && "black" || "white"; // variant

  if (variant === "text") {
    backgroundColor = backgroundColor || contrastColor;
  } else {
    backgroundColor = color;
  }

  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_IconButton.default, {
    ref: anchorRef,
    onClick: _onToggle
  }, /*#__PURE__*/_react.default.createElement(_ColorSettingsIcon.default, {
    variant: variant,
    backgroundColor: backgroundColor,
    color: color
  })), /*#__PURE__*/_react.default.createElement(_core.Dropdown, {
    expanded: expanded,
    anchorEl: anchorRef.current,
    backgroundColor: backgroundColor,
    onClickAway: _onCollapse
  }, /*#__PURE__*/_react.default.createElement(_ColorSettingsPicker.default, {
    variant: variant,
    backgroundColor: backgroundColor,
    color: color,
    colors: colors,
    onChange: _onChange
  })));
};

ColorSettings.defaultProps = {
  variant: "default",
  value: "#666"
};
ColorSettings.propTypes = process.env.NODE_ENV !== "production" ? {
  value: _propTypes.default.string,
  colors: _propTypes.default.array
} : {};
var _default = ColorSettings;
exports.default = _default;
module.exports = exports.default;
