// @kit-ui/schema/lib/ColorSettings/ColorSettings.stories.js
"use strict";

exports.__esModule = true;
exports.Default = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _SchemaEditor = _interopRequireDefault(require("../SchemaEditor/"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _default = {
  title: 'Settings/ColorSettings',
  component: _SchemaEditor.default,
  args: {
    schema: {
      type: "object",
      properties: {
        color: {
          "type": "string",
          "format": "color"
        },
        colorEnum: {
          "type": "string",
          "format": "color",
          "enum": ["#ff0", "#f0f", "#00f", "red", "blue", "green", "orange", "pink"]
        }
      }
    },
    uiSchema: {
      "ui:layout": "module",
      "ui:options": {
        "editable": true
      },
      "ui:settings": ["color", "colorEnum"],
      "color": {}
    }
  },
  argTypes: {}
};
exports.default = _default;

var Template = function Template(args) {
  return /*#__PURE__*/_react.default.createElement(_SchemaEditor.default, args);
};

var Default = Template.bind({});
exports.Default = Default;
Default.args = {};
