// @rjsf/playground/lib/DemoFrame.js
"use strict";

var _interopRequireDefault = require("@babel/runtime-corejs2/helpers/interopRequireDefault");

var _Object$defineProperty = require("@babel/runtime-corejs2/core-js/object/define-property");

_Object$defineProperty(exports, "__esModule", {
  value: true
});

exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime-corejs2/helpers/extends"));

var _map = _interopRequireDefault(require("@babel/runtime-corejs2/core-js/map"));

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime-corejs2/helpers/slicedToArray"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime-corejs2/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _jss = require("jss");

var _styles = require("@material-ui/core/styles");

var _reactFrameComponent = _interopRequireDefault(require("react-frame-component"));

/*
Adapted from https://github.com/mui-org/material-ui/blob/master/docs/src/modules/components/DemoSandboxed.js

The MIT License (MIT)

Copyright (c) 2014 Call-Em-All

Permission is hereby granted, free of charge, to any person obtaining a copy
of this software and associated documentation files (the "Software"), to deal
in the Software without restriction, including without limitation the rights
to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
copies of the Software, and to permit persons to whom the Software is
furnished to do so, subject to the following conditions:

The above copyright notice and this permission notice shall be included in all
copies or substantial portions of the Software.

THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
SOFTWARE.
*/
function DemoFrame(props) {
  var children = props.children,
      classes = props.classes,
      theme = props.theme,
      other = (0, _objectWithoutProperties2["default"])(props, ["children", "classes", "theme"]);

  var _React$useState = _react["default"].useState({
    ready: false
  }),
      _React$useState2 = (0, _slicedToArray2["default"])(_React$useState, 2),
      state = _React$useState2[0],
      setState = _React$useState2[1];

  var instanceRef = _react["default"].useRef();

  var handleRef = _react["default"].useCallback(function (ref) {
    instanceRef.current = {
      contentDocument: ref ? ref.node.contentDocument : null,
      contentWindow: ref ? ref.node.contentWindow : null
    };
  }, []);

  var onContentDidMount = function onContentDidMount() {
    setState({
      ready: true,
      jss: (0, _jss.create)({
        plugins: (0, _styles.jssPreset)().plugins,
        insertionPoint: instanceRef.current.contentWindow["demo-frame-jss"]
      }),
      sheetsManager: new _map["default"](),
      container: instanceRef.current.contentDocument.body,
      window: function window() {
        return instanceRef.current.contentWindow;
      }
    });
  };

  return _react["default"].createElement(_reactFrameComponent["default"], (0, _extends2["default"])({
    ref: handleRef,
    contentDidMount: onContentDidMount
  }, other), _react["default"].createElement("div", {
    id: "demo-frame-jss"
  }), theme === "material-ui" ? state.ready ? _react["default"].createElement(_styles.StylesProvider, {
    jss: state.jss,
    sheetsManager: state.sheetsManager
  }, _react["default"].cloneElement(children, {
    container: state.container,
    window: state.window
  })) : null : children);
}

var _default = DemoFrame;
exports["default"] = _default;
