// @rjsf/playground/lib/index.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime-corejs2/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime-corejs2/helpers/interopRequireDefault");

var _Object$defineProperty = require("@babel/runtime-corejs2/core-js/object/define-property");

_Object$defineProperty(exports, "__esModule", {
  value: true
});

exports["default"] = void 0;

var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime-corejs2/helpers/assertThisInitialized"));

var _setImmediate2 = _interopRequireDefault(require("@babel/runtime-corejs2/core-js/set-immediate"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime-corejs2/helpers/defineProperty"));

var _parseFloat2 = _interopRequireDefault(require("@babel/runtime-corejs2/core-js/parse-float"));

var _objectSpread2 = _interopRequireDefault(require("@babel/runtime-corejs2/helpers/objectSpread"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime-corejs2/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime-corejs2/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime-corejs2/helpers/possibleConstructorReturn"));

var _getPrototypeOf3 = _interopRequireDefault(require("@babel/runtime-corejs2/helpers/getPrototypeOf"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime-corejs2/helpers/inherits"));

var _stringify = _interopRequireDefault(require("@babel/runtime-corejs2/core-js/json/stringify"));

var _keys = _interopRequireDefault(require("@babel/runtime-corejs2/core-js/object/keys"));

var _typeof2 = _interopRequireDefault(require("@babel/runtime-corejs2/helpers/typeof"));

var _react = _interopRequireWildcard(require("react"));

var _reactMonacoEditor = _interopRequireDefault(require("react-monaco-editor"));

var _samples = require("./samples");

require("react-app-polyfill/ie11");

var _reactJsonschemaForm = _interopRequireWildcard(require("react-jsonschema-form"));

var _DemoFrame = _interopRequireDefault(require("./DemoFrame"));

// deepEquals and shouldRender and isArguments are copied from rjsf-core. TODO: unify these utility functions.
function isArguments(object) {
  return Object.prototype.toString.call(object) === "[object Arguments]";
}

function deepEquals(a, b) {
  var ca = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : [];
  var cb = arguments.length > 3 && arguments[3] !== undefined ? arguments[3] : [];

  // Partially extracted from node-deeper and adapted to exclude comparison
  // checks for functions.
  // https://github.com/othiym23/node-deeper
  if (a === b) {
    return true;
  } else if (typeof a === "function" || typeof b === "function") {
    // Assume all functions are equivalent
    // see https://github.com/mozilla-services/react-jsonschema-form/issues/255
    return true;
  } else if ((0, _typeof2["default"])(a) !== "object" || (0, _typeof2["default"])(b) !== "object") {
    return false;
  } else if (a === null || b === null) {
    return false;
  } else if (a instanceof Date && b instanceof Date) {
    return a.getTime() === b.getTime();
  } else if (a instanceof RegExp && b instanceof RegExp) {
    return a.source === b.source && a.global === b.global && a.multiline === b.multiline && a.lastIndex === b.lastIndex && a.ignoreCase === b.ignoreCase;
  } else if (isArguments(a) || isArguments(b)) {
    if (!(isArguments(a) && isArguments(b))) {
      return false;
    }

    var slice = Array.prototype.slice;
    return deepEquals(slice.call(a), slice.call(b), ca, cb);
  } else {
    if (a.constructor !== b.constructor) {
      return false;
    }

    var ka = (0, _keys["default"])(a);
    var kb = (0, _keys["default"])(b); // don't bother with stack acrobatics if there's nothing there

    if (ka.length === 0 && kb.length === 0) {
      return true;
    }

    if (ka.length !== kb.length) {
      return false;
    }

    var cal = ca.length;

    while (cal--) {
      if (ca[cal] === a) {
        return cb[cal] === b;
      }
    }

    ca.push(a);
    cb.push(b);
    ka.sort();
    kb.sort();

    for (var j = ka.length - 1; j >= 0; j--) {
      if (ka[j] !== kb[j]) {
        return false;
      }
    }

    var key;

    for (var k = ka.length - 1; k >= 0; k--) {
      key = ka[k];

      if (!deepEquals(a[key], b[key], ca, cb)) {
        return false;
      }
    }

    ca.pop();
    cb.pop();
    return true;
  }
}

function shouldRender(comp, nextProps, nextState) {
  var props = comp.props,
      state = comp.state;
  return !deepEquals(props, nextProps) || !deepEquals(state, nextState);
}

var log = function log(type) {
  return console.log.bind(console, type);
};

var toJson = function toJson(val) {
  return (0, _stringify["default"])(val, null, 2);
};

var liveSettingsSchema = {
  type: "object",
  properties: {
    validate: {
      type: "boolean",
      title: "Live validation"
    },
    disable: {
      type: "boolean",
      title: "Disable whole form"
    },
    omitExtraData: {
      type: "boolean",
      title: "Omit extra data"
    },
    liveOmit: {
      type: "boolean",
      title: "Live omit"
    }
  }
};
var monacoEditorOptions = {
  minimap: {
    enabled: false
  },
  automaticLayout: true
};

var GeoPosition =
/*#__PURE__*/
function (_Component) {
  (0, _inherits2["default"])(GeoPosition, _Component);

  function GeoPosition(props) {
    var _this;

    (0, _classCallCheck2["default"])(this, GeoPosition);
    _this = (0, _possibleConstructorReturn2["default"])(this, (0, _getPrototypeOf3["default"])(GeoPosition).call(this, props));
    _this.state = (0, _objectSpread2["default"])({}, props.formData);
    return _this;
  }

  (0, _createClass2["default"])(GeoPosition, [{
    key: "onChange",
    value: function onChange(name) {
      var _this2 = this;

      return function (event) {
        _this2.setState((0, _defineProperty2["default"])({}, name, (0, _parseFloat2["default"])(event.target.value)));

        (0, _setImmediate2["default"])(function () {
          return _this2.props.onChange(_this2.state);
        });
      };
    }
  }, {
    key: "render",
    value: function render() {
      var _this$state = this.state,
          lat = _this$state.lat,
          lon = _this$state.lon;
      return _react["default"].createElement("div", {
        className: "geo"
      }, _react["default"].createElement("h3", null, "Hey, I'm a custom component"), _react["default"].createElement("p", null, "I'm registered as ", _react["default"].createElement("code", null, "geo"), " and referenced in", _react["default"].createElement("code", null, "uiSchema"), " as the ", _react["default"].createElement("code", null, "ui:field"), " to use for this schema."), _react["default"].createElement("div", {
        className: "row"
      }, _react["default"].createElement("div", {
        className: "col-sm-6"
      }, _react["default"].createElement("label", null, "Latitude"), _react["default"].createElement("input", {
        className: "form-control",
        type: "number",
        value: lat,
        step: "0.00001",
        onChange: this.onChange("lat")
      })), _react["default"].createElement("div", {
        className: "col-sm-6"
      }, _react["default"].createElement("label", null, "Longitude"), _react["default"].createElement("input", {
        className: "form-control",
        type: "number",
        value: lon,
        step: "0.00001",
        onChange: this.onChange("lon")
      }))));
    }
  }]);
  return GeoPosition;
}(_react.Component);

var Editor =
/*#__PURE__*/
function (_Component2) {
  (0, _inherits2["default"])(Editor, _Component2);

  function Editor(props) {
    var _this3;

    (0, _classCallCheck2["default"])(this, Editor);
    _this3 = (0, _possibleConstructorReturn2["default"])(this, (0, _getPrototypeOf3["default"])(Editor).call(this, props));
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this3), "onCodeChange", function (code) {
      try {
        var parsedCode = JSON.parse(code);

        _this3.setState({
          valid: true,
          code: code
        }, function () {
          return _this3.props.onChange(parsedCode);
        });
      } catch (err) {
        _this3.setState({
          valid: false,
          code: code
        });
      }
    });
    _this3.state = {
      valid: true,
      code: props.code
    };
    return _this3;
  }

  (0, _createClass2["default"])(Editor, [{
    key: "UNSAFE_componentWillReceiveProps",
    value: function UNSAFE_componentWillReceiveProps(props) {
      this.setState({
        valid: true,
        code: props.code
      });
    }
  }, {
    key: "shouldComponentUpdate",
    value: function shouldComponentUpdate(nextProps, nextState) {
      if (this.state.valid) {
        return (0, _stringify["default"])(JSON.parse(nextProps.code)) !== (0, _stringify["default"])(JSON.parse(this.state.code));
      }

      return false;
    }
  }, {
    key: "render",
    value: function render() {
      var title = this.props.title;
      var icon = this.state.valid ? "ok" : "remove";
      var cls = this.state.valid ? "valid" : "invalid";
      return _react["default"].createElement("div", {
        className: "panel panel-default"
      }, _react["default"].createElement("div", {
        className: "panel-heading"
      }, _react["default"].createElement("span", {
        className: "".concat(cls, " glyphicon glyphicon-").concat(icon)
      }), " " + title), _react["default"].createElement(_reactMonacoEditor["default"], {
        language: "json",
        value: this.state.code,
        theme: "vs-light",
        onChange: this.onCodeChange,
        height: 400,
        options: monacoEditorOptions
      }));
    }
  }]);
  return Editor;
}(_react.Component);

var Selector =
/*#__PURE__*/
function (_Component3) {
  (0, _inherits2["default"])(Selector, _Component3);

  function Selector(props) {
    var _this4;

    (0, _classCallCheck2["default"])(this, Selector);
    _this4 = (0, _possibleConstructorReturn2["default"])(this, (0, _getPrototypeOf3["default"])(Selector).call(this, props));
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this4), "onLabelClick", function (label) {
      return function (event) {
        event.preventDefault();

        _this4.setState({
          current: label
        });

        (0, _setImmediate2["default"])(function () {
          return _this4.props.onSelected(_samples.samples[label]);
        });
      };
    });
    _this4.state = {
      current: "Simple"
    };
    return _this4;
  }

  (0, _createClass2["default"])(Selector, [{
    key: "shouldComponentUpdate",
    value: function shouldComponentUpdate(nextProps, nextState) {
      return shouldRender(this, nextProps, nextState);
    }
  }, {
    key: "render",
    value: function render() {
      var _this5 = this;

      return _react["default"].createElement("ul", {
        className: "nav nav-pills"
      }, (0, _keys["default"])(_samples.samples).map(function (label, i) {
        return _react["default"].createElement("li", {
          key: i,
          role: "presentation",
          className: _this5.state.current === label ? "active" : ""
        }, _react["default"].createElement("a", {
          href: "#",
          onClick: _this5.onLabelClick(label)
        }, label));
      }));
    }
  }]);
  return Selector;
}(_react.Component);

function ThemeSelector(_ref) {
  var theme = _ref.theme,
      themes = _ref.themes,
      select = _ref.select;
  var themeSchema = {
    type: "string",
    "enum": (0, _keys["default"])(themes)
  };
  return _react["default"].createElement(_reactJsonschemaForm["default"], {
    className: "form_rjsf_themeSelector",
    idPrefix: "rjsf_themeSelector",
    schema: themeSchema,
    formData: theme,
    onChange: function onChange(_ref2) {
      var formData = _ref2.formData;
      return select(formData, themes[formData]);
    }
  }, _react["default"].createElement("div", null));
}

var CopyLink =
/*#__PURE__*/
function (_Component4) {
  (0, _inherits2["default"])(CopyLink, _Component4);

  function CopyLink() {
    var _getPrototypeOf2;

    var _this6;

    (0, _classCallCheck2["default"])(this, CopyLink);

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this6 = (0, _possibleConstructorReturn2["default"])(this, (_getPrototypeOf2 = (0, _getPrototypeOf3["default"])(CopyLink)).call.apply(_getPrototypeOf2, [this].concat(args)));
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this6), "onCopyClick", function (event) {
      _this6.input.select();

      document.execCommand("copy");
    });
    return _this6;
  }

  (0, _createClass2["default"])(CopyLink, [{
    key: "render",
    value: function render() {
      var _this7 = this;

      var _this$props = this.props,
          shareURL = _this$props.shareURL,
          onShare = _this$props.onShare;

      if (!shareURL) {
        return _react["default"].createElement("button", {
          className: "btn btn-default",
          type: "button",
          onClick: onShare
        }, "Share");
      }

      return _react["default"].createElement("div", {
        className: "input-group"
      }, _react["default"].createElement("input", {
        type: "text",
        ref: function ref(input) {
          return _this7.input = input;
        },
        className: "form-control",
        defaultValue: shareURL
      }), _react["default"].createElement("span", {
        className: "input-group-btn"
      }, _react["default"].createElement("button", {
        className: "btn btn-default",
        type: "button",
        onClick: this.onCopyClick
      }, _react["default"].createElement("i", {
        className: "glyphicon glyphicon-copy"
      }))));
    }
  }]);
  return CopyLink;
}(_react.Component);

var Playground =
/*#__PURE__*/
function (_Component5) {
  (0, _inherits2["default"])(Playground, _Component5);

  function Playground(props) {
    var _this8;

    (0, _classCallCheck2["default"])(this, Playground);
    _this8 = (0, _possibleConstructorReturn2["default"])(this, (0, _getPrototypeOf3["default"])(Playground).call(this, props)); // initialize state with Simple data sample

    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this8), "load", function (data) {
      // Reset the ArrayFieldTemplate whenever you load new data
      var ArrayFieldTemplate = data.ArrayFieldTemplate,
          ObjectFieldTemplate = data.ObjectFieldTemplate,
          extraErrors = data.extraErrors; // uiSchema is missing on some examples. Provide a default to
      // clear the field in all cases.

      var _data$uiSchema = data.uiSchema,
          uiSchema = _data$uiSchema === void 0 ? {} : _data$uiSchema;
      var _data$theme = data.theme,
          theme = _data$theme === void 0 ? _this8.state.theme : _data$theme;
      var themes = _this8.props.themes;

      _this8.onThemeSelected(theme, themes[theme]); // force resetting form component instance


      _this8.setState({
        form: false
      }, function () {
        return _this8.setState((0, _objectSpread2["default"])({}, data, {
          form: true,
          ArrayFieldTemplate: ArrayFieldTemplate,
          ObjectFieldTemplate: ObjectFieldTemplate,
          uiSchema: uiSchema,
          extraErrors: extraErrors
        }));
      });
    });
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this8), "onSchemaEdited", function (schema) {
      return _this8.setState({
        schema: schema,
        shareURL: null
      });
    });
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this8), "onUISchemaEdited", function (uiSchema) {
      return _this8.setState({
        uiSchema: uiSchema,
        shareURL: null
      });
    });
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this8), "onFormDataEdited", function (formData) {
      return _this8.setState({
        formData: formData,
        shareURL: null
      });
    });
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this8), "onExtraErrorsEdited", function (extraErrors) {
      return _this8.setState({
        extraErrors: extraErrors,
        shareURL: null
      });
    });
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this8), "onThemeSelected", function (theme) {
      var _ref3 = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : {},
          stylesheet = _ref3.stylesheet,
          themeObj = _ref3.theme,
          editor = _ref3.editor;

      _this8.setState({
        theme: theme,
        themeObj: themeObj,
        stylesheet: stylesheet,
        editor: editor ? editor : "default"
      });
    });
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this8), "setLiveSettings", function (_ref4) {
      var formData = _ref4.formData;
      return _this8.setState({
        liveSettings: formData
      });
    });
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this8), "onFormDataChange", function (_ref5) {
      var formData = _ref5.formData;
      return _this8.setState({
        formData: formData,
        shareURL: null
      });
    });
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this8), "onShare", function () {
      var _this8$state = _this8.state,
          formData = _this8$state.formData,
          schema = _this8$state.schema,
          uiSchema = _this8$state.uiSchema,
          liveSettings = _this8$state.liveSettings,
          errorSchema = _this8$state.errorSchema,
          theme = _this8$state.theme;
      var _document = document,
          _document$location = _document.location,
          origin = _document$location.origin,
          pathname = _document$location.pathname;

      try {
        var hash = btoa((0, _stringify["default"])({
          formData: formData,
          schema: schema,
          uiSchema: uiSchema,
          theme: theme,
          liveSettings: liveSettings,
          errorSchema: errorSchema
        }));

        _this8.setState({
          shareURL: "".concat(origin).concat(pathname, "#").concat(hash)
        });
      } catch (err) {
        _this8.setState({
          shareURL: null
        });
      }
    });
    var _samples$Simple = _samples.samples.Simple,
        _schema = _samples$Simple.schema,
        _uiSchema = _samples$Simple.uiSchema,
        _formData = _samples$Simple.formData,
        validate = _samples$Simple.validate;
    _this8.state = {
      form: false,
      schema: _schema,
      uiSchema: _uiSchema,
      formData: _formData,
      validate: validate,
      theme: "default",
      liveSettings: {
        validate: true,
        disable: false,
        omitExtraData: false,
        liveOmit: false
      },
      shareURL: null,
      themeObj: {}
    };
    return _this8;
  }

  (0, _createClass2["default"])(Playground, [{
    key: "componentDidMount",
    value: function componentDidMount() {
      var hash = document.location.hash.match(/#(.*)/);

      if (hash && typeof hash[1] === "string" && hash[1].length > 0) {
        try {
          this.load(JSON.parse(atob(hash[1])));
        } catch (err) {
          alert("Unable to load form setup data.");
        }
      } else {
        this.load({
          theme: (0, _keys["default"])(this.props.themes)[0]
        });
      }
    }
  }, {
    key: "shouldComponentUpdate",
    value: function shouldComponentUpdate(nextProps, nextState) {
      return shouldRender(this, nextProps, nextState);
    }
  }, {
    key: "render",
    value: function render() {
      var _this$state2 = this.state,
          schema = _this$state2.schema,
          uiSchema = _this$state2.uiSchema,
          formData = _this$state2.formData,
          extraErrors = _this$state2.extraErrors,
          liveSettings = _this$state2.liveSettings,
          validate = _this$state2.validate,
          theme = _this$state2.theme,
          themeObj = _this$state2.themeObj,
          ArrayFieldTemplate = _this$state2.ArrayFieldTemplate,
          ObjectFieldTemplate = _this$state2.ObjectFieldTemplate,
          transformErrors = _this$state2.transformErrors;
      var themes = this.props.themes;
      var FormComponent = (0, _reactJsonschemaForm.withTheme)(themeObj);
      return _react["default"].createElement("div", {
        className: "container-fluid"
      }, _react["default"].createElement("div", {
        className: "page-header"
      }, _react["default"].createElement("h1", null, "react-jsonschema-form"), _react["default"].createElement("div", {
        className: "row"
      }, _react["default"].createElement("div", {
        className: "col-sm-8"
      }, _react["default"].createElement(Selector, {
        onSelected: this.load
      })), _react["default"].createElement("div", {
        className: "col-sm-2"
      }, _react["default"].createElement(_reactJsonschemaForm["default"], {
        idPrefix: "rjsf_options",
        schema: liveSettingsSchema,
        formData: liveSettings,
        onChange: this.setLiveSettings
      }, _react["default"].createElement("div", null))), _react["default"].createElement("div", {
        className: "col-sm-2"
      }, _react["default"].createElement(ThemeSelector, {
        themes: themes,
        theme: theme,
        select: this.onThemeSelected
      }), _react["default"].createElement("br", null), _react["default"].createElement("br", null), _react["default"].createElement(CopyLink, {
        shareURL: this.state.shareURL,
        onShare: this.onShare
      })))), _react["default"].createElement("div", {
        className: "col-sm-7"
      }, _react["default"].createElement(Editor, {
        title: "JSONSchema",
        code: toJson(schema),
        onChange: this.onSchemaEdited
      }), _react["default"].createElement("div", {
        className: "row"
      }, _react["default"].createElement("div", {
        className: "col-sm-6"
      }, _react["default"].createElement(Editor, {
        title: "UISchema",
        code: toJson(uiSchema),
        onChange: this.onUISchemaEdited
      })), _react["default"].createElement("div", {
        className: "col-sm-6"
      }, _react["default"].createElement(Editor, {
        title: "formData",
        code: toJson(formData),
        onChange: this.onFormDataEdited
      }))), extraErrors && _react["default"].createElement("div", {
        className: "row"
      }, _react["default"].createElement("div", {
        className: "col"
      }, _react["default"].createElement(Editor, {
        title: "extraErrors",
        code: toJson(extraErrors || {}),
        onChange: this.onExtraErrorsEdited
      })))), _react["default"].createElement("div", {
        className: "col-sm-5"
      }, this.state.form && _react["default"].createElement(_DemoFrame["default"], {
        head: _react["default"].createElement("link", {
          rel: "stylesheet",
          id: "theme",
          href: this.state.stylesheet || ""
        }),
        style: {
          width: "100%",
          height: 1000,
          border: 0
        },
        theme: theme
      }, _react["default"].createElement(FormComponent, {
        ArrayFieldTemplate: ArrayFieldTemplate,
        ObjectFieldTemplate: ObjectFieldTemplate,
        liveValidate: liveSettings.validate,
        disabled: liveSettings.disable,
        omitExtraData: liveSettings.omitExtraData,
        liveOmit: liveSettings.liveOmit,
        schema: schema,
        uiSchema: uiSchema,
        formData: formData,
        extraErrors: extraErrors,
        onChange: this.onFormDataChange,
        onSubmit: function onSubmit(_ref6, e) {
          var formData = _ref6.formData;
          console.log("submitted formData", formData);
          console.log("submit event", e);
        },
        fields: {
          geo: GeoPosition
        },
        validate: validate,
        onBlur: function onBlur(id, value) {
          return console.log("Touched ".concat(id, " with value ").concat(value));
        },
        onFocus: function onFocus(id, value) {
          return console.log("Focused ".concat(id, " with value ").concat(value));
        },
        transformErrors: transformErrors,
        onError: log("errors")
      }))), _react["default"].createElement("div", {
        className: "col-sm-12"
      }, _react["default"].createElement("p", {
        style: {
          textAlign: "center"
        }
      }, "Powered by", " ", _react["default"].createElement("a", {
        href: "https://github.com/mozilla-services/react-jsonschema-form"
      }, "react-jsonschema-form"), ".", process.env.SHOW_NETLIFY_BADGE === "true" && _react["default"].createElement("div", {
        style: {
          "float": "right"
        }
      }, _react["default"].createElement("a", {
        href: "https://www.netlify.com"
      }, _react["default"].createElement("img", {
        src: "https://www.netlify.com/img/global/badges/netlify-color-accent.svg"
      }))))));
    }
  }]);
  return Playground;
}(_react.Component);

var _default = Playground;
exports["default"] = _default;
