// @kit-ui/admin/lib/NavSettings/NavSettings.stories.js
"use strict";

exports.__esModule = true;
exports.Default = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _NavSettings = _interopRequireDefault(require("./NavSettings"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _default = {
  title: 'Navigation/NavSettings',
  component: _NavSettings.default,
  args: {},
  argTypes: {}
};
exports.default = _default;

var Template = function Template(args) {
  return /*#__PURE__*/_react.default.createElement(_NavSettings.default, args);
};

var Default = Template.bind({});
exports.Default = Default;
Default.args = {
  settings: [{
    template: "grid",
    name: "grid",
    options: ["1:1", "2:2"]
  }, {
    template: "color",
    name: "color",
    label: "Red",
    value: "#ff0",
    options: ["red", "blue"]
  }, {
    name: "color",
    label: "Red",
    value: "#ff0",
    options: ["red", "blue"]
  }, {
    name: "color",
    label: "Red",
    value: "#ff0",
    options: [{
      value: "red"
    }, {
      value: "blue"
    }, {
      value: "green"
    }]
  }]
};
