// @kit-ui/schema/lib/Field/FieldTemplate.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _FormControl = _interopRequireDefault(require("@material-ui/core/FormControl"));

var _FormHelperText = _interopRequireDefault(require("@material-ui/core/FormHelperText"));

var _List = _interopRequireDefault(require("@material-ui/core/List"));

var _ListItem = _interopRequireDefault(require("@material-ui/core/ListItem"));

var _Typography = _interopRequireDefault(require("@material-ui/core/Typography"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var FieldTemplate = function FieldTemplate(_ref) {
  var id = _ref.id,
      children = _ref.children,
      displayLabel = _ref.displayLabel,
      required = _ref.required,
      _ref$rawErrors = _ref.rawErrors,
      rawErrors = _ref$rawErrors === void 0 ? [] : _ref$rawErrors,
      rawHelp = _ref.rawHelp,
      rawDescription = _ref.rawDescription,
      props = _objectWithoutPropertiesLoose(_ref, ["id", "children", "displayLabel", "required", "rawErrors", "rawHelp", "rawDescription"]);

  return children;
  return /*#__PURE__*/_react.default.createElement("div", null, /*#__PURE__*/_react.default.createElement(_FormControl.default, {
    fullWidth: true,
    error: rawErrors.length ? true : false,
    required: required
  }, children, displayLabel && rawDescription ? /*#__PURE__*/_react.default.createElement(_Typography.default, {
    variant: "caption",
    color: "textSecondary"
  }, rawDescription) : null, rawErrors.length > 0 && /*#__PURE__*/_react.default.createElement(_List.default, {
    dense: true,
    disablePadding: true
  }, rawErrors.map(function (error, i) {
    return /*#__PURE__*/_react.default.createElement(_ListItem.default, {
      key: i,
      disableGutters: true
    }, /*#__PURE__*/_react.default.createElement(_FormHelperText.default, {
      id: id
    }, error));
  })), rawHelp && /*#__PURE__*/_react.default.createElement(_FormHelperText.default, {
    id: id
  }, rawHelp)));
};

var _default = FieldTemplate;
exports.default = _default;
module.exports = exports.default;
