// @kit-ui/admin/lib/PreviewMedia/PreviewMedia.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireWildcard(require("react"));

var _styles = require("@material-ui/core/styles");

var _reactI18next = require("react-i18next");

var _ = require("../");

var _MediaPreview = _interopRequireDefault(require("../Media/MediaPreview"));

var _MediaMetadata = _interopRequireDefault(require("../Media/MediaMetadata"));

var _Typography = _interopRequireDefault(require("@material-ui/core/Typography"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var useStyles = (0, _styles.makeStyles)(function (theme) {
  return {
    preview: {
      backgroundColor: theme.palette.background.default,
      color: theme.palette.text.primary,
      position: "absolute",
      top: 0,
      right: 0,
      bottom: 0,
      left: 0,
      display: "flex",
      flexDirection: "column",
      width: "100%"
    },
    media: {
      flexBasis: "50%",
      position: "relative",
      display: "flex",
      alignItems: "center",
      justifyContent: "center",
      margin: theme.spacing(1)
    },
    content: {
      display: "flex",
      flexDirection: "column",
      alignItems: "center"
    },
    title: {
      fontSize: "24px"
    },
    metadata: {
      width: "100%",
      fontFamily: "Akkurat, sans-serif",
      fontSize: "16px",
      lineHeight: "24px",
      "& b": {
        fontWeight: "bold"
      },
      "& > p": {
        padding: theme.spacing(1, 0)
      },
      "& > * + *": {
        borderTop: "1px solid",
        borderColor: theme.palette.divider
      }
    }
  };
});

var PreviewMedia = function PreviewMedia(props) {
  var _useTranslation = (0, _reactI18next.useTranslation)(),
      t = _useTranslation.t,
      i18n = _useTranslation.i18n;

  var classes = useStyles();
  var title = props.title,
      filename = props.filename,
      mimeType = props.mimeType;
  return /*#__PURE__*/_react.default.createElement("article", {
    className: classes.preview
  }, /*#__PURE__*/_react.default.createElement(_MediaPreview.default, _extends({}, props, {
    className: classes.media,
    maxWidth: "100%",
    minHeight: 192
  })), /*#__PURE__*/_react.default.createElement("div", {
    className: classes.content
  }, /*#__PURE__*/_react.default.createElement(_Typography.default, {
    className: classes.title
  }, title || filename), /*#__PURE__*/_react.default.createElement(_.ModuleLabel, null, mimeType), /*#__PURE__*/_react.default.createElement(_MediaMetadata.default, props)));
};

var _default = PreviewMedia;
exports.default = _default;
module.exports = exports.default;
