// @kit-ui/admin/lib/PreviewMedia/PreviewMedia.stories.js
"use strict";

exports.__esModule = true;
exports.Pdf = exports.Sketchfab = exports.Vimeo = exports.Youtube = exports.Audio = exports.Video = exports.Image = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _PreviewMedia = _interopRequireDefault(require("./PreviewMedia"));

var _image = _interopRequireDefault(require("../data/media/image"));

var _video = _interopRequireDefault(require("../data/media/video"));

var _audio = _interopRequireDefault(require("../data/media/audio"));

var _youtube = _interopRequireDefault(require("../data/media/youtube"));

var _vimeo = _interopRequireDefault(require("../data/media/vimeo"));

var _sketchfab = _interopRequireDefault(require("../data/media/sketchfab"));

var _pdf = _interopRequireDefault(require("../data/media/pdf"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var _default = {
  title: 'Preview/PreviewMedia',
  component: _PreviewMedia.default,
  args: {
    mediaType: "mediaType"
  },
  argTypes: {}
};
exports.default = _default;

var Template = function Template(args) {
  return /*#__PURE__*/_react.default.createElement(_PreviewMedia.default, args);
};

var Image = Template.bind({});
exports.Image = Image;
Image.args = _extends({}, _image.default);
var Video = Template.bind({});
exports.Video = Video;
Video.args = _extends({}, _video.default);
var Audio = Template.bind({});
exports.Audio = Audio;
Audio.args = _extends({}, _audio.default);
var Youtube = Template.bind({});
exports.Youtube = Youtube;
Youtube.args = _extends({}, _youtube.default);
var Vimeo = Template.bind({});
exports.Vimeo = Vimeo;
Vimeo.args = _extends({}, _vimeo.default);
var Sketchfab = Template.bind({});
exports.Sketchfab = Sketchfab;
Sketchfab.args = _extends({}, _sketchfab.default);
var Pdf = Template.bind({});
exports.Pdf = Pdf;
Pdf.args = _extends({}, _pdf.default);
