// @kit-ui/schema/lib/utils/__getUiFieldset BACKUP.js
"use strict";

exports.__esModule = true;
exports.getUiFieldset = getUiFieldset;
exports.default = void 0;

var _2 = require("./");

var _lodash = _interopRequireDefault(require("lodash"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function getUiFieldset(props) {
  var id = props.id,
      idSchema = props.idSchema,
      schema = props.schema,
      uiSchema = props.uiSchema,
      formData = props.formData,
      registry = props.registry;
  var definitions = registry.definitions,
      formContext = registry.formContext;
  var uiOptions = (0, _2.getUiOptions)(uiSchema);
  var fieldset = uiOptions.fieldset || false;

  if (!fieldset) {
    return false;
  }

  var uiFieldset = [];

  if (fieldset) {
    fieldset.forEach(function (name) {
      // get fieldId
      var fieldId;

      if (id && name) {
        fieldId = id + '_' + name;
      } else {
        fieldId = idSchema.$id;
      } // selected?


      var selected = false;

      if (formContext.currentId && formContext.currentId.startsWith(fieldId)) {
        selected = true;
      } // schema, uiSchema, formData


      var fieldSchema = schema && schema.properties && schema.properties[name] && (0, _2.retrieveSchema)(schema.properties[name], definitions, formData[name]);
      var fieldFormData = fieldSchema && formData && formData[name] && (0, _2.getDefaultFormState)(fieldSchema, formData[name]);
      var fieldUiSchema = uiSchema && uiSchema[name] || {};
      var fieldUiOptions = fieldUiSchema && (0, _2.getUiOptions)(fieldUiSchema);
      uiFieldset.push({
        id: fieldId,
        name: name,
        selected: selected,
        schema: fieldSchema,
        uiSchema: fieldUiSchema,
        uiOptions: fieldUiOptions,
        formData: fieldFormData,
        registry: registry
      });
    });
  }

  var selected = _lodash.default.find(uiFieldset, function (field) {
    return field.selected === true;
  });

  if (uiFieldset[0] && !selected) {
    uiFieldset[0].selected = true;
  }

  return uiFieldset;
}

var _default = getUiFieldset;
exports.default = _default;
