// @kit-ui/schema/lib/utils/getUiMenu.js
"use strict";

exports.__esModule = true;
exports.default = exports.getUiMenu = void 0;

var _lodash = _interopRequireDefault(require("lodash"));

var _2 = require("./");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var getUiMenu = function getUiMenu(props) {
  var formContext = props.formContext;
  var onExpand = formContext && formContext.onExpand;
  var uiFieldset = (0, _2.getUiFieldset)(props);
  var menu = [];

  if (uiFieldset) {
    menu = uiFieldset.map(function (field) {
      var name = field.name,
          idPrefix = field.idPrefix,
          uiSchema = field.uiSchema;
      var uiOptions = (0, _2.getUiOptions)(uiSchema);
      return {
        id: idPrefix,
        name: name,
        icon: uiOptions['icon'] || undefined,
        label: uiOptions['title'] || name,
        selected: uiOptions.selected,
        onClick: function onClick() {
          return onExpand({
            id: idPrefix
          });
        }
      };
    });
  }

  if (!menu.length) {
    return false;
  }

  return menu;
};

exports.getUiMenu = getUiMenu;
var _default = getUiMenu;
exports.default = _default;
