// @kit-ui/schema/lib/utils/getUiSettings.js
"use strict";

exports.__esModule = true;
exports.default = exports.getUiSettings = void 0;

var _react = _interopRequireDefault(require("react"));

var _ = require(".");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var getUiSettings = function getUiSettings(props) {
  var id = props.id,
      schema = props.schema,
      uiSchema = props.uiSchema,
      formData = props.formData,
      registry = props.registry;
  var uiOptions = (0, _.getUiOptions)(uiSchema);
  var settings = uiOptions && uiOptions.settings;

  if (!settings) {
    return undefined;
  }

  var uiActions = (0, _.getUiActions)(props);
  var onPropertyChange = uiActions.onPropertyChange;
  var SettingsField = registry.fields.SettingsField;

  var SettingsTemplate = function SettingsTemplate(props) {
    return /*#__PURE__*/_react.default.createElement(SettingsField, _extends({}, props, {
      registry: registry
    }));
  };

  var uiSettings = [];
  settings.map(function (name) {
    if (typeof name === "string" && schema.properties[name]) {
      var itemId = id + '_settings_' + name;
      var itemSchema = schema.properties[name] || {};
      var itemUiSchema = uiSchema[name] || {};
      var itemFormData = formData[name] || undefined;
      uiSettings.push({
        id: itemId,
        name: name,
        schema: itemSchema,
        uiSchema: itemUiSchema,
        formData: itemFormData,
        template: SettingsTemplate,
        onChange: function onChange(value) {
          return onPropertyChange(name, value);
        }
      });
    }
  });
  return uiSettings;
};

exports.getUiSettings = getUiSettings;
var _default = getUiSettings;
exports.default = _default;
