// @kit-ui/admin/es/NavSettings/ValignSettings.js
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import React, { forwardRef, useState, useEffect, useRef } from 'react';
import SettingsButton from "./SettingsButton";
import SettingsDropdown from "./SettingsDropdown";
import _ from "lodash";
import SettingsList from "./SettingsList";
var defaultOptions = ["top", "center", "bottom"];
var defaultProps = {
  "top": {
    "order": 1,
    "icon": "vertical_align_top"
  },
  "center": {
    "order": 2,
    "icon": "vertical_align_center"
  },
  "bottom": {
    "order": 3,
    "icon": "vertical_align_bottom"
  }
};

var getOptions = function getOptions(_ref) {
  var _ref$options = _ref.options,
      options = _ref$options === void 0 ? [] : _ref$options;
  var _options = [],
      _byValue = {};

  if (!options.length) {
    options = defaultOptions;
  }

  options.map(function (item, index) {
    if (typeof item === "string") {
      item = {
        label: item,
        value: item
      };
    }

    var itemProps = defaultProps[item.value];
    item = _extends({}, item, itemProps);

    _options.push(item);

    _byValue[item.value] = item;
  });
  _options = _.orderBy(_options, 'order', 'desc');
  return {
    byValue: _byValue,
    options: _options
  };
};

var SettingsButtonRef = forwardRef(function (props, ref) {
  return /*#__PURE__*/React.createElement(SettingsButton, _extends({}, props, {
    forwardedRef: ref
  }));
});

var ValignSettings = function ValignSettings(_ref2) {
  var _ref2$type = _ref2.type,
      type = _ref2$type === void 0 ? "grid" : _ref2$type,
      name = _ref2.name,
      label = _ref2.label,
      value = _ref2.value,
      onChange = _ref2.onChange,
      props = _objectWithoutPropertiesLoose(_ref2, ["type", "name", "label", "value", "onChange"]);

  var _useState = useState(false),
      expanded = _useState[0],
      setExpanded = _useState[1];

  var anchorRef = useRef(null);

  var _getOptions = getOptions(props),
      options = _getOptions.options,
      byValue = _getOptions.byValue;

  var current = byValue[value] || options[0];

  var onToggle = function onToggle(event) {
    setExpanded(function (prevExpanded) {
      return !prevExpanded;
    });
  };

  var onSelect = function onSelect(item) {
    if (onChange) {
      onChange(name, item.value);
    }

    setExpanded(false);
  };

  var onCollapse = function onCollapse() {
    setExpanded(false);
  };

  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(SettingsButtonRef, _extends({}, current, {
    type: type,
    onToggle: onToggle,
    expanded: expanded,
    ref: anchorRef
  })), /*#__PURE__*/React.createElement(SettingsDropdown, {
    expanded: expanded,
    anchorEl: anchorRef.current,
    onCollapse: onCollapse
  }, /*#__PURE__*/React.createElement(SettingsList, {
    value: value,
    options: options,
    onSelect: onSelect
  })));
};

export default ValignSettings;
