// @kit-ui/admin/es/Masonry/Masonry.js
import React from 'react';
import PropTypes from "prop-types";
import { makeStyles } from '@material-ui/core/styles';
import { withResizeDetector } from 'react-resize-detector';
var useStyles = makeStyles(function (theme) {
  return {
    masonry: {
      width: "100%",
      overflow: "hidden"
    },
    row: {
      display: "flex",
      width: "auto",
      margin: function margin(props) {
        return theme.spacing(props.spacing / 2 * -1);
      },
      "& > *": {
        flexBasis: 0,
        flexGrow: 1
      }
    },
    col: {
      display: "flex",
      flexDirection: "column"
    },
    item: {
      backgroundColor: "red",
      margin: function margin(props) {
        return theme.spacing(props.spacing / 2);
      },
      "& > *": {
        display: "block",
        width: "100%"
      }
    }
  };
});

var Masonry = function Masonry(_ref) {
  var width = _ref.width,
      _ref$columns = _ref.columns,
      columns = _ref$columns === void 0 ? 3 : _ref$columns,
      _ref$spacing = _ref.spacing,
      spacing = _ref$spacing === void 0 ? 1 : _ref$spacing,
      _ref$padding = _ref.padding,
      padding = _ref$padding === void 0 ? 0 : _ref$padding,
      children = _ref.children,
      _ref$debug = _ref.debug,
      debug = _ref$debug === void 0 ? false : _ref$debug;
  var classes = useStyles({
    padding: padding,
    spacing: spacing
  });
  var items = [].concat(children || []);

  var getItemsByColumn = function getItemsByColumn() {
    var itemsByColumn = {};
    items.map(function (item, index) {
      var c = index % columns;

      if (!itemsByColumn[c]) {
        itemsByColumn[c] = [];
      }

      itemsByColumn[c].push(item);
    });
    return itemsByColumn;
  };

  var itemsByColumn = getItemsByColumn();
  return /*#__PURE__*/React.createElement("div", {
    className: classes.masonry
  }, /*#__PURE__*/React.createElement("div", {
    className: classes.row
  }, itemsByColumn && Object.values(itemsByColumn).map(function (items) {
    return /*#__PURE__*/React.createElement("div", {
      className: classes.col
    }, items.map(function (item) {
      return /*#__PURE__*/React.createElement("div", {
        className: classes.item
      }, item);
    }));
  })));
};

export default withResizeDetector(Masonry);
