// @kit-ui/admin/es/Masonry/Masonry.stories.js
import React from 'react';
import Masonry from './Masonry';
export default {
  title: 'Layout/Masonry',
  component: Masonry,
  args: {},
  argTypes: {}
};

var Template = function Template(args) {
  return /*#__PURE__*/React.createElement(Masonry, args, /*#__PURE__*/React.createElement("img", {
    src: "https://w3bits.com/files/img/cherry-plant.jpg"
  }), /*#__PURE__*/React.createElement("img", {
    src: "https://w3bits.com/files/img/blueberries.jpg"
  }), /*#__PURE__*/React.createElement("img", {
    src: "https://w3bits.com/files/img/cherry-plant.jpg"
  }), /*#__PURE__*/React.createElement("img", {
    src: "https://w3bits.com/files/img/blueberries.jpg"
  }), /*#__PURE__*/React.createElement("img", {
    src: "https://w3bits.com/files/img/cherry-plant.jpg"
  }), /*#__PURE__*/React.createElement("img", {
    src: "https://w3bits.com/files/img/blueberries.jpg"
  }), /*#__PURE__*/React.createElement("img", {
    src: "https://w3bits.com/files/img/cherry-plant.jpg"
  }), /*#__PURE__*/React.createElement("img", {
    src: "https://w3bits.com/files/img/blueberries.jpg"
  }), /*#__PURE__*/React.createElement("img", {
    src: "https://w3bits.com/files/img/cherry-plant.jpg"
  }), /*#__PURE__*/React.createElement("img", {
    src: "https://w3bits.com/files/img/blueberries.jpg"
  }));
};

export var Default = Template.bind({});
Default.args = {};
export var TwoColumns = Template.bind({});
TwoColumns.args = {
  columns: 2
};
export var ThreeColumns = Template.bind({});
ThreeColumns.args = {
  columns: 3
};
export var FourColumns = Template.bind({});
FourColumns.args = {
  columns: 4
};
