// @kit-ui/schema/es/PageEditor/PageContentLayout.stories.js
import React from 'react';
import SchemaEditor from "../SchemaEditor";

var DialogTemplate = function DialogTemplate(_ref) {
  var formData = _ref.formData;
  return /*#__PURE__*/React.createElement("p", null, "Dialg template");
};

var PreviewTemplate = function PreviewTemplate(_ref2) {
  var formData = _ref2.formData;
  return /*#__PURE__*/React.createElement("p", null, "Preview template");
};

export default {
  title: 'PageEditor/PageContentLayout',
  component: SchemaEditor,
  args: {
    "schema": {
      "type": "object",
      "properties": {
        "title": {
          "type": "string"
        },
        "content": {
          "type": "object",
          "properties": {
            "title": {
              "type": "string"
            },
            "description": {
              "type": "string"
            },
            "array": {
              "type": "array",
              "items": {
                "type": "string"
              }
            }
          }
        }
      }
    },
    "uiSchema": {
      "ui:layout": "pageContent",
      "ui:fieldset": ["content"],
      "content": {
        "array": {
          "ui:buttons": ["sidebar"]
        }
      }
    },
    "formContext": {
      "preview": {
        "template": PreviewTemplate
      },
      "title": "Document title",
      "parents": [{
        "title": "Parent"
      }]
    }
  },
  argTypes: {}
};

var Template = function Template(args) {
  return /*#__PURE__*/React.createElement(SchemaEditor, args);
};

export var Default = Template.bind({});
Default.args = {};
export var WithMenu = Template.bind({});
WithMenu.args = {
  "uiSchema": {
    "ui:options": {
      "nav": true
    },
    "ui:fieldset": ["menu1", "menu2"],
    "menu1": {
      "ui:fieldset": ["title", "description"]
    },
    "menu2": {
      "ui:fieldset": ["array", "description"]
    }
  }
};
