// @kit-ui/schema/es/PageEditor/PageSettingsLayout.stories.js
import React from 'react';
import SchemaEditor from "../SchemaEditor";

var DialogTemplate = function DialogTemplate(_ref) {
  var formData = _ref.formData;
  return /*#__PURE__*/React.createElement("p", null, "Dialg template");
};

var PreviewTemplate = function PreviewTemplate(_ref2) {
  var formData = _ref2.formData;
  return /*#__PURE__*/React.createElement("p", null, "Preview template");
};

export default {
  title: 'PageEditor/PageSettingsLayout',
  component: SchemaEditor,
  args: {
    "schema": {
      "type": "object",
      "properties": {
        "title": {
          "type": "string"
        },
        "array": {
          "type": "array",
          "items": {
            "type": "string"
          }
        }
      }
    },
    "uiSchema": {
      "ui:layout": "pageSettings",
      "ui:fieldset": ["array", "location"],
      "array": {
        "ui:buttons": ["dialog"]
      },
      "location": {
        "lat": {
          "type": "number"
        },
        "lng": {
          "type": "number"
        }
      }
    },
    "formContext": {
      "preview": {
        "template": PreviewTemplate
      },
      "title": "Document title",
      "parents": [{
        "title": "Parent"
      }]
    }
  },
  argTypes: {}
};

var Template = function Template(args) {
  return /*#__PURE__*/React.createElement(SchemaEditor, args);
};

export var Default = Template.bind({});
Default.args = {};
