// @kit-ui/schema/es/PageEditor/PageMenu.js
import React from 'react';
import { NavMenu, NavMenuList, NavMenuItem, NavMenuLink } from "@kit-ui/admin";
import { makeStyles } from '@material-ui/core/styles';
var useStyles = makeStyles(function (theme) {
  return {
    menu: {
      position: "absolute",
      zIndex: 1,
      top: 0,
      right: "auto",
      left: 0,
      bottom: 0,
      margin: 0,
      width: function width(props) {
        return props.menuWidth;
      },
      transition: ".125s ease-out",
      "& + *": {
        position: "absolute",
        zIndex: 1,
        top: 0,
        right: 0,
        left: 0,
        bottom: 0,
        margin: 0,
        overflowY: "scroll"
      },
      "&[aria-expanded=true] + *": {
        marginLeft: function marginLeft(props) {
          return props.menuWidth;
        }
      }
    },
    menuList: {
      margin: theme.spacing(2, 0),
      "& > li + li": {
        marginTop: theme.spacing(1)
      }
    }
  };
});

var PageMenu = function PageMenu(_ref) {
  var className = _ref.className,
      _ref$menuWidth = _ref.menuWidth,
      menuWidth = _ref$menuWidth === void 0 ? 224 : _ref$menuWidth,
      _ref$menu = _ref.menu,
      menu = _ref$menu === void 0 ? undefined : _ref$menu;
  var classes = useStyles({
    menuWidth: menuWidth
  });

  if (!menu) {
    return false;
  }

  return /*#__PURE__*/React.createElement("aside", {
    className: className || classes.menu,
    "aria-expanded": true
  }, /*#__PURE__*/React.createElement(NavMenuList, {
    className: classes.menuList
  }, menu.map(function (item, index) {
    return /*#__PURE__*/React.createElement(NavMenuItem, null, /*#__PURE__*/React.createElement(NavMenuLink, item));
  })));
};

export default PageMenu;
