// @kit-ui/schema/es/PageEditor/PageSettingsLayout.js
import React from 'react';
import { EditorSection, EditorContent, EditorOverlay } from "@kit-ui/admin";
import PageMenu from "./PageMenu";
import { makeStyles } from '@material-ui/core/styles';
var useStyles = makeStyles(function (theme) {
  return {
    settings: {
      backgroundColor: theme.palette.background.default,
      zIndex: 1,
      position: "absolute",
      top: 0,
      right: 0,
      bottom: 0,
      left: 0,
      "&[aria-expanded=true]": {
        top: 0
      }
    }
  };
});

var PageSettings = function PageSettings(_ref) {
  var id = _ref.id,
      _ref$expanded = _ref.expanded,
      expanded = _ref$expanded === void 0 ? true : _ref$expanded,
      menuWidth = _ref.menuWidth,
      _ref$menu = _ref.menu,
      menu = _ref$menu === void 0 ? undefined : _ref$menu,
      children = _ref.children;
  var classes = useStyles();
  /*
   const renderDialog = () => {
      const DialogTemplate = dialog && dialog.template;
       return (
          <EditorOverlay position="fixed" expanded={dialog && dialog.expanded}>
              { DialogTemplate && <DialogTemplate {...dialog} /> }
          </EditorOverlay>
      )
  }
   */

  return /*#__PURE__*/React.createElement(EditorSection, {
    id: id,
    elevation: 0,
    className: classes.settings,
    "aria-expanded": expanded
  }, menu && /*#__PURE__*/React.createElement(PageMenu, {
    menuWidth: menuWidth,
    menu: menu
  }), /*#__PURE__*/React.createElement(EditorContent, {
    expanded: true
  }, children));
};

export default PageSettings;
