// @kit-ui/schema/es/PageEditor/PageEditorField.stories.js
import React from 'react';
import SchemaEditor from "../SchemaEditor/";
import PageEditorField from "./PageEditorField";

var PreviewTemplate = function PreviewTemplate(_ref) {
  var formData = _ref.formData;
  return /*#__PURE__*/React.createElement("p", null, "Preview template");
};

export default {
  title: 'PageEditor/PageEditorField',
  component: SchemaEditor,
  args: {
    "schema": {
      "type": "object",
      "properties": {
        "title": {
          "type": "string"
        },
        "arrayWithSidebar": {
          "type": "array",
          "items": {
            "type": "string"
          }
        },
        "settings": {
          "type": "object",
          "properties": {
            "setting1": {
              "type": "string"
            },
            "arrayWithOverlay": {
              "type": "array",
              "items": {
                "type": "string"
              }
            }
          }
        }
      }
    },
    "uiSchema": {
      "ui:field": "pageEditor",
      "ui:collapsible": true,
      "ui:fieldset": ["content", "settings"],
      "content": {
        "ui:fieldset": ["title", "arrayWithSidebar"],
        "ui:layout": "pageContent",
        "arrayWithSidebar": {
          "ui:dialog": true
        },
        "ui:buttons": ["sidebar"]
      },
      "settings": {
        "ui:layout": "pageSettings",
        "arrayWithOverlay": {
          "ui:dialog": true
        }
      }
    },
    "formContext": {
      "preview": {
        "template": PreviewTemplate
      },
      "parents": [{
        title: "Parent"
      }]
    }
  },
  argTypes: {}
};

var Template = function Template(args) {
  return /*#__PURE__*/React.createElement(SchemaEditor, args);
};

export var Default = Template.bind({});
Default.args = {};
export var Collapsible = Template.bind({});
Collapsible.args = {
  "ui:schema": {
    "ui:collapsible": true
  }
};
