// @kit-ui/schema/es/PageEditor/PageEditorLayout.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import React from 'react';
import { EditorBase, EditorHeader, EditorBody, EditorFooter, EditorOverlay } from "@kit-ui/admin";

var PageEditor = function PageEditor(_ref) {
  var collapsible = _ref.collapsible,
      expanded = _ref.expanded,
      overlay = _ref.overlay,
      onToggle = _ref.onToggle,
      onSelect = _ref.onSelect,
      children = _ref.children,
      props = _objectWithoutPropertiesLoose(_ref, ["collapsible", "expanded", "overlay", "onToggle", "onSelect", "children"]);

  var renderOverlay = function renderOverlay() {
    var OverlayTemplate = overlay && overlay.template;
    return /*#__PURE__*/React.createElement(EditorOverlay, {
      position: "fixed",
      expanded: overlay && overlay.expanded
    }, OverlayTemplate && /*#__PURE__*/React.createElement(OverlayTemplate, overlay));
  };

  return /*#__PURE__*/React.createElement(EditorBase, null, /*#__PURE__*/React.createElement(EditorHeader, _extends({}, props, {
    collapsible: collapsible,
    expanded: expanded,
    onToggle: onToggle,
    onSelect: onSelect
  })), /*#__PURE__*/React.createElement(EditorBody, {
    elevation: 1,
    expanded: expanded
  }, children), /*#__PURE__*/React.createElement(EditorFooter, props), overlay && renderOverlay());
};

export default PageEditor;
