// @kit-ui/schema/es/PageEditor/PageContentLayout.js
import React from 'react';
import { EditorSection, EditorContent, EditorPreview, EditorSidebar } from "@kit-ui/admin";
import PageMenu from "./PageMenu";
import { makeStyles } from '@material-ui/core/styles';
var useStyles = makeStyles(function (theme) {
  return {
    content: {
      backgroundColor: theme.palette.background.default,
      position: "absolute",
      zIndex: 2,
      top: 0,
      right: 0,
      bottom: 0,
      left: 0,
      width: "100%",
      overflow: "hidden",
      transition: ".125s ease-out",
      transform: "translateY(100%)",
      "&[aria-expanded=true]": {
        transform: "translateY(0)"
      }
    },
    wrapper: {
      backgroundColor: "inherit",
      position: "absolute",
      top: 0,
      right: 0,
      bottom: 0,
      left: 0
    }
  };
});

var PageContent = function PageContent(_ref) {
  var id = _ref.id,
      _ref$selected = _ref.selected,
      selected = _ref$selected === void 0 ? true : _ref$selected,
      sidebar = _ref.sidebar,
      preview = _ref.preview,
      menuWidth = _ref.menuWidth,
      _ref$menu = _ref.menu,
      menu = _ref$menu === void 0 ? undefined : _ref$menu,
      children = _ref.children;
  var classes = useStyles();

  var renderSidebar = function renderSidebar() {
    var SidebarTemplate = sidebar && sidebar.template;
    return /*#__PURE__*/React.createElement(EditorSidebar, {
      expanded: sidebar && sidebar.expanded
    }, SidebarTemplate && /*#__PURE__*/React.createElement(SidebarTemplate, sidebar));
  };

  var renderPreview = function renderPreview() {
    var PreviewTemplate = preview && preview.template;
    return /*#__PURE__*/React.createElement(EditorPreview, {
      expanded: true
    }, PreviewTemplate && /*#__PURE__*/React.createElement(PreviewTemplate, preview));
  };

  return /*#__PURE__*/React.createElement(EditorSection, {
    id: id,
    expanded: selected,
    className: classes.content
  }, sidebar && renderSidebar(), /*#__PURE__*/React.createElement(EditorSection, {
    elevation: 0,
    className: classes.wrapper
  }, preview && renderPreview(), /*#__PURE__*/React.createElement(EditorContent, {
    elevation: 0
  }, menu && /*#__PURE__*/React.createElement(PageMenu, {
    menuWidth: menuWidth,
    menu: menu
  }), children)));
};

export default PageContent;
