// @kit-ui/schema/es/utils/__getUiFieldset BACKUP.js
import { getUiOptions, retrieveSchema, getDefaultFormState } from './';
import _ from "lodash";
export function getUiFieldset(props) {
  var id = props.id,
      idSchema = props.idSchema,
      schema = props.schema,
      uiSchema = props.uiSchema,
      formData = props.formData,
      registry = props.registry;
  var definitions = registry.definitions,
      formContext = registry.formContext;
  var uiOptions = getUiOptions(uiSchema);
  var fieldset = uiOptions.fieldset || false;

  if (!fieldset) {
    return false;
  }

  var uiFieldset = [];

  if (fieldset) {
    fieldset.forEach(function (name) {
      // get fieldId
      var fieldId;

      if (id && name) {
        fieldId = id + '_' + name;
      } else {
        fieldId = idSchema.$id;
      } // selected?


      var selected = false;

      if (formContext.currentId && formContext.currentId.startsWith(fieldId)) {
        selected = true;
      } // schema, uiSchema, formData


      var fieldSchema = schema && schema.properties && schema.properties[name] && retrieveSchema(schema.properties[name], definitions, formData[name]);
      var fieldFormData = fieldSchema && formData && formData[name] && getDefaultFormState(fieldSchema, formData[name]);
      var fieldUiSchema = uiSchema && uiSchema[name] || {};
      var fieldUiOptions = fieldUiSchema && getUiOptions(fieldUiSchema);
      uiFieldset.push({
        id: fieldId,
        name: name,
        selected: selected,
        schema: fieldSchema,
        uiSchema: fieldUiSchema,
        uiOptions: fieldUiOptions,
        formData: fieldFormData,
        registry: registry
      });
    });
  }

  var selected = _.find(uiFieldset, function (field) {
    return field.selected === true;
  });

  if (uiFieldset[0] && !selected) {
    uiFieldset[0].selected = true;
  }

  return uiFieldset;
}
export default getUiFieldset;
