// @kit-ui/admin/es/NavToolbar/NavToolbar.stories.js
import React from 'react';
import NavToolbar from './NavToolbar';
export default {
  title: 'Navigation/NavToolbar',
  component: NavToolbar,
  args: {},
  argTypes: {}
};

var Template = function Template(args) {
  return /*#__PURE__*/React.createElement(NavToolbar, args);
};

export var Default = Template.bind({});
Default.args = {};
export var Editable = Template.bind({});
Editable.args = {
  editable: true
};
export var EditableDeletable = Template.bind({});
EditableDeletable.args = {
  editable: true,
  deletable: true
};
export var Deleted = Template.bind({});
Deleted.args = {
  editable: true,
  deletable: true,
  deleted: true
};
export var DeletedErasable = Template.bind({});
DeletedErasable.args = {
  editable: true,
  deletable: true,
  deleted: true,
  erasable: true
};
export var DeletedRestorable = Template.bind({});
DeletedRestorable.args = {
  deleted: true,
  restorable: true
};
export var DeletedRestorableErasable = Template.bind({});
DeletedRestorableErasable.args = {
  deleted: true,
  restorable: true,
  erasable: true
};
export var Hideable = Template.bind({});
Hideable.args = {
  hideable: true
};
export var HideableHidden = Template.bind({});
HideableHidden.args = {
  hideable: true,
  hidden: true
};
export var MultipleOptions = Template.bind({});
MultipleOptions.args = {
  hideable: true,
  deletable: true,
  restoreable: true
};
export var CustomToolbar = Template.bind({});
CustomToolbar.args = {
  toolbar: [{
    name: "size",
    value: "small",
    options: [{
      value: "small"
    }, {
      value: "medium"
    }, {
      value: "large"
    }]
  }, {
    name: "color",
    value: "#ff0",
    options: [{
      value: "red"
    }, {
      value: "blue"
    }, {
      value: "green"
    }]
  }]
};
