// @rjsf/playground/lib/samples/customArray.js
"use strict";

var _interopRequireDefault = require("@babel/runtime-corejs2/helpers/interopRequireDefault");

var _Object$defineProperty = require("@babel/runtime-corejs2/core-js/object/define-property");

_Object$defineProperty(exports, "__esModule", {
  value: true
});

exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

function ArrayFieldTemplate(props) {
  return _react["default"].createElement("div", {
    className: props.className
  }, props.items && props.items.map(function (element) {
    return _react["default"].createElement("div", {
      key: element.key,
      className: element.className
    }, _react["default"].createElement("div", null, element.children), element.hasMoveDown && _react["default"].createElement("button", {
      onClick: element.onReorderClick(element.index, element.index + 1)
    }, "Down"), element.hasMoveUp && _react["default"].createElement("button", {
      onClick: element.onReorderClick(element.index, element.index - 1)
    }, "Up"), _react["default"].createElement("button", {
      onClick: element.onDropIndexClick(element.index)
    }, "Delete"), _react["default"].createElement("hr", null));
  }), props.canAdd && _react["default"].createElement("div", {
    className: "row"
  }, _react["default"].createElement("p", {
    className: "col-xs-3 col-xs-offset-9 array-item-add text-right"
  }, _react["default"].createElement("button", {
    onClick: props.onAddClick,
    type: "button"
  }, "Custom +"))));
}

var _default = {
  schema: {
    title: "Custom array of strings",
    type: "array",
    items: {
      type: "string"
    }
  },
  formData: ["react", "jsonschema", "form"],
  ArrayFieldTemplate: ArrayFieldTemplate
};
exports["default"] = _default;
