// @rjsf/playground/lib/samples/customObject.js
"use strict";

var _interopRequireDefault = require("@babel/runtime-corejs2/helpers/interopRequireDefault");

var _Object$defineProperty = require("@babel/runtime-corejs2/core-js/object/define-property");

_Object$defineProperty(exports, "__esModule", {
  value: true
});

exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

function ObjectFieldTemplate(_ref) {
  var TitleField = _ref.TitleField,
      properties = _ref.properties,
      title = _ref.title,
      description = _ref.description;
  return _react["default"].createElement("div", null, _react["default"].createElement(TitleField, {
    title: title
  }), _react["default"].createElement("div", {
    className: "row"
  }, properties.map(function (prop) {
    return _react["default"].createElement("div", {
      className: "col-lg-2 col-md-4 col-sm-6 col-xs-12",
      key: prop.content.key
    }, prop.content);
  })), description);
}

var _default = {
  schema: {
    title: "A registration form",
    description: "This is the same as the simple form, but it is rendered as a bootstrap grid. Try shrinking the browser window to see it in action.",
    type: "object",
    required: ["firstName", "lastName"],
    properties: {
      firstName: {
        type: "string",
        title: "First name"
      },
      lastName: {
        type: "string",
        title: "Last name"
      },
      age: {
        type: "integer",
        title: "Age"
      },
      bio: {
        type: "string",
        title: "Bio"
      },
      password: {
        type: "string",
        title: "Password",
        minLength: 3
      },
      telephone: {
        type: "string",
        title: "Telephone",
        minLength: 10
      }
    }
  },
  formData: {
    firstName: "Chuck",
    lastName: "Norris",
    age: 75,
    bio: "Roundhouse kicking asses since 1940",
    password: "noneed"
  },
  ObjectFieldTemplate: ObjectFieldTemplate
};
exports["default"] = _default;
