// @kit-ui/admin/es/Calendar/Schedule.js
import React from 'react';
import moment from 'moment';
import { getSchedule } from '../utils/';
import { makeStyles } from '@material-ui/core/styles';
var useStyles = makeStyles(function (theme) {
  return {
    schedule: {
      display: "flex",
      flexDirection: "row",
      alignItems: "flex-start",
      paddingLeft: theme.spacing(7),
      "& > *": {
        flexBasis: 0,
        flexGrow: 1
      }
    },
    date: {
      display: "flex",
      flexDirection: "column",
      alignItems: "flex-start",
      "& $hourLabel": {
        display: "none"
      },
      "&:first-child $hourLabel": {
        display: "block"
      }
    },
    dateLabel: {
      position: "sticky",
      zIndex: 2,
      top: 0,
      padding: theme.spacing(1, 0)
    }
  };
});

var ScheduleDate = function ScheduleDate(_ref) {
  var datetime = _ref.datetime,
      _ref$events = _ref.events,
      events = _ref$events === void 0 ? [] : _ref$events;
  var date = moment(datetime).format('LL');
  var classes = useStyles();
  return /*#__PURE__*/React.createElement("div", {
    className: classes.date
  }, /*#__PURE__*/React.createElement("date", {
    className: classes.dateLabel
  }, date), /*#__PURE__*/React.createElement(ScheduleEvents, {
    events: events
  }));
};

var ScheduleEvent = function ScheduleEvent(_ref2) {
  var dtStart = _ref2.dtStart,
      title = _ref2.title;
  var classes = useStyles();
  var hour = moment(dtStart).format('HH:mm');
  return /*#__PURE__*/React.createElement("div", null, /*#__PURE__*/React.createElement("time", {
    className: classes.timeLabel
  }, hour), title);
};

var ScheduleEvents = function ScheduleEvents(_ref3) {
  var _ref3$events = _ref3.events,
      events = _ref3$events === void 0 ? [] : _ref3$events;
  var classes = useStyles();
  return /*#__PURE__*/React.createElement("div", null, events && events.map(function (event) {
    return /*#__PURE__*/React.createElement(ScheduleEvent, event);
  }));
};

var Schedule = function Schedule(_ref4) {
  var date = _ref4.date,
      _ref4$items = _ref4.items,
      items = _ref4$items === void 0 ? [] : _ref4$items,
      _ref4$days = _ref4.days,
      days = _ref4$days === void 0 ? 4 : _ref4$days,
      _ref4$timeFrom = _ref4.timeFrom,
      timeFrom = _ref4$timeFrom === void 0 ? 8 : _ref4$timeFrom,
      _ref4$timeTo = _ref4.timeTo,
      timeTo = _ref4$timeTo === void 0 ? 23 : _ref4$timeTo;
  var timetable = getSchedule({
    date: date,
    items: items,
    days: days,
    timeFrom: timeFrom,
    timeTo: timeTo
  });
  var rows = timetable.rows,
      events = timetable.events;
  var classes = useStyles();
  return /*#__PURE__*/React.createElement("div", {
    className: classes.schedule
  }, rows.map(function (row) {
    return /*#__PURE__*/React.createElement(ScheduleDate, row);
  }));
};

export default Schedule;
