// @kit-ui/admin/es/Calendar/Timetable.js
import React from 'react';
import PropTypes from "prop-types";
import moment from 'moment';
import { getTimetable } from '../utils/';
import { makeStyles } from '@material-ui/core/styles';
var useStyles = makeStyles(function (theme) {
  return {
    timetable: {
      display: "flex",
      flexDirection: "row",
      alignItems: "flex-start",
      paddingLeft: theme.spacing(7),
      "& > *": {
        flexBasis: 0,
        flexGrow: 1
      }
    },
    date: {
      display: "flex",
      flexDirection: "column",
      alignItems: "flex-start",
      "& $hourLabel": {
        display: "none"
      },
      "&:first-child $hourLabel": {
        display: "block"
      }
    },
    dateLabel: {
      position: "sticky",
      zIndex: 2,
      top: 0,
      padding: theme.spacing(1, 0)
    },
    hour: {
      position: "relative",
      width: "100%",
      height: theme.spacing(7),
      borderTop: "1px solid",
      bordercolor: theme.palette.divider
    },
    hourLabel: {
      position: "absolute",
      width: theme.spacing(6),
      padding: theme.spacing(1, 0),
      display: "block",
      marginTop: "-1px",
      marginLeft: theme.spacing(-7),
      borderTop: "1px solid"
    }
  };
});

var TimetableDate = function TimetableDate(_ref) {
  var datetime = _ref.datetime,
      children = _ref.children;
  var date = moment(datetime).format('LL');
  var classes = useStyles();
  return /*#__PURE__*/React.createElement("div", {
    className: classes.date
  }, /*#__PURE__*/React.createElement("date", {
    className: classes.dateLabel
  }, date), children);
};

var TimetableHour = function TimetableHour(_ref2) {
  var datetime = _ref2.datetime,
      _ref2$events = _ref2.events,
      events = _ref2$events === void 0 ? [] : _ref2$events;
  var hour = moment(datetime).format('HH:mm');
  var classes = useStyles();
  return /*#__PURE__*/React.createElement("div", {
    className: classes.hour
  }, /*#__PURE__*/React.createElement("time", {
    className: classes.hourLabel
  }, hour), /*#__PURE__*/React.createElement(TimetableEvents, {
    events: events
  }));
};

var TimetableEvents = function TimetableEvents(_ref3) {
  var _ref3$events = _ref3.events,
      events = _ref3$events === void 0 ? [] : _ref3$events;
  return /*#__PURE__*/React.createElement("div", null, events && events.map(function (event) {
    return /*#__PURE__*/React.createElement("div", null, event && event.title);
  }));
};

var Timetable = function Timetable(_ref4) {
  var date = _ref4.date,
      _ref4$items = _ref4.items,
      items = _ref4$items === void 0 ? [] : _ref4$items,
      days = _ref4.days,
      timeFrom = _ref4.timeFrom,
      timeTo = _ref4.timeTo;
  var timetable = getTimetable({
    date: date,
    items: items,
    days: days,
    timeFrom: timeFrom,
    timeTo: timeTo
  });
  var cols = timetable.cols;
  var classes = useStyles();
  return /*#__PURE__*/React.createElement("div", {
    className: classes.timetable
  }, cols.map(function (rows, index) {
    return /*#__PURE__*/React.createElement(TimetableDate, rows[0], rows.map(function (row) {
      return /*#__PURE__*/React.createElement(TimetableHour, row);
    }));
  }));
};

Timetable.defaultProps = {
  days: 4,
  timeFrom: 0,
  timeTo: 23
};
Timetable.propTypes = process.env.NODE_ENV !== "production" ? {
  date: PropTypes.string,
  items: PropTypes.array,
  days: PropTypes.number,
  timeFrom: PropTypes.number,
  timeTo: PropTypes.number
} : {};
export default Timetable;
