// @kit-ui/admin/es/Calendar/TimetableModule.js
function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _inheritsLoose(subClass, superClass) { subClass.prototype = Object.create(superClass.prototype); subClass.prototype.constructor = subClass; subClass.__proto__ = superClass; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

import React from 'react';
import moment from 'moment';
import { ModuleTitle, ModuleDate } from "../Module"; // import './ModuleTimetable.scss';

var ModuleTimetable = /*#__PURE__*/function (_React$Component) {
  _inheritsLoose(ModuleTimetable, _React$Component);

  function ModuleTimetable() {
    var _this;

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = _React$Component.call.apply(_React$Component, [this].concat(args)) || this;

    _defineProperty(_assertThisInitialized(_this), "onToggle", function (item) {
      _this.props.action.onToggle(item);
    });

    _defineProperty(_assertThisInitialized(_this), "getStyle", function (start, duration) {
      var minutesHeight = 100 / 60;
      var top = Math.floor(start * minutesHeight);
      var height = Math.floor(duration * minutesHeight);
      var style = {
        top: top + '%',
        height: height + '%'
      };
      return style;
    });

    return _this;
  }

  var _proto = ModuleTimetable.prototype;

  _proto.render = function render() {
    var _this2 = this;

    var item = this.props.item;
    var title = item.title,
        _dtStart = item._dtStart,
        _dtEnd = item._dtEnd;
    var start = moment(_dtStart).format('mm');
    var duration = moment(_dtEnd).diff(_dtStart, 'minutes');
    var style = this.getStyle(start, duration);
    return /*#__PURE__*/React.createElement("div", {
      className: "module  module--timetable",
      style: style,
      onClick: function onClick() {
        return _this2.onToggle(item);
      }
    }, /*#__PURE__*/React.createElement(ModuleDate, {
      dt: _dtStart,
      format: "HH:mm"
    }), /*#__PURE__*/React.createElement(ModuleTitle, {
      title: title
    }));
  };

  return ModuleTimetable;
}(React.Component);

export default ModuleTimetable;
