// @kit-ui/schema/lib/CustomFields/MeasureField/MeasureField.stories.js
"use strict";

exports.__esModule = true;
exports.Default = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _SchemaEditor = _interopRequireDefault(require("../../SchemaEditor"));

var _MeasureField = _interopRequireDefault(require("./MeasureField"));

var _MeasureField2 = _interopRequireDefault(require("./MeasureField.model"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var _default = {
  title: 'Fields/CustomFields/MeasureField',
  component: _SchemaEditor.default,
  args: _extends({}, _MeasureField2.default, {
    "uiSchema": _extends({}, _MeasureField2.default.uiSchema, {
      "ui:field": _MeasureField.default
    })
  }),
  argTypes: {}
};
exports.default = _default;

var Template = function Template(args) {
  return /*#__PURE__*/_react.default.createElement(_SchemaEditor.default, args);
};

var Default = Template.bind({});
exports.Default = Default;
Default.args = {
  formData: {
    text: "Forside:  12x12x12cm, Bakside: 12x12x12cm, Topp: 12x12x12cm"
  }
};
