// @kit-ui/core/lib/Button/Button.template.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _Button = _interopRequireDefault(require("./Button"));

var _styles = require("@material-ui/core/styles");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var useStyles = (0, _styles.makeStyles)(function (theme) {
  return {
    root: {
      '& > *': {
        margin: theme.spacing(1)
      }
    }
  };
});

var Template = function Template(args) {
  var classes = useStyles();
  return /*#__PURE__*/_react.default.createElement("div", {
    className: classes.root
  }, /*#__PURE__*/_react.default.createElement(_Button.default, args, "Default"), /*#__PURE__*/_react.default.createElement(_Button.default, _extends({}, args, {
    color: "primary"
  }), "Primary"), /*#__PURE__*/_react.default.createElement(_Button.default, _extends({}, args, {
    color: "secondary"
  }), "Secondary"), /*#__PURE__*/_react.default.createElement(_Button.default, _extends({}, args, {
    disabled: true
  }), "Disabled"), /*#__PURE__*/_react.default.createElement(_Button.default, _extends({}, args, {
    color: "primary",
    href: "#contained-buttons"
  }), "Link"));
};

var _default = Template;
exports.default = _default;
module.exports = exports.default;
