// @kit-ui/admin/lib/Finder/FinderSidebar.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _styles = require("@material-ui/core/styles");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var useStyles = (0, _styles.makeStyles)(function (theme) {
  var _ariaExpandedTrue, _ariaExpandedTrue2;

  return {
    sidebar: {
      position: "absolute",
      zIndex: 1,
      top: 0,
      right: "auto",
      left: 0,
      bottom: 0,
      width: function width(props) {
        return props.width;
      },
      marginLeft: function marginLeft(props) {
        return "-" + props.width + "px";
      },
      transition: ".125s ease-out",
      "& + *": {
        transition: ".125s ease-out"
      },
      "&[aria-expanded=true]": (_ariaExpandedTrue = {}, _ariaExpandedTrue[theme.breakpoints.up('sm')] = {
        marginLeft: 0
      }, _ariaExpandedTrue),
      "&[aria-expanded=true] + *": (_ariaExpandedTrue2 = {}, _ariaExpandedTrue2[theme.breakpoints.up('sm')] = {
        marginLeft: function marginLeft(props) {
          return props.width;
        }
      }, _ariaExpandedTrue2)
    }
  };
});

var FinderSidebar = function FinderSidebar(_ref) {
  var className = _ref.className,
      children = _ref.children,
      _ref$expanded = _ref.expanded,
      expanded = _ref$expanded === void 0 ? false : _ref$expanded,
      _ref$width = _ref.width,
      width = _ref$width === void 0 ? 240 : _ref$width;
  var classes = useStyles({
    width: width
  });
  return /*#__PURE__*/_react.default.createElement("aside", {
    className: className || classes.sidebar,
    "aria-expanded": expanded
  }, children);
};

var _default = FinderSidebar;
exports.default = _default;
module.exports = exports.default;
