// @kit-ui/admin/lib/Finder/FinderLayout.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _FinderBase = _interopRequireDefault(require("./FinderBase"));

var _FinderHeader = _interopRequireDefault(require("./FinderHeader"));

var _FinderToolbar = _interopRequireDefault(require("./FinderToolbar"));

var _FinderBody = _interopRequireDefault(require("./FinderBody"));

var _NavPath = _interopRequireDefault(require("../NavPath/NavPath"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/** FinderLayout represents a view within an app. Normally this would be wrapped in AppLayout */
var FinderLayout = function FinderLayout(_ref) {
  var backgroundColor = _ref.backgroundColor,
      expanded = _ref.expanded,
      title = _ref.title,
      parents = _ref.parents,
      filters = _ref.filters,
      action = _ref.action,
      views = _ref.views,
      search = _ref.search,
      children = _ref.children,
      onSelect = _ref.onSelect;
  return /*#__PURE__*/_react.default.createElement(_FinderBase.default, {
    backgroundColor: backgroundColor,
    expanded: expanded
  }, /*#__PURE__*/_react.default.createElement(_FinderHeader.default, {
    backgroundColor: backgroundColor
  }, /*#__PURE__*/_react.default.createElement(_NavPath.default, {
    parents: parents,
    title: title,
    filters: filters,
    search: search,
    onSelect: onSelect
  }), action && /*#__PURE__*/_react.default.createElement(_FinderToolbar.default, {
    buttons: action
  })), /*#__PURE__*/_react.default.createElement(_FinderBody.default, null, children));
};

FinderLayout.propTypes = process.env.NODE_ENV !== "production" ? {
  backgroundColor: _propTypes.default.string,
  expanded: _propTypes.default.bool,
  title: _propTypes.default.string,

  /** Array of parents */
  parents: _propTypes.default.array,
  filters: _propTypes.default.array,
  action: _propTypes.default.array,
  views: _propTypes.default.array,
  search: _propTypes.default.object,

  /** Finder content */
  children: _propTypes.default.node
} : {};
var _default = FinderLayout;
exports.default = _default;
module.exports = exports.default;
