// @kit-ui/admin/lib/Finder/FinderHeader.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _color = _interopRequireDefault(require("color"));

var _styles = require("@material-ui/core/styles");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var getBackgroundImage = function getBackgroundImage(backgroundColor) {
  var color = (0, _color.default)(backgroundColor);
  var c1 = color.alpha(0).rgb();
  var c2 = color.alpha(.75).rgb();
  var c3 = color.alpha(1).rgb();
  return "linear-gradient(to bottom, " + c1 + ") 0%, " + c2 + " 75%, " + c3 + " 100%)";
};

var useStyles = (0, _styles.makeStyles)(function (theme) {
  return {
    root: {
      position: "absolute",
      zIndex: 2,
      top: 0,
      right: 0,
      bottom: "auto",
      left: 0,
      backgroundColor: function backgroundColor(props) {
        return props.backgroundColor || theme.palette.background.default;
      },
      backgroundImage: function backgroundImage(props) {
        return getBackgroundImage(props.backgroundColor || theme.palette.background.default);
      },
      height: theme.spacing(8),
      paddingLeft: theme.spacing(2),
      paddingRight: theme.spacing(2),
      display: "flex",
      alignItems: "center",
      justifyContent: "space-between",
      "& + *": {
        marginTop: theme.spacing(6)
      }
    }
  };
});

var FinderHeader = function FinderHeader(_ref) {
  var backgroundColor = _ref.backgroundColor,
      className = _ref.className,
      children = _ref.children;
  var classes = useStyles({
    backgroundColor: backgroundColor
  });
  return /*#__PURE__*/_react.default.createElement("header", {
    className: className || classes.root
  }, children);
};

var _default = FinderHeader;
exports.default = _default;
module.exports = exports.default;
