// @kit-ui/admin/lib/Finder/README.md.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _components = require("@frontend-components/styleguide/lib/components/");

var _utils = require("./../../.styleguide/utils.js");

var _tag = require("@mdx-js/tag");

var _ = require("../");

var _SchemaWithState = _interopRequireDefault(require("../Schema/SchemaWithState"));

var _data = require("../../resources/data");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _inheritsLoose(subClass, superClass) { subClass.prototype = Object.create(superClass.prototype); subClass.prototype.constructor = subClass; subClass.__proto__ = superClass; }

var layoutProps = {};

var MDXContent = /*#__PURE__*/function (_React$Component) {
  _inheritsLoose(MDXContent, _React$Component);

  function MDXContent(props) {
    var _this;

    _this = _React$Component.call(this, props) || this;
    _this.layout = null;
    return _this;
  }

  var _proto = MDXContent.prototype;

  _proto.render = function render() {
    var _this$props = this.props,
        components = _this$props.components,
        props = _objectWithoutPropertiesLoose(_this$props, ["components"]);

    return /*#__PURE__*/_react.default.createElement(_tag.MDXTag, {
      name: "wrapper",
      components: components
    }, /*#__PURE__*/_react.default.createElement(_tag.MDXTag, {
      name: "h1",
      components: components
    }, "Finder"), /*#__PURE__*/_react.default.createElement(_tag.MDXTag, {
      name: "h2",
      components: components
    }, "FinderLayout + FinderModule"), /*#__PURE__*/_react.default.createElement(_tag.MDXTag, {
      name: "p",
      components: components
    }, "FinderLayout with parents and FinderModule as children:"), /*#__PURE__*/_react.default.createElement(_components.Playground, null, /*#__PURE__*/_react.default.createElement(_.FinderLayout, {
      parents: [{
        "title": "App"
      }, {
        "title": "Parent"
      }, {
        "title": "Page"
      }]
    }, /*#__PURE__*/_react.default.createElement(_.FinderModule, {
      title: "Document 1",
      description: "description",
      label: "Document"
    }), /*#__PURE__*/_react.default.createElement(_.FinderModule, {
      title: "Document 2",
      description: "description",
      label: "Document"
    }), /*#__PURE__*/_react.default.createElement(_.FinderModule, {
      title: "Document 3",
      description: "description",
      label: "Document"
    }))), /*#__PURE__*/_react.default.createElement(_tag.MDXTag, {
      name: "h2",
      components: components
    }, "FinderLayout with grid + FinderModule"), /*#__PURE__*/_react.default.createElement(_tag.MDXTag, {
      name: "p",
      components: components
    }, "FinderLayout with parents and FinderModule as children:"), /*#__PURE__*/_react.default.createElement(_components.Playground, null, /*#__PURE__*/_react.default.createElement(_.FinderLayout, {
      layout: "grid",
      parents: [{
        "title": "App"
      }, {
        "title": "Parent"
      }, {
        "title": "Page"
      }]
    }, /*#__PURE__*/_react.default.createElement(_.FinderModule, {
      title: "Document 1",
      description: "description",
      label: "Document",
      moduleLayout: "grid"
    }), /*#__PURE__*/_react.default.createElement(_.FinderModule, {
      title: "Document 2",
      description: "description",
      label: "Document",
      moduleLayout: "grid"
    }), /*#__PURE__*/_react.default.createElement(_.FinderModule, {
      title: "Document 3",
      description: "description",
      label: "Document",
      moduleLayout: "grid"
    }), /*#__PURE__*/_react.default.createElement(_.FinderModule, {
      title: "Document 1",
      description: "description",
      label: "Document",
      moduleLayout: "grid"
    }), /*#__PURE__*/_react.default.createElement(_.FinderModule, {
      title: "Document 2",
      description: "description",
      label: "Document",
      moduleLayout: "grid"
    }), /*#__PURE__*/_react.default.createElement(_.FinderModule, {
      title: "Document 3",
      description: "description",
      label: "Document",
      moduleLayout: "grid"
    }), /*#__PURE__*/_react.default.createElement(_.FinderModule, {
      title: "Document 1",
      description: "description",
      label: "Document",
      moduleLayout: "grid"
    }), /*#__PURE__*/_react.default.createElement(_.FinderModule, {
      title: "Document 2",
      description: "description",
      label: "Document",
      moduleLayout: "grid"
    }), /*#__PURE__*/_react.default.createElement(_.FinderModule, {
      title: "Document 3",
      description: "description",
      label: "Document",
      moduleLayout: "grid"
    }))), /*#__PURE__*/_react.default.createElement(_tag.MDXTag, {
      name: "h2",
      components: components
    }, "Finder app"), /*#__PURE__*/_react.default.createElement(_tag.MDXTag, {
      name: "p",
      components: components
    }, "FinderLayout within LayoutAdmin, makes up a finder view:"), /*#__PURE__*/_react.default.createElement(_components.Playground, null, /*#__PURE__*/_react.default.createElement(_.LayoutAdmin, {
      app: _data.appInbox
    }, /*#__PURE__*/_react.default.createElement(_.FinderLayout, _data.appInbox, /*#__PURE__*/_react.default.createElement(_.FinderModule, {
      title: "Document 1",
      description: "description",
      label: "Document"
    }), /*#__PURE__*/_react.default.createElement(_.FinderModule, {
      title: "Document 2",
      description: "description",
      label: "Document"
    }), /*#__PURE__*/_react.default.createElement(_.FinderModule, {
      title: "Document 3",
      description: "description",
      label: "Document"
    })))), /*#__PURE__*/_react.default.createElement(_tag.MDXTag, {
      name: "h2",
      components: components
    }, "Schema example"), /*#__PURE__*/_react.default.createElement(_tag.MDXTag, {
      name: "p",
      components: components
    }, "Finder layout, schema example."), /*#__PURE__*/_react.default.createElement(_components.Playground, null, /*#__PURE__*/_react.default.createElement(_SchemaWithState.default, {
      schema: {
        "type": "object",
        "properties": {
          "default": {
            "type": "object",
            "properties": {
              "text": {
                "type": "string"
              },
              "switch": {
                "type": "boolean"
              }
            }
          }
        }
      },
      uiSchema: {
        "default": {
          "ui:layout": "finder"
        }
      }
    })), /*#__PURE__*/_react.default.createElement(_tag.MDXTag, {
      name: "h2",
      components: components
    }, "Static examples"), /*#__PURE__*/_react.default.createElement(_tag.MDXTag, {
      name: "p",
      components: components
    }, "Finder header + body."), /*#__PURE__*/_react.default.createElement(_components.Playground, null, /*#__PURE__*/_react.default.createElement(_.Finder, null, /*#__PURE__*/_react.default.createElement(_.FinderHeader, null, ".finder__header"), /*#__PURE__*/_react.default.createElement(_.FinderBody, null, ".finder__body"))), /*#__PURE__*/_react.default.createElement(_tag.MDXTag, {
      name: "p",
      components: components
    }, "Finder header with title and action."), /*#__PURE__*/_react.default.createElement(_components.Playground, null, /*#__PURE__*/_react.default.createElement(_.Finder, null, /*#__PURE__*/_react.default.createElement(_.FinderHeader, null, /*#__PURE__*/_react.default.createElement(_.FinderTitle, {
      title: "Title"
    }), /*#__PURE__*/_react.default.createElement(_.NavAction, {
      button: {
        title: "Submit"
      }
    })), /*#__PURE__*/_react.default.createElement(_.FinderBody, null, ".finder__body"))), /*#__PURE__*/_react.default.createElement(_tag.MDXTag, {
      name: "p",
      components: components
    }, "Finder with path."), /*#__PURE__*/_react.default.createElement(_components.Playground, null, /*#__PURE__*/_react.default.createElement(_.Finder, null, /*#__PURE__*/_react.default.createElement(_.FinderHeader, null, /*#__PURE__*/_react.default.createElement(_.NavPath, {
      parents: [{
        title: "Grandparent"
      }, {
        title: "Parent"
      }]
    })), /*#__PURE__*/_react.default.createElement(_.FinderBody, null, ".finder__body"))), /*#__PURE__*/_react.default.createElement(_tag.MDXTag, {
      name: "p",
      components: components
    }, "LayoutAdmin + FinderLayout."), /*#__PURE__*/_react.default.createElement(_components.Playground, null, /*#__PURE__*/_react.default.createElement(_.Admin, null, /*#__PURE__*/_react.default.createElement(_.AdminHeader, null, ".admin__header"), /*#__PURE__*/_react.default.createElement(_.AdminBody, null, /*#__PURE__*/_react.default.createElement(_.AdminSidebar, null, ".admin__sidebar"), /*#__PURE__*/_react.default.createElement(_.AdminContent, null, /*#__PURE__*/_react.default.createElement(_.Finder, null, /*#__PURE__*/_react.default.createElement(_.FinderHeader, null, ".finder__header"), /*#__PURE__*/_react.default.createElement(_.FinderBody, null, ".finder__body")))))), /*#__PURE__*/_react.default.createElement(_tag.MDXTag, {
      name: "p",
      components: components
    }, "LayoutAdmin + Finder content + action."), /*#__PURE__*/_react.default.createElement(_components.Playground, null, /*#__PURE__*/_react.default.createElement(_.Admin, null, /*#__PURE__*/_react.default.createElement(_.AdminHeader, null, ".admin__header"), /*#__PURE__*/_react.default.createElement(_.AdminBody, null, /*#__PURE__*/_react.default.createElement(_.AdminSidebar, {
      expanded: false
    }, ".admin__sidebar"), /*#__PURE__*/_react.default.createElement(_.AdminContent, {
      expanded: false
    }, /*#__PURE__*/_react.default.createElement(_.FinderLayout, {
      title: "Content"
    }, ".admin__content")), /*#__PURE__*/_react.default.createElement(_.AdminAction, {
      expanded: true
    }, /*#__PURE__*/_react.default.createElement(_.FinderLayout, {
      title: "Action"
    }, ".admin__action"))))), /*#__PURE__*/_react.default.createElement(_tag.MDXTag, {
      name: "p",
      components: components
    }, "LayoutAdmin + Finder + Upload + Action."), /*#__PURE__*/_react.default.createElement(_components.Playground, null, /*#__PURE__*/_react.default.createElement(_.Admin, null, /*#__PURE__*/_react.default.createElement(_.AdminHeader, null, ".admin__header"), /*#__PURE__*/_react.default.createElement(_.AdminBody, null, /*#__PURE__*/_react.default.createElement(_.AdminSidebar, {
      expanded: false
    }, ".admin__sidebar"), /*#__PURE__*/_react.default.createElement(_.AdminContent, {
      expanded: false
    }, /*#__PURE__*/_react.default.createElement(_.LayoutUpload, null)), /*#__PURE__*/_react.default.createElement(_.AdminAction, {
      expanded: true
    }, ".admin__action")))));
  };

  return MDXContent;
}(_react.default.Component);

exports.default = MDXContent;
module.exports = exports.default;
