// @kit-ui/schema/es/CustomFields/RRulesField/EventField.stories.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import React from 'react';
import Schema from "../../SchemaEditor";
import model from "./EventField.model";
import EventField from "./EventField";
export default {
  title: 'Fields/CustomFields/EventField',
  component: Schema,
  args: _extends({}, model, {
    uiSchema: {
      "ui:field": EventField
    }
  }),
  argTypes: {}
};

var Template = function Template(args) {
  return /*#__PURE__*/React.createElement(Schema, args);
};

export var Default = Template.bind({});
Default.args = {};
export var EveryWednesday = Template.bind({});
EveryWednesday.args = {
  formData: {
    rrule: "FREQ=WEEKLY;BYDAY=WE"
  }
};
export var EveryFebruary6 = Template.bind({});
EveryFebruary6.args = {
  formData: {
    date: "1958-02-06",
    title: "Munich anniversary",
    allDay: true,
    rrule: "FREQ=YEARLY;BYMONTH=2;BYMONTHDAY=6"
  }
};
export var CustomRRule = Template.bind({});
CustomRRule.args = {
  formData: {
    date: "1958-02-06",
    title: "Munich anniversary",
    allDay: true,
    rrule: "custom",
    rrules: {
      "FREQ": "WEEKLY"
    }
  }
};
