// @kit-ui/schema/es/CustomFields/RRulesField/EventField.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import React, { useState, useEffect } from "react";
import moment from "moment";
import { getRRuleFromFormData, getFormDataFromRRule, getRRuleText } from "./utils";
import model from "./EventField.model";
import { utils } from "@rjsf/core";
import { is } from "core-js/fn/object";
var getUiOptions = utils.getUiOptions,
    getDefaultFormState = utils.getDefaultFormState;
var byWeekday = ["MO", "TU", "WE", "TH", "FR", "SA", "SU"];

var EventField = function EventField(props) {
  var formData = props.formData;

  var schema = _extends({}, model.schema, props.schema);

  var uiSchema = _extends({}, model.uiSchema, props.uiSchema); // set initial dtStart + rrule + rrules


  var _useState = useState(formData.rrule),
      rrule = _useState[0],
      setRRule = _useState[1];

  var _useState2 = useState(rrule),
      rruleText = _useState2[0],
      setRRuleText = _useState2[1];

  var _useState3 = useState(formData.rrules),
      rrules = _useState3[0],
      setRRules = _useState3[1]; // onChange


  var _onChange = function _onChange(formData) {
    var _formData = formData,
        allDay = _formData.allDay,
        date = _formData.date,
        _formData$startTime = _formData.startTime,
        startTime = _formData$startTime === void 0 ? "00:00" : _formData$startTime,
        _formData$endTime = _formData.endTime,
        endTime = _formData$endTime === void 0 ? "00:00" : _formData$endTime,
        dtStart = _formData.dtStart,
        dtEnd = _formData.dtEnd;

    if (date) {
      dtStart = date + " " + startTime;
      dtEnd = date + " " + endTime;
    }

    var formatDate = function formatDate(date) {
      return date && allDay && moment(date).format('YYYY-MM-DD') || date && moment(date).format('YYYY-MM-DDTHH:mm:ss') || undefined;
    };

    if (!dtEnd && dtStart || moment(dtEnd).isBefore(moment(dtStart))) {
      dtEnd = dtStart;
    }

    var time = allDay && "Hele dagen" || startTime !== endTime && startTime + "–" + endTime || startTime;
    formData = _extends({}, formData, {
      time: time,
      dtStart: formatDate(dtStart),
      dtEnd: formatDate(dtEnd)
    });
    props.onChange(formData);
  }; // rrules from rrule


  useEffect(function () {
    var rrules;

    if (formData.rrule && formData.rrule !== "custom") {
      rrules = getFormDataFromRRule(formData.rrule);
    } else if (!formData.rrule) {
      rrules = getDefaultFormState({
        schema: schema.properties.rrules
      });
    }

    rrules && setRRules(rrules);
    setRRule(formData.rrule);
  }, [formData.rrule]); // rrule from rrules

  useEffect(function () {
    var rrule;

    if (formData.rrule === "custom") {
      rrule = getRRuleFromFormData(formData.rrules);
    }

    rrule && setRRule(rrule);
    rrule && setRRuleText(getRRuleText(rrule));
    setRRules(formData.rrules);
  }, [formData.rrules]); // get dtStart rules

  var _useState4 = useState([]),
      dtEnum = _useState4[0],
      setDtEnum = _useState4[1];

  var _useState5 = useState([]),
      dtEnumNames = _useState5[0],
      setDtEnumNames = _useState5[1];

  useEffect(function () {
    var isoWeekday = moment(formData.dtStart).isoWeekday();
    var isoMonth = moment(formData.dtStart).month() + 1;
    var dayOfMonth = moment(formData.dtStart).format('D');
    var weekly = "FREQ=WEEKLY;BYDAY=" + byWeekday[isoWeekday - 1];
    var monthly = "FREQ=MONTHLY;BYMONTHDAY=" + dayOfMonth;
    var yearly = "FREQ=YEARLY;BYMONTH=" + isoMonth + ";BYMONTHDAY=" + dayOfMonth;
    setDtEnum([weekly, monthly, yearly]);
  }, [formData.dtStart]); // get rrules enum

  var getRRuleSchema = function getRRuleSchema() {
    var rruleSchema = schema.properties.rrule;
    var defaultRRuleEnum = rruleSchema && rruleSchema.enum;
    var defaultRRuleEnumNames = rruleSchema && rruleSchema.enumNames; //        const customRule = defaultRRuleEnum.includes("custom")
    //        const customRuleIndex = customRule && defaultRRuleEnum.indexOf("custom")
    //        const customRulesName = customRuleIndex && defaultRRuleEnumNames[customRuleIndex] || customRule

    var rruleEnum = defaultRRuleEnum.concat(dtEnum);

    if (rrule && rrule !== "custom" && !rruleEnum.includes(rrule)) {
      rruleEnum.push(rrule);
    }

    var rruleEnumNames = rruleEnum.map(function (rrule, index) {
      if (defaultRRuleEnumNames && defaultRRuleEnumNames[index]) {
        return defaultRRuleEnumNames[index];
      }

      return getRRuleText(rrule);
    });
    return _extends({}, rruleSchema, {
      enum: rruleEnum,
      enumNames: rruleEnumNames
    });
  }; // schema, formData, etc


  var newSchema = _extends({}, schema, {
    properties: _extends({}, schema.properties, {
      rrule: getRRuleSchema()
    })
  });

  var newFormData = _extends({}, formData, {
    //        rrule: rrule,
    rrules: rrules
  }); // formData


  var ObjectField = props.registry.fields.ObjectField;
  return /*#__PURE__*/React.createElement("div", null, /*#__PURE__*/React.createElement(ObjectField, _extends({}, props, {
    schema: newSchema,
    uiSchema: uiSchema,
    formData: newFormData,
    onChange: _onChange
  })), /*#__PURE__*/React.createElement("hr", null), JSON.stringify(dtEnum), /*#__PURE__*/React.createElement("hr", null), "RRULE: ", JSON.stringify(rrule), /*#__PURE__*/React.createElement("hr", null), "TEXT: ", rruleText, /*#__PURE__*/React.createElement("hr", null), "RRULES: ", JSON.stringify(rrules));
};

EventField.defaultProps = {
  formData: {}
};
export default EventField;
