// @kit-ui/schema/es/CustomFields/RRulesField/RRulesField.stories.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import React from 'react';
import Schema from "../../SchemaEditor";
import RRulesField from "./RRulesField";
import model from "./RRulesField.model";
export default {
  title: 'Fields/CustomFields/RRulesField',
  component: Schema,
  args: _extends({}, model),
  argTypes: {}
};

var Template = function Template(args) {
  return /*#__PURE__*/React.createElement(Schema, args);
};

export var Default = Template.bind({});
Default.args = {};
export var YearlyOnJanuary2nd = Template.bind({});
YearlyOnJanuary2nd.args = {
  schema: {},
  uiSchema: {},
  formData: {
    "FREQ": "YEARLY",
    "BYMONTH": ["1"],
    "BYMONTHDAY": ["2"]
  }
};
export var YearlyOnFirstMondayOfFebruary = Template.bind({});
YearlyOnFirstMondayOfFebruary.args = {
  schema: {},
  uiSchema: {},
  formData: {
    "FREQ": "YEARLY",
    "BYDAY": ["MO"],
    "BYSETPOS": [1],
    "BYMONTH": [2]
  }
};
export var MonthlyOnFirstSunday = Template.bind({});
MonthlyOnFirstSunday.args = {
  schema: {},
  uiSchema: {},
  formData: {
    "FREQ": "MONTHLY",
    "BYSETPOS": [1],
    "BYDAY": ["SU"],
    "INTERVAL": 2
  }
};
export var MonthlyOnDay5 = Template.bind({});
MonthlyOnDay5.args = {
  schema: {},
  uiSchema: {},
  formData: {
    "FREQ": "MONTHLY",
    "BYMONTHDAY": [5],
    "INTERVAL": 2
  }
};
export var WeeklyOnFridays = Template.bind({});
WeeklyOnFridays.args = {
  schema: {},
  uiSchema: {},
  formData: {
    "FREQ": "WEEKLY",
    "BYDAY": ["FR"],
    "INTERVAL": 2
  }
};
export var WeeklyOnWeekends = Template.bind({});
WeeklyOnWeekends.args = {
  schema: {},
  uiSchema: {},
  formData: {
    "FREQ": "WEEKLY",
    "BYDAY": ["SA", "SU"],
    "INTERVAL": 2
  }
};
