// @kit-ui/admin/es/PreviewJSON/PreviewJSON.stories.js
var _glossary;

import React from 'react';
import PreviewJSON from './PreviewJSON';
export default {
  title: 'Preview/PreviewJSON',
  component: PreviewJSON,
  args: {
    data: {
      "glossary": (_glossary = {
        "title": "example glossary",
        "string": "A string",
        "boolean": true
      }, _glossary["boolean"] = false, _glossary["number"] = 5, _glossary["GlossDiv"] = {
        "title": "S",
        "GlossList": {
          "GlossEntry": {
            "ID": "SGML",
            "SortAs": "SGML",
            "GlossTerm": "Standard Generalized Markup Language",
            "Acronym": "SGML",
            "Abbrev": "ISO 8879:1986",
            "GlossDef": {
              "para": "A meta-markup language, used to create markup languages such as DocBook.",
              "GlossSeeAlso": ["GML", "XML"]
            },
            "GlossSee": "markup"
          }
        }
      }, _glossary)
    }
  },
  argTypes: {}
};

var Template = function Template(args) {
  return /*#__PURE__*/React.createElement(PreviewJSON, args);
};

export var Default = Template.bind({});
Default.args = {};
export var DarkTheme = Template.bind({});
DarkTheme.args = {
  theme: "dark"
};
