// @kit-ui/admin/lib/NavUser/UserInfo.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Typography = _interopRequireDefault(require("@material-ui/core/Typography"));

var _icons = require("@kit-ui/icons");

var _styles = require("@material-ui/core/styles");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

var useStyles = (0, _styles.makeStyles)(function (theme) {
  return {
    root: {
      position: "relative",
      display: "flex",
      alignItems: "center",
      justifyContent: "flex-start",
      width: "320px",
      padding: "6px"
    },
    figure: {
      position: "relative",
      display: "block",
      width: "72px",
      height: "72px",
      margin: "6px"
    },
    header: {
      margin: "6px"
    },
    icon: {
      width: "72px",
      height: "72px"
    },
    name: {
      fontFamily: "Akkurat, sans-serif",
      fontWeight: "bold",
      fontSize: "16px"
    },
    email: {
      fontFamily: "Akkurat, sans-serif",
      fontWeight: "normal",
      fontSize: "14px"
    }
  };
});

var UserInfo = function UserInfo(_ref) {
  var imageUrl = _ref.imageUrl,
      name = _ref.name,
      email = _ref.email,
      initials = _ref.initials;
  var classes = useStyles();
  return /*#__PURE__*/_react.default.createElement("div", {
    className: classes.root
  }, /*#__PURE__*/_react.default.createElement("figure", {
    className: classes.figure
  }, /*#__PURE__*/_react.default.createElement(_icons.UserIcon, {
    className: classes.icon,
    name: name,
    initials: initials,
    imageUrl: imageUrl
  })), /*#__PURE__*/_react.default.createElement("header", {
    className: classes.header
  }, /*#__PURE__*/_react.default.createElement(_Typography.default, {
    className: classes.name,
    nowrap: true
  }, name), /*#__PURE__*/_react.default.createElement(_Typography.default, {
    className: classes.email,
    nowrap: true
  }, email)));
};

UserInfo.propTypes = process.env.NODE_ENV !== "production" ? {
  imageUrl: _propTypes.default.string,
  name: _propTypes.default.string,
  email: _propTypes.default.string
} : {};
var _default = UserInfo;
exports.default = _default;
module.exports = exports.default;
