// @kit-ui/schema/es/SchemaFieldset/SchemaFieldset.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import React from "react";
import { getDefaultRegistry, getUiFieldset } from "../utils";

var SchemaFieldset = function SchemaFieldset(props) {
  var schema = props.schema,
      uiSchema = props.uiSchema,
      formData = props.formData,
      idSchema = props.idSchema,
      name = props.name,
      required = props.required,
      disabled = props.disabled,
      readonly = props.readonly,
      onBlur = props.onBlur,
      onFocus = props.onFocus,
      _props$registry = props.registry,
      registry = _props$registry === void 0 ? getDefaultRegistry() : _props$registry;
  var uiFieldset = getUiFieldset(props);
  var rootSchema = registry.rootSchema,
      fields = registry.fields,
      formContext = registry.formContext;
  var SchemaField = fields.SchemaField;
  var ObjectFieldTemplate = registry.ObjectFieldTemplate;
  var templateProps = {
    name: name,
    properties: uiFieldset.map(function (field) {
      return _extends({}, field, {
        content: /*#__PURE__*/React.createElement(SchemaField, _extends({}, field, {
          onBlur: onBlur,
          onFocus: onFocus,
          registry: registry,
          disabled: disabled,
          readonly: readonly
        }))
      });
    }),
    readonly: readonly,
    disabled: disabled,
    required: required,
    idSchema: idSchema,
    uiSchema: uiSchema,
    schema: schema,
    formData: formData,
    formContext: formContext
  };
  return /*#__PURE__*/React.createElement(ObjectFieldTemplate, templateProps);
};

export default SchemaFieldset;
