// @kit-ui/admin/lib/utils/getTimetable.js
"use strict";

exports.__esModule = true;
exports.getTimetable = void 0;

var _lodash = _interopRequireDefault(require("lodash"));

var _moment = _interopRequireDefault(require("moment"));

var _2 = require("..");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var getTimetable = function getTimetable(_ref) {
  var date = _ref.date,
      days = _ref.days,
      _ref$items = _ref.items,
      items = _ref$items === void 0 ? [] : _ref$items,
      _ref$timeFrom = _ref.timeFrom,
      timeFrom = _ref$timeFrom === void 0 ? 0 : _ref$timeFrom,
      _ref$timeTo = _ref.timeTo,
      timeTo = _ref$timeTo === void 0 ? 23 : _ref$timeTo;

  if (!days) {
    days = 1;
  }

  var dt = (0, _moment.default)(date).format('YYYY-MM-DD');
  var dtStart = (0, _moment.default)(dt).format('YYYY-MM-DD');
  var dtEnd = (0, _moment.default)(dt).add(days, 'days').format('YYYY-MM-DD');
  var events = {};

  if (items.length) {
    events = (0, _2.getEvents)({
      items: items,
      dtStart: dtStart,
      dtEnd: dtEnd
    });
  }

  var cols = [];

  for (var d = 0; d < days; d++) {
    var datetime = (0, _moment.default)(dtStart).add(d, 'days');
    datetime = (0, _moment.default)(datetime).set('hour', 0);
    datetime = (0, _moment.default)(datetime).set('minute', 0);
    datetime = (0, _moment.default)(datetime).set('second', 0);
    var rows = [];

    for (var h = timeFrom; h <= timeTo; h++) {
      datetime = (0, _moment.default)(datetime).set('hour', h);
      datetime = (0, _moment.default)(datetime).set('minute', 0);
      datetime = (0, _moment.default)(datetime).set('second', 0);
      var hour = (0, _2.getEventsByHour)(datetime, events);
      rows.push(hour);
    }

    cols.push(rows);
  }

  var timetable = {
    dt: dt,
    dtStart: dtStart,
    dtEnd: dtEnd,
    cols: cols,
    events: events
  };
  return timetable;
};

exports.getTimetable = getTimetable;
