// @kit-ui/admin/lib/Media/ImageFocalpoint.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireWildcard(require("react"));

var _Typography = _interopRequireDefault(require("@material-ui/core/Typography"));

var _styles = require("@material-ui/core/styles");

var _ImageFocalpointBase = _interopRequireDefault(require("./ImageFocalpointBase"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var useStyles = (0, _styles.makeStyles)(function (theme) {
  return {
    root: {
      position: "absolute",
      top: 0,
      right: 0,
      bottom: 0,
      left: 0
    },
    preview: {
      position: "absolute",
      top: 0,
      right: 0,
      bottom: 0,
      left: 0,
      marginBottom: theme.spacing(6)
    },
    footer: {
      position: "absolute",
      top: "auto",
      right: 0,
      left: 0,
      bottom: 0,
      paddingTop: theme.spacing(1),
      height: theme.spacing(5)
    },
    description: {
      fontFamily: "Akkurat, sans-serif",
      fontSize: "12px",
      textAlign: "center"
    }
  };
});

var EditImageFocalpoint = function EditImageFocalpoint(_ref) {
  var className = _ref.className,
      imageUrl = _ref.imageUrl,
      _ref$imageFocalpoint = _ref.imageFocalpoint,
      imageFocalpoint = _ref$imageFocalpoint === void 0 ? {} : _ref$imageFocalpoint,
      imageCropdata = _ref.imageCropdata,
      imageFilters = _ref.imageFilters,
      onChange = _ref.onChange,
      props = _objectWithoutPropertiesLoose(_ref, ["className", "imageUrl", "imageFocalpoint", "imageCropdata", "imageFilters", "onChange"]);

  var classes = useStyles();

  var handleChange = function handleChange(imageFocalpoint) {
    onChange && onChange(imageFocalpoint);
  };

  return /*#__PURE__*/_react.default.createElement("div", {
    className: className || classes.root
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: classes.preview
  }, /*#__PURE__*/_react.default.createElement(_ImageFocalpointBase.default, {
    imageUrl: imageUrl,
    imageCropdata: imageCropdata,
    imageFilters: imageFilters,
    imageFocalpoint: imageFocalpoint,
    onChange: handleChange
  })), /*#__PURE__*/_react.default.createElement("footer", {
    className: classes.footer
  }, /*#__PURE__*/_react.default.createElement(_Typography.default, {
    nowrap: true,
    className: classes.description
  }, "Klikk p\xE5 bildet for \xE5 sette fokuspunkt")));
};

var _default = EditImageFocalpoint;
exports.default = _default;
module.exports = exports.default;
