// @kit-ui/admin/lib/Media/ImageAnnotations.example.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireWildcard(require("react"));

var _ImageAnnotations = _interopRequireDefault(require("./ImageAnnotations"));

var _ImagePreview = _interopRequireDefault(require("./ImagePreview"));

var _styles = require("@material-ui/core/styles");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

var useStyles = (0, _styles.makeStyles)(function (theme) {
  return {
    root: {
      position: "absolute",
      top: 0,
      right: 0,
      bottom: 0,
      left: 0
    },
    header: {
      position: "absolute",
      top: 0,
      right: 0,
      bottom: "50%",
      left: 0
    },
    footer: {
      position: "absolute",
      top: "50%",
      right: 0,
      bottom: 0,
      left: 0
    },
    image: {
      position: "absolute",
      top: 0,
      right: "50%",
      bottom: 0,
      left: 0
    },
    preview: {
      position: "absolute",
      top: 0,
      right: 0,
      bottom: 0,
      left: "50%"
    }
  };
});

var ExampleAnnotations = function ExampleAnnotations(_ref) {
  var _ref$imageUrl = _ref.imageUrl,
      imageUrl = _ref$imageUrl === void 0 ? undefined : _ref$imageUrl,
      _ref$imageAnnotations = _ref.imageAnnotations,
      imageAnnotations = _ref$imageAnnotations === void 0 ? [] : _ref$imageAnnotations;
  var classes = useStyles();

  var _useState = (0, _react.useState)(imageAnnotations),
      formData = _useState[0],
      setFormData = _useState[1];

  var _onChange = function onChange(formData) {
    setFormData(formData);
  };

  return /*#__PURE__*/_react.default.createElement("div", {
    className: classes.root
  }, /*#__PURE__*/_react.default.createElement("header", {
    className: classes.header
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: classes.image
  }, /*#__PURE__*/_react.default.createElement(_ImageAnnotations.default, {
    imageUrl: imageUrl,
    imageAnnotations: formData,
    onChange: function onChange(formData) {
      return _onChange(formData);
    }
  })), /*#__PURE__*/_react.default.createElement("div", {
    className: classes.preview
  }, /*#__PURE__*/_react.default.createElement(_ImagePreview.default, {
    imageUrl: imageUrl,
    imageAnnotations: formData
  }))), /*#__PURE__*/_react.default.createElement("footer", {
    className: classes.footer
  }, JSON.stringify(formData)));
};

var _default = ExampleAnnotations;
exports.default = _default;
module.exports = exports.default;
