// @kit-ui/admin/lib/Media/MediaPlayPauseButton.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _styles = require("@material-ui/core/styles");

var _IconButton = _interopRequireDefault(require("@material-ui/core/IconButton"));

var _PlayArrow = _interopRequireDefault(require("@material-ui/icons/PlayArrow"));

var _PlayCircleFilled = _interopRequireDefault(require("@material-ui/icons/PlayCircleFilled"));

var _Pause = _interopRequireDefault(require("@material-ui/icons/Pause"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var useStyles = (0, _styles.makeStyles)(function (theme) {
  return {
    root: {
      display: "flex",
      alignItems: "center",
      justifyContent: "center"
    }
  };
});

var MediaButtonPlayPause = function MediaButtonPlayPause(_ref) {
  var className = _ref.className,
      _ref$filled = _ref.filled,
      filled = _ref$filled === void 0 ? true : _ref$filled,
      _ref$color = _ref.color,
      color = _ref$color === void 0 ? "primary" : _ref$color,
      paused = _ref.paused,
      onClick = _ref.onClick;
  var classes = useStyles({
    filled: filled
  });

  if (paused) {
    return /*#__PURE__*/_react.default.createElement(_IconButton.default, {
      color: color,
      className: className || classes.root,
      onClick: onClick
    }, /*#__PURE__*/_react.default.createElement(_PlayArrow.default, null));
  }

  return /*#__PURE__*/_react.default.createElement(_IconButton.default, {
    color: color,
    className: className || classes.root,
    onClick: onClick
  }, /*#__PURE__*/_react.default.createElement(_Pause.default, null));
};

var _default = MediaButtonPlayPause;
exports.default = _default;
module.exports = exports.default;
