// @kit-ui/admin/lib/Media/MediaMetadata.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _styles = require("@material-ui/core/styles");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var useStyles = (0, _styles.makeStyles)(function (theme) {
  return {
    metadata: {
      listStyle: "none",
      display: "flex",
      flexDirection: "column",
      fontFamily: "Akkurat, sans-serif",
      fontSize: "16px",
      lineHeight: "24px",
      "& > li": {
        padding: theme.spacing(1, 0)
      },
      "& > * + *": {
        borderTop: "1px solid",
        borderColor: theme.palette.divider
      },
      "& b": {
        fontWeight: "bold"
      }
    }
  };
});

var MediaMetadata = function MediaMetadata(_ref) {
  var filename = _ref.filename,
      mimeType = _ref.mimeType,
      uniqueId = _ref.uniqueId,
      sourceId = _ref.sourceId,
      dimensions = _ref.dimensions,
      aspectRatio = _ref.aspectRatio,
      duration = _ref.duration,
      size = _ref.size,
      props = _objectWithoutPropertiesLoose(_ref, ["filename", "mimeType", "uniqueId", "sourceId", "dimensions", "aspectRatio", "duration", "size"]);

  var classes = useStyles();
  return /*#__PURE__*/_react.default.createElement("ul", {
    className: classes.metadata
  }, mimeType && /*#__PURE__*/_react.default.createElement("li", null, /*#__PURE__*/_react.default.createElement("b", null, "Format:"), " ", mimeType), uniqueId && /*#__PURE__*/_react.default.createElement("li", null, /*#__PURE__*/_react.default.createElement("b", null, "UniqueId:"), " ", uniqueId), sourceId && /*#__PURE__*/_react.default.createElement("li", null, /*#__PURE__*/_react.default.createElement("b", null, "SourceId:"), " ", sourceId), dimensions && /*#__PURE__*/_react.default.createElement("li", null, /*#__PURE__*/_react.default.createElement("b", null, "Dimensions:"), " ", dimensions), aspectRatio && /*#__PURE__*/_react.default.createElement("li", null, /*#__PURE__*/_react.default.createElement("b", null, "Format:"), " ", aspectRatio), duration && /*#__PURE__*/_react.default.createElement("li", null, /*#__PURE__*/_react.default.createElement("b", null, "Duration:"), " ", duration), size && /*#__PURE__*/_react.default.createElement("li", null, /*#__PURE__*/_react.default.createElement("b", null, "Size:"), " ", size));
};

var _default = MediaMetadata;
exports.default = _default;
module.exports = exports.default;
