// @kit-ui/admin/lib/Media/MediaVolumeSlider.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _styles = require("@material-ui/core/styles");

var _Slider = _interopRequireDefault(require("@material-ui/core/Slider"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var useStyles = (0, _styles.makeStyles)(function (theme) {
  return {
    root: {
      flexGrow: 1,
      margin: theme.spacing(2)
    }
  };
});

var MediaVolumeSlider = function MediaVolumeSlider(_ref) {
  var className = _ref.className,
      _ref$min = _ref.min,
      min = _ref$min === void 0 ? 0 : _ref$min,
      _ref$max = _ref.max,
      max = _ref$max === void 0 ? 1 : _ref$max,
      _ref$value = _ref.value,
      value = _ref$value === void 0 ? 0 : _ref$value,
      onChange = _ref.onChange;
  var classes = useStyles();

  var handleChange = function handleChange(event, value) {
    onChange && onChange(value);
  };

  return /*#__PURE__*/_react.default.createElement(_Slider.default, {
    className: className || classes.volume,
    step: .1,
    marks: true,
    min: min,
    max: max,
    value: value,
    onChange: handleChange
  });
};

var _default = MediaVolumeSlider;
exports.default = _default;
module.exports = exports.default;
