// @kit-ui/admin/lib/Media/SourcePreview.stories.js
"use strict";

exports.__esModule = true;
exports.Sketchfab = exports.Vimeo = exports.Youtube = exports.Default = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _SourcePreview = _interopRequireDefault(require("./SourcePreview"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _default = {
  title: 'Media/SourcePreview',
  component: _SourcePreview.default,
  args: {},
  argTypes: {}
};
exports.default = _default;

var Template = function Template(args) {
  return /*#__PURE__*/_react.default.createElement(_SourcePreview.default, args);
};

var Default = Template.bind({});
exports.Default = Default;
Default.args = {};
var Youtube = Template.bind({});
exports.Youtube = Youtube;
Youtube.args = {
  source: "youtube",
  sourceId: "RnTbc9wyl8s"
};
var Vimeo = Template.bind({});
exports.Vimeo = Vimeo;
Vimeo.args = {
  source: "vimeo",
  sourceId: "304002147"
};
var Sketchfab = Template.bind({});
exports.Sketchfab = Sketchfab;
Sketchfab.args = {
  source: "sketchfab",
  sourceId: "adb409f636e64140a2704aba5a8417cf"
};
