// @kit-ui/core/es/Dropdown/Dropdown.position.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import React, { useState } from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Grid from '@material-ui/core/Grid';
import Button from '@material-ui/core/Button';
import Dropdown from "./Dropdown";
var useStyles = makeStyles(function (theme) {
  return {
    root: {},
    content: {
      width: 200,
      height: 200
    }
  };
});

var DropdownPosition = function DropdownPosition(props) {
  var _useState = useState(null),
      anchorEl = _useState[0],
      setAnchorEl = _useState[1];

  var _useState2 = useState(false),
      expanded = _useState2[0],
      setExpanded = _useState2[1];

  var _useState3 = useState(),
      placement = _useState3[0],
      setPlacement = _useState3[1];

  var classes = useStyles();

  var handleClick = function handleClick(newPlacement) {
    return function (event) {
      setAnchorEl(event.currentTarget);
      setExpanded(function (prev) {
        return placement !== newPlacement || !prev;
      });
      setPlacement(newPlacement);
    };
  };

  return /*#__PURE__*/React.createElement("div", {
    className: classes.root
  }, /*#__PURE__*/React.createElement(Dropdown, _extends({}, props, {
    expanded: expanded,
    anchorEl: anchorEl,
    placement: placement
  }), /*#__PURE__*/React.createElement("div", {
    className: classes.content
  }, JSON.stringify(props, 0, 2))), /*#__PURE__*/React.createElement(Grid, {
    container: true,
    justify: "center"
  }, /*#__PURE__*/React.createElement(Grid, {
    item: true
  }, /*#__PURE__*/React.createElement(Button, {
    onClick: handleClick('top-start')
  }, "top-start"), /*#__PURE__*/React.createElement(Button, {
    onClick: handleClick('top')
  }, "top"), /*#__PURE__*/React.createElement(Button, {
    onClick: handleClick('top-end')
  }, "top-end"))), /*#__PURE__*/React.createElement(Grid, {
    container: true,
    justify: "center"
  }, /*#__PURE__*/React.createElement(Grid, {
    item: true,
    xs: 6
  }, /*#__PURE__*/React.createElement(Button, {
    onClick: handleClick('left-start')
  }, "left-start"), /*#__PURE__*/React.createElement("br", null), /*#__PURE__*/React.createElement(Button, {
    onClick: handleClick('left')
  }, "left"), /*#__PURE__*/React.createElement("br", null), /*#__PURE__*/React.createElement(Button, {
    onClick: handleClick('left-end')
  }, "left-end")), /*#__PURE__*/React.createElement(Grid, {
    item: true,
    container: true,
    xs: 6,
    alignItems: "flex-end",
    direction: "column"
  }, /*#__PURE__*/React.createElement(Grid, {
    item: true
  }, /*#__PURE__*/React.createElement(Button, {
    onClick: handleClick('right-start')
  }, "right-start")), /*#__PURE__*/React.createElement(Grid, {
    item: true
  }, /*#__PURE__*/React.createElement(Button, {
    onClick: handleClick('right')
  }, "right")), /*#__PURE__*/React.createElement(Grid, {
    item: true
  }, /*#__PURE__*/React.createElement(Button, {
    onClick: handleClick('right-end')
  }, "right-end")))), /*#__PURE__*/React.createElement(Grid, {
    container: true,
    justify: "center"
  }, /*#__PURE__*/React.createElement(Grid, {
    item: true
  }, /*#__PURE__*/React.createElement(Button, {
    onClick: handleClick('bottom-start')
  }, "bottom-start"), /*#__PURE__*/React.createElement(Button, {
    onClick: handleClick('bottom')
  }, "bottom"), /*#__PURE__*/React.createElement(Button, {
    onClick: handleClick('bottom-end')
  }, "bottom-end"))));
};

export default DropdownPosition;
