// @kit-ui/admin/es/Editor/EditorBase.js
import React from 'react';
import Paper from '@material-ui/core/Paper';
import { makeStyles } from '@material-ui/core/styles';
var useStyles = makeStyles(function (theme) {
  return {
    root: {
      backgroundColor: "inherit",
      position: "absolute",
      top: 0,
      right: 0,
      bottom: 0,
      left: 0
    }
  };
});

var EditorBase = function EditorBase(_ref) {
  var className = _ref.className,
      _ref$elevation = _ref.elevation,
      elevation = _ref$elevation === void 0 ? 1 : _ref$elevation,
      _ref$expanded = _ref.expanded,
      expanded = _ref$expanded === void 0 ? true : _ref$expanded,
      children = _ref.children;
  var classes = useStyles();

  if (!elevation) {
    return /*#__PURE__*/React.createElement("div", {
      className: className || classes.root,
      "aria-expanded": expanded
    }, children);
  }

  return /*#__PURE__*/React.createElement(Paper, {
    elevation: elevation,
    square: true,
    className: className || classes.root,
    "aria-expanded": expanded
  }, children);
};

export default EditorBase;
