// @kit-ui/admin/es/Editor/EditorPreview.stories.js
import React from 'react';
import EditorBase from './EditorBase';
import EditorHeader from './EditorHeader';
import EditorSection from './EditorSection';
import EditorSidebar from './EditorSidebar';
import EditorPreview from './EditorPreview';
import EditorContent from './EditorContent';
export default {
  title: 'Editor/EditorPreview',
  component: EditorPreview,
  args: {
    header: {
      title: "Preview"
    },
    sidebar: {
      expanded: false
    }
  },
  argTypes: {}
};

var Template = function Template(args) {
  return /*#__PURE__*/React.createElement(EditorBase, null, /*#__PURE__*/React.createElement(EditorHeader, args.header), /*#__PURE__*/React.createElement(EditorSection, null, /*#__PURE__*/React.createElement(EditorSidebar, args.sidebar, "Sidebar"), /*#__PURE__*/React.createElement(EditorSection, null, /*#__PURE__*/React.createElement(EditorPreview, null, "Preview"), /*#__PURE__*/React.createElement(EditorContent, null, "Content"))));
};

export var Default = Template.bind({});
Default.args = {};
export var ExpandedSidebar = Template.bind({});
ExpandedSidebar.args = {
  header: {
    title: "Sidebar expanded"
  },
  sidebar: {
    expanded: true
  }
};
