// @kit-ui/admin/es/Editor/EditorOverlay.js
import React from 'react';
import Paper from "@material-ui/core/Paper";
import { makeStyles } from '@material-ui/core/styles';
var useStyles = makeStyles(function (theme) {
  return {
    overlay: {
      position: function position(props) {
        return props.position;
      },
      zIndex: 2000,
      backgroundColor: "rgba(0,0,0,.25)",
      top: 0,
      right: 0,
      bottom: 0,
      left: 0,
      transition: ".25s ease-out",
      opacity: 0,
      pointerEvents: "none",
      "&[aria-expanded=true]": {
        pointerEvents: "all",
        opacity: 1
      }
    },
    content: {
      position: function position(props) {
        return props.position;
      },
      top: 0,
      right: 0,
      bottom: 0,
      left: 0,
      backgroundColor: theme.palette.background.paper,
      margin: theme.spacing(2)
    }
  };
});

var EditorOverlay = function EditorOverlay(_ref) {
  var className = _ref.className,
      _ref$position = _ref.position,
      position = _ref$position === void 0 ? "absolute" : _ref$position,
      _ref$elevation = _ref.elevation,
      elevation = _ref$elevation === void 0 ? 12 : _ref$elevation,
      expanded = _ref.expanded,
      children = _ref.children;
  var classes = useStyles({
    position: position
  });
  return /*#__PURE__*/React.createElement("aside", {
    className: classes.overlay,
    "aria-expanded": expanded
  }, /*#__PURE__*/React.createElement(Paper, {
    className: className || classes.content,
    elevation: elevation,
    square: true
  }, children));
};

EditorOverlay.defaultProps = {
  position: "absolute",
  expanded: false
};
export default EditorOverlay;
