// @kit-ui/admin/es/Editor/EditorSidebar.stories.js
import React from 'react';
import EditorBase from './EditorBase';
import EditorHeader from './EditorHeader';
import EditorBody from './EditorBody';
import EditorSection from './EditorSection';
import EditorSidebar from './EditorSidebar';
export default {
  title: 'Editor/EditorSidebar',
  component: EditorSidebar,
  args: {
    header: {
      title: "Sidebar"
    },
    sidebar: {
      expanded: true
    }
  },
  argTypes: {}
};

var Template = function Template(args) {
  return /*#__PURE__*/React.createElement(EditorBase, null, /*#__PURE__*/React.createElement(EditorHeader, args.header), /*#__PURE__*/React.createElement(EditorBody, null, /*#__PURE__*/React.createElement(EditorSidebar, args.sidebar, "Sidebar"), /*#__PURE__*/React.createElement(EditorSection, null)));
};

export var Expanded = Template.bind({});
Expanded.args = {
  header: {
    title: "Sidebar expanded"
  },
  sidebar: {
    expanded: true
  }
};
export var Collapsed = Template.bind({});
Collapsed.args = {
  header: {
    title: "Sidebar collapsed"
  },
  sidebar: {
    expanded: false
  }
};
