// @kit-ui/schema/lib/PageEditor/PageEditorLayout.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _admin = require("@kit-ui/admin");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var PageEditor = function PageEditor(_ref) {
  var collapsible = _ref.collapsible,
      expanded = _ref.expanded,
      overlay = _ref.overlay,
      onToggle = _ref.onToggle,
      onSelect = _ref.onSelect,
      children = _ref.children,
      props = _objectWithoutPropertiesLoose(_ref, ["collapsible", "expanded", "overlay", "onToggle", "onSelect", "children"]);

  var renderOverlay = function renderOverlay() {
    var OverlayTemplate = overlay && overlay.template;
    return /*#__PURE__*/_react.default.createElement(_admin.EditorOverlay, {
      position: "fixed",
      expanded: overlay && overlay.expanded
    }, OverlayTemplate && /*#__PURE__*/_react.default.createElement(OverlayTemplate, overlay));
  };

  return /*#__PURE__*/_react.default.createElement(_admin.EditorBase, null, /*#__PURE__*/_react.default.createElement(_admin.EditorHeader, _extends({}, props, {
    collapsible: collapsible,
    expanded: expanded,
    onToggle: onToggle,
    onSelect: onSelect
  })), /*#__PURE__*/_react.default.createElement(_admin.EditorBody, {
    elevation: 1,
    expanded: expanded
  }, children), /*#__PURE__*/_react.default.createElement(_admin.EditorFooter, props), overlay && renderOverlay());
};

var _default = PageEditor;
exports.default = _default;
module.exports = exports.default;
