// @kit-ui/schema/lib/PageEditor/PageContentLayout.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _admin = require("@kit-ui/admin");

var _PageMenu = _interopRequireDefault(require("./PageMenu"));

var _styles = require("@material-ui/core/styles");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var useStyles = (0, _styles.makeStyles)(function (theme) {
  return {
    content: {
      backgroundColor: theme.palette.background.default,
      position: "absolute",
      zIndex: 2,
      top: 0,
      right: 0,
      bottom: 0,
      left: 0,
      width: "100%",
      overflow: "hidden",
      transition: ".125s ease-out",
      transform: "translateY(100%)",
      "&[aria-expanded=true]": {
        transform: "translateY(0)"
      }
    },
    wrapper: {
      backgroundColor: "inherit",
      position: "absolute",
      top: 0,
      right: 0,
      bottom: 0,
      left: 0
    }
  };
});

var PageContent = function PageContent(_ref) {
  var id = _ref.id,
      _ref$selected = _ref.selected,
      selected = _ref$selected === void 0 ? true : _ref$selected,
      sidebar = _ref.sidebar,
      preview = _ref.preview,
      menuWidth = _ref.menuWidth,
      _ref$menu = _ref.menu,
      menu = _ref$menu === void 0 ? undefined : _ref$menu,
      children = _ref.children;
  var classes = useStyles();

  var renderSidebar = function renderSidebar() {
    var SidebarTemplate = sidebar && sidebar.template;
    return /*#__PURE__*/_react.default.createElement(_admin.EditorSidebar, {
      expanded: sidebar && sidebar.expanded
    }, SidebarTemplate && /*#__PURE__*/_react.default.createElement(SidebarTemplate, sidebar));
  };

  var renderPreview = function renderPreview() {
    var PreviewTemplate = preview && preview.template;
    return /*#__PURE__*/_react.default.createElement(_admin.EditorPreview, {
      expanded: true
    }, PreviewTemplate && /*#__PURE__*/_react.default.createElement(PreviewTemplate, preview));
  };

  return /*#__PURE__*/_react.default.createElement(_admin.EditorSection, {
    id: id,
    expanded: selected,
    className: classes.content
  }, sidebar && renderSidebar(), /*#__PURE__*/_react.default.createElement(_admin.EditorSection, {
    elevation: 0,
    className: classes.wrapper
  }, preview && renderPreview(), /*#__PURE__*/_react.default.createElement(_admin.EditorContent, {
    elevation: 0
  }, menu && /*#__PURE__*/_react.default.createElement(_PageMenu.default, {
    menuWidth: menuWidth,
    menu: menu
  }), children)));
};

var _default = PageContent;
exports.default = _default;
module.exports = exports.default;
