// @kit-ui/schema/lib/PageEditor/PageSettingsLayout.stories.js
"use strict";

exports.__esModule = true;
exports.Default = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _SchemaEditor = _interopRequireDefault(require("../SchemaEditor"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var DialogTemplate = function DialogTemplate(_ref) {
  var formData = _ref.formData;
  return /*#__PURE__*/_react.default.createElement("p", null, "Dialg template");
};

var PreviewTemplate = function PreviewTemplate(_ref2) {
  var formData = _ref2.formData;
  return /*#__PURE__*/_react.default.createElement("p", null, "Preview template");
};

var _default = {
  title: 'PageEditor/PageSettingsLayout',
  component: _SchemaEditor.default,
  args: {
    "schema": {
      "type": "object",
      "properties": {
        "title": {
          "type": "string"
        },
        "array": {
          "type": "array",
          "items": {
            "type": "string"
          }
        }
      }
    },
    "uiSchema": {
      "ui:layout": "pageSettings",
      "ui:fieldset": ["array", "location"],
      "array": {
        "ui:buttons": ["dialog"]
      },
      "location": {
        "lat": {
          "type": "number"
        },
        "lng": {
          "type": "number"
        }
      }
    },
    "formContext": {
      "preview": {
        "template": PreviewTemplate
      },
      "title": "Document title",
      "parents": [{
        "title": "Parent"
      }]
    }
  },
  argTypes: {}
};
exports.default = _default;

var Template = function Template(args) {
  return /*#__PURE__*/_react.default.createElement(_SchemaEditor.default, args);
};

var Default = Template.bind({});
exports.Default = Default;
Default.args = {};
