// @kit-ui/schema/lib/PageEditor/PageEditorField.stories.js
"use strict";

exports.__esModule = true;
exports.Collapsible = exports.Default = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _SchemaEditor = _interopRequireDefault(require("../SchemaEditor/"));

var _PageEditorField = _interopRequireDefault(require("./PageEditorField"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var PreviewTemplate = function PreviewTemplate(_ref) {
  var formData = _ref.formData;
  return /*#__PURE__*/_react.default.createElement("p", null, "Preview template");
};

var _default = {
  title: 'PageEditor/PageEditorField',
  component: _SchemaEditor.default,
  args: {
    "schema": {
      "type": "object",
      "properties": {
        "title": {
          "type": "string"
        },
        "arrayWithSidebar": {
          "type": "array",
          "items": {
            "type": "string"
          }
        },
        "settings": {
          "type": "object",
          "properties": {
            "setting1": {
              "type": "string"
            },
            "arrayWithOverlay": {
              "type": "array",
              "items": {
                "type": "string"
              }
            }
          }
        }
      }
    },
    "uiSchema": {
      "ui:field": "pageEditor",
      "ui:collapsible": true,
      "ui:fieldset": ["content", "settings"],
      "content": {
        "ui:fieldset": ["title", "arrayWithSidebar"],
        "ui:layout": "pageContent",
        "arrayWithSidebar": {
          "ui:dialog": true
        },
        "ui:buttons": ["sidebar"]
      },
      "settings": {
        "ui:layout": "pageSettings",
        "arrayWithOverlay": {
          "ui:dialog": true
        }
      }
    },
    "formContext": {
      "preview": {
        "template": PreviewTemplate
      },
      "parents": [{
        title: "Parent"
      }]
    }
  },
  argTypes: {}
};
exports.default = _default;

var Template = function Template(args) {
  return /*#__PURE__*/_react.default.createElement(_SchemaEditor.default, args);
};

var Default = Template.bind({});
exports.Default = Default;
Default.args = {};
var Collapsible = Template.bind({});
exports.Collapsible = Collapsible;
Collapsible.args = {
  "ui:schema": {
    "ui:collapsible": true
  }
};
