// @kit-ui/schema/lib/PageEditor/PageSettingsLayout.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _admin = require("@kit-ui/admin");

var _PageMenu = _interopRequireDefault(require("./PageMenu"));

var _styles = require("@material-ui/core/styles");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var useStyles = (0, _styles.makeStyles)(function (theme) {
  return {
    settings: {
      backgroundColor: theme.palette.background.default,
      zIndex: 1,
      position: "absolute",
      top: 0,
      right: 0,
      bottom: 0,
      left: 0,
      "&[aria-expanded=true]": {
        top: 0
      }
    }
  };
});

var PageSettings = function PageSettings(_ref) {
  var id = _ref.id,
      _ref$expanded = _ref.expanded,
      expanded = _ref$expanded === void 0 ? true : _ref$expanded,
      menuWidth = _ref.menuWidth,
      _ref$menu = _ref.menu,
      menu = _ref$menu === void 0 ? undefined : _ref$menu,
      children = _ref.children;
  var classes = useStyles();
  /*
   const renderDialog = () => {
      const DialogTemplate = dialog && dialog.template;
       return (
          <EditorOverlay position="fixed" expanded={dialog && dialog.expanded}>
              { DialogTemplate && <DialogTemplate {...dialog} /> }
          </EditorOverlay>
      )
  }
   */

  return /*#__PURE__*/_react.default.createElement(_admin.EditorSection, {
    id: id,
    elevation: 0,
    className: classes.settings,
    "aria-expanded": expanded
  }, menu && /*#__PURE__*/_react.default.createElement(_PageMenu.default, {
    menuWidth: menuWidth,
    menu: menu
  }), /*#__PURE__*/_react.default.createElement(_admin.EditorContent, {
    expanded: true
  }, children));
};

var _default = PageSettings;
exports.default = _default;
module.exports = exports.default;
