// @kit-ui/schema/lib/PageEditor/PageEditorLayout.stories.js
"use strict";

exports.__esModule = true;
exports.Default = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _SchemaEditor = _interopRequireDefault(require("../SchemaEditor"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _default = {
  title: 'PageEditor/PageEditorLayout',
  component: _SchemaEditor.default,
  args: {
    "schema": {
      "type": "object",
      "properties": {
        "title": {
          "type": "string"
        },
        "content": {
          "type": "object",
          "properties": {
            "title": {
              "type": "string"
            }
          }
        }
      }
    },
    "uiSchema": {
      "ui:layout": "pageEditor",
      "ui:fieldset": ["content"]
    },
    "formContext": {
      "title": "Document title",
      "parents": [{
        "title": "Parent"
      }]
    }
  },
  argTypes: {}
};
exports.default = _default;

var Template = function Template(args) {
  return /*#__PURE__*/_react.default.createElement(_SchemaEditor.default, args);
};

var Default = Template.bind({});
exports.Default = Default;
Default.args = {};
