// @kit-ui/schema/lib/PageEditor/PageMenu.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _admin = require("@kit-ui/admin");

var _styles = require("@material-ui/core/styles");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var useStyles = (0, _styles.makeStyles)(function (theme) {
  return {
    menu: {
      position: "absolute",
      zIndex: 1,
      top: 0,
      right: "auto",
      left: 0,
      bottom: 0,
      margin: 0,
      width: function width(props) {
        return props.menuWidth;
      },
      transition: ".125s ease-out",
      "& + *": {
        position: "absolute",
        zIndex: 1,
        top: 0,
        right: 0,
        left: 0,
        bottom: 0,
        margin: 0,
        overflowY: "scroll"
      },
      "&[aria-expanded=true] + *": {
        marginLeft: function marginLeft(props) {
          return props.menuWidth;
        }
      }
    },
    menuList: {
      margin: theme.spacing(2, 0),
      "& > li + li": {
        marginTop: theme.spacing(1)
      }
    }
  };
});

var PageMenu = function PageMenu(_ref) {
  var className = _ref.className,
      _ref$menuWidth = _ref.menuWidth,
      menuWidth = _ref$menuWidth === void 0 ? 224 : _ref$menuWidth,
      _ref$menu = _ref.menu,
      menu = _ref$menu === void 0 ? undefined : _ref$menu;
  var classes = useStyles({
    menuWidth: menuWidth
  });

  if (!menu) {
    return false;
  }

  return /*#__PURE__*/_react.default.createElement("aside", {
    className: className || classes.menu,
    "aria-expanded": true
  }, /*#__PURE__*/_react.default.createElement(_admin.NavMenuList, {
    className: classes.menuList
  }, menu.map(function (item, index) {
    return /*#__PURE__*/_react.default.createElement(_admin.NavMenuItem, null, /*#__PURE__*/_react.default.createElement(_admin.NavMenuLink, item));
  })));
};

var _default = PageMenu;
exports.default = _default;
module.exports = exports.default;
