// @kit-ui/admin/es/AppHeader/AppNav.js
import React from 'react';
import NavUser from "../NavUser/NavUser";
import NavApps from "../AppSwitcher/AppSwitcher";
import { makeStyles } from '@material-ui/core/styles';
var useStyles = makeStyles(function (theme) {
  var _root;

  return {
    root: (_root = {
      position: "absolute",
      top: 0,
      right: 0,
      zIndex: 2000,
      display: "flex",
      marginTop: theme.spacing(-8),
      "&[aria-expanded=true]": {
        marginTop: 0
      },
      height: theme.spacing(8),
      marginLeft: theme.spacing(1),
      marginRight: theme.spacing(1)
    }, _root["display"] = "flex", _root.alignItems = "center", _root.justifyContent = "flex-end", _root),
    button: {
      width: theme.spacing(6),
      height: theme.spacing(6),
      display: "flex",
      alignItems: "center",
      justifyContent: "center"
    }
  };
});

var AppNav = function AppNav(_ref) {
  var className = _ref.className,
      _ref$expanded = _ref.expanded,
      expanded = _ref$expanded === void 0 ? true : _ref$expanded,
      app = _ref.app,
      user = _ref.user,
      owner = _ref.owner,
      onLogin = _ref.onLogin,
      onLogout = _ref.onLogout,
      onSelect = _ref.onSelect;
  var classes = useStyles();
  return /*#__PURE__*/React.createElement("section", {
    className: className || classes.root,
    "aria-expanded": expanded
  }, user && user.uniqueId && /*#__PURE__*/React.createElement(NavApps, {
    className: classes.button,
    user: user,
    owners: user && user.owners,
    owner: owner,
    onSelect: onSelect
  }), /*#__PURE__*/React.createElement(NavUser, {
    className: classes.button,
    app: app,
    user: user,
    onLogin: onLogin,
    onLogout: onLogout
  }));
};

export default AppNav;
