// @kit-ui/admin/es/NavLocale/NavLocale.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import React, { useRef, useState } from 'react';
import { Dropdown } from "@kit-ui/core";
import Button from '@material-ui/core/Button';
import IconButton from "@material-ui/core/IconButton";
import Icon from "@material-ui/core/Icon";
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemText from '@material-ui/core/ListItemText';
import ISO6391 from 'iso-639-1';
import LocaleIcon from "./LocaleIcon";

var NavLocale = function NavLocale(_ref) {
  var _ref$options = _ref.options,
      options = _ref$options === void 0 ? [] : _ref$options,
      value = _ref.value,
      onChange = _ref.onChange;

  var _useState = useState(false),
      expanded = _useState[0],
      setExpanded = _useState[1];

  var anchorRef = useRef(null);

  var _onToggle = function _onToggle(event) {
    setExpanded(function (prevExpanded) {
      return !prevExpanded;
    });
  };

  var _onCollapse = function _onCollapse(event) {
    setExpanded(false);
  };

  var _onChange = function _onChange(value) {
    onChange && onChange(value);
    setExpanded(false);
  };

  var ListOption = function ListOption(option) {
    var title = option.title,
        value = option.value,
        onClick = option.onClick;
    return /*#__PURE__*/React.createElement(ListItem, {
      button: true,
      onClick: onClick
    }, /*#__PURE__*/React.createElement(ListItemIcon, null, /*#__PURE__*/React.createElement(Icon, null, /*#__PURE__*/React.createElement(LocaleIcon, {
      value: value
    }))), /*#__PURE__*/React.createElement(ListItemText, {
      primary: title || value
    }));
  };

  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(IconButton, {
    onClick: _onToggle,
    ref: anchorRef
  }, /*#__PURE__*/React.createElement(LocaleIcon, {
    value: value
  })), /*#__PURE__*/React.createElement(Dropdown, {
    expanded: expanded,
    anchorEl: anchorRef.current,
    onClickAway: _onCollapse
  }, /*#__PURE__*/React.createElement(List, null, options && options.map(function (option, index) {
    if (typeof option === "string") {
      option = {
        title: ISO6391.getNativeName(option) || option,
        value: option
      };
    }

    var value = option.value;
    return /*#__PURE__*/React.createElement(ListOption, _extends({}, option, {
      key: index,
      onClick: function onClick() {
        return _onChange(value);
      }
    }));
  }))));
};

export default NavLocale;
