// @kit-ui/admin/es/Section/ButtonCollapsible.js
import React, { Component } from 'react';
import IconButton from '@material-ui/core/IconButton';
import ToggleIcon from '@material-ui/icons/MoreHoriz';
import { makeStyles } from '@material-ui/core/styles';
var useStyles = makeStyles(function (theme) {
  return {
    button: {},
    icon: {
      transition: ".125s ease-in-out",
      transform: "rotate(0deg)",
      "&[aria-expanded=true]": {
        transition: ".125s ease-in-out",
        transform: "rotate(90deg)"
      }
    }
  };
});

var ButtonCollapsible = function ButtonCollapsible(_ref) {
  var className = _ref.className,
      expanded = _ref.expanded,
      onClick = _ref.onClick;
  var classes = useStyles();
  return /*#__PURE__*/React.createElement(IconButton, {
    className: className || classes.button,
    color: "inherit",
    "aria-label": "toggle",
    "aria-expanded": expanded,
    onClick: onClick
  }, /*#__PURE__*/React.createElement(ToggleIcon, {
    className: classes.icon,
    "aria-expanded": expanded
  }));
};

export default ButtonCollapsible;
