// @kit-ui/admin/es/Section/SectionButtons.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import React from 'react';
import PropTypes from "prop-types";
import ButtonGroup from "@material-ui/core/ButtonGroup";
import Button from '@material-ui/core/Button';
import Icon from '@material-ui/core/Icon';
import { makeStyles } from '@material-ui/core/styles';
var useStyles = makeStyles(function (theme) {
  return {
    buttongroup: {
      margin: function margin(props) {
        return theme.spacing(props.spacing / 2 * -1);
      },
      "& > *": {
        margin: function margin(props) {
          return theme.spacing(props.spacing / 2);
        }
      }
    },
    button: {
      fontFamily: "Akkurat, sans-serif",
      fontSize: "14px",
      fontWeight: "bold",
      textTransform: "none",
      letterSpacing: 0,
      borderRadius: 1,
      minWidth: theme.spacing(12),
      minHeight: theme.spacing(7)
    },
    icon: {
      marginRight: theme.spacing(1),
      "& + $label": {
        marginRight: theme.spacing(1)
      }
    },
    label: {}
  };
});

var SectionButton = function SectionButton(_ref) {
  var icon = _ref.icon,
      title = _ref.title,
      _ref$color = _ref.color,
      color = _ref$color === void 0 ? "primary" : _ref$color,
      _ref$variant = _ref.variant,
      variant = _ref$variant === void 0 ? "outlined" : _ref$variant,
      _ref$size = _ref.size,
      size = _ref$size === void 0 ? "large" : _ref$size,
      props = _objectWithoutPropertiesLoose(_ref, ["icon", "title", "color", "variant", "size"]);

  var classes = useStyles();
  return /*#__PURE__*/React.createElement(Button, _extends({}, props, {
    type: "button",
    "data-size": size,
    "data-variant": variant,
    disableElevation: true,
    color: color,
    variant: variant,
    size: size,
    className: classes.button
  }), icon && /*#__PURE__*/React.createElement(Icon, {
    className: classes.icon
  }, icon), /*#__PURE__*/React.createElement("b", {
    className: classes.label
  }, title));
};

var SectionButtons = function SectionButtons(_ref2) {
  var className = _ref2.className,
      _ref2$spacing = _ref2.spacing,
      spacing = _ref2$spacing === void 0 ? 1 : _ref2$spacing,
      _ref2$buttons = _ref2.buttons,
      buttons = _ref2$buttons === void 0 ? [] : _ref2$buttons;
  var classes = useStyles({
    spacing: spacing
  });

  if (!buttons.length) {
    return null;
  }

  return /*#__PURE__*/React.createElement(ButtonGroup, {
    color: "primary",
    className: className || classes.buttongroup
  }, buttons && buttons.map(function (button, index) {
    return /*#__PURE__*/React.createElement(SectionButton, _extends({}, button, {
      key: index
    }));
  }));
};

SectionButtons.defaultProps = {
  "spacing": 1
};
SectionButtons.propTypes = process.env.NODE_ENV !== "production" ? {
  buttons: PropTypes.array,
  buttonGroupProps: PropTypes.object,
  buttonProps: PropTypes.object
} : {};
export default SectionButtons;
