// @kit-ui/admin/es/Section/SectionBody.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import React, { Component } from 'react';
import { makeStyles } from '@material-ui/core/styles';
var useStyles = makeStyles(function (theme) {
  return {
    sectionBody: {
      position: function position(props) {
        return props.position;
      },
      top: function top(props) {
        return props.top;
      },
      right: function right(props) {
        return props.right;
      },
      bottom: function bottom(props) {
        return props.bottom;
      },
      left: function left(props) {
        return props.left;
      },
      margin: function margin(props) {
        return props.position === "absolute" && theme.spacing(props.padding);
      },
      marginTop: function marginTop(props) {
        return props.position === "absolute" && theme.spacing(8);
      },
      paddingLeft: function paddingLeft(props) {
        return theme.spacing(props.padding);
      },
      paddingRight: function paddingRight(props) {
        return theme.spacing(props.padding);
      },
      paddingBottom: function paddingBottom(props) {
        return theme.spacing(props.padding);
      },
      "& > * + *": {
        marginTop: function marginTop(props) {
          return theme.spacing(props.spacing);
        }
      }
    }
  };
});

var SectionBody = function SectionBody(_ref) {
  var className = _ref.className,
      _ref$position = _ref.position,
      position = _ref$position === void 0 ? "relative" : _ref$position,
      _ref$grid = _ref.grid,
      grid = _ref$grid === void 0 ? false : _ref$grid,
      _ref$padding = _ref.padding,
      padding = _ref$padding === void 0 ? 2 : _ref$padding,
      _ref$spacing = _ref.spacing,
      spacing = _ref$spacing === void 0 ? 0 : _ref$spacing,
      children = _ref.children,
      props = _objectWithoutPropertiesLoose(_ref, ["className", "position", "grid", "padding", "spacing", "children"]);

  if (position === "absolute") {
    props.top = 0;
    props.right = 0;
    props.bottom = 0;
    props.left = 0;
  }

  var classes = useStyles(_extends({
    position: position,
    padding: padding,
    spacing: spacing
  }, props));

  if (!children) {
    return "";
  }

  return /*#__PURE__*/React.createElement("div", {
    className: className || classes.sectionBody
  }, children);
};

export default SectionBody;
