// @kit-ui/schema/lib/__SchemaFieldset/SchemaFieldset.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _utils = require("../utils");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var SchemaFieldset = function SchemaFieldset(props) {
  var id = props.id,
      idPrefix = props.idPrefix,
      schema = props.schema,
      uiSchema = props.uiSchema,
      formData = props.formData,
      children = props.children;
  var uiOptions = (0, _utils.getUiOptions)(props);
  var fieldset = uiOptions && uiOptions.fieldset;

  if (!fieldset) {
    return children;
  }

  var renderFieldset = function renderFieldset(name, index) {
    var fieldSchema = schema && schema.properties && schema.properties[name] && (0, _utils.retrieveSchema)(schema.properties[name], props.registry.definitions, formData[name]);
    var fieldFormData = fieldSchema && formData && formData[name] && (0, _utils.getDefaultFormState)(fieldSchema, formData[name]);
    var fieldUiSchema = uiSchema && uiSchema[name] || {};
    var fieldUiOptions = fieldUiSchema && (0, _utils.getUiOptions)(fieldUiSchema);
    var SchemaField = props.registry.fields.SchemaField;

    if (fieldSchema) {
      return /*#__PURE__*/_react.default.createElement(SchemaField, _extends({}, props, {
        key: index,
        idPrefix: id,
        name: name,
        schema: fieldSchema,
        uiSchema: fieldUiSchema,
        formData: fieldFormData,
        onChange: function onChange(value) {
          var _extends2;

          return props.onChange(_extends({}, formData, (_extends2 = {}, _extends2[name] = value, _extends2)));
        }
      }));
    }

    if (fieldUiSchema && name) {
      return /*#__PURE__*/_react.default.createElement(SchemaField, _extends({}, props, {
        key: index,
        idPrefix: id,
        name: name,
        schema: schema,
        uiSchema: fieldUiSchema,
        formData: formData
      }));
      return /*#__PURE__*/_react.default.createElement("p", null, "fieldset:", name);
    }

    return /*#__PURE__*/_react.default.createElement("p", null, "fieldset:", name);
  };

  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, fieldset && fieldset.map(renderFieldset));
};

var _default = SchemaFieldset;
exports.default = _default;
module.exports = exports.default;
