// @kit-ui/schema/lib/Layout/FlexItemLayout.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _styles = require("@material-ui/core/styles");

var _Grid = _interopRequireDefault(require("@material-ui/core/Grid"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var useStyles = (0, _styles.makeStyles)(function (theme) {
  return {
    fieldset: {
      padding: function padding(props) {
        return theme.spacing(props.padding);
      }
    },
    nav: {
      "& > [aria-hidden=true]": {
        display: "none"
      }
    },
    flex: {
      display: "flex",
      flexDirection: "column",
      margin: function margin(props) {
        return theme.spacing(props.spacing * -1 / 2);
      },
      "&[role=tablist]": {
        "& > [role=tab][aria-hidden=true]": {
          display: "none"
        }
      }
    },
    flexItem: {
      margin: function margin(props) {
        return theme.spacing(props.spacing * 1 / 2);
      }
    }
  };
});

var FlexLayout = function FlexLayout(_ref) {
  var menu = _ref.menu,
      padding = _ref.padding,
      spacing = _ref.spacing,
      children = _ref.children;
  var classes = useStyles();
  return /*#__PURE__*/_react.default.createElement("div", {
    className: classes.fieldset
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: classes.flex,
    role: menu && "tablist"
  }, children));
};

var _default = FlexLayout;
exports.default = _default;
module.exports = exports.default;
