// @kit-ui/schema/lib/Layout/ModuleLayout.stories.js
"use strict";

exports.__esModule = true;
exports.EditableWithTabs = exports.EditableWithSettings = exports.Editable = exports.Default = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _ModuleLayout = _interopRequireDefault(require("./ModuleLayout"));

var _SchemaEditor = _interopRequireDefault(require("../SchemaEditor/SchemaEditor"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _default = {
  title: 'Layouts/ModuleLayout',
  component: _ModuleLayout.default,
  args: {
    schema: {
      type: "object",
      properties: {
        title: {
          type: "string"
        },
        description: {
          type: "string"
        },
        size: {
          "type": "string",
          "enum": ["s", "m", "l"]
        }
      }
    },
    uiSchema: {
      "ui:layout": _ModuleLayout.default
    },
    formData: {
      title: "My title",
      description: "My description"
    }
  },
  argTypes: {}
};
exports.default = _default;

var Template = function Template(args) {
  return /*#__PURE__*/_react.default.createElement(_SchemaEditor.default, args);
};

var Default = Template.bind({});
exports.Default = Default;
Default.args = {};
var Editable = Template.bind({});
exports.Editable = Editable;
Editable.args = {
  uiSchema: {
    "ui:editable": true
  }
};
var EditableWithSettings = Template.bind({});
exports.EditableWithSettings = EditableWithSettings;
EditableWithSettings.args = {
  uiSchema: {
    "ui:editable": true,
    "ui:options": {
      "spacing": 2,
      "grid": true
    },
    "ui:settings": ["size"]
  }
};
var EditableWithTabs = Template.bind({});
exports.EditableWithTabs = EditableWithTabs;
EditableWithTabs.args = {
  uiSchema: {
    "ui:editable": true,
    "ui:fieldset": ["tab1", "tab2"],
    "tab1": {
      "ui:spacing": 2,
      "ui:fieldset": ["title", "description"]
    },
    "tab2": {
      "ui:spacing": 2,
      "ui:fieldset": ["title", "size"]
    },
    "ui:options": {
      "grid": true,
      "nav": "tabs"
    },
    "ui:settings": ["size"]
  }
};
