// @kit-ui/schema/lib/Layout/SectionLayout.stories.js
"use strict";

exports.__esModule = true;
exports.EditableExpanded = exports.Editable = exports.CollapsibleExpanded = exports.Collapsible = exports.DefaultWithIcon = exports.Default = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _SchemaEditor = _interopRequireDefault(require("../SchemaEditor/SchemaEditor"));

var _SectionLayout = _interopRequireDefault(require("./SectionLayout"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _default = {
  title: 'Layouts/SectionLayout',
  component: _SchemaEditor.default,
  args: {
    schema: {
      type: "object",
      properties: {
        title: {
          type: "string"
        },
        description: {
          type: "string"
        }
      }
    },
    uiSchema: {
      "ui:layout": "section"
    },
    formData: {
      title: "Title",
      description: "description"
    }
  },
  argTypes: {}
};
exports.default = _default;

var Template = function Template(args) {
  return /*#__PURE__*/_react.default.createElement(_SchemaEditor.default, args);
};

var Default = Template.bind({});
exports.Default = Default;
Default.args = {};
var DefaultWithIcon = Template.bind({});
exports.DefaultWithIcon = DefaultWithIcon;
DefaultWithIcon.args = {
  uiSchema: {
    "ui:icon": "people"
  }
};
var Collapsible = Template.bind({});
exports.Collapsible = Collapsible;
Collapsible.args = {
  uiSchema: {
    "ui:options": {
      collapsible: true,
      expanded: false,
      buttons: [{
        title: "Click"
      }, {
        title: "Dont click",
        disabled: true
      }]
    }
  }
};
var CollapsibleExpanded = Template.bind({});
exports.CollapsibleExpanded = CollapsibleExpanded;
CollapsibleExpanded.args = {
  uiSchema: {
    "ui:options": {
      collapsible: true,
      expanded: true,
      buttons: [{
        title: "Click"
      }, {
        title: "Dont click",
        disabled: true
      }]
    }
  }
};
var Editable = Template.bind({});
exports.Editable = Editable;
Editable.args = {
  uiSchema: {
    "ui:options": {
      editable: true,
      expanded: false,
      buttons: [{
        title: "Click"
      }, {
        title: "Dont click",
        disabled: true
      }]
    }
  }
};
var EditableExpanded = Template.bind({});
exports.EditableExpanded = EditableExpanded;
EditableExpanded.args = {
  uiSchema: {
    "ui:options": {
      editable: true,
      expanded: true,
      buttons: [{
        title: "Click"
      }, {
        title: "Dont click",
        disabled: true
      }]
    }
  }
};
