// @kit-ui/schema/lib/Layout/SectionLayout.stories.js
"use strict";

exports.__esModule = true;
exports.ExpandedWithButtons = exports.Collapsible = exports.Default = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _SectionLayout = _interopRequireDefault(require("./SectionLayout"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _default = {
  title: 'Layouts/SectionLayout',
  component: _SectionLayout.default,
  args: {
    title: "Title",
    description: "description"
  },
  argTypes: {}
};
exports.default = _default;

var Template = function Template(args) {
  return /*#__PURE__*/_react.default.createElement(_SectionLayout.default, args, "Content");
};

var Default = Template.bind({});
exports.Default = Default;
Default.args = {};
var Collapsible = Template.bind({});
exports.Collapsible = Collapsible;
Collapsible.args = {
  collapsible: true,
  expanded: false,
  buttons: [{
    title: "Click"
  }, {
    title: "Dont click",
    disabled: true
  }]
};
var ExpandedWithButtons = Template.bind({});
exports.ExpandedWithButtons = ExpandedWithButtons;
ExpandedWithButtons.args = {
  collapsible: true,
  expanded: true,
  buttons: [{
    title: "Click"
  }, {
    title: "Dont click",
    disabled: true
  }]
};
