// @kit-ui/core/es/Button/Button.template.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import React from "react";
import Button from './Button';
import { makeStyles } from '@material-ui/core/styles';
var useStyles = makeStyles(function (theme) {
  return {
    root: {
      '& > *': {
        margin: theme.spacing(1)
      }
    }
  };
});

var Template = function Template(args) {
  var classes = useStyles();
  return /*#__PURE__*/React.createElement("div", {
    className: classes.root
  }, /*#__PURE__*/React.createElement(Button, args, "Default"), /*#__PURE__*/React.createElement(Button, _extends({}, args, {
    color: "primary"
  }), "Primary"), /*#__PURE__*/React.createElement(Button, _extends({}, args, {
    color: "secondary"
  }), "Secondary"), /*#__PURE__*/React.createElement(Button, _extends({}, args, {
    disabled: true
  }), "Disabled"), /*#__PURE__*/React.createElement(Button, _extends({}, args, {
    color: "primary",
    href: "#contained-buttons"
  }), "Link"));
};

export default Template;
