// @kit-ui/admin/lib/NavPath/PathFiltersList.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _styles = require("@material-ui/core/styles");

var _ButtonBase = _interopRequireDefault(require("@material-ui/core/ButtonBase"));

var _Check = _interopRequireDefault(require("@material-ui/icons/Check"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var useStyles = (0, _styles.makeStyles)(function (theme) {
  return {
    list: {
      position: "relative",
      listStyle: "none",
      padding: 0,
      margin: 0
    },
    item: {
      display: "block",
      padding: theme.spacing(0.25),
      minWidth: theme.spacing(20)
    },
    button: {
      width: "100%",
      display: "flex",
      justifyContent: "flex-start",
      padding: theme.spacing(0.5),
      "&:hover": {
        backgroundColor: "rgba(0,0,0,.15)"
      },
      "&[aria-selected=true]": {
        color: "blue"
      }
    },
    icon: {
      color: "inherit",
      minWidth: theme.spacing(2),
      marginLeft: theme.spacing(0.5),
      marginRight: theme.spacing(0.5),
      opacity: 0,
      "&[aria-selected=true]": {
        opacity: 1
      }
    },
    label: {
      fontFamily: "Akkurat, sans-serif",
      fontSize: '14px',
      lineHeight: '24px',
      margin: theme.spacing(1)
    }
  };
});

var ListItem = function ListItem(_ref) {
  var style = _ref.style,
      value = _ref.value,
      icon = _ref.icon,
      label = _ref.label,
      title = _ref.title,
      selected = _ref.selected,
      onClick = _ref.onClick,
      props = _objectWithoutPropertiesLoose(_ref, ["style", "value", "icon", "label", "title", "selected", "onClick"]);

  var classes = useStyles(props);
  return /*#__PURE__*/_react.default.createElement("li", {
    className: classes.item
  }, /*#__PURE__*/_react.default.createElement(_ButtonBase.default, {
    className: classes.button,
    "aria-selected": selected,
    onClick: onClick
  }, selected && /*#__PURE__*/_react.default.createElement(_Check.default, {
    className: classes.icon,
    "aria-selected": selected
  }), !selected && /*#__PURE__*/_react.default.createElement(_Check.default, {
    className: classes.icon,
    "aria-selected": selected
  }), /*#__PURE__*/_react.default.createElement("b", {
    className: classes.label
  }, label || title)));
};

var List = function List(_ref2) {
  var children = _ref2.children,
      props = _objectWithoutPropertiesLoose(_ref2, ["children"]);

  var classes = useStyles(props);
  return /*#__PURE__*/_react.default.createElement("ul", {
    className: classes.list
  }, children);
};

var PathFilters = function PathFilters(_ref3) {
  var cols = _ref3.cols,
      rows = _ref3.rows,
      value = _ref3.value,
      _ref3$options = _ref3.options,
      options = _ref3$options === void 0 ? [] : _ref3$options,
      onSelect = _ref3.onSelect,
      props = _objectWithoutPropertiesLoose(_ref3, ["cols", "rows", "value", "options", "onSelect"]);

  return /*#__PURE__*/_react.default.createElement(List, null, options.map(function (item, index) {
    return /*#__PURE__*/_react.default.createElement(ListItem, _extends({}, item, {
      selected: item.value === value,
      key: index,
      onClick: function onClick() {
        return onSelect(item);
      }
    }), item.value);
  }));
};

var _default = PathFilters;
exports.default = _default;
module.exports = exports.default;
