// @kit-ui/admin/lib/Module/EditableModule.stories.js
"use strict";

exports.__esModule = true;
exports.EditingButtons = exports.EditingMenu = exports.Editing = exports.Default = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _EditableModule = _interopRequireDefault(require("./EditableModule"));

var _ = require("../");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _default = {
  title: 'Module/Variant/EditableModule',
  component: _EditableModule.default,
  args: {
    imageUrl: "https://dms-test.dimu.org/image/012uNXVpKSZo?mediaType=image/png",
    title: "Title",
    description: "description",
    metadata: ["25kb"],
    status: "publish",
    type: "document"
  },
  argTypes: {}
};
exports.default = _default;

var Template = function Template(args) {
  return /*#__PURE__*/_react.default.createElement(_EditableModule.default, args, /*#__PURE__*/_react.default.createElement("div", null, JSON.stringify(args, 0, 2)));
};

var Default = Template.bind({});
exports.Default = Default;
Default.args = {};
var Editing = Template.bind({});
exports.Editing = Editing;
Editing.args = {
  editing: true
};
var EditingMenu = Template.bind({});
exports.EditingMenu = EditingMenu;
EditingMenu.args = {
  editing: true,
  menu: [{
    title: "Menu 1"
  }, {
    title: "Menu 2"
  }]
};
var EditingButtons = Template.bind({});
exports.EditingButtons = EditingButtons;
EditingButtons.args = {
  editing: true,
  buttons: [{
    title: "Button 1"
  }, {
    title: "Button 2"
  }]
};
