// @kit-ui/admin/lib/Module/ExpandedModule.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _ModuleFormat = _interopRequireDefault(require("./ModuleFormat"));

var _ModuleBase = _interopRequireDefault(require("./ModuleBase"));

var _ModuleHeader = _interopRequireDefault(require("./ModuleHeader"));

var _ModuleBody = _interopRequireDefault(require("./ModuleBody"));

var _ModuleFooter = _interopRequireDefault(require("./ModuleFooter"));

var _ModuleButtons = _interopRequireDefault(require("./ModuleButtons"));

var _styles = require("@material-ui/core/styles");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var useStyles = (0, _styles.makeStyles)(function (theme) {
  var _header;

  return {
    module: {
      position: "relative",
      zIndex: 2,
      "& + *": {
        zIndex: 1
      },
      display: "flex",
      flexDirection: "column",
      minHeight: function minHeight(props) {
        return props.minHeight;
      }
    },
    header: (_header = {
      position: "absolute",
      zIndex: 2,
      top: 0,
      right: 0,
      bottom: "auto",
      left: 0,
      display: "flex"
    }, _header["zIndex"] = 2, _header.alignItems = "center", _header.justifyContent = "flex-start", _header.height = theme.spacing(6), _header.marginLeft = theme.spacing(1), _header.marginRight = theme.spacing(1), _header.color = "inherit", _header["& + *"] = {
      zIndex: 1,
      marginTop: theme.spacing(6)
    }, _header),
    body: {
      minHeight: function minHeight(props) {
        return props.minHeight;
      },
      marginTop: function marginTop(props) {
        return props.minHeight && 0;
      },
      paddingRight: theme.spacing(2),
      paddingBottom: theme.spacing(2),
      paddingLeft: theme.spacing(2),
      flexGrow: 1
    },
    footer: {
      borderTop: "1px solid",
      borderColor: theme.palette.divider,
      padding: theme.spacing(2)
    }
  };
});

var ExpandedModule = function ExpandedModule(_ref) {
  var id = _ref.id,
      buttons = _ref.buttons,
      children = _ref.children,
      props = _objectWithoutPropertiesLoose(_ref, ["id", "buttons", "children"]);

  var classes = useStyles(props);
  return /*#__PURE__*/_react.default.createElement(_ModuleBase.default, {
    className: classes.module,
    id: id,
    expanded: true,
    elevation: 4
  }, /*#__PURE__*/_react.default.createElement(_ModuleFormat.default, props, /*#__PURE__*/_react.default.createElement(_ModuleHeader.default, _extends({}, props, {
    className: classes.header
  })), /*#__PURE__*/_react.default.createElement(_ModuleBody.default, {
    className: classes.body
  }, children)), /*#__PURE__*/_react.default.createElement(_ModuleFooter.default, _extends({}, props, {
    className: classes.footer
  }), buttons && /*#__PURE__*/_react.default.createElement(_ModuleButtons.default, {
    buttons: buttons
  })));
};

ExpandedModule.propTypes = process.env.NODE_ENV !== "production" ? {
  title: _propTypes.default.string,
  menu: _propTypes.default.array,
  buttons: _propTypes.default.array,
  children: _propTypes.default.node
} : {};
var _default = ExpandedModule;
exports.default = _default;
module.exports = exports.default;
