// @kit-ui/admin/lib/Module/GridModule.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _ModuleBase = _interopRequireDefault(require("./ModuleBase"));

var _ModulePrefix = _interopRequireDefault(require("./ModulePrefix"));

var _ModuleMedia = _interopRequireDefault(require("./ModuleMedia"));

var _ModuleTitle = _interopRequireDefault(require("./ModuleTitle"));

var _ModuleLabel = _interopRequireDefault(require("./ModuleLabel"));

var _ModuleStatus = _interopRequireDefault(require("./ModuleStatus"));

var _ModuleByline = _interopRequireDefault(require("./ModuleByline"));

var _NavToolbar = _interopRequireDefault(require("../NavToolbar/NavToolbar"));

var _styles = require("@material-ui/core/styles");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var useStyles = (0, _styles.makeStyles)(function (theme) {
  return {
    root: {
      backgroundColor: "transparent",
      position: "relative",
      width: function width(props) {
        return props.width;
      },
      userSelect: "none",
      "& $toolbar": {
        opacity: 0
      },
      "&:hover": {
        "& $toolbar": {
          opacity: 1
        }
      },
      "&[data-status=trash]": {
        "& $media": {
          opacity: .5
        },
        "& $toolbar": {
          opacity: 1
        }
      },
      "&[data-status=erased]": {
        "& $media": {
          opacity: .25
        }
      }
    },
    grid: {
      display: "block",
      backgroundColor: "transparent",
      color: "white",
      position: "relative",
      width: "100%",
      "&[role=button]": {
        cursor: "pointer"
      },
      "&[aria-selected=true]": {
        backgroundColor: theme.palette.action.selected,
        color: "currentColor",
        "& $primaryAction": {
          opacity: 1
        },
        "& $media": {//                opacity: .75
        }
      }
    },
    media: {
      display: "block",
      position: "relative",
      width: "100%",
      height: function height(props) {
        return props.mediaHeight;
      },
      margin: "0"
    },
    primaryAction: {
      position: "absolute",
      zIndex: 3,
      top: 0,
      right: 'auto',
      bottom: "auto",
      left: 0,
      "& button": {
        color: "white"
      }
    },
    toolbar: {
      position: "absolute",
      zIndex: 2,
      top: 0,
      right: 0,
      bottom: "auto",
      left: 0,
      width: "100%",
      height: function height(props) {
        return props.mediaHeight;
      },
      display: "flex",
      alignItems: "center",
      justifyContent: "center",
      "& button": {
        color: "white",
        opacity: .5,
        "&:hover": {
          opacity: 1
        }
      },
      "& button[name=select]": {
        position: "absolute",
        top: 0,
        right: 0
      },
      "& button[name=toggle]": {
        position: "absolute",
        top: 0,
        right: 0
      }
    },
    content: {
      flexDirection: "column",
      display: "flex",
      alignItems: "left",
      justifyContent: "flex-start"
    },
    header: {
      width: "100%",
      flexBasis: "100%",
      display: "flex",
      alignItems: "center",
      justifyContent: "flex-start",
      marginTop: theme.spacing(.5)
    },
    body: {
      width: "100%",
      flexBasis: "100%",
      display: "flex",
      alignItems: "center",
      justifyContent: "flex-start",
      "& > * + *": {
        marginLeft: theme.spacing(.5)
      }
    },
    footer: {
      display: "flex",
      alignItems: "baseline",
      width: "100%",
      flexBasis: "100%",
      justifyContent: "flex-start",
      "& > * + *": {
        marginLeft: theme.spacing(.5)
      }
    }
  };
});

var GridModule = function GridModule(_ref) {
  var _ref$width = _ref.width,
      width = _ref$width === void 0 ? 180 : _ref$width,
      maxWidth = _ref.maxWidth,
      mediaStyle = _ref.mediaStyle,
      _ref$mediaHeight = _ref.mediaHeight,
      mediaHeight = _ref$mediaHeight === void 0 ? 180 : _ref$mediaHeight,
      _ref$mediaLayout = _ref.mediaLayout,
      mediaLayout = _ref$mediaLayout === void 0 ? "contain" : _ref$mediaLayout,
      selected = _ref.selected,
      imageUrl = _ref.imageUrl,
      title = _ref.title,
      description = _ref.description,
      type = _ref.type,
      typeLabel = _ref.typeLabel,
      status = _ref.status,
      statusLabel = _ref.statusLabel,
      author = _ref.author,
      datetime = _ref.datetime,
      onClick = _ref.onClick,
      onEdit = _ref.onEdit,
      props = _objectWithoutPropertiesLoose(_ref, ["width", "maxWidth", "mediaStyle", "mediaHeight", "mediaLayout", "selected", "imageUrl", "title", "description", "type", "typeLabel", "status", "statusLabel", "author", "datetime", "onClick", "onEdit"]);

  if (maxWidth) {
    width = maxWidth;
  }

  var classes = useStyles({
    width: width,
    mediaHeight: mediaHeight
  });
  var media = {
    elevation: 1,
    selected: selected,
    layout: mediaLayout,
    imageUrl: imageUrl || "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mNk+P+/HgAFhAJ/wlseKgAAAABJRU5ErkJggg=="
  };
  return /*#__PURE__*/_react.default.createElement(_ModuleBase.default, _extends({}, props, {
    className: classes.root,
    selected: selected,
    status: status,
    onClick: undefined
  }), /*#__PURE__*/_react.default.createElement("div", {
    className: classes.grid,
    onClick: onClick,
    role: onClick && "button",
    "aria-selected": selected
  }, /*#__PURE__*/_react.default.createElement(_ModuleMedia.default, _extends({}, media, {
    className: classes.media
  }))), /*#__PURE__*/_react.default.createElement(_NavToolbar.default, _extends({}, props, {
    className: classes.toolbar
  })), /*#__PURE__*/_react.default.createElement("div", {
    className: classes.content
  }, /*#__PURE__*/_react.default.createElement("header", {
    className: classes.header
  }, /*#__PURE__*/_react.default.createElement(_ModuleTitle.default, {
    title: title,
    status: status,
    onClick: onClick && undefined || onEdit
  })), /*#__PURE__*/_react.default.createElement("div", {
    className: classes.body
  }, /*#__PURE__*/_react.default.createElement(_ModuleLabel.default, {
    label: typeLabel || type
  }), /*#__PURE__*/_react.default.createElement(_ModuleStatus.default, {
    status: statusLabel || status
  })), /*#__PURE__*/_react.default.createElement("footer", {
    className: classes.footer
  }, /*#__PURE__*/_react.default.createElement(_ModuleByline.default, {
    author: author,
    datetime: datetime
  }))), /*#__PURE__*/_react.default.createElement(_ModulePrefix.default, _extends({
    className: classes.primaryAction,
    selected: selected
  }, props)));
};

GridModule.propTypes = process.env.NODE_ENV !== "production" ? {
  imageUrl: _propTypes.default.string,
  title: _propTypes.default.string,
  description: _propTypes.default.string,
  type: _propTypes.default.string,
  typeLabel: _propTypes.default.string,
  status: _propTypes.default.string,
  statusLabel: _propTypes.default.string,
  author: _propTypes.default.string,
  datetime: _propTypes.default.string,
  onClick: _propTypes.default.func,
  onEdit: _propTypes.default.func
} : {};
var _default = GridModule;
exports.default = _default;
module.exports = exports.default;
