// @kit-ui/admin/lib/Module/ListModule.stories.js
"use strict";

exports.__esModule = true;
exports.Capabilities = exports.WithSettings = exports.Status = exports.LongDescription = exports.Default = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _ListModule = _interopRequireDefault(require("./ListModule"));

var _ = require("../");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var _default = {
  title: 'Module/Variant/ListModule',
  component: _ListModule.default,
  args: {
    imageUrl: "https://dms-test.dimu.org/image/012uNXVpKSZo?mediaType=image/png",
    title: "Title",
    description: "description",
    metadata: ["25kb"],
    status: "publish",
    type: "document"
  },
  argTypes: {}
};
exports.default = _default;

var Template = function Template(args) {
  return /*#__PURE__*/_react.default.createElement(_ListModule.default, args);
};

var Default = Template.bind({});
exports.Default = Default;
Default.args = {};
var LongDescription = Template.bind({});
exports.LongDescription = LongDescription;
LongDescription.args = {
  description: "Nullam id dolor id nibh ultricies vehicula ut id elit. Praesent commodo cursus magna, vel scelerisque nisl consectetur et. Aenean eu leo quam. Pellentesque ornare sem lacinia quam venenatis vestibulum. Duis mollis, est non commodo luctus, nisi erat porttitor ligula, eget lacinia odio sem nec elit. Vivamus sagittis lacus vel augue laoreet rutrum faucibus dolor auctor.",
  editable: true,
  deletable: true
};

var StatusTemplate = function StatusTemplate(args) {
  return /*#__PURE__*/_react.default.createElement("div", null, /*#__PURE__*/_react.default.createElement(_ListModule.default, _extends({}, args, {
    status: "publish"
  })), /*#__PURE__*/_react.default.createElement(_ListModule.default, _extends({}, args, {
    status: "draft"
  })), /*#__PURE__*/_react.default.createElement(_ListModule.default, _extends({}, args, {
    status: "trash"
  })), /*#__PURE__*/_react.default.createElement(_ListModule.default, _extends({}, args, {
    status: "erased"
  })), /*#__PURE__*/_react.default.createElement(_ListModule.default, _extends({}, args, {
    selectable: true
  })), /*#__PURE__*/_react.default.createElement(_ListModule.default, _extends({}, args, {
    selectable: true,
    selected: true
  })));
};

var Status = StatusTemplate.bind({});
exports.Status = Status;
Default.args = {};

var WithSettingsTemplate = function WithSettingsTemplate(args) {
  return /*#__PURE__*/_react.default.createElement("div", null, /*#__PURE__*/_react.default.createElement(_ListModule.default, _extends({}, args, {
    status: "publish"
  })), /*#__PURE__*/_react.default.createElement(_ListModule.default, _extends({}, args, {
    status: "draft"
  })), /*#__PURE__*/_react.default.createElement(_ListModule.default, _extends({}, args, {
    status: "trash"
  })), /*#__PURE__*/_react.default.createElement(_ListModule.default, _extends({}, args, {
    status: "erased"
  })), /*#__PURE__*/_react.default.createElement(_ListModule.default, _extends({}, args, {
    selectable: true
  })), /*#__PURE__*/_react.default.createElement(_ListModule.default, _extends({}, args, {
    selectable: true,
    selected: true
  })));
};

var WithSettings = WithSettingsTemplate.bind({});
exports.WithSettings = WithSettings;
WithSettings.args = {
  settings: [{
    template: "color"
  }]
};

var CapabilitiesTemplate = function CapabilitiesTemplate(args) {
  return /*#__PURE__*/_react.default.createElement("div", null, /*#__PURE__*/_react.default.createElement(_ListModule.default, _extends({}, args, {
    selectable: true
  })), /*#__PURE__*/_react.default.createElement(_ListModule.default, _extends({}, args, {
    selectable: true,
    selected: true
  })), /*#__PURE__*/_react.default.createElement(_ListModule.default, _extends({}, args, {
    editable: true
  })), /*#__PURE__*/_react.default.createElement(_ListModule.default, _extends({}, args, {
    editable: true,
    deletable: true
  })), /*#__PURE__*/_react.default.createElement(_ListModule.default, _extends({}, args, {
    editable: true
  })), /*#__PURE__*/_react.default.createElement(_ListModule.default, _extends({}, args, {
    restorable: true
  })));
};

var Capabilities = CapabilitiesTemplate.bind({});
exports.Capabilities = Capabilities;
Capabilities.args = {};
