// @kit-ui/admin/lib/Module/Module.stories.js
"use strict";

exports.__esModule = true;
exports.EditableExpanded = exports.Editable = exports.Expanded = exports.Elevation = exports.Default = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _Module = _interopRequireDefault(require("./Module"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _default = {
  title: 'Module/Module',
  component: _Module.default,
  args: {
    imageUrl: "https://dms-test.dimu.org/image/012uNXVpKSZo?mediaType=image/png",
    title: "Title",
    description: "description",
    metadata: ["25kb"],
    status: "publish",
    type: "document"
  },
  argTypes: {}
};
exports.default = _default;

var Template = function Template(args) {
  return /*#__PURE__*/_react.default.createElement(_Module.default, args);
};

var Default = Template.bind({});
exports.Default = Default;
Default.args = {};
var Elevation = Template.bind({});
exports.Elevation = Elevation;
Elevation.args = {
  elevation: 1
};
var Expanded = Template.bind({});
exports.Expanded = Expanded;
Expanded.args = {
  expanded: true,
  children: /*#__PURE__*/_react.default.createElement("p", null, "children")
};
var Editable = Template.bind({});
exports.Editable = Editable;
Editable.args = {
  editable: true
};
var EditableExpanded = Template.bind({});
exports.EditableExpanded = EditableExpanded;
EditableExpanded.args = {
  editable: true,
  expanded: true,
  children: /*#__PURE__*/_react.default.createElement("p", null, "children")
};
