// @kit-ui/admin/lib/Module/ModuleActionbar.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _styles = require("@material-ui/core/styles");

var _propTypes = _interopRequireDefault(require("prop-types"));

var _NavSettings = _interopRequireDefault(require("../NavSettings/NavSettings"));

var _NavToolbar = _interopRequireDefault(require("../NavToolbar/NavToolbar"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var useStyles = (0, _styles.makeStyles)(function (theme) {
  return {
    actionbar: {
      display: "flex",
      alignItems: "center",
      justifyContent: "center"
    },
    settings: {
      display: "flex",
      alignItems: "center",
      "& > * + *": {
        marginLeft: theme.spacing(-1)
      }
    },
    toolbar: {
      display: "flex",
      alignItems: "center",
      "& > * + *": {
        marginLeft: theme.spacing(-1)
      },
      "& button": {
        opacity: .5,
        "&:hover": {
          opacity: 1
        }
      }
    }
  };
});

var ModuleActionbar = function ModuleActionbar(_ref) {
  var className = _ref.className,
      settings = _ref.settings,
      toolbar = _ref.toolbar,
      children = _ref.children,
      props = _objectWithoutPropertiesLoose(_ref, ["className", "settings", "toolbar", "children"]);

  var classes = useStyles();

  if (children) {
    return /*#__PURE__*/_react.default.createElement("div", {
      className: className || classes.actionbar
    }, children);
  }

  return /*#__PURE__*/_react.default.createElement("div", {
    className: className || classes.actionbar
  }, /*#__PURE__*/_react.default.createElement(_NavSettings.default, {
    className: classes.settings,
    settings: settings
  }), /*#__PURE__*/_react.default.createElement(_NavToolbar.default, _extends({}, props, {
    className: classes.toolbar,
    toolbar: toolbar
  })));
};

ModuleActionbar.propTypes = process.env.NODE_ENV !== "production" ? {
  settings: _propTypes.default.array,
  toolbar: _propTypes.default.array
} : {};
var _default = ModuleActionbar;
exports.default = _default;
module.exports = exports.default;
