// @kit-ui/admin/lib/Module/ModuleDescription.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Typography = _interopRequireDefault(require("@material-ui/core/Typography"));

var _styles = require("@material-ui/core/styles");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var useStyles = (0, _styles.makeStyles)(function (theme) {
  return {
    root: {
      maxWidth: "100%",
      fontFamily: "Akkurat, sans-serif",
      fontSize: "14px",
      lineHeight: "24px",
      fontWeight: "normal",
      color: function color(props) {
        return props.color || theme.palette.text.secondary;
      }
    }
  };
});

var ModuleDescription = function ModuleDescription(_ref) {
  var className = _ref.className,
      color = _ref.color,
      description = _ref.description,
      children = _ref.children;
  var classes = useStyles({
    color: color
  });

  if (!description && children) {
    description = children;
  }

  if (!description || typeof description === "undefined") {
    return false;
  }

  return /*#__PURE__*/_react.default.createElement(_Typography.default, {
    className: className || classes.root,
    noWrap: true,
    component: "h3"
  }, description);
};

ModuleDescription.propTypes = process.env.NODE_ENV !== "production" ? {
  description: _propTypes.default.string
} : {};
var _default = ModuleDescription;
exports.default = _default;
module.exports = exports.default;
