// @kit-ui/admin/lib/Module/ModuleTitle.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Typography = _interopRequireDefault(require("@material-ui/core/Typography"));

var _Link = _interopRequireDefault(require("@material-ui/core/Link"));

var _styles = require("@material-ui/core/styles");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var useStyles = (0, _styles.makeStyles)(function (theme) {
  return {
    root: {
      fontFamily: "Akkurat, sans-serif",
      fontSize: "16px",
      lineHeight: "24px",
      fontWeight: "bold",
      maxWidth: "100%",
      color: theme.palette.text.primary,
      "&[data-status=trash]": {
        color: theme.palette.text.disabled,
        textDecoration: "line-through"
      },
      "&[data-status=erased]": {
        color: theme.palette.text.disabled,
        textDecoration: "line-through"
      }
    },
    link: {
      color: "inherit",
      "&:hover": {
        cursor: "pointer"
      }
    }
  };
});

var ModuleTitle = function ModuleTitle(_ref) {
  var className = _ref.className,
      title = _ref.title,
      _ref$untitled = _ref.untitled,
      untitled = _ref$untitled === void 0 ? "Untitled" : _ref$untitled,
      status = _ref.status,
      onClick = _ref.onClick,
      children = _ref.children;
  var classes = useStyles();

  if (!title && children) {
    title = children;
  }

  if (!title && untitled) {
    title = untitled;
  }

  if (onClick) {
    return /*#__PURE__*/_react.default.createElement(_Typography.default, {
      "data-status": status,
      className: className || classes.root,
      component: "h2",
      noWrap: true
    }, /*#__PURE__*/_react.default.createElement(_Link.default, {
      className: classes.link,
      onClick: onClick
    }, title));
  }

  return /*#__PURE__*/_react.default.createElement(_Typography.default, {
    "data-status": status,
    className: className || classes.root,
    component: "h2",
    noWrap: true
  }, title);
};

ModuleTitle.propTypes = process.env.NODE_ENV !== "production" ? {
  untitled: _propTypes.default.string,
  title: _propTypes.default.string,
  onClick: _propTypes.default.func
} : {};
ModuleTitle.defaultProps = {
  onClick: undefined
};
var _default = ModuleTitle;
exports.default = _default;
module.exports = exports.default;
