// @kit-ui/admin/lib/Module/ModuleMetadata.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Typography = _interopRequireDefault(require("@material-ui/core/Typography"));

var _styles = require("@material-ui/core/styles");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var useStyles = (0, _styles.makeStyles)(function (theme) {
  return {
    root: {
      maxWidth: "100%",
      fontFamily: "Akkurat, sans-serif",
      fontSize: "14px",
      lineHeight: "24px",
      fontWeight: "normal",
      color: theme.palette.text.secondary
    }
  };
});
/** ModuleMetadata takes an array of strings and displays them on a single line */

var ModuleMetadata = function ModuleMetadata(_ref) {
  var className = _ref.className,
      _ref$metadata = _ref.metadata,
      metadata = _ref$metadata === void 0 ? [] : _ref$metadata;
  var classes = useStyles();

  if (!metadata) {
    return false;
  }

  metadata = metadata.join(' – ');
  return /*#__PURE__*/_react.default.createElement(_Typography.default, {
    className: className || classes.root,
    noWrap: true,
    component: "h3"
  }, metadata);
};

ModuleMetadata.propTypes = process.env.NODE_ENV !== "production" ? {
  component: _propTypes.default.string,
  metadata: _propTypes.default.array
} : {};
var _default = ModuleMetadata;
exports.default = _default;
module.exports = exports.default;
