// @kit-ui/admin/lib/Module/ModuleFooter.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _styles = require("@material-ui/core/styles");

var _ButtonGroup = _interopRequireDefault(require("@material-ui/core/ButtonGroup"));

var _Button = _interopRequireDefault(require("@material-ui/core/Button"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var useStyles = (0, _styles.makeStyles)(function (theme) {
  return {
    footer: {
      borderTop: "1px solid",
      borderColor: theme.palette.divider,
      padding: function padding(props) {
        return theme.spacing(props.padding);
      }
    }
  };
});
/** ModuleFooter */

var ModuleFooter = function ModuleFooter(_ref) {
  var className = _ref.className,
      _ref$padding = _ref.padding,
      padding = _ref$padding === void 0 ? 2 : _ref$padding,
      children = _ref.children;
  var classes = useStyles({
    padding: padding
  });

  if (!children) {
    return false;
  }

  return /*#__PURE__*/_react.default.createElement("footer", {
    className: className || classes.footer
  }, children);
};

var _default = ModuleFooter;
exports.default = _default;
module.exports = exports.default;
