// @kit-ui/admin/lib/Module/Module.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _CollapsedModule = _interopRequireDefault(require("./CollapsedModule"));

var _ExpandedModule = _interopRequireDefault(require("./ExpandedModule"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

/** Module, use for documents or media objects */
var Module = function Module(_ref) {
  var editable = _ref.editable,
      expanded = _ref.expanded,
      buttons = _ref.buttons,
      onSave = _ref.onSave,
      children = _ref.children,
      props = _objectWithoutPropertiesLoose(_ref, ["editable", "expanded", "buttons", "onSave", "children"]);

  if (editable && expanded) {
    if (!buttons || !buttons.length) {
      buttons = [{
        title: "Save",
        onClick: onSave
      }];
    }
  }

  if (expanded) {
    return /*#__PURE__*/_react.default.createElement(_ExpandedModule.default, _extends({}, props, {
      buttons: buttons
    }), children);
  }

  return /*#__PURE__*/_react.default.createElement(_CollapsedModule.default, _extends({}, props, {
    editable: editable
  }));
};

Module.defaultProps = {
  padding: 2,
  editable: false
};
Module.propTypes = process.env.NODE_ENV !== "production" ? {
  editable: _propTypes.default.bool,
  editing: _propTypes.default.bool
} : {};
var _default = Module;
exports.default = _default;
module.exports = exports.default;
