// @kit-ui/admin/lib/Module/TableModule.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _ModuleBase = _interopRequireDefault(require("./ModuleBase"));

var _ModulePrefix = _interopRequireDefault(require("./ModulePrefix"));

var _ModuleActionbar = _interopRequireDefault(require("./ModuleActionbar"));

var _ModuleMedia = _interopRequireDefault(require("./ModuleMedia"));

var _ModuleTitle = _interopRequireDefault(require("./ModuleTitle"));

var _ModuleLabel = _interopRequireDefault(require("./ModuleLabel"));

var _ModuleDescription = _interopRequireDefault(require("./ModuleDescription"));

var _ModuleMetadata = _interopRequireDefault(require("./ModuleMetadata"));

var _ModuleStatus = _interopRequireDefault(require("./ModuleStatus"));

var _ModuleByline = _interopRequireDefault(require("./ModuleByline"));

var _styles = require("@material-ui/core/styles");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var useStyles = (0, _styles.makeStyles)(function (theme) {
  return {
    root: {
      userSelect: "none",
      "& > td": {
        minHeight: theme.spacing(8)
      },
      "&[data-status=trash]": {
        "& figure": {
          opacity: "0.5"
        },
        "& $content": {
          opacity: "0.5"
        }
      },
      "& h2": {
        fontSize: "16px",
        lineHeight: "24px"
      },
      "& $primaryAction": {
        opacity: .5
      },
      "& $actionbar": {
        opacity: .5
      },
      "&[aria-selected=true]": {
        backgroundColor: theme.palette.action.selected,
        "& $primaryAction": {
          opacity: 1,
          "& svg": {
            opacity: 1
          }
        }
      },
      "&:hover": {
        "& $primaryAction": {
          opacity: 1,
          "& svg": {
            opacity: 1
          }
        },
        "& $actionbar": {
          opacity: 1,
          "& button": {
            opacity: .5,
            "&:hover": {
              opacity: 1
            }
          }
        }
      },
      "&[data-role=button]": {
        "&:hover": {
          cursor: "pointer"
        },
        "& *": {//                pointerEvents: "none"
        },
        "& $primaryAction": {
          opacity: .5
          /*
          "& svg": {
              opacity: 0
          }*/

        },
        "&[aria-selected=true]": {}
      }
    },
    headerCell: {
      width: "50%",
      verticalAlign: "middle",
      padding: theme.spacing(1)
    },
    footerCell: {
      width: "50%",
      verticalAlign: "middle",
      padding: theme.spacing(1),
      position: "relative"
    },
    footer: {
      display: "flex",
      width: "100%"
    },
    actionCell: {
      width: theme.spacing(6),
      verticalAlign: "middle"
    },
    mediaCell: {
      width: theme.spacing(7),
      textAlign: "center",
      verticalAlign: "middle"
    },
    media: {
      position: "relative",
      width: theme.spacing(7),
      height: theme.spacing(7)
    },
    content: {
      width: "100%",
      display: "flex",
      flexGrow: 1,
      flexDirection: "column",
      alignItems: "flex-start",
      justifyContent: "flex-start"
    },
    metadata: {
      width: "100%",
      display: "flex",
      alignItems: "baseline",
      justifyContent: "flex-start",
      overflow: "hidden",
      textOverflow: "ellipsis",
      "& * + *": {
        marginLeft: theme.spacing(.5)
      },
      "& > *": {
        lineHeight: "24px"
      }
    },
    primaryAction: {
      display: "flex",
      alignItems: "center",
      justifyContent: "center"
    },
    actionbar: {
      position: "absolute",
      top: 0,
      right: 0,
      bottom: 0,
      left: "auto",
      display: "flex",
      alignItems: "center",
      justifyContent: "flex-start"
    }
  };
});

var TableModule = function TableModule(_ref) {
  var imageUrl = _ref.imageUrl,
      title = _ref.title,
      description = _ref.description,
      metadata = _ref.metadata,
      type = _ref.type,
      typeLabel = _ref.typeLabel,
      status = _ref.status,
      statusLabel = _ref.statusLabel,
      author = _ref.author,
      datetime = _ref.datetime,
      selected = _ref.selected,
      onClick = _ref.onClick,
      onEdit = _ref.onEdit,
      props = _objectWithoutPropertiesLoose(_ref, ["imageUrl", "title", "description", "metadata", "type", "typeLabel", "status", "statusLabel", "author", "datetime", "selected", "onClick", "onEdit"]);

  var classes = useStyles();
  var defaultImageUrl = "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mNk+P+/HgAFhAJ/wlseKgAAAABJRU5ErkJggg==";
  return /*#__PURE__*/_react.default.createElement(_ModuleBase.default, _extends({}, props, {
    component: "tr",
    className: classes.root,
    status: status,
    selected: selected,
    role: onClick && "button"
  }), /*#__PURE__*/_react.default.createElement("td", {
    className: classes.actionCell,
    onClick: onClick
  }, /*#__PURE__*/_react.default.createElement(_ModulePrefix.default, _extends({
    className: classes.primaryAction,
    selected: selected
  }, props))), /*#__PURE__*/_react.default.createElement("td", {
    className: classes.mediaCell,
    onClick: onClick
  }, /*#__PURE__*/_react.default.createElement(_ModuleMedia.default, {
    backgroundColor: "transparent",
    imageUrl: imageUrl || defaultImageUrl,
    className: classes.media
  })), /*#__PURE__*/_react.default.createElement("td", {
    className: classes.headerCell,
    onClick: onClick
  }, /*#__PURE__*/_react.default.createElement("header", {
    className: classes.content
  }, /*#__PURE__*/_react.default.createElement(_ModuleTitle.default, {
    status: status,
    title: title,
    onClick: !onClick && onEdit || undefined
  }), /*#__PURE__*/_react.default.createElement("div", {
    className: classes.metadata
  }, /*#__PURE__*/_react.default.createElement(_ModuleLabel.default, {
    label: typeLabel || type
  }), /*#__PURE__*/_react.default.createElement(_ModuleDescription.default, {
    description: description
  }), /*#__PURE__*/_react.default.createElement(_ModuleMetadata.default, {
    metadata: metadata
  })))), /*#__PURE__*/_react.default.createElement("td", {
    className: classes.footerCell,
    onClick: onClick
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: classes.content
  }, /*#__PURE__*/_react.default.createElement(_ModuleStatus.default, {
    component: "h2",
    status: statusLabel || status
  }), /*#__PURE__*/_react.default.createElement("div", {
    className: classes.metadata
  }, /*#__PURE__*/_react.default.createElement(_ModuleByline.default, {
    author: author,
    datetime: datetime
  }))), /*#__PURE__*/_react.default.createElement(_ModuleActionbar.default, _extends({}, props, {
    className: classes.actionbar
  }))));
};

TableModule.propTypes = process.env.NODE_ENV !== "production" ? {
  imageUrl: _propTypes.default.string,
  title: _propTypes.default.string,
  description: _propTypes.default.string,
  type: _propTypes.default.string,
  typeLabel: _propTypes.default.string,
  status: _propTypes.default.string,
  statusLabel: _propTypes.default.string,
  author: _propTypes.default.string,
  datetime: _propTypes.default.string,
  onClick: _propTypes.default.func,
  onEdit: _propTypes.default.func
} : {};
var _default = TableModule;
exports.default = _default;
module.exports = exports.default;
