// @kit-ui/schema/lib/TagsWidget/TagsWidget.stories.js
"use strict";

exports.__esModule = true;
exports.WithEnumOptions = exports.Default = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _SchemaEditor = _interopRequireDefault(require("../SchemaEditor/"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _default = {
  title: 'Widgets/TagsWidget',
  component: _SchemaEditor.default,
  args: {
    formData: [],
    schema: {
      "type": "array",
      "items": {
        "type": "string",
        "enum": []
      },
      "uniqueItems": true
    },
    uiSchema: {
      "ui:widget": "tags"
    }
  },
  argTypes: {}
};
exports.default = _default;

var Template = function Template(args) {
  return /*#__PURE__*/_react.default.createElement(_SchemaEditor.default, args);
};

var Default = Template.bind({});
exports.Default = Default;
Default.args = {};
var WithEnumOptions = Template.bind({});
exports.WithEnumOptions = WithEnumOptions;
WithEnumOptions.args = {
  schema: {
    "type": "array",
    "items": {
      "type": "string",
      "enum": ["foo", "bar", "fuzz", "qux"]
    }
  },
  uiSchema: {
    "ui:title": "Title",
    "ui:help": "Help text",
    "ui:inline": true
  }
};
