// @kit-ui/schema/lib/Reference/Reference.uiPreview.js
"use strict";

exports.__esModule = true;
exports.default = void 0;
var _default = {
  "select": {
    "_action": "_action",
    "type": "type",
    "title": "title",
    "description": "description",
    "imageUrl": "imageUrl",
    "status": "status",
    "referenceId": "referenceId",
    "reference": "reference"
  },
  prepare: function prepare(_ref) {
    var _ref$formData = _ref.formData,
        formData = _ref$formData === void 0 ? {} : _ref$formData;
    var _action = formData._action,
        reference = formData.reference,
        referenceId = formData.referenceId;
    var status = reference && reference.status;

    if (_action === "delete") {
      status = "trash";
    }

    return {
      _action: _action,
      type: reference && reference.documentType,
      title: reference && reference.title,
      description: reference && reference.description,
      imageUrl: reference && reference.imageUrl,
      status: status,
      referenceId: referenceId,
      reference: reference
    };
  }
};
exports.default = _default;
module.exports = exports.default;
