// @kit-ui/schema/lib/Reference/ReferenceField.stories.js
"use strict";

exports.__esModule = true;
exports.ReferenceArray = exports.ReferenceType = exports.WithProperties = exports.WithContent = exports.DocumentType = exports.Default = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _ReferenceField = _interopRequireDefault(require("./ReferenceField"));

var _SchemaEditor = _interopRequireDefault(require("../SchemaEditor"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _default = {
  title: 'Fields/ReferenceField',
  component: _ReferenceField.default,
  args: {},
  argTypes: {}
};
exports.default = _default;

var Template = function Template(args) {
  return /*#__PURE__*/_react.default.createElement(_SchemaEditor.default, args);
};

var Default = Template.bind({});
exports.Default = Default;
Default.args = {
  schema: {
    type: "reference"
  }
};
var DocumentType = Template.bind({});
exports.DocumentType = DocumentType;
DocumentType.args = {
  schema: {
    type: "reference",
    documentType: "article"
  },
  formData: {
    referenceId: "1",
    reference: {
      title: "Referanse 1"
    }
  }
};
var WithContent = Template.bind({});
exports.WithContent = WithContent;
WithContent.args = {
  schema: {
    type: "reference",
    documentType: "article",
    content: {
      properties: {
        title: {
          "type": "string"
        }
      }
    }
  },
  formData: {
    referenceId: "1",
    reference: {
      title: "Referanse 1"
    }
  }
};
var WithProperties = Template.bind({});
exports.WithProperties = WithProperties;
WithProperties.args = {
  schema: {
    type: "reference",
    documentType: "article",
    content: {
      properties: {
        title: {
          "type": "string"
        }
      }
    },
    properties: {
      custom: {
        type: "object",
        properties: {
          title: {
            "type": "string"
          },
          description: {
            "type": "string"
          }
        }
      }
    }
  },
  formData: {
    referenceId: "1",
    reference: {
      title: "Referanse 1"
    }
  }
};
var ReferenceType = Template.bind({});
exports.ReferenceType = ReferenceType;
ReferenceType.args = {
  schema: {
    type: "reference",
    documentType: "article",
    referenceType: "pageHasArticles",
    formData: {
      referenceId: "1",
      reference: {
        title: "Referanse 1"
      }
    }
  }
};
var ReferenceArray = Template.bind({});
exports.ReferenceArray = ReferenceArray;
ReferenceArray.args = {
  schema: {
    type: "object",
    properties: {
      "array": {
        type: "array",
        items: {
          type: "reference",
          documentType: "article",
          referenceType: "pageHasArticles"
        }
      }
    }
  },
  uiSchema: {
    "ui:field": "pageEditor",
    "ui:fieldset": ["content"],
    "content": {
      "ui:field": "pageContent",
      "ui:fieldset": ["array"],
      "array": {
        "ui:sortable": true,
        "ui:dialog": true,
        "items": {
          "ui:layout": "module",
          "ui:editable": true
        }
      }
    }
  },
  formData: {
    array: [{
      referenceId: "1",
      reference: {
        title: "Referanse 1"
      }
    }]
  }
};
