// @kit-ui/admin/lib/utils/getCalendar.js
"use strict";

exports.__esModule = true;
exports.default = exports.getCalendar = void 0;

var _moment = _interopRequireDefault(require("moment"));

require("moment/locale/nb");

require("moment/locale/nn");

require("moment/locale/sv");

var _ = require("..");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var getCalendar = function getCalendar(_ref) {
  var date = _ref.date,
      _ref$items = _ref.items,
      items = _ref$items === void 0 ? [] : _ref$items;
  var dt = (0, _moment.default)(date).format('YYYY-MM-DD');
  var dtStart = (0, _moment.default)(dt).startOf('month').startOf('isoWeek').format('YYYY-MM-DD');
  var dtEnd = (0, _moment.default)(dt).endOf('month').endOf('isoWeek').format('YYYY-MM-DD');
  var weeks = (0, _moment.default)(dtEnd).diff(dtStart, 'week');
  var events = [];

  if (items.length) {
    events = (0, _.getEvents)({
      items: items,
      dtStart: dtStart,
      dtEnd: dtEnd
    });
  }

  var rows = [];

  for (var w = 0; w <= weeks; w++) {
    var datetime = (0, _moment.default)(dtStart).add(w, 'weeks');
    datetime = (0, _moment.default)(datetime).set('hour', 0);
    datetime = (0, _moment.default)(datetime).set('minute', 0);
    datetime = (0, _moment.default)(datetime).set('second', 0);
    var cols = [];
    var week = datetime;

    for (var d = 0; d <= 6; d++) {
      datetime = (0, _moment.default)(week).add(d, 'days');

      var _date = (0, _.getEventsByDate)(datetime, events);

      cols.push(_date);
    }

    rows.push(cols);
  }

  return {
    dt: dt,
    dtStart: dtStart,
    dtEnd: dtEnd,
    rows: rows,
    events: events
  };
};

exports.getCalendar = getCalendar;
var _default = getCalendar;
exports.default = _default;
