// @kit-ui/admin/lib/utils/getEvents.js
"use strict";

exports.__esModule = true;
exports.getEventsByHour = getEventsByHour;
exports.getEventsByDate = getEventsByDate;
exports.getEvents = getEvents;

var _lodash = _interopRequireDefault(require("lodash"));

var _moment = _interopRequireDefault(require("moment"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function getEventsByHour(datetime, events) {
  datetime = (0, _moment.default)(datetime).format();
  var date = {
    date: (0, _moment.default)(datetime).format('YYYY-MM-DD'),
    datetime: datetime,
    events: []
  };

  if (events && events.byHour && events.byHour[datetime]) {
    date.events = events.byHour[datetime].map(function (id) {
      return events.byId[id];
    });
  }

  return date;
}

function getEventsByDate(datetime, events) {
  datetime = (0, _moment.default)(datetime).format('YYYY-MM-DD');
  var date = {
    date: (0, _moment.default)(datetime).format('YYYY-MM-DD'),
    datetime: datetime,
    events: []
  };

  if (events && events.byDate && events.byDate[datetime]) {
    date.events = events.byDate[datetime].map(function (id) {
      return events.byId[id];
    });
  }

  return date;
}

function getEvents(_ref) {
  var items = _ref.items,
      dtStart = _ref.dtStart,
      dtEnd = _ref.dtEnd;
  var dates = [];
  var hours = [];
  var byId = {};
  var byDate = {};
  var byHour = {};

  var sortedEvents = _lodash.default.sortBy(items, 'dtStart', 'dtEnd');

  for (var i = 0; i < sortedEvents.length; i++) {
    var item = sortedEvents[i];
    var id = item['uniqueId'] || i;
    var date = (0, _moment.default)(item['dtStart']).format('YYYY-MM-DD');
    var hour = (0, _moment.default)(item['dtStart']);
    hour = (0, _moment.default)(hour).set('minute', 0);
    hour = (0, _moment.default)(hour).set('second', 0);
    hour = (0, _moment.default)(hour).format();
    byId[id] = item;

    if (byDate[date]) {
      byDate[date].push(id);
    } else {
      byDate[date] = [id];
      dates.push(date);
    }

    if (byHour[hour]) {
      byHour[hour].push(id);
    } else {
      byHour[hour] = [id];
      hours.push(hour);
    }
  }

  var events = {
    dates: dates,
    hours: hours,
    byId: byId,
    byDate: byDate,
    byHour: byHour
  };
  return events;
}
