// @kit-ui/admin/lib/utils/getRandomOwner.js
"use strict";

exports.__esModule = true;
exports.default = exports.getRandomOwners = exports.getRandomOwner = exports.getOwner = exports.getOwnerIcon = void 0;

var _react = _interopRequireDefault(require("react"));

var _icons = require("@kit-ui/icons");

var _ = require("./");

var _owner = _interopRequireDefault(require("../data/owner"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var owners = [{
  name: "val",
  title: "Valdresmusea"
}, {
  name: "byg",
  title: "Bygdøy Kongsgård"
}, {
  name: "nmm",
  title: "Norsk Maritimt Museum"
}, {
  name: "kyst",
  title: "Kystmyseet"
}, {
  name: "nkim",
  title: "Nordenfjeldske Kunstindustrimuseum"
}, {
  name: "kvi",
  title: "Kvinnemuseet"
}, {
  name: "fet",
  title: "Fetsund Lenser"
}];
var ownerImages = ["https://dms-cf-09.dimu.org/image/032yiVGVkzHu", "https://dms-cf-09.dimu.org/image/032yiziWuETr"];

var getOwnerIcon = function getOwnerIcon(_ref) {
  var name = _ref.name,
      title = _ref.title,
      theme = _ref.theme;
  var color = theme && theme.primaryColor;
  var label = name || title;
  var text = label && label.substr(0, 3).toUpperCase();
  return /*#__PURE__*/_react.default.createElement(_icons.OwnerIcon, {
    color: color,
    text: text
  });
};

exports.getOwnerIcon = getOwnerIcon;

var getOwner = function getOwner(owner) {
  if (!owner) {
    owner = _owner.default;
  }

  return _extends({}, owner, {
    icon: getOwnerIcon(owner)
  });
};

exports.getOwner = getOwner;

var getRandomOwner = function getRandomOwner() {
  var _getRandomArrayItem = (0, _.getRandomArrayItem)(owners),
      name = _getRandomArrayItem.name,
      title = _getRandomArrayItem.title;

  var imageUrl = (0, _.getRandomArrayItem)(ownerImages);

  var owner = _extends({}, _owner.default, {
    name: name,
    title: title,
    imageUrl: imageUrl,
    theme: (0, _.getRandomTheme)()
  });

  return getOwner(owner);
};

exports.getRandomOwner = getRandomOwner;

var getRandomOwners = function getRandomOwners(_ref2) {
  var _ref2$max = _ref2.max,
      max = _ref2$max === void 0 ? 20 : _ref2$max;
  var owners = [];

  for (var i = 0; i < max; i++) {
    var owner = getRandomOwner();
    owner.uniqueid = "owner-" + i;
    owners.push(owner);
  }

  return owners;
};

exports.getRandomOwners = getRandomOwners;
var _default = getRandomOwner;
exports.default = _default;
