// @kit-ui/admin/lib/utils/getSchedule.js
"use strict";

exports.__esModule = true;
exports.getSchedule = getSchedule;
exports.default = void 0;

var _lodash = _interopRequireDefault(require("lodash"));

var _moment = _interopRequireDefault(require("moment"));

var _2 = require("..");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function getSchedule(_ref) {
  var date = _ref.date,
      _ref$items = _ref.items,
      items = _ref$items === void 0 ? [] : _ref$items,
      months = _ref.months;

  if (!months) {
    months = 3;
  }

  var dt = (0, _moment.default)(date).format('YYYY-MM-DD');
  var dtStart = (0, _moment.default)(dt).format('YYYY-MM-DD');
  var dtEnd = (0, _moment.default)(dt).add(months, 'months').format('YYYY-MM-DD');
  var events = {
    dates: []
  };

  if (items) {
    events = (0, _2.getEvents)({
      items: items,
      dtStart: dtStart,
      dtEnd: dtEnd
    });
  }

  var rows = [];

  for (var i = 0; i < events.dates.length; i++) {
    var _dt = events.dates[i];
    var datetime = (0, _moment.default)(_dt);
    datetime = (0, _moment.default)(datetime).set('hour', 0);
    datetime = (0, _moment.default)(datetime).set('minute', 0);
    datetime = (0, _moment.default)(datetime).set('second', 0);

    var _date = (0, _2.getEventsByDate)(datetime, events);

    rows.push(_date);
  }

  var schedule = {
    dt: dt,
    dtStart: dtStart,
    dtEnd: dtEnd,
    rows: rows,
    events: events
  };
  return schedule;
}

var _default = getSchedule;
exports.default = _default;
