// @kit-ui/schema/lib/Playground/Playground.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireWildcard(require("react"));

var _reactMonacoEditor = _interopRequireDefault(require("react-monaco-editor"));

var _samples = require("@rjsf/playground/lib/samples");

require("react-app-polyfill/ie11");

var _core = _interopRequireWildcard(require("@rjsf/core"));

var _DemoFrame = _interopRequireDefault(require("./DemoFrame"));

var _admin = require("@kit-ui/admin");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _inheritsLoose(subClass, superClass) { subClass.prototype = Object.create(superClass.prototype); subClass.prototype.constructor = subClass; subClass.__proto__ = superClass; }

// deepEquals and shouldRender and isArguments are copied from rjsf-core. TODO: unify these utility functions.
function isArguments(object) {
  return Object.prototype.toString.call(object) === "[object Arguments]";
}

function deepEquals(a, b, ca, cb) {
  if (ca === void 0) {
    ca = [];
  }

  if (cb === void 0) {
    cb = [];
  }

  // Partially extracted from node-deeper and adapted to exclude comparison
  // checks for functions.
  // https://github.com/othiym23/node-deeper
  if (a === b) {
    return true;
  } else if (typeof a === "function" || typeof b === "function") {
    // Assume all functions are equivalent
    // see https://github.com/rjsf-team/react-jsonschema-form/issues/255
    return true;
  } else if (typeof a !== "object" || typeof b !== "object") {
    return false;
  } else if (a === null || b === null) {
    return false;
  } else if (a instanceof Date && b instanceof Date) {
    return a.getTime() === b.getTime();
  } else if (a instanceof RegExp && b instanceof RegExp) {
    return a.source === b.source && a.global === b.global && a.multiline === b.multiline && a.lastIndex === b.lastIndex && a.ignoreCase === b.ignoreCase;
  } else if (isArguments(a) || isArguments(b)) {
    if (!(isArguments(a) && isArguments(b))) {
      return false;
    }

    var slice = Array.prototype.slice;
    return deepEquals(slice.call(a), slice.call(b), ca, cb);
  } else {
    if (a.constructor !== b.constructor) {
      return false;
    }

    var ka = Object.keys(a);
    var kb = Object.keys(b); // don't bother with stack acrobatics if there's nothing there

    if (ka.length === 0 && kb.length === 0) {
      return true;
    }

    if (ka.length !== kb.length) {
      return false;
    }

    var cal = ca.length;

    while (cal--) {
      if (ca[cal] === a) {
        return cb[cal] === b;
      }
    }

    ca.push(a);
    cb.push(b);
    ka.sort();
    kb.sort();

    for (var j = ka.length - 1; j >= 0; j--) {
      if (ka[j] !== kb[j]) {
        return false;
      }
    }

    var key;

    for (var k = ka.length - 1; k >= 0; k--) {
      key = ka[k];

      if (!deepEquals(a[key], b[key], ca, cb)) {
        return false;
      }
    }

    ca.pop();
    cb.pop();
    return true;
  }
}

function shouldRender(comp, nextProps, nextState) {
  var props = comp.props,
      state = comp.state;
  return !deepEquals(props, nextProps) || !deepEquals(state, nextState);
}

var log = function log(type) {
  return console.log.bind(console, type);
};

var toJson = function toJson(val) {
  return JSON.stringify(val, null, 2);
};

var liveSettingsSchema = {
  type: "object",
  properties: {
    validate: {
      type: "boolean",
      title: "Live validation"
    },
    disable: {
      type: "boolean",
      title: "Disable whole form"
    },
    omitExtraData: {
      type: "boolean",
      title: "Omit extra data"
    },
    liveOmit: {
      type: "boolean",
      title: "Live omit"
    }
  }
};
var monacoEditorOptions = {
  minimap: {
    enabled: false
  },
  automaticLayout: true
};

var GeoPosition = /*#__PURE__*/function (_Component) {
  _inheritsLoose(GeoPosition, _Component);

  function GeoPosition(props) {
    var _this;

    _this = _Component.call(this, props) || this;
    _this.state = _extends({}, props.formData);
    return _this;
  }

  var _proto = GeoPosition.prototype;

  _proto.onChange = function onChange(name) {
    var _this2 = this;

    return function (event) {
      var _this2$setState;

      _this2.setState((_this2$setState = {}, _this2$setState[name] = parseFloat(event.target.value), _this2$setState));

      setImmediate(function () {
        return _this2.props.onChange(_this2.state);
      });
    };
  };

  _proto.render = function render() {
    var _this$state = this.state,
        lat = _this$state.lat,
        lon = _this$state.lon;
    return /*#__PURE__*/_react.default.createElement("div", {
      className: "geo"
    }, /*#__PURE__*/_react.default.createElement("h3", null, "Hey, I'm a custom component"), /*#__PURE__*/_react.default.createElement("p", null, "I'm registered as ", /*#__PURE__*/_react.default.createElement("code", null, "geo"), " and referenced in", /*#__PURE__*/_react.default.createElement("code", null, "uiSchema"), " as the ", /*#__PURE__*/_react.default.createElement("code", null, "ui:field"), " to use for this schema."), /*#__PURE__*/_react.default.createElement("div", {
      className: "row"
    }, /*#__PURE__*/_react.default.createElement("div", {
      className: "col-sm-6"
    }, /*#__PURE__*/_react.default.createElement("label", null, "Latitude"), /*#__PURE__*/_react.default.createElement("input", {
      className: "form-control",
      type: "number",
      value: lat,
      step: "0.00001",
      onChange: this.onChange("lat")
    })), /*#__PURE__*/_react.default.createElement("div", {
      className: "col-sm-6"
    }, /*#__PURE__*/_react.default.createElement("label", null, "Longitude"), /*#__PURE__*/_react.default.createElement("input", {
      className: "form-control",
      type: "number",
      value: lon,
      step: "0.00001",
      onChange: this.onChange("lon")
    }))));
  };

  return GeoPosition;
}(_react.Component);

var Editor = /*#__PURE__*/function (_Component2) {
  _inheritsLoose(Editor, _Component2);

  function Editor(props) {
    var _this3;

    _this3 = _Component2.call(this, props) || this;

    _defineProperty(_assertThisInitialized(_this3), "onCodeChange", function (code) {
      try {
        var parsedCode = JSON.parse(code);

        _this3.setState({
          valid: true,
          code: code
        }, function () {
          return _this3.props.onChange(parsedCode);
        });
      } catch (err) {
        _this3.setState({
          valid: false,
          code: code
        });
      }
    });

    _this3.state = {
      valid: true,
      code: props.code
    };
    return _this3;
  }

  var _proto2 = Editor.prototype;

  _proto2.UNSAFE_componentWillReceiveProps = function UNSAFE_componentWillReceiveProps(props) {
    this.setState({
      valid: true,
      code: props.code
    });
  };

  _proto2.shouldComponentUpdate = function shouldComponentUpdate(nextProps, nextState) {
    if (this.state.valid) {
      return JSON.stringify(JSON.parse(nextProps.code)) !== JSON.stringify(JSON.parse(this.state.code));
    }

    return false;
  };

  _proto2.render = function render() {
    var title = this.props.title;
    var icon = this.state.valid ? "ok" : "remove";
    var cls = this.state.valid ? "valid" : "invalid";
    return /*#__PURE__*/_react.default.createElement("div", {
      className: "panel panel-default"
    }, /*#__PURE__*/_react.default.createElement("div", {
      className: "panel-heading"
    }, /*#__PURE__*/_react.default.createElement("span", {
      className: cls + " glyphicon glyphicon-" + icon
    }), " " + title), /*#__PURE__*/_react.default.createElement(_reactMonacoEditor.default, {
      language: "json",
      value: this.state.code,
      theme: "vs-light",
      onChange: this.onCodeChange,
      height: 400,
      options: monacoEditorOptions
    }));
  };

  return Editor;
}(_react.Component);

var Selector = /*#__PURE__*/function (_Component3) {
  _inheritsLoose(Selector, _Component3);

  function Selector(props) {
    var _this4;

    _this4 = _Component3.call(this, props) || this;

    _defineProperty(_assertThisInitialized(_this4), "onLabelClick", function (label) {
      return function (event) {
        event.preventDefault();

        _this4.setState({
          current: label
        });

        setImmediate(function () {
          return _this4.props.onSelected(_samples.samples[label]);
        });
      };
    });

    _this4.state = {
      current: "Simple"
    };
    return _this4;
  }

  var _proto3 = Selector.prototype;

  _proto3.shouldComponentUpdate = function shouldComponentUpdate(nextProps, nextState) {
    return shouldRender(this, nextProps, nextState);
  };

  _proto3.render = function render() {
    var _this5 = this;

    return /*#__PURE__*/_react.default.createElement("ul", {
      className: "nav nav-pills"
    }, Object.keys(_samples.samples).map(function (label, i) {
      return /*#__PURE__*/_react.default.createElement("li", {
        key: i,
        role: "presentation",
        className: _this5.state.current === label ? "active" : ""
      }, /*#__PURE__*/_react.default.createElement("a", {
        href: "#",
        onClick: _this5.onLabelClick(label)
      }, label));
    }));
  };

  return Selector;
}(_react.Component);

function ThemeSelector(_ref) {
  var theme = _ref.theme,
      themes = _ref.themes,
      select = _ref.select;
  var schema = {
    type: "string",
    enum: Object.keys(themes)
  };
  var uiSchema = {
    "ui:placeholder": "Select theme"
  };
  return /*#__PURE__*/_react.default.createElement(_core.default, {
    className: "form_rjsf_themeSelector",
    idPrefix: "rjsf_themeSelector",
    schema: schema,
    uiSchema: uiSchema,
    formData: theme,
    onChange: function onChange(_ref2) {
      var formData = _ref2.formData;
      return formData && select(formData, themes[formData]);
    }
  }, /*#__PURE__*/_react.default.createElement("div", null));
}

function SubthemeSelector(_ref3) {
  var subtheme = _ref3.subtheme,
      subthemes = _ref3.subthemes,
      select = _ref3.select;
  var schema = {
    type: "string",
    enum: Object.keys(subthemes)
  };
  var uiSchema = {
    "ui:placeholder": "Select subtheme"
  };
  return /*#__PURE__*/_react.default.createElement(_core.default, {
    className: "form_rjsf_subthemeSelector",
    idPrefix: "rjsf_subthemeSelector",
    schema: schema,
    uiSchema: uiSchema,
    formData: subtheme,
    onChange: function onChange(_ref4) {
      var formData = _ref4.formData;
      return formData && select(formData, subthemes[formData]);
    }
  }, /*#__PURE__*/_react.default.createElement("div", null));
}

var CopyLink = /*#__PURE__*/function (_Component4) {
  _inheritsLoose(CopyLink, _Component4);

  function CopyLink() {
    var _this6;

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this6 = _Component4.call.apply(_Component4, [this].concat(args)) || this;

    _defineProperty(_assertThisInitialized(_this6), "onCopyClick", function (event) {
      _this6.input.select();

      document.execCommand("copy");
    });

    return _this6;
  }

  var _proto4 = CopyLink.prototype;

  _proto4.render = function render() {
    var _this7 = this;

    var _this$props = this.props,
        shareURL = _this$props.shareURL,
        onShare = _this$props.onShare;

    if (!shareURL) {
      return /*#__PURE__*/_react.default.createElement("button", {
        className: "btn btn-default",
        type: "button",
        onClick: onShare
      }, "Share");
    }

    return /*#__PURE__*/_react.default.createElement("div", {
      className: "input-group"
    }, /*#__PURE__*/_react.default.createElement("input", {
      type: "text",
      ref: function ref(input) {
        return _this7.input = input;
      },
      className: "form-control",
      defaultValue: shareURL
    }), /*#__PURE__*/_react.default.createElement("span", {
      className: "input-group-btn"
    }, /*#__PURE__*/_react.default.createElement("button", {
      className: "btn btn-default",
      type: "button",
      onClick: this.onCopyClick
    }, /*#__PURE__*/_react.default.createElement("i", {
      className: "glyphicon glyphicon-copy"
    }))));
  };

  return CopyLink;
}(_react.Component);

var Playground = /*#__PURE__*/function (_Component5) {
  _inheritsLoose(Playground, _Component5);

  function Playground(props) {
    var _this8;

    _this8 = _Component5.call(this, props) || this; // set default theme

    _defineProperty(_assertThisInitialized(_this8), "load", function (data) {
      // Reset the ArrayFieldTemplate whenever you load new data
      var ArrayFieldTemplate = data.ArrayFieldTemplate,
          ObjectFieldTemplate = data.ObjectFieldTemplate,
          extraErrors = data.extraErrors; // uiSchema is missing on some examples. Provide a default to
      // clear the field in all cases.

      var _data$uiSchema = data.uiSchema,
          uiSchema = _data$uiSchema === void 0 ? {} : _data$uiSchema;
      var _data$theme = data.theme,
          theme = _data$theme === void 0 ? _this8.state.theme : _data$theme;
      var themes = _this8.props.themes;

      _this8.onThemeSelected(theme, themes[theme]); // force resetting form component instance


      _this8.setState({
        form: false
      }, function () {
        return _this8.setState(_extends({}, data, {
          form: true,
          ArrayFieldTemplate: ArrayFieldTemplate,
          ObjectFieldTemplate: ObjectFieldTemplate,
          uiSchema: uiSchema,
          extraErrors: extraErrors
        }));
      });
    });

    _defineProperty(_assertThisInitialized(_this8), "onSchemaEdited", function (schema) {
      return _this8.setState({
        schema: schema,
        shareURL: null
      });
    });

    _defineProperty(_assertThisInitialized(_this8), "onUISchemaEdited", function (uiSchema) {
      return _this8.setState({
        uiSchema: uiSchema,
        shareURL: null
      });
    });

    _defineProperty(_assertThisInitialized(_this8), "onFormDataEdited", function (formData) {
      return _this8.setState({
        formData: formData,
        shareURL: null
      });
    });

    _defineProperty(_assertThisInitialized(_this8), "onExtraErrorsEdited", function (extraErrors) {
      return _this8.setState({
        extraErrors: extraErrors,
        shareURL: null
      });
    });

    _defineProperty(_assertThisInitialized(_this8), "onThemeSelected", function (theme, _temp) {
      var _ref5 = _temp === void 0 ? {} : _temp,
          subthemes = _ref5.subthemes,
          stylesheet = _ref5.stylesheet,
          themeObj = _ref5.theme;

      _this8.setState({
        theme: theme,
        subthemes: subthemes,
        subtheme: null,
        FormComponent: (0, _core.withTheme)(themeObj),
        stylesheet: stylesheet
      });
    });

    _defineProperty(_assertThisInitialized(_this8), "onSubthemeSelected", function (subtheme, _ref6) {
      var stylesheet = _ref6.stylesheet;

      _this8.setState({
        subtheme: subtheme,
        stylesheet: stylesheet
      });
    });

    _defineProperty(_assertThisInitialized(_this8), "setLiveSettings", function (_ref7) {
      var formData = _ref7.formData;
      return _this8.setState({
        liveSettings: formData
      });
    });

    _defineProperty(_assertThisInitialized(_this8), "onFormDataChange", function (_ref8) {
      var formData = _ref8.formData;
      return _this8.setState({
        formData: formData,
        shareURL: null
      });
    });

    _defineProperty(_assertThisInitialized(_this8), "onShare", function () {
      var _this8$state = _this8.state,
          formData = _this8$state.formData,
          schema = _this8$state.schema,
          uiSchema = _this8$state.uiSchema,
          liveSettings = _this8$state.liveSettings,
          errorSchema = _this8$state.errorSchema,
          theme = _this8$state.theme;
      var _document = document,
          _document$location = _document.location,
          origin = _document$location.origin,
          pathname = _document$location.pathname;

      try {
        var hash = btoa(JSON.stringify({
          formData: formData,
          schema: schema,
          uiSchema: uiSchema,
          theme: theme,
          liveSettings: liveSettings,
          errorSchema: errorSchema
        }));

        _this8.setState({
          shareURL: "" + origin + pathname + "#" + hash
        });
      } catch (err) {
        _this8.setState({
          shareURL: null
        });
      }
    });

    var _theme = "default"; // initialize state with Simple data sample

    var _samples$Simple = _samples.samples.Simple,
        _schema = _samples$Simple.schema,
        _uiSchema = _samples$Simple.uiSchema,
        _formData = _samples$Simple.formData,
        validate = _samples$Simple.validate;
    _this8.state = {
      form: false,
      schema: _schema,
      uiSchema: _uiSchema,
      formData: _formData,
      validate: validate,
      theme: _theme,
      subtheme: null,
      liveSettings: {
        validate: false,
        disable: false,
        omitExtraData: false,
        liveOmit: false
      },
      shareURL: null,
      FormComponent: (0, _core.withTheme)({})
    };
    return _this8;
  }

  var _proto5 = Playground.prototype;

  _proto5.componentDidMount = function componentDidMount() {
    var themes = this.props.themes;
    var theme = this.state.theme;
    var hash = document.location.hash.match(/#(.*)/);

    if (hash && typeof hash[1] === "string" && hash[1].length > 0) {
      try {
        this.load(JSON.parse(atob(hash[1])));
      } catch (err) {
        alert("Unable to load form setup data.");
      }
    } else {
      // initialize theme
      this.onThemeSelected(theme, themes[theme]);
      this.setState({
        form: true
      });
    }
  };

  _proto5.shouldComponentUpdate = function shouldComponentUpdate(nextProps, nextState) {
    return shouldRender(this, nextProps, nextState);
  };

  _proto5.render = function render() {
    var _this$state2 = this.state,
        schema = _this$state2.schema,
        uiSchema = _this$state2.uiSchema,
        formData = _this$state2.formData,
        extraErrors = _this$state2.extraErrors,
        liveSettings = _this$state2.liveSettings,
        validate = _this$state2.validate,
        theme = _this$state2.theme,
        subtheme = _this$state2.subtheme,
        FormComponent = _this$state2.FormComponent,
        ArrayFieldTemplate = _this$state2.ArrayFieldTemplate,
        ObjectFieldTemplate = _this$state2.ObjectFieldTemplate,
        transformErrors = _this$state2.transformErrors;
    var themes = this.props.themes;
    var templateProps = {};

    if (ArrayFieldTemplate) {
      templateProps.ArrayFieldTemplate = ArrayFieldTemplate;
    }

    if (ObjectFieldTemplate) {
      templateProps.ObjectFieldTemplate = ObjectFieldTemplate;
    }

    if (extraErrors) {
      templateProps.extraErrors = extraErrors;
    }
    /*
    return (
        <EditorBase>
             <EditorHeader>
            </EditorHeader>
            <EditorBody>
                 <EditorContent>
                    <Form
                        idPrefix="rjsf_options"
                        schema={liveSettingsSchema}
                        formData={liveSettings}
                        onChange={this.setLiveSettings}>
                        <div />
                    </Form>
                </EditorContent>
                <EditorPreview>
                 <Selector onSelected={this.load} />
                 <ThemeSelector
                    themes={themes}
                    theme={theme}
                    select={this.onThemeSelected}
                />
                   </EditorPreview>
             </EditorBody>
          </EditorBase>
    )*/


    return /*#__PURE__*/_react.default.createElement("div", {
      className: "container-fluid"
    }, /*#__PURE__*/_react.default.createElement("div", {
      className: "page-header"
    }, /*#__PURE__*/_react.default.createElement("h1", null, "react-jsonschema-form"), /*#__PURE__*/_react.default.createElement("div", {
      className: "row"
    }, /*#__PURE__*/_react.default.createElement("div", {
      className: "col-sm-8"
    }, /*#__PURE__*/_react.default.createElement(Selector, {
      onSelected: this.load
    })), /*#__PURE__*/_react.default.createElement("div", {
      className: "col-sm-2"
    }, /*#__PURE__*/_react.default.createElement(_core.default, {
      idPrefix: "rjsf_options",
      schema: liveSettingsSchema,
      formData: liveSettings,
      onChange: this.setLiveSettings
    }, /*#__PURE__*/_react.default.createElement("div", null))), /*#__PURE__*/_react.default.createElement("div", {
      className: "col-sm-2"
    }, /*#__PURE__*/_react.default.createElement(ThemeSelector, {
      themes: themes,
      theme: theme,
      select: this.onThemeSelected
    }), themes[theme].subthemes && /*#__PURE__*/_react.default.createElement(SubthemeSelector, {
      subthemes: themes[theme].subthemes,
      subtheme: subtheme,
      select: this.onSubthemeSelected
    }), /*#__PURE__*/_react.default.createElement(CopyLink, {
      shareURL: this.state.shareURL,
      onShare: this.onShare
    })))), /*#__PURE__*/_react.default.createElement("div", {
      className: "col-sm-7"
    }, /*#__PURE__*/_react.default.createElement(Editor, {
      title: "JSONSchema",
      code: toJson(schema),
      onChange: this.onSchemaEdited
    }), /*#__PURE__*/_react.default.createElement("div", {
      className: "row"
    }, /*#__PURE__*/_react.default.createElement("div", {
      className: "col-sm-6"
    }, /*#__PURE__*/_react.default.createElement(Editor, {
      title: "UISchema",
      code: toJson(uiSchema),
      onChange: this.onUISchemaEdited
    })), /*#__PURE__*/_react.default.createElement("div", {
      className: "col-sm-6"
    }, /*#__PURE__*/_react.default.createElement(Editor, {
      title: "formData",
      code: toJson(formData),
      onChange: this.onFormDataEdited
    }))), extraErrors && /*#__PURE__*/_react.default.createElement("div", {
      className: "row"
    }, /*#__PURE__*/_react.default.createElement("div", {
      className: "col"
    }, /*#__PURE__*/_react.default.createElement(Editor, {
      title: "extraErrors",
      code: toJson(extraErrors || {}),
      onChange: this.onExtraErrorsEdited
    })))), /*#__PURE__*/_react.default.createElement("div", {
      className: "col-sm-5"
    }, this.state.form && /*#__PURE__*/_react.default.createElement(_DemoFrame.default, {
      head: /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("link", {
        rel: "stylesheet",
        id: "theme",
        href: this.state.stylesheet || ""
      }), theme === "antd" && /*#__PURE__*/_react.default.createElement("div", {
        dangerouslySetInnerHTML: {
          __html: document.getElementById("antd-styles-iframe").contentDocument.head.innerHTML
        }
      })),
      style: {
        width: "100%",
        height: 1000,
        border: 0
      },
      theme: theme
    }, /*#__PURE__*/_react.default.createElement(FormComponent, _extends({}, templateProps, {
      liveValidate: liveSettings.validate,
      disabled: liveSettings.disable,
      omitExtraData: liveSettings.omitExtraData,
      liveOmit: liveSettings.liveOmit,
      schema: schema,
      uiSchema: uiSchema,
      formData: formData,
      onChange: this.onFormDataChange,
      noHtml5Validate: true,
      onSubmit: function onSubmit(_ref9, e) {
        var formData = _ref9.formData;
        console.log("submitted formData", formData);
        console.log("submit event", e);
      },
      fields: {
        geo: GeoPosition
      },
      validate: validate,
      onBlur: function onBlur(id, value) {
        return console.log("Touched " + id + " with value " + value);
      },
      onFocus: function onFocus(id, value) {
        return console.log("Focused " + id + " with value " + value);
      },
      transformErrors: transformErrors,
      onError: log("errors")
    })))), /*#__PURE__*/_react.default.createElement("div", {
      className: "col-sm-12"
    }, /*#__PURE__*/_react.default.createElement("p", {
      style: {
        textAlign: "center"
      }
    }, "Powered by", " ", /*#__PURE__*/_react.default.createElement("a", {
      href: "https://github.com/rjsf-team/react-jsonschema-form"
    }, "react-jsonschema-form"), ".", process.env.SHOW_NETLIFY_BADGE === "true" && /*#__PURE__*/_react.default.createElement("div", {
      style: {
        float: "right"
      }
    }, /*#__PURE__*/_react.default.createElement("a", {
      href: "https://www.netlify.com"
    }, /*#__PURE__*/_react.default.createElement("img", {
      src: "https://www.netlify.com/img/global/badges/netlify-color-accent.svg"
    }))))));
  };

  return Playground;
}(_react.Component);

var _default = Playground;
exports.default = _default;
module.exports = exports.default;
