// @kit-ui/schema/es/ColorSettings/ColorInput.js
import React from 'react';
import { CustomPicker } from 'react-color';
import { EditableInput } from 'react-color/lib/components/common';
import TextField from '@material-ui/core/TextField';
import color from 'react-color/lib/helpers/color';
import { makeStyles } from '@material-ui/core/styles';
var useStyles = makeStyles(function (theme) {
  return {
    root: {
      fontFamily: "Akkurat Mono, monospace",
      backgroundColor: function backgroundColor(props) {
        return props.backgroundColor;
      },
      color: function color(props) {
        return props.color;
      },
      "& input": {
        border: "none",
        backgroundColor: "transparent",
        color: "inherit",
        fontFamily: "inherit",
        fontSize: "inherit",
        lineHeight: "1",
        width: "100%",
        minHeight: theme.spacing(7),
        textAlign: "center"
      }
    }
  };
});

var ColorInput = function ColorInput(_ref) {
  var className = _ref.className,
      hex = _ref.hex,
      rgb = _ref.rgb,
      hsl = _ref.hsl,
      hsv = _ref.hsv,
      onChange = _ref.onChange;

  var _onChange = function _onChange(_ref2, e) {
    var hex = _ref2.hex;

    if (!color.isValidHex(hex)) {
      return false;
    }

    onChange && onChange({
      hex: hex
    });
  };

  var getContrastColor = function getContrastColor() {
    if (!color.isValidHex(hex)) {
      return false;
    }

    if (hex === 'transparent') {
      return 'rgba(0,0,0,0.4)';
    }

    var yiq = (rgb.r * 299 + rgb.g * 587 + rgb.b * 114) / 1000;
    return yiq >= 128 ? '#000' : '#fff';
  };

  var classes = useStyles({
    backgroundColor: hex,
    color: getContrastColor(hex)
  });
  return /*#__PURE__*/React.createElement("div", {
    className: className || classes.root
  }, /*#__PURE__*/React.createElement(EditableInput, {
    hideLabel: true,
    label: "hex",
    placeholder: "HEX",
    value: hex,
    onChange: _onChange
  }));
};

export default CustomPicker(ColorInput);
