// @kit-ui/schema/es/ColorSettings/ColorSwatches.js
import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Color from 'color';
import IconButton from "@material-ui/core/IconButton";
import ColorIcon from "@material-ui/icons/Lens";
var useStyles = makeStyles(function (theme) {
  return {
    swatches: {
      display: "flex",
      flexWrap: "wrap",
      padding: theme.spacing(.5)
    },
    icon: {
      //        backgroundColor: props => { return props.backgroundColor },
      color: function color(props) {
        return props.color;
      }
    }
  };
});

var SwatchButton = function SwatchButton(_ref) {
  var color = _ref.color,
      backgroundColor = _ref.backgroundColor,
      contrastColor = _ref.contrastColor,
      onClick = _ref.onClick;
  var classes = useStyles({
    backgroundColor: backgroundColor,
    color: color
  });
  return /*#__PURE__*/React.createElement(IconButton, {
    onClick: onClick
  }, /*#__PURE__*/React.createElement(ColorIcon, {
    className: classes.icon
  }));
};

var ColorSwatches = function ColorSwatches(_ref2) {
  var variant = _ref2.variant,
      backgroundColor = _ref2.backgroundColor,
      _ref2$colors = _ref2.colors,
      colors = _ref2$colors === void 0 ? [] : _ref2$colors,
      color = _ref2.color,
      onChange = _ref2.onChange;
  var classes = useStyles();

  var _onChange = function _onChange(hex, event) {
    onChange && onChange({
      hex: hex
    }, event);
  };

  if (!colors.length) {
    return false;
  }

  return /*#__PURE__*/React.createElement("div", {
    className: classes.swatches,
    "data-size": colors.length
  }, colors.map(function (color, index) {
    var primaryColor = Color(color);
    var contrastColor = primaryColor.isLight() && "black" || "white";
    return /*#__PURE__*/React.createElement(SwatchButton, {
      color: color,
      backgroundColor: backgroundColor,
      contrastColor: contrastColor,
      key: index,
      onClick: function onClick(event) {
        return _onChange(color, event);
      }
    });
  }));
};

export default ColorSwatches;
