// @kit-ui/admin/es/Widget/SearchWidget.js
import React, { useRef } from 'react';
import { WidgetSearch, WidgetScroller } from "./";
import { makeStyles } from '@material-ui/core/styles';
var useStyles = makeStyles(function (theme) {
  return {
    root: {
      position: "absolute",
      width: "100%",
      top: 0,
      right: 0,
      bottom: 0,
      left: 0
    },
    search: {
      display: "flex",
      position: "relative",
      backgroundColor: "transparent",
      color: "black",
      height: "36px",
      overflow: "hidden",
      fontFamily: "Akkurat, sans-serif",
      margin: "12px"
    },
    scroller: {
      position: "absolute",
      zIndex: 1,
      top: 0,
      right: 0,
      bottom: 0,
      left: 0
    },
    results: {
      paddingTop: theme.spacing(7),
      display: "block",
      width: "100%"
    },
    nohits: {
      fontFamily: "Akkurat, sans-serif",
      fontSize: "14px",
      padding: theme.spacing(1.5)
    }
  };
});

var SearchWidget = function SearchWidget(_ref) {
  var className = _ref.className,
      placeholder = _ref.placeholder,
      q = _ref.q,
      onSearch = _ref.onSearch,
      children = _ref.children;
  var classes = useStyles();
  return /*#__PURE__*/React.createElement("div", {
    className: className || classes.root
  }, /*#__PURE__*/React.createElement(WidgetSearch, {
    placeholder: placeholder,
    value: q,
    onChange: onSearch
  }), /*#__PURE__*/React.createElement(WidgetScroller, {
    className: classes.scroller
  }, children));
};

SearchWidget.defaultProps = {
  placeholder: "Search"
};
export default SearchWidget;
