// @kit-ui/admin/es/Widget/WidgetMedia.js
function _inheritsLoose(subClass, superClass) { subClass.prototype = Object.create(superClass.prototype); subClass.prototype.constructor = subClass; subClass.__proto__ = superClass; }

import React, { Component } from 'react';

var WidgetMedia = /*#__PURE__*/function (_Component) {
  _inheritsLoose(WidgetMedia, _Component);

  function WidgetMedia() {
    return _Component.apply(this, arguments) || this;
  }

  var _proto = WidgetMedia.prototype;

  _proto.render = function render() {
    var _this$props = this.props,
        mediaType = _this$props.mediaType,
        backgroundImage = _this$props.backgroundImage,
        imageUrl = _this$props.imageUrl;

    if (backgroundImage) {
      return /*#__PURE__*/React.createElement("figure", {
        className: "admin-widget__media",
        style: {
          backgroundImage: "url(" + backgroundImage + ")"
        }
      });
    }

    if (imageUrl) {
      return /*#__PURE__*/React.createElement("figure", {
        className: "admin-widget__media"
      }, /*#__PURE__*/React.createElement("img", {
        src: imageUrl
      }));
    }

    return false;
  };

  return WidgetMedia;
}(Component);

export default WidgetMedia;
