// @kit-ui/schema/lib/Media/MediaButtons.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _ButtonGroup = _interopRequireDefault(require("@material-ui/core/ButtonGroup"));

var _Button = _interopRequireDefault(require("@material-ui/core/Button"));

var _styles = require("@material-ui/core/styles");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var useStyles = (0, _styles.makeStyles)(function (theme) {
  return {
    buttongroup: {
      position: "absolute",
      top: 0,
      right: 0,
      bottom: 0,
      left: 0,
      display: "flex",
      alignItems: "center",
      justifyContent: "center",
      width: "100%",
      height: "100%",
      borderColor: "inherit",
      color: "inherit"
    },
    button: {
      fontFamily: "Akkurat, sans-serif",
      fontSize: "14px",
      fontWeight: "bold",
      textTransform: "none",
      letterSpacing: 0,
      borderRadius: 1,
      minWidth: theme.spacing(16),
      minHeight: theme.spacing(7),
      borderColor: "inherit",
      color: "inherit"
    }
  };
});

var MediaButtons = function MediaButtons(_ref) {
  var _ref$buttons = _ref.buttons,
      buttons = _ref$buttons === void 0 ? [] : _ref$buttons;
  var classes = useStyles();

  var MediaButton = function MediaButton(_ref2) {
    var title = _ref2.title,
        _ref2$color = _ref2.color,
        color = _ref2$color === void 0 ? "primary" : _ref2$color,
        _ref2$variant = _ref2.variant,
        variant = _ref2$variant === void 0 ? "outlined" : _ref2$variant,
        props = _objectWithoutPropertiesLoose(_ref2, ["title", "color", "variant"]);

    return /*#__PURE__*/_react.default.createElement(_Button.default, _extends({
      className: classes.button,
      variant: variant
    }, props), title);
  };

  return /*#__PURE__*/_react.default.createElement("footer", {
    className: classes.buttongroup
  }, buttons && buttons.map(function (button, index) {
    return /*#__PURE__*/_react.default.createElement(MediaButton, _extends({}, button, {
      key: index
    }));
  }));
};

var _default = MediaButtons;
exports.default = _default;
module.exports = exports.default;
