// @kit-ui/schema/lib/Media/MediaButtons.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _ButtonBase = _interopRequireDefault(require("@material-ui/core/ButtonBase"));

var _styles = require("@material-ui/core/styles");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var useStyles = (0, _styles.makeStyles)(function (theme) {
  return {
    mediaButtons: {
      position: "absolute",
      top: 0,
      right: 0,
      bottom: 0,
      left: 0,
      display: "flex",
      alignItems: "center",
      justifyContent: "center",
      width: "100%",
      height: "100%"
    },
    button: {
      "&.MuiButtonBase-root": {
        fontFamily: "Akkurat, sans-serif",
        fontSize: "14px",
        fontWeight: "bold",
        border: "1px solid",
        borderRadius: 1,
        minWidth: theme.spacing(12),
        minHeight: theme.spacing(7)
      }
    }
  };
});

var MediaButton = function MediaButton(_ref) {
  var title = _ref.title,
      button = _objectWithoutPropertiesLoose(_ref, ["title"]);

  var classes = useStyles();
  return /*#__PURE__*/_react.default.createElement(_ButtonBase.default, _extends({
    className: classes.button
  }, button), title);
};

var MediaButtons = function MediaButtons(_ref2) {
  var _ref2$buttons = _ref2.buttons,
      buttons = _ref2$buttons === void 0 ? [] : _ref2$buttons;
  var classes = useStyles();
  return /*#__PURE__*/_react.default.createElement("footer", {
    className: classes.mediaButtons
  }, buttons && buttons.map(function (button, index) {
    return /*#__PURE__*/_react.default.createElement(MediaButton, _extends({}, button, {
      key: index
    }));
  }));
};

var _default = MediaButtons;
exports.default = _default;
module.exports = exports.default;
