// @kit-ui/schema/lib/Media/MediaField.stories.js
"use strict";

exports.__esModule = true;
exports.Audio = exports.EmptyAudio = exports.Video = exports.EmptyVideo = exports.ImageFilters = exports.ImageAnnotations = exports.ImageFocalpoint = exports.ImageCropdata = exports.MediaArrayWithDialog = exports.MediaArray = exports.ImageModule = exports.Image = exports.EmptyImage = exports.MediaWithDialog = exports.EmptyMedia = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _MediaField = _interopRequireDefault(require("./MediaField"));

var _SchemaEditor = _interopRequireDefault(require("../SchemaEditor"));

var _formData = _interopRequireDefault(require("../data/media/image/formData"));

var _formData2 = _interopRequireDefault(require("../data/media/video/formData"));

var _formData3 = _interopRequireDefault(require("../data/media/audio/formData"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _default = {
  title: 'Fields/MediaField',
  component: _MediaField.default,
  args: {},
  argTypes: {}
};
exports.default = _default;

var Template = function Template(args) {
  return /*#__PURE__*/_react.default.createElement(_SchemaEditor.default, args);
};

var EmptyMedia = Template.bind({});
exports.EmptyMedia = EmptyMedia;
EmptyMedia.args = {
  schema: {
    type: "media"
  }
};
var MediaWithDialog = Template.bind({});
exports.MediaWithDialog = MediaWithDialog;
MediaWithDialog.args = {
  schema: {
    type: "object",
    "properties": {
      "media": {
        type: "media",
        mediaType: "image"
      }
    }
  },
  uiSchema: {
    "ui:field": "pageEditor",
    "ui:fieldset": ["content"],
    "content": {
      "ui:field": "pageContent",
      "ui:fieldset": ["media"],
      "media": {
        "ui:dialog": true
      }
    }
  }
};
var EmptyImage = Template.bind({});
exports.EmptyImage = EmptyImage;
EmptyImage.args = {
  schema: {
    type: "media",
    mediaType: "image"
  }
};
var Image = Template.bind({});
exports.Image = Image;
Image.args = {
  schema: {
    type: "media",
    mediaType: "image"
  },
  formData: {
    mediaId: 1,
    media: _formData.default
  }
};
var ImageModule = Template.bind({});
exports.ImageModule = ImageModule;
ImageModule.args = {
  schema: {
    type: "media",
    mediaType: "image"
  },
  uiSchema: {
    "ui:layout": "module",
    "ui:editable": true
  },
  formData: {
    mediaId: 1,
    media: _formData.default
  }
};
var MediaArray = Template.bind({});
exports.MediaArray = MediaArray;
MediaArray.args = {
  schema: {
    type: "array",
    items: {
      type: "media",
      mediaType: "image",
      cropdata: true
    }
  },
  uiSchema: {
    "items": {
      "ui:layout": "module",
      "ui:editable": true
    }
  },
  formData: [{
    mediaId: 1,
    media: _formData.default
  }, {
    mediaId: 2,
    media: _formData2.default
  }, {
    mediaId: 3,
    media: _formData3.default
  }]
};
var MediaArrayWithDialog = Template.bind({});
exports.MediaArrayWithDialog = MediaArrayWithDialog;
MediaArrayWithDialog.args = {
  schema: {
    "type": "object",
    "properties": {
      "array": {
        "type": "array",
        "items": {
          "type": "media",
          "mediaType": "image"
        }
      }
    }
  },
  uiSchema: {
    "ui:field": "pageEditor",
    "ui:fieldset": ["content"],
    "content": {
      "ui:field": "pageContent",
      "ui:fieldset": ["array"],
      "array": {
        "ui:sortable": true,
        "ui:dialog": true,
        "items": {
          "ui:layout": "module",
          "ui:editable": true
        }
      }
    }
  },
  formData: {
    array: [{
      mediaId: 1,
      media: _formData.default
    }, {
      mediaId: 2,
      media: _formData2.default
    }, {
      mediaId: 3,
      media: _formData3.default
    }]
  }
};
var ImageCropdata = Template.bind({});
exports.ImageCropdata = ImageCropdata;
ImageCropdata.args = {
  schema: {
    type: "media",
    mediaType: "image",
    cropdata: true
  },
  formData: {
    mediaId: 1,
    media: _formData.default
  }
};
var ImageFocalpoint = Template.bind({});
exports.ImageFocalpoint = ImageFocalpoint;
ImageFocalpoint.args = {
  schema: {
    type: "media",
    mediaType: "image",
    focalpoint: true
  },
  formData: {
    mediaId: 1,
    media: _formData.default
  }
};
var ImageAnnotations = Template.bind({});
exports.ImageAnnotations = ImageAnnotations;
ImageAnnotations.args = {
  schema: {
    type: "media",
    mediaType: "image",
    annotations: true
  },
  formData: {
    mediaId: 1,
    media: _formData.default
  }
};
var ImageFilters = Template.bind({});
exports.ImageFilters = ImageFilters;
ImageFilters.args = {
  schema: {
    type: "media",
    mediaType: "image",
    filters: ["brightness"]
  },
  formData: {
    mediaId: 1,
    media: _formData.default
  }
};
var EmptyVideo = Template.bind({});
exports.EmptyVideo = EmptyVideo;
EmptyVideo.args = {
  schema: {
    type: "media",
    mediaType: "video"
  }
};
var Video = Template.bind({});
exports.Video = Video;
Video.args = {
  schema: {
    type: "media",
    mediaType: "video"
  },
  formData: {
    mediaId: 1,
    media: _formData2.default
  }
};
var EmptyAudio = Template.bind({});
exports.EmptyAudio = EmptyAudio;
EmptyAudio.args = {
  schema: {
    type: "media",
    mediaType: "audio"
  }
};
var Audio = Template.bind({});
exports.Audio = Audio;
Audio.args = {
  schema: {
    type: "media",
    mediaType: "audio"
  },
  formData: {
    mediaId: 1,
    media: _formData3.default
  }
};
