// @kit-ui/core/lib/Dropdown/Dropdown.position.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireWildcard(require("react"));

var _styles = require("@material-ui/core/styles");

var _Grid = _interopRequireDefault(require("@material-ui/core/Grid"));

var _Button = _interopRequireDefault(require("@material-ui/core/Button"));

var _Dropdown = _interopRequireDefault(require("./Dropdown"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var useStyles = (0, _styles.makeStyles)(function (theme) {
  return {
    root: {},
    content: {
      width: 200,
      height: 200
    }
  };
});

var DropdownPosition = function DropdownPosition(props) {
  var _useState = (0, _react.useState)(null),
      anchorEl = _useState[0],
      setAnchorEl = _useState[1];

  var _useState2 = (0, _react.useState)(false),
      expanded = _useState2[0],
      setExpanded = _useState2[1];

  var _useState3 = (0, _react.useState)(),
      placement = _useState3[0],
      setPlacement = _useState3[1];

  var classes = useStyles();

  var handleClick = function handleClick(newPlacement) {
    return function (event) {
      setAnchorEl(event.currentTarget);
      setExpanded(function (prev) {
        return placement !== newPlacement || !prev;
      });
      setPlacement(newPlacement);
    };
  };

  return /*#__PURE__*/_react.default.createElement("div", {
    className: classes.root
  }, /*#__PURE__*/_react.default.createElement(_Dropdown.default, _extends({}, props, {
    expanded: expanded,
    anchorEl: anchorEl,
    placement: placement
  }), /*#__PURE__*/_react.default.createElement("div", {
    className: classes.content
  }, JSON.stringify(props, 0, 2))), /*#__PURE__*/_react.default.createElement(_Grid.default, {
    container: true,
    justify: "center"
  }, /*#__PURE__*/_react.default.createElement(_Grid.default, {
    item: true
  }, /*#__PURE__*/_react.default.createElement(_Button.default, {
    onClick: handleClick('top-start')
  }, "top-start"), /*#__PURE__*/_react.default.createElement(_Button.default, {
    onClick: handleClick('top')
  }, "top"), /*#__PURE__*/_react.default.createElement(_Button.default, {
    onClick: handleClick('top-end')
  }, "top-end"))), /*#__PURE__*/_react.default.createElement(_Grid.default, {
    container: true,
    justify: "center"
  }, /*#__PURE__*/_react.default.createElement(_Grid.default, {
    item: true,
    xs: 6
  }, /*#__PURE__*/_react.default.createElement(_Button.default, {
    onClick: handleClick('left-start')
  }, "left-start"), /*#__PURE__*/_react.default.createElement("br", null), /*#__PURE__*/_react.default.createElement(_Button.default, {
    onClick: handleClick('left')
  }, "left"), /*#__PURE__*/_react.default.createElement("br", null), /*#__PURE__*/_react.default.createElement(_Button.default, {
    onClick: handleClick('left-end')
  }, "left-end")), /*#__PURE__*/_react.default.createElement(_Grid.default, {
    item: true,
    container: true,
    xs: 6,
    alignItems: "flex-end",
    direction: "column"
  }, /*#__PURE__*/_react.default.createElement(_Grid.default, {
    item: true
  }, /*#__PURE__*/_react.default.createElement(_Button.default, {
    onClick: handleClick('right-start')
  }, "right-start")), /*#__PURE__*/_react.default.createElement(_Grid.default, {
    item: true
  }, /*#__PURE__*/_react.default.createElement(_Button.default, {
    onClick: handleClick('right')
  }, "right")), /*#__PURE__*/_react.default.createElement(_Grid.default, {
    item: true
  }, /*#__PURE__*/_react.default.createElement(_Button.default, {
    onClick: handleClick('right-end')
  }, "right-end")))), /*#__PURE__*/_react.default.createElement(_Grid.default, {
    container: true,
    justify: "center"
  }, /*#__PURE__*/_react.default.createElement(_Grid.default, {
    item: true
  }, /*#__PURE__*/_react.default.createElement(_Button.default, {
    onClick: handleClick('bottom-start')
  }, "bottom-start"), /*#__PURE__*/_react.default.createElement(_Button.default, {
    onClick: handleClick('bottom')
  }, "bottom"), /*#__PURE__*/_react.default.createElement(_Button.default, {
    onClick: handleClick('bottom-end')
  }, "bottom-end"))));
};

var _default = DropdownPosition;
exports.default = _default;
module.exports = exports.default;
