// @kit-ui/admin/lib/Masonry/Masonry.stories.js
"use strict";

exports.__esModule = true;
exports.FourColumns = exports.ThreeColumns = exports.TwoColumns = exports.Default = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _Masonry = _interopRequireDefault(require("./Masonry"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _default = {
  title: 'Layout/Masonry',
  component: _Masonry.default,
  args: {},
  argTypes: {}
};
exports.default = _default;

var Template = function Template(args) {
  return /*#__PURE__*/_react.default.createElement(_Masonry.default, args, /*#__PURE__*/_react.default.createElement("img", {
    src: "https://w3bits.com/files/img/cherry-plant.jpg"
  }), /*#__PURE__*/_react.default.createElement("img", {
    src: "https://w3bits.com/files/img/blueberries.jpg"
  }), /*#__PURE__*/_react.default.createElement("img", {
    src: "https://w3bits.com/files/img/cherry-plant.jpg"
  }), /*#__PURE__*/_react.default.createElement("img", {
    src: "https://w3bits.com/files/img/blueberries.jpg"
  }), /*#__PURE__*/_react.default.createElement("img", {
    src: "https://w3bits.com/files/img/cherry-plant.jpg"
  }), /*#__PURE__*/_react.default.createElement("img", {
    src: "https://w3bits.com/files/img/blueberries.jpg"
  }), /*#__PURE__*/_react.default.createElement("img", {
    src: "https://w3bits.com/files/img/cherry-plant.jpg"
  }), /*#__PURE__*/_react.default.createElement("img", {
    src: "https://w3bits.com/files/img/blueberries.jpg"
  }), /*#__PURE__*/_react.default.createElement("img", {
    src: "https://w3bits.com/files/img/cherry-plant.jpg"
  }), /*#__PURE__*/_react.default.createElement("img", {
    src: "https://w3bits.com/files/img/blueberries.jpg"
  }));
};

var Default = Template.bind({});
exports.Default = Default;
Default.args = {};
var TwoColumns = Template.bind({});
exports.TwoColumns = TwoColumns;
TwoColumns.args = {
  columns: 2
};
var ThreeColumns = Template.bind({});
exports.ThreeColumns = ThreeColumns;
ThreeColumns.args = {
  columns: 3
};
var FourColumns = Template.bind({});
exports.FourColumns = FourColumns;
FourColumns.args = {
  columns: 4
};
