// @kit-ui/schema/es/SchemaEditor/SchemaEditor.stories.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import React from 'react';
import SchemaEditor from "./SchemaEditor";
import article from "../data/article";
import articleSection from "../data/article/section";
import kpSchemasArticle from "../data/kp/article";
import imageData from "../data/media/image/formData";
import schemasByName from "../schemasByName";
export default {
  title: 'Schema/SchemaEditor',
  component: SchemaEditor,
  args: {},
  argTypes: {}
};

var Template = function Template(args) {
  return /*#__PURE__*/React.createElement(SchemaEditor, args);
};

export var Article = Template.bind({});
Article.args = _extends({}, article);
export var Section = Template.bind({});
Section.args = _extends({}, articleSection);
export var KpArticle = Template.bind({});
KpArticle.args = _extends({}, kpSchemasArticle);
export var MediaImage = Template.bind({});
MediaImage.args = _extends({}, schemasByName["media/image"], {
  formData: imageData
});
export var MediaVideo = Template.bind({});
MediaVideo.args = _extends({}, schemasByName["media/video"]);
export var MediaAudio = Template.bind({});
MediaAudio.args = _extends({}, schemasByName["media/audio"]);
export var MediaYoutube = Template.bind({});
MediaYoutube.args = _extends({}, schemasByName["media/video/youtube"]);
export var MediaVimeo = Template.bind({});
MediaVimeo.args = _extends({}, schemasByName["media/video/vimeo"]);
export var MediaSketchfab = Template.bind({});
MediaSketchfab.args = _extends({}, schemasByName["media/application/sketchfab"]);
