// @kit-ui/admin/lib/Editor/EditorHeader.stories.js
"use strict";

exports.__esModule = true;
exports.CollapsibleExpanded = exports.Collapsible = exports.ParentsTitle = exports.ParentsUntitled = exports.Default = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _EditorHeader = _interopRequireDefault(require("./EditorHeader"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _default = {
  title: 'Components/EditorHeader',
  component: _EditorHeader.default,
  args: {},
  argTypes: {}
};
exports.default = _default;

var Template = function Template(args) {
  return /*#__PURE__*/_react.default.createElement(_EditorHeader.default, args);
};

var Default = Template.bind({});
exports.Default = Default;
Default.args = {
  title: "Document",
  parents: [{
    "title": "Grandparent"
  }, {
    "title": "Parent"
  }],
  primaryAction: {
    title: "Save"
  },
  status: "draft",
  statusDate: "1999-05-26"
};
var ParentsUntitled = Template.bind({});
exports.ParentsUntitled = ParentsUntitled;
ParentsUntitled.args = {
  untitled: "No title",
  parents: [{
    "title": "Grandparent"
  }, {
    "title": "Parent"
  }]
};
var ParentsTitle = Template.bind({});
exports.ParentsTitle = ParentsTitle;
ParentsTitle.args = {
  untitled: "No title",
  title: "Document title",
  parents: [{
    "title": "Grandparent"
  }, {
    "title": "Parent"
  }]
};
var Collapsible = Template.bind({});
exports.Collapsible = Collapsible;
Collapsible.args = {
  untitled: "No title",
  title: "Document title",
  collapsible: true,
  parents: [{
    "title": "Grandparent"
  }, {
    "title": "Parent"
  }]
};
var CollapsibleExpanded = Template.bind({});
exports.CollapsibleExpanded = CollapsibleExpanded;
CollapsibleExpanded.args = {
  untitled: "No title",
  title: "Document title",
  collapsible: true,
  expanded: true,
  parents: [{
    "title": "Grandparent"
  }, {
    "title": "Parent"
  }]
};
