// @kit-ui/admin/lib/Editor/EditorToogle.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _IconButton = _interopRequireDefault(require("@material-ui/core/IconButton"));

var _MoreHoriz = _interopRequireDefault(require("@material-ui/icons/MoreHoriz"));

var _styles = require("@material-ui/core/styles");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var useStyles = (0, _styles.makeStyles)(function (theme) {
  return {
    root: {
      transition: ".125s ease-in-out",
      transform: "rotate(0deg)",
      "&[aria-expanded=true]": {
        transition: ".125s ease-in-out",
        transform: "rotate(90deg)"
      }
    }
  };
});

var EditorToggle = function EditorToggle(_ref) {
  var expanded = _ref.expanded,
      onClick = _ref.onClick;
  var classes = useStyles();
  return /*#__PURE__*/_react.default.createElement(_IconButton.default, {
    className: classes.root,
    color: "inherit",
    "aria-label": "toggle",
    "aria-expanded": expanded,
    onClick: onClick
  }, /*#__PURE__*/_react.default.createElement(_MoreHoriz.default, null));
};

var _default = EditorToggle;
exports.default = _default;
module.exports = exports.default;
