// @kit-ui/admin/lib/Editor/EditorOverlay.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _Paper = _interopRequireDefault(require("@material-ui/core/Paper"));

var _styles = require("@material-ui/core/styles");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var useStyles = (0, _styles.makeStyles)(function (theme) {
  return {
    overlay: {
      position: function position(props) {
        return props.position;
      },
      zIndex: 2000,
      backgroundColor: "rgba(0,0,0,.25)",
      top: 0,
      right: 0,
      bottom: 0,
      left: 0,
      transition: ".25s ease-out",
      opacity: 0,
      pointerEvents: "none",
      "&[aria-expanded=true]": {
        pointerEvents: "all",
        opacity: 1
      }
    },
    content: {
      position: function position(props) {
        return props.position;
      },
      top: 0,
      right: 0,
      bottom: 0,
      left: 0,
      backgroundColor: theme.palette.background.paper,
      margin: theme.spacing(2)
    }
  };
});

var EditorOverlay = function EditorOverlay(_ref) {
  var className = _ref.className,
      _ref$position = _ref.position,
      position = _ref$position === void 0 ? "absolute" : _ref$position,
      _ref$elevation = _ref.elevation,
      elevation = _ref$elevation === void 0 ? 12 : _ref$elevation,
      expanded = _ref.expanded,
      children = _ref.children;
  var classes = useStyles({
    position: position
  });
  return /*#__PURE__*/_react.default.createElement("aside", {
    className: classes.overlay,
    "aria-expanded": expanded
  }, /*#__PURE__*/_react.default.createElement(_Paper.default, {
    className: className || classes.content,
    elevation: elevation,
    square: true
  }, children));
};

EditorOverlay.defaultProps = {
  position: "absolute",
  expanded: false
};
var _default = EditorOverlay;
exports.default = _default;
module.exports = exports.default;
