// @kit-ui/admin/lib/Editor/EditorBase.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _Paper = _interopRequireDefault(require("@material-ui/core/Paper"));

var _styles = require("@material-ui/core/styles");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var useStyles = (0, _styles.makeStyles)(function (theme) {
  return {
    root: {
      backgroundColor: "inherit",
      position: "absolute",
      top: 0,
      right: 0,
      bottom: 0,
      left: 0
    }
  };
});

var EditorBase = function EditorBase(_ref) {
  var className = _ref.className,
      _ref$elevation = _ref.elevation,
      elevation = _ref$elevation === void 0 ? 1 : _ref$elevation,
      _ref$expanded = _ref.expanded,
      expanded = _ref$expanded === void 0 ? true : _ref$expanded,
      children = _ref.children;
  var classes = useStyles();

  if (!elevation) {
    return /*#__PURE__*/_react.default.createElement("div", {
      className: className || classes.root,
      "aria-expanded": expanded
    }, children);
  }

  return /*#__PURE__*/_react.default.createElement(_Paper.default, {
    elevation: elevation,
    square: true,
    className: className || classes.root,
    "aria-expanded": expanded
  }, children);
};

var _default = EditorBase;
exports.default = _default;
module.exports = exports.default;
