// @kit-ui/admin/lib/Editor/EditorLayout.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _EditorBase = _interopRequireDefault(require("./EditorBase"));

var _EditorHeader = _interopRequireDefault(require("./EditorHeader"));

var _EditorBody = _interopRequireDefault(require("./EditorBody"));

var _EditorContent = _interopRequireDefault(require("./EditorContent"));

var _EditorSidebar = _interopRequireDefault(require("./EditorSidebar"));

var _EditorPreview = _interopRequireDefault(require("./EditorPreview"));

var _styles = require("@material-ui/core/styles");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var useStyles = (0, _styles.makeStyles)(function (theme) {
  return {
    app: {
      position: "absolute",
      backgroundColor: theme.palette.background.default,
      zIndex: 1,
      top: 0,
      right: 0,
      left: 0,
      bottom: 0,
      overflow: "hidden",
      display: "flex",
      flexDirection: "column"
    },
    header: {
      position: "relative",
      height: theme.spacing(8),
      overflow: "hidden",
      marginTop: theme.spacing(-8),
      transition: ".125s ease-out",
      "&[aria-expanded=true]": {
        marginTop: 0
      }
    }
  };
});
/** AppLayout is a wrapper for all apps using AppHeader */

var EditorLayout = function EditorLayout(_ref) {
  var _ref$app = _ref.app,
      app = _ref$app === void 0 ? {} : _ref$app,
      _ref$theme = _ref.theme,
      theme = _ref$theme === void 0 ? {} : _ref$theme,
      _ref$header = _ref.header,
      header = _ref$header === void 0 ? {} : _ref$header,
      _ref$sidebar = _ref.sidebar,
      sidebar = _ref$sidebar === void 0 ? {} : _ref$sidebar,
      _ref$search = _ref.search,
      search = _ref$search === void 0 ? undefined : _ref$search,
      _ref$subview = _ref.subview,
      subview = _ref$subview === void 0 ? undefined : _ref$subview,
      menu = _ref.menu,
      _ref$menuByUrl = _ref.menuByUrl,
      menuByUrl = _ref$menuByUrl === void 0 ? {} : _ref$menuByUrl,
      preview = _ref.preview,
      children = _ref.children;

  if (!theme && app.theme) {
    theme = app.theme;
  }

  header = _extends({
    title: app && app.title,
    subtitle: app && app.subtitle
  }, header, {
    subview: subview,
    menu: menu,
    search: search
  });
  sidebar = _extends({}, sidebar, {
    primaryAction: app.primaryAction,
    menu: app.menu,
    menuByUrl: menuByUrl
  });
  var classes = useStyles({
    header: header
  });
  return /*#__PURE__*/_react.default.createElement(_EditorBase.default, null, /*#__PURE__*/_react.default.createElement(_EditorHeader.default, null), /*#__PURE__*/_react.default.createElement(_EditorBody.default, null, /*#__PURE__*/_react.default.createElement(_EditorSidebar.default, sidebar), /*#__PURE__*/_react.default.createElement(_EditorContent.default, null, children), preview && /*#__PURE__*/_react.default.createElement(_EditorPreview.default, null, preview && preview.children)));
};

EditorLayout.defaultProps = {
  app: {},
  header: {
    expanded: true
  }
};
EditorLayout.propTypes = process.env.NODE_ENV !== "production" ? {
  /** App properties */
  app: _propTypes.default.shape({
    uniqueId: _propTypes.default.string,
    title: _propTypes.default.string
  }),

  /** Header properties */
  header: _propTypes.default.shape({
    expanded: _propTypes.default.bool
  }),

  /** Search properties */
  search: _propTypes.default.shape({
    expanded: _propTypes.default.bool
  }),

  /** Subview properties */
  subview: _propTypes.default.shape({
    expanded: _propTypes.default.bool
  })
} : {};
var _default = EditorLayout;
exports.default = _default;
module.exports = exports.default;
