// @kit-ui/admin/lib/AppSearch/AppSearch.stories.js
"use strict";

exports.__esModule = true;
exports.GrowingExpanded = exports.GrowingCollapsed = exports.Collapsed = exports.Default = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _AppSearch = _interopRequireDefault(require("./AppSearch"));

var _AppSearch2 = _interopRequireDefault(require("./AppSearch.example"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _default = {
  title: 'App/AppSearch',
  component: _AppSearch.default,
  args: {
    placeholder: "Search app"
  },
  argTypes: {
    collapsible: {
      control: 'boolean'
    },
    expanded: {
      control: 'boolean'
    }
  }
};
exports.default = _default;

var Template = function Template(args) {
  return /*#__PURE__*/_react.default.createElement(_AppSearch2.default, args);
};

var Default = Template.bind({});
exports.Default = Default;
var Collapsed = Template.bind({});
exports.Collapsed = Collapsed;
Collapsed.args = {
  expanded: false
};
var GrowingCollapsed = Template.bind({});
exports.GrowingCollapsed = GrowingCollapsed;
GrowingCollapsed.args = {
  variant: "growing",
  expanded: false
};
var GrowingExpanded = Template.bind({});
exports.GrowingExpanded = GrowingExpanded;
GrowingExpanded.args = {
  variant: "growing",
  expanded: true
};
