// @kit-ui/admin/lib/AppSearch/AppSearchDefault.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _ButtonBase = _interopRequireDefault(require("@material-ui/core/ButtonBase"));

var _IconButton = _interopRequireDefault(require("@material-ui/core/IconButton"));

var _Close = _interopRequireDefault(require("@material-ui/icons/Close"));

var _Search = _interopRequireDefault(require("@material-ui/icons/Search"));

var _styles = require("@material-ui/core/styles");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

var useStyles = (0, _styles.makeStyles)(function (theme) {
  return {
    search: {
      display: "flex",
      position: "relative",
      backgroundColor: "white",
      color: "black",
      width: "100%",
      height: "64px",
      overflow: "hidden",
      "& $field": {
        display: "none"
      },
      "&[aria-expanded=true] & $field": {
        display: "flex"
      }
    },
    field: {},
    input: {
      backgroundColor: "transparent",
      flexBasis: "100%",
      height: "100%",
      paddingLeft: theme.spacing(8),
      paddingRight: theme.spacing(8),
      fontFamily: "Akkurat, sans-serif",
      fontSize: "18px",
      lineHeight: "1",
      color: "inherit",
      border: "none",
      outline: "none",
      "&:focus + *": {
        opacity: 1
      }
    },
    toggleButton: {
      position: "absolute",
      zIndex: "2",
      top: "0",
      left: "0",
      margin: theme.spacing(1),
      "&[disabled]": {
        color: theme.palette.text.secondary
      }
    },
    resetButton: {
      position: "absolute",
      zIndex: "2",
      top: "0",
      right: "0",
      margin: theme.spacing(1)
    }
  };
});

var ButtonReset = function ButtonReset(_ref) {
  var className = _ref.className,
      onClick = _ref.onClick;
  return /*#__PURE__*/_react.default.createElement(_IconButton.default, {
    className: className,
    onClick: onClick
  }, /*#__PURE__*/_react.default.createElement(_Close.default, null));
};

var ButtonToggle = function ButtonToggle(_ref2) {
  var className = _ref2.className,
      disabled = _ref2.disabled,
      onClick = _ref2.onClick;
  return /*#__PURE__*/_react.default.createElement(_IconButton.default, {
    disableRipple: true,
    className: className,
    disabled: disabled,
    onClick: onClick
  }, /*#__PURE__*/_react.default.createElement(_Search.default, null));
};

var NavSearch = function NavSearch(_ref3) {
  var className = _ref3.className,
      variant = _ref3.variant,
      _ref3$expanded = _ref3.expanded,
      expanded = _ref3$expanded === void 0 ? true : _ref3$expanded,
      _ref3$placeholder = _ref3.placeholder,
      placeholder = _ref3$placeholder === void 0 ? "Search" : _ref3$placeholder,
      q = _ref3.q,
      onFocus = _ref3.onFocus,
      onChange = _ref3.onChange,
      onToggle = _ref3.onToggle,
      onReset = _ref3.onReset;

  var _useState = (0, _react.useState)(q),
      value = _useState[0],
      setValue = _useState[1];

  var inputRef = (0, _react.useRef)(null);
  var classes = useStyles();

  var handleToggle = function handleToggle(event) {
    onToggle && onToggle(event);

    if (!expanded) {
      inputRef.current.focus();
    }
  };

  var handleChange = function handleChange(event) {
    onChange && onChange(event.target.value, event) || setValue(event.target.value);
    onFocus && onFocus(event);
  };

  var handleReset = function handleReset(event) {
    setValue("");

    if (onReset) {
      onReset(event);
    } else if (onChange) {
      onChange("", event);
    }
  };

  (0, _react.useEffect)(function () {
    setValue(q);
  }, [q]);
  return /*#__PURE__*/_react.default.createElement("div", {
    className: className || classes.search,
    "aria-expanded": expanded
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: classes.field
  }, /*#__PURE__*/_react.default.createElement(ButtonToggle, {
    className: classes.toggleButton,
    onClick: handleToggle,
    disabled: !onToggle && true || false
  }), /*#__PURE__*/_react.default.createElement("input", {
    ref: inputRef,
    className: classes.input,
    type: "text",
    value: value,
    placeholder: placeholder,
    onChange: handleChange
  }), /*#__PURE__*/_react.default.createElement(ButtonReset, {
    className: classes.resetButton,
    onClick: handleReset
  })));
};

NavSearch.propTypes = process.env.NODE_ENV !== "production" ? {
  /** Placeholder */
  placeholder: _propTypes.default.string,

  /** Query */
  q: _propTypes.default.string,

  /** onChange */
  onChange: _propTypes.default.func,

  /** onToggle */
  onToggle: _propTypes.default.func,

  /** onReset */
  onReset: _propTypes.default.func
} : {};
var _default = NavSearch;
exports.default = _default;
module.exports = exports.default;
