// @kit-ui/admin/lib/AppSearch/AppSearchBase.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _ButtonBase = _interopRequireDefault(require("@material-ui/core/ButtonBase"));

var _IconButton = _interopRequireDefault(require("@material-ui/core/IconButton"));

var _Close = _interopRequireDefault(require("@material-ui/icons/Close"));

var _Search = _interopRequireDefault(require("@material-ui/icons/Search"));

var _styles = require("@material-ui/core/styles");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var useStyles = (0, _styles.makeStyles)(function (theme) {
  var _root;

  return {
    root: (_root = {
      display: "flex",
      position: "relative",
      backgroundColor: "white"
    }, _root["backgroundColor"] = "#ccc", _root.color = "black", _root.width = "100%", _root.height = "64px", _root.overflow = "hidden", _root),
    searchInput: {
      backgroundColor: "transparent",
      flexBasis: "100%",
      height: "100%",
      paddingLeft: theme.spacing(8),
      paddingRight: theme.spacing(8),
      fontFamily: "Akkurat, sans-serif",
      fontSize: "18px",
      lineHeight: "1",
      color: "inherit",
      border: "none",
      outline: "none",
      "&:focus + *": {
        opacity: 1
      }
    },
    buttonToggle: {
      position: "absolute",
      zIndex: "2",
      top: "0",
      left: "0",
      margin: theme.spacing(1),
      "&[disabled]": {
        color: theme.palette.text.secondary
      }
    },
    buttonReset: {
      position: "absolute",
      zIndex: "2",
      top: "0",
      right: "0",
      margin: theme.spacing(1)
    }
  };
});

var ButtonReset = function ButtonReset(_ref) {
  var className = _ref.className,
      onClick = _ref.onClick;
  return /*#__PURE__*/_react.default.createElement(_IconButton.default, {
    className: className,
    onClick: onClick
  }, /*#__PURE__*/_react.default.createElement(_Close.default, null));
};

var ButtonToggle = function ButtonToggle(_ref2) {
  var className = _ref2.className,
      disabled = _ref2.disabled,
      onClick = _ref2.onClick;
  return /*#__PURE__*/_react.default.createElement(_IconButton.default, {
    disableRipple: true,
    className: className,
    disabled: disabled,
    onClick: onClick
  }, /*#__PURE__*/_react.default.createElement(_Search.default, null));
};

var AppSearchBase = function AppSearchBase(_ref3) {
  var className = _ref3.className,
      expanded = _ref3.expanded,
      variant = _ref3.variant,
      _ref3$placeholder = _ref3.placeholder,
      placeholder = _ref3$placeholder === void 0 ? "Search" : _ref3$placeholder,
      value = _ref3.value,
      onFocus = _ref3.onFocus,
      onChange = _ref3.onChange,
      onToggle = _ref3.onToggle,
      onReset = _ref3.onReset,
      props = _objectWithoutPropertiesLoose(_ref3, ["className", "expanded", "variant", "placeholder", "value", "onFocus", "onChange", "onToggle", "onReset"]);

  var inputRef = (0, _react.useRef)(null);
  var classes = useStyles();
  return /*#__PURE__*/_react.default.createElement("div", {
    className: className || classes.root,
    "data-variant": variant,
    "aria-expanded": expanded
  }, /*#__PURE__*/_react.default.createElement(ButtonToggle, {
    className: classes.buttonToggle,
    onClick: onToggle,
    disabled: !onToggle && true || false
  }), /*#__PURE__*/_react.default.createElement("input", {
    ref: inputRef,
    className: classes.searchInput,
    type: "text",
    value: value,
    placeholder: placeholder,
    onChange: onChange
  }), /*#__PURE__*/_react.default.createElement(ButtonReset, {
    className: classes.buttonReset,
    onClick: onReset
  }));
};

AppSearchBase.propTypes = process.env.NODE_ENV !== "production" ? {
  /** Expanded */
  expanded: _propTypes.default.bool,

  /** Placeholder */
  placeholder: _propTypes.default.string,

  /** Query */
  q: _propTypes.default.string,

  /** onChange */
  onChange: _propTypes.default.func,

  /** onToggle */
  onToggle: _propTypes.default.func,

  /** onReset */
  onReset: _propTypes.default.func
} : {};
var _default = AppSearchBase;
exports.default = _default;
module.exports = exports.default;
