// @kit-ui/schema/lib/CustomLayout/CustomLayout.stories.js
"use strict";

exports.__esModule = true;
exports.XsmallXlarge = exports.Default = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _SchemaEditor = _interopRequireDefault(require("../SchemaEditor/"));

var _CustomLayout = _interopRequireDefault(require("./CustomLayout"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _default = {
  title: 'Layouts/CustomLayouts/CustomLayout',
  component: _SchemaEditor.default,
  args: {
    schema: {
      type: "object",
      properties: {
        size: {
          "type": "string",
          "enum": ["s", "m", "l"]
        }
      }
    },
    uiSchema: {
      "ui:layout": _CustomLayout.default,
      "ui:options": {
        "editable": true
      },
      "ui:settings": ["size"],
      "size": {
        "ui:widget": "list"
      }
    },
    formData: {}
  },
  argTypes: {}
};
exports.default = _default;

var Template = function Template(args) {
  return /*#__PURE__*/_react.default.createElement(_SchemaEditor.default, args);
};

var Default = Template.bind({});
exports.Default = Default;
Default.args = {};
var XsmallXlarge = Template.bind({});
exports.XsmallXlarge = XsmallXlarge;
XsmallXlarge.args = {
  schema: {
    type: "object",
    properties: {
      size: {
        "type": "string",
        "enum": ["xs", "s", "s-m", "m-s", "m", "m-l", "l-m", "l", "xl"]
      }
    }
  }
};
