// @kit-ui/admin/es/App/AppBase.js
import React, { useState, useEffect } from 'react';
import { ThemeProvider } from '@material-ui/core/styles';
import { makeStyles } from '@material-ui/core/styles';
import { createAppTheme } from "./utils/createAppTheme";
var useStyles = makeStyles(function (theme) {
  return {
    root: {
      position: "absolute",
      backgroundColor: theme.palette.background.default,
      zIndex: 1,
      top: 0,
      right: 0,
      left: 0,
      bottom: 0,
      overflow: "hidden"
    }
  };
});
/** AppBase sets a backgroundColor and supplies a theme to the application */

var AppBase = function AppBase(_ref) {
  var className = _ref.className,
      theme = _ref.theme,
      children = _ref.children;
  var classes = useStyles();
  return /*#__PURE__*/React.createElement(ThemeProvider, {
    theme: createAppTheme(theme)
  }, /*#__PURE__*/React.createElement("main", {
    className: className || classes.root
  }, children));
};

AppBase.defaultProps = {
  /** MUI theme or primaryColor */
  theme: {
    primaryColor: "blue"
  }
};
export default AppBase;
