// @kit-ui/admin/es/App/AppDrawer.js
import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
var useStyles = makeStyles(function (theme) {
  var _ariaExpandedTrue, _ariaExpandedTrue2;

  return {
    root: {
      position: "absolute",
      backgroundColor: theme.palette.background.default,
      zIndex: 1,
      top: 0,
      right: "auto",
      left: 0,
      bottom: 0,
      width: function width(props) {
        return props.width;
      },
      marginLeft: function marginLeft(props) {
        return "-" + props.width + "px";
      },
      transition: ".125s ease-out",
      "&[aria-expanded=true]": (_ariaExpandedTrue = {}, _ariaExpandedTrue[theme.breakpoints.up('sm')] = {
        marginLeft: 0
      }, _ariaExpandedTrue),
      "&[aria-expanded=true] + *": (_ariaExpandedTrue2 = {}, _ariaExpandedTrue2[theme.breakpoints.up('sm')] = {
        marginLeft: function marginLeft(props) {
          return props.width;
        }
      }, _ariaExpandedTrue2)
    }
  };
});

var AppDrawer = function AppDrawer(_ref) {
  var _ref$expanded = _ref.expanded,
      expanded = _ref$expanded === void 0 ? true : _ref$expanded,
      _ref$width = _ref.width,
      width = _ref$width === void 0 ? 224 : _ref$width,
      children = _ref.children;
  var classes = useStyles({
    width: width
  });
  return /*#__PURE__*/React.createElement("aside", {
    className: classes.root,
    "aria-expanded": expanded
  }, children);
};

export default AppDrawer;
