// @kit-ui/admin/es/Module/ExpandedModule.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import ModuleFormat from "./ModuleFormat";
import ModuleBase from "./ModuleBase";
import ModuleHeader from "./ModuleHeader";
import ModuleBody from "./ModuleBody";
import ModuleFooter from "./ModuleFooter";
import ModuleButtons from "./ModuleButtons";
import { makeStyles } from '@material-ui/core/styles';
var useStyles = makeStyles(function (theme) {
  var _header;

  return {
    module: {
      position: "relative",
      zIndex: 2,
      "& + *": {
        zIndex: 1
      },
      display: "flex",
      flexDirection: "column",
      minHeight: function minHeight(props) {
        return props.minHeight;
      }
    },
    header: (_header = {
      position: "absolute",
      zIndex: 2,
      top: 0,
      right: 0,
      bottom: "auto",
      left: 0,
      display: "flex"
    }, _header["zIndex"] = 2, _header.alignItems = "center", _header.justifyContent = "flex-start", _header.height = theme.spacing(6), _header.marginLeft = theme.spacing(1), _header.marginRight = theme.spacing(1), _header.color = "inherit", _header["& + *"] = {
      zIndex: 1,
      marginTop: theme.spacing(6)
    }, _header),
    body: {
      minHeight: function minHeight(props) {
        return props.minHeight;
      },
      marginTop: function marginTop(props) {
        return props.minHeight && 0;
      },
      paddingRight: theme.spacing(2),
      paddingBottom: theme.spacing(2),
      paddingLeft: theme.spacing(2),
      flexGrow: 1
    },
    footer: {
      borderTop: "1px solid",
      borderColor: theme.palette.divider,
      padding: theme.spacing(2)
    }
  };
});

var ExpandedModule = function ExpandedModule(_ref) {
  var id = _ref.id,
      buttons = _ref.buttons,
      children = _ref.children,
      props = _objectWithoutPropertiesLoose(_ref, ["id", "buttons", "children"]);

  var classes = useStyles(props);
  return /*#__PURE__*/React.createElement(ModuleBase, {
    className: classes.module,
    id: id,
    expanded: true,
    elevation: 4
  }, /*#__PURE__*/React.createElement(ModuleFormat, props, /*#__PURE__*/React.createElement(ModuleHeader, _extends({}, props, {
    className: classes.header
  })), /*#__PURE__*/React.createElement(ModuleBody, {
    className: classes.body
  }, children)), /*#__PURE__*/React.createElement(ModuleFooter, _extends({}, props, {
    className: classes.footer
  }), buttons && /*#__PURE__*/React.createElement(ModuleButtons, {
    buttons: buttons
  })));
};

ExpandedModule.propTypes = process.env.NODE_ENV !== "production" ? {
  title: PropTypes.string,
  menu: PropTypes.array,
  buttons: PropTypes.array,
  children: PropTypes.node
} : {};
export default ExpandedModule;
