// @kit-ui/admin/es/Module/GalleryModule.stories.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import React from 'react';
import Gallery from "../Gallery/Gallery";
import GalleryModule from './GalleryModule';
export default {
  title: 'Module/Variant/GalleryModule',
  component: GalleryModule,
  args: {
    imageUrl: "https://dms-test.dimu.org/image/012uNXVpKSZo?mediaType=image/png",
    title: "Title",
    description: "description",
    metadata: ["25kb"],
    status: "publish",
    type: "document"
  },
  argTypes: {}
};

var Template = function Template(args) {
  return /*#__PURE__*/React.createElement(GalleryModule, args);
};

export var Default = Template.bind({});
Default.args = {};

var StatusTemplate = function StatusTemplate(args) {
  return /*#__PURE__*/React.createElement(Gallery, null, /*#__PURE__*/React.createElement(GalleryModule, _extends({}, args, {
    status: "publish"
  })), /*#__PURE__*/React.createElement(GalleryModule, _extends({}, args, {
    status: "draft"
  })), /*#__PURE__*/React.createElement(GalleryModule, _extends({}, args, {
    status: "trash"
  })), /*#__PURE__*/React.createElement(GalleryModule, _extends({}, args, {
    status: "erased"
  })), /*#__PURE__*/React.createElement(GalleryModule, _extends({}, args, {
    selectable: true
  })), /*#__PURE__*/React.createElement(GalleryModule, _extends({}, args, {
    selectable: true,
    selected: true
  })));
};

export var Status = StatusTemplate.bind({});
Default.args = {};

var WithSettingsTemplate = function WithSettingsTemplate(args) {
  return /*#__PURE__*/React.createElement(Gallery, null, /*#__PURE__*/React.createElement(GalleryModule, _extends({}, args, {
    status: "publish"
  })), /*#__PURE__*/React.createElement(GalleryModule, _extends({}, args, {
    status: "draft"
  })), /*#__PURE__*/React.createElement(GalleryModule, _extends({}, args, {
    status: "trash"
  })), /*#__PURE__*/React.createElement(GalleryModule, _extends({}, args, {
    status: "erased"
  })), /*#__PURE__*/React.createElement(GalleryModule, _extends({}, args, {
    selectable: true
  })), /*#__PURE__*/React.createElement(GalleryModule, _extends({}, args, {
    selectable: true,
    selected: true
  })));
};

export var WithSettings = WithSettingsTemplate.bind({});
WithSettings.args = {
  settings: [{
    template: "color"
  }]
};

var CapabilitiesTemplate = function CapabilitiesTemplate(args) {
  return /*#__PURE__*/React.createElement(Gallery, null, /*#__PURE__*/React.createElement(GalleryModule, _extends({}, args, {
    selectable: true
  })), /*#__PURE__*/React.createElement(GalleryModule, _extends({}, args, {
    selectable: true,
    selected: true
  })), /*#__PURE__*/React.createElement(GalleryModule, _extends({}, args, {
    editable: true
  })), /*#__PURE__*/React.createElement(GalleryModule, _extends({}, args, {
    editable: true,
    deletable: true
  })), /*#__PURE__*/React.createElement(GalleryModule, _extends({}, args, {
    editable: true
  })), /*#__PURE__*/React.createElement(GalleryModule, _extends({}, args, {
    restorable: true
  })));
};

export var Capabilities = CapabilitiesTemplate.bind({});
Capabilities.args = {};
