// @kit-ui/admin/es/Module/GridModule.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import ModuleBase from "./ModuleBase";
import ModulePrefix from "./ModulePrefix";
import ModuleMedia from './ModuleMedia';
import ModuleTitle from './ModuleTitle';
import ModuleLabel from './ModuleLabel';
import ModuleStatus from './ModuleStatus';
import ModuleByline from './ModuleByline';
import NavToolbar from "../NavToolbar/NavToolbar";
import { makeStyles } from '@material-ui/core/styles';
var useStyles = makeStyles(function (theme) {
  return {
    root: {
      backgroundColor: "transparent",
      position: "relative",
      width: function width(props) {
        return props.width;
      },
      userSelect: "none",
      "& $toolbar": {
        opacity: 0
      },
      "&:hover": {
        "& $toolbar": {
          opacity: 1
        }
      },
      "&[data-status=trash]": {
        "& $media": {
          opacity: .5
        },
        "& $toolbar": {
          opacity: 1
        }
      },
      "&[data-status=erased]": {
        "& $media": {
          opacity: .25
        }
      }
    },
    grid: {
      display: "block",
      backgroundColor: "transparent",
      color: "white",
      position: "relative",
      width: "100%",
      "&[role=button]": {
        cursor: "pointer"
      },
      "&[aria-selected=true]": {
        backgroundColor: theme.palette.action.selected,
        color: "currentColor",
        "& $primaryAction": {
          opacity: 1
        },
        "& $media": {//                opacity: .75
        }
      }
    },
    media: {
      display: "block",
      position: "relative",
      width: "100%",
      height: function height(props) {
        return props.mediaHeight;
      },
      margin: "0"
    },
    primaryAction: {
      position: "absolute",
      zIndex: 3,
      top: 0,
      right: 'auto',
      bottom: "auto",
      left: 0,
      "& button": {
        color: "white"
      }
    },
    toolbar: {
      position: "absolute",
      zIndex: 2,
      top: 0,
      right: 0,
      bottom: "auto",
      left: 0,
      width: "100%",
      height: function height(props) {
        return props.mediaHeight;
      },
      display: "flex",
      alignItems: "center",
      justifyContent: "center",
      "& button": {
        color: "white",
        opacity: .5,
        "&:hover": {
          opacity: 1
        }
      },
      "& button[name=select]": {
        position: "absolute",
        top: 0,
        right: 0
      },
      "& button[name=toggle]": {
        position: "absolute",
        top: 0,
        right: 0
      }
    },
    content: {
      flexDirection: "column",
      display: "flex",
      alignItems: "left",
      justifyContent: "flex-start"
    },
    header: {
      width: "100%",
      flexBasis: "100%",
      display: "flex",
      alignItems: "center",
      justifyContent: "flex-start",
      marginTop: theme.spacing(.5)
    },
    body: {
      width: "100%",
      flexBasis: "100%",
      display: "flex",
      alignItems: "center",
      justifyContent: "flex-start",
      "& > * + *": {
        marginLeft: theme.spacing(.5)
      }
    },
    footer: {
      display: "flex",
      alignItems: "baseline",
      width: "100%",
      flexBasis: "100%",
      justifyContent: "flex-start",
      "& > * + *": {
        marginLeft: theme.spacing(.5)
      }
    }
  };
});

var GridModule = function GridModule(_ref) {
  var _ref$width = _ref.width,
      width = _ref$width === void 0 ? 180 : _ref$width,
      maxWidth = _ref.maxWidth,
      mediaStyle = _ref.mediaStyle,
      _ref$mediaHeight = _ref.mediaHeight,
      mediaHeight = _ref$mediaHeight === void 0 ? 180 : _ref$mediaHeight,
      _ref$mediaLayout = _ref.mediaLayout,
      mediaLayout = _ref$mediaLayout === void 0 ? "contain" : _ref$mediaLayout,
      selected = _ref.selected,
      imageUrl = _ref.imageUrl,
      title = _ref.title,
      description = _ref.description,
      type = _ref.type,
      typeLabel = _ref.typeLabel,
      status = _ref.status,
      statusLabel = _ref.statusLabel,
      author = _ref.author,
      datetime = _ref.datetime,
      onClick = _ref.onClick,
      onEdit = _ref.onEdit,
      props = _objectWithoutPropertiesLoose(_ref, ["width", "maxWidth", "mediaStyle", "mediaHeight", "mediaLayout", "selected", "imageUrl", "title", "description", "type", "typeLabel", "status", "statusLabel", "author", "datetime", "onClick", "onEdit"]);

  if (maxWidth) {
    width = maxWidth;
  }

  var classes = useStyles({
    width: width,
    mediaHeight: mediaHeight
  });
  var media = {
    elevation: 1,
    selected: selected,
    layout: mediaLayout,
    imageUrl: imageUrl || "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mNk+P+/HgAFhAJ/wlseKgAAAABJRU5ErkJggg=="
  };
  return /*#__PURE__*/React.createElement(ModuleBase, _extends({}, props, {
    className: classes.root,
    selected: selected,
    status: status,
    onClick: undefined
  }), /*#__PURE__*/React.createElement("div", {
    className: classes.grid,
    onClick: onClick,
    role: onClick && "button",
    "aria-selected": selected
  }, /*#__PURE__*/React.createElement(ModuleMedia, _extends({}, media, {
    className: classes.media
  }))), /*#__PURE__*/React.createElement(NavToolbar, _extends({}, props, {
    className: classes.toolbar
  })), /*#__PURE__*/React.createElement("div", {
    className: classes.content
  }, /*#__PURE__*/React.createElement("header", {
    className: classes.header
  }, /*#__PURE__*/React.createElement(ModuleTitle, {
    title: title,
    status: status,
    onClick: onClick && undefined || onEdit
  })), /*#__PURE__*/React.createElement("div", {
    className: classes.body
  }, /*#__PURE__*/React.createElement(ModuleLabel, {
    label: typeLabel || type
  }), /*#__PURE__*/React.createElement(ModuleStatus, {
    status: statusLabel || status
  })), /*#__PURE__*/React.createElement("footer", {
    className: classes.footer
  }, /*#__PURE__*/React.createElement(ModuleByline, {
    author: author,
    datetime: datetime
  }))), /*#__PURE__*/React.createElement(ModulePrefix, _extends({
    className: classes.primaryAction,
    selected: selected
  }, props)));
};

GridModule.propTypes = process.env.NODE_ENV !== "production" ? {
  imageUrl: PropTypes.string,
  title: PropTypes.string,
  description: PropTypes.string,
  type: PropTypes.string,
  typeLabel: PropTypes.string,
  status: PropTypes.string,
  statusLabel: PropTypes.string,
  author: PropTypes.string,
  datetime: PropTypes.string,
  onClick: PropTypes.func,
  onEdit: PropTypes.func
} : {};
export default GridModule;
