// @kit-ui/admin/es/Module/ListModule.stories.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import React from 'react';
import ListModule from './ListModule';
import { ColorSettings } from "../";
export default {
  title: 'Module/Variant/ListModule',
  component: ListModule,
  args: {
    imageUrl: "https://dms-test.dimu.org/image/012uNXVpKSZo?mediaType=image/png",
    title: "Title",
    description: "description",
    metadata: ["25kb"],
    status: "publish",
    type: "document"
  },
  argTypes: {}
};

var Template = function Template(args) {
  return /*#__PURE__*/React.createElement(ListModule, args);
};

export var Default = Template.bind({});
Default.args = {};
export var LongDescription = Template.bind({});
LongDescription.args = {
  description: "Nullam id dolor id nibh ultricies vehicula ut id elit. Praesent commodo cursus magna, vel scelerisque nisl consectetur et. Aenean eu leo quam. Pellentesque ornare sem lacinia quam venenatis vestibulum. Duis mollis, est non commodo luctus, nisi erat porttitor ligula, eget lacinia odio sem nec elit. Vivamus sagittis lacus vel augue laoreet rutrum faucibus dolor auctor.",
  editable: true,
  deletable: true
};

var StatusTemplate = function StatusTemplate(args) {
  return /*#__PURE__*/React.createElement("div", null, /*#__PURE__*/React.createElement(ListModule, _extends({}, args, {
    status: "publish"
  })), /*#__PURE__*/React.createElement(ListModule, _extends({}, args, {
    status: "draft"
  })), /*#__PURE__*/React.createElement(ListModule, _extends({}, args, {
    status: "trash"
  })), /*#__PURE__*/React.createElement(ListModule, _extends({}, args, {
    status: "erased"
  })), /*#__PURE__*/React.createElement(ListModule, _extends({}, args, {
    selectable: true
  })), /*#__PURE__*/React.createElement(ListModule, _extends({}, args, {
    selectable: true,
    selected: true
  })));
};

export var Status = StatusTemplate.bind({});
Default.args = {};

var WithSettingsTemplate = function WithSettingsTemplate(args) {
  return /*#__PURE__*/React.createElement("div", null, /*#__PURE__*/React.createElement(ListModule, _extends({}, args, {
    status: "publish"
  })), /*#__PURE__*/React.createElement(ListModule, _extends({}, args, {
    status: "draft"
  })), /*#__PURE__*/React.createElement(ListModule, _extends({}, args, {
    status: "trash"
  })), /*#__PURE__*/React.createElement(ListModule, _extends({}, args, {
    status: "erased"
  })), /*#__PURE__*/React.createElement(ListModule, _extends({}, args, {
    selectable: true
  })), /*#__PURE__*/React.createElement(ListModule, _extends({}, args, {
    selectable: true,
    selected: true
  })));
};

export var WithSettings = WithSettingsTemplate.bind({});
WithSettings.args = {
  settings: [{
    template: "color"
  }]
};

var CapabilitiesTemplate = function CapabilitiesTemplate(args) {
  return /*#__PURE__*/React.createElement("div", null, /*#__PURE__*/React.createElement(ListModule, _extends({}, args, {
    selectable: true
  })), /*#__PURE__*/React.createElement(ListModule, _extends({}, args, {
    selectable: true,
    selected: true
  })), /*#__PURE__*/React.createElement(ListModule, _extends({}, args, {
    editable: true
  })), /*#__PURE__*/React.createElement(ListModule, _extends({}, args, {
    editable: true,
    deletable: true
  })), /*#__PURE__*/React.createElement(ListModule, _extends({}, args, {
    editable: true
  })), /*#__PURE__*/React.createElement(ListModule, _extends({}, args, {
    restorable: true
  })));
};

export var Capabilities = CapabilitiesTemplate.bind({});
Capabilities.args = {};
