// @kit-ui/admin/es/Module/ModuleActionbar.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import PropTypes from "prop-types";
import NavSettings from "../NavSettings/NavSettings";
import NavToolbar from "../NavToolbar/NavToolbar";
var useStyles = makeStyles(function (theme) {
  return {
    actionbar: {
      display: "flex",
      alignItems: "center",
      justifyContent: "center"
    },
    settings: {
      display: "flex",
      alignItems: "center",
      "& > * + *": {
        marginLeft: theme.spacing(-1)
      }
    },
    toolbar: {
      display: "flex",
      alignItems: "center",
      "& > * + *": {
        marginLeft: theme.spacing(-1)
      },
      "& button": {
        opacity: .5,
        "&:hover": {
          opacity: 1
        }
      }
    }
  };
});

var ModuleActionbar = function ModuleActionbar(_ref) {
  var className = _ref.className,
      settings = _ref.settings,
      toolbar = _ref.toolbar,
      children = _ref.children,
      props = _objectWithoutPropertiesLoose(_ref, ["className", "settings", "toolbar", "children"]);

  var classes = useStyles();

  if (children) {
    return /*#__PURE__*/React.createElement("div", {
      className: className || classes.actionbar
    }, children);
  }

  return /*#__PURE__*/React.createElement("div", {
    className: className || classes.actionbar
  }, /*#__PURE__*/React.createElement(NavSettings, {
    className: classes.settings,
    settings: settings
  }), /*#__PURE__*/React.createElement(NavToolbar, _extends({}, props, {
    className: classes.toolbar,
    toolbar: toolbar
  })));
};

ModuleActionbar.propTypes = process.env.NODE_ENV !== "production" ? {
  settings: PropTypes.array,
  toolbar: PropTypes.array
} : {};
export default ModuleActionbar;
