// @kit-ui/admin/es/Module/ModuleStatus.js
import React from 'react';
import PropTypes from "prop-types";
import Typography from '@material-ui/core/Typography';
import { makeStyles } from '@material-ui/core/styles';
var useStyles = makeStyles(function (theme) {
  return {
    label: {
      fontFamily: "Akkurat, sans-serif",
      fontSize: "14px",
      lineHeight: "24px",
      fontWeight: "bold",
      color: theme.palette.text.secondary
    }
  };
});
/** ModuleStatus, draft, publish, trash, erased */

var ModuleStatus = function ModuleStatus(_ref) {
  var _ref$component = _ref.component,
      component = _ref$component === void 0 ? "h3" : _ref$component,
      status = _ref.status,
      children = _ref.children;
  var classes = useStyles();

  if (!status && children) {
    status = children;
  }

  if (!status) {
    return false;
  }

  return /*#__PURE__*/React.createElement(Typography, {
    className: classes.label,
    noWrap: true,
    component: component
  }, status);
};

ModuleStatus.propTypes = process.env.NODE_ENV !== "production" ? {
  status: PropTypes.string
} : {};
export default ModuleStatus;
