// @kit-ui/admin/es/Module/Module.stories.js
import React from 'react';
import Module from './Module';
export default {
  title: 'Module/Module',
  component: Module,
  args: {
    imageUrl: "https://dms-test.dimu.org/image/012uNXVpKSZo?mediaType=image/png",
    title: "Title",
    description: "description",
    metadata: ["25kb"],
    status: "publish",
    type: "document"
  },
  argTypes: {}
};

var Template = function Template(args) {
  return /*#__PURE__*/React.createElement(Module, args);
};

export var Default = Template.bind({});
Default.args = {};
export var Elevation = Template.bind({});
Elevation.args = {
  elevation: 1
};
export var Expanded = Template.bind({});
Expanded.args = {
  expanded: true,
  children: /*#__PURE__*/React.createElement("p", null, "children")
};
export var Editable = Template.bind({});
Editable.args = {
  editable: true
};
export var EditableExpanded = Template.bind({});
EditableExpanded.args = {
  editable: true,
  expanded: true,
  children: /*#__PURE__*/React.createElement("p", null, "children")
};
