// @kit-ui/schema/es/TextWidget/TextWidget.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import React from "react";
import TextField from "@material-ui/core/TextField";
import TextWidgetPrefix from "./TextWidgetPrefix";
import TextWidgetSuffix from "./TextWidgetSuffix";
import { utils } from "@rjsf/core";
var getDisplayLabel = utils.getDisplayLabel;

var TextWidget = function TextWidget(_ref) {
  var id = _ref.id,
      required = _ref.required,
      readonly = _ref.readonly,
      disabled = _ref.disabled,
      type = _ref.type,
      value = _ref.value,
      onChange = _ref.onChange,
      onBlur = _ref.onBlur,
      onFocus = _ref.onFocus,
      autofocus = _ref.autofocus,
      options = _ref.options,
      schema = _ref.schema,
      uiSchema = _ref.uiSchema,
      _ref$rawErrors = _ref.rawErrors,
      rawErrors = _ref$rawErrors === void 0 ? [] : _ref$rawErrors,
      formContext = _ref.formContext,
      props = _objectWithoutPropertiesLoose(_ref, ["id", "required", "readonly", "disabled", "type", "value", "onChange", "onBlur", "onFocus", "autofocus", "options", "schema", "uiSchema", "rawErrors", "formContext"]);

  var _onChange = function _onChange(event) {
    return onChange(event.target.value === "" ? options.emptyValue : event.target.value);
  };

  var _onBlur = function _onBlur(event) {
    return onBlur(id, event.target.value);
  };

  var _onFocus = function _onFocus(event) {
    return onFocus(id, event.target.value);
  };

  var displayLabel = getDisplayLabel(schema, uiSchema
  /* TODO: , rootSchema */
  );
  var label = options.title || props.title || props.label || schema.title;
  var variant = options.variant || "filled";
  var size = options.size;
  var helperText = options.help || null;
  return /*#__PURE__*/React.createElement(TextField, _extends({}, props, {
    id: id,
    variant: variant,
    size: size,
    fullWidth: true,
    label: displayLabel && label,
    helperText: helperText,
    autoFocus: autofocus,
    required: required,
    disabled: disabled || readonly,
    type: type || schema.type,
    value: value || value === 0 ? value : "",
    error: rawErrors.length > 0,
    onChange: _onChange,
    onBlur: _onBlur,
    onFocus: _onFocus,
    InputProps: {
      startAdornment: /*#__PURE__*/React.createElement(TextWidgetPrefix, options),
      endAdornment: /*#__PURE__*/React.createElement(TextWidgetSuffix, _extends({}, options, {
        uiSchema: uiSchema,
        formContext: formContext
      }))
    }
  }));
};

export default TextWidget;
