// @kit-ui/schema/es/TextWidget/TextWidgetSuffix.js
import React from "react";
import InputAdornment from '@material-ui/core/InputAdornment';
import IconButton from '@material-ui/core/IconButton';
import Icon from '@material-ui/core/Icon';
import { NavSettings } from "@kit-ui/admin";
import { getParentUiSettings } from "../utils/";
import { makeStyles } from '@material-ui/core/styles';
var useStyles = makeStyles(function (theme) {
  return {
    settings: {
      display: "flex",
      flexWrap: "nowrap",
      "& > * + *": {
        marginLeft: theme.spacing(-1)
      }
    }
  };
});

var TextWidgetSuffix = function TextWidgetSuffix(_ref) {
  var endAdornment = _ref.endAdornment,
      suffix = _ref.suffix,
      toolbar = _ref.toolbar,
      settings = _ref.settings,
      _ref$uiSchema = _ref.uiSchema,
      uiSchema = _ref$uiSchema === void 0 ? {} : _ref$uiSchema,
      _ref$formContext = _ref.formContext,
      formContext = _ref$formContext === void 0 ? {} : _ref$formContext;
  var classes = useStyles();
  var parentSchema = formContext.parentSchema,
      parentFormData = formContext.parentFormData,
      onParentChange = formContext.onParentChange;

  if (parentSchema) {
    settings = getParentUiSettings({
      schema: parentSchema,
      formData: parentFormData,
      uiSchema: uiSchema,
      onChange: onParentChange
    });
  }

  if (!suffix && !toolbar && !settings && !endAdornment) {
    return false;
  }

  var WidgetToolbar = function WidgetToolbar(_ref2) {
    var _ref2$toolbar = _ref2.toolbar,
        toolbar = _ref2$toolbar === void 0 ? [] : _ref2$toolbar;

    if (!toolbar.length) {
      return false;
    }

    return toolbar.map(function (button, index) {
      var icon = button.icon,
          onClick = button.onClick;
      return /*#__PURE__*/React.createElement(IconButton, {
        onClick: onClick,
        key: index
      }, /*#__PURE__*/React.createElement(Icon, null, icon));
    });
  };

  return /*#__PURE__*/React.createElement(InputAdornment, {
    position: "end"
  }, suffix, /*#__PURE__*/React.createElement(NavSettings, {
    settings: settings,
    className: classes.settings
  }), /*#__PURE__*/React.createElement(WidgetToolbar, {
    toolbar: toolbar
  }), endAdornment);
};

export default TextWidgetSuffix;
