// @kit-ui/schema/es/TextWidget/TextWidget.stories.js
import React from 'react';
import Schema from "../SchemaEditor/";
export default {
  title: 'Widgets/TextWidget',
  component: Schema,
  args: {
    schema: {
      type: "string"
    },
    uiSchema: {},
    formData: "Text"
  },
  argTypes: {}
};

var Template = function Template(args) {
  return /*#__PURE__*/React.createElement(Schema, args);
};

export var Default = Template.bind({});
Default.args = {};
export var WithLabel = Template.bind({});
WithLabel.args = {
  uiSchema: {
    "ui:title": "Title",
    "ui:help": "Write a title"
  }
};
export var WithPrefix = Template.bind({});
WithPrefix.args = {
  uiSchema: {
    "ui:title": "Title",
    "ui:help": "Write a title",
    "ui:prefix": "Kr."
  }
};
export var WithSuffix = Template.bind({});
WithSuffix.args = {
  uiSchema: {
    "ui:title": "Title",
    "ui:help": "Write a title",
    "ui:suffix": "Suffix"
  }
};
export var OutlinedVariant = Template.bind({});
OutlinedVariant.args = {
  uiSchema: {
    "ui:title": "Title",
    "ui:variant": "outlined",
    "ui:help": "Write a title"
  }
};
export var StandardVariant = Template.bind({});
StandardVariant.args = {
  uiSchema: {
    "ui:title": "Title",
    "ui:variant": "standard",
    "ui:help": "Write a title"
  }
};
