// @kit-ui/schema/es/widgets.js
import CheckboxWidget from "./CheckboxWidget/CheckboxWidget";
import CheckboxesWidget from "./CheckboxesWidget/CheckboxesWidget";
import ColorWidget from "./ColorWidget/ColorWidget";
import DateWidget from "./DateWidget/DateWidget";
import DateTimeWidget from "./DateTimeWidget/DateTimeWidget";
import TimeWidget from "./TimeWidget/TimeWidget";
import EmailWidget from "./EmailWidget/EmailWidget";
import PasswordWidget from "./PasswordWidget/PasswordWidget";
import RadioWidget from "./RadioWidget/RadioWidget";
import RangeWidget from "./RangeWidget/RangeWidget";
import SelectWidget from "./SelectWidget/SelectWidget";
import TextareaWidget from "./TextareaWidget/TextareaWidget";
import TextWidget from "./TextWidget/TextWidget";
import UpDownWidget from "./UpDownWidget/UpDownWidget";
import URLWidget from "./URLWidget/URLWidget.js";
import SwitchWidget from "./SwitchWidget/SwitchWidget"; // import AutocompleteWidget from "./AutocompleteWidget/AutocompleteWidget"

import TagsWidget from "./TagsWidget/TagsWidget"; // settings widgets

import ListSettingsWidget from "./ListSettings/ListSettingsWidget";
import ColorSettingsWidget from "./ColorSettings/ColorSettingsWidget";
import SizeSettingsWidget from "./SizeSettings/SizeSettingsWidget";
import GridSettingsWidget from "./GridSettings/GridSettingsWidget";
import PlacementSettingsWidget from "./PlacementSettings/PlacementSettingsWidget";
export default {
  CheckboxWidget: CheckboxWidget,
  CheckboxesWidget: CheckboxesWidget,
  ColorWidget: ColorWidget,
  DateWidget: DateWidget,
  DateTimeWidget: DateTimeWidget,
  EmailWidget: EmailWidget,
  PasswordWidget: PasswordWidget,
  RadioWidget: RadioWidget,
  RangeWidget: RangeWidget,
  SelectWidget: SelectWidget,
  TextareaWidget: TextareaWidget,
  TextWidget: TextWidget,
  UpDownWidget: UpDownWidget,
  URLWidget: URLWidget,
  "time": TimeWidget,
  "texteditor": TextareaWidget,
  "switch": SwitchWidget,
  //    "autocomplete": AutocompleteWidget,
  "tags": TagsWidget,
  "colorSettings": ColorSettingsWidget,
  "list": SelectWidget,
  "listSettings": ListSettingsWidget,
  "size": SelectWidget,
  "sizeSettings": SizeSettingsWidget,
  "grid": SelectWidget,
  "gridSettings": GridSettingsWidget,
  "placement": SelectWidget,
  "placementSettings": PlacementSettingsWidget
};
