// @kit-ui/admin/es/NavAction/NavAction.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import React, { useState, useRef, forwardRef } from 'react';
import { makeStyles } from '@material-ui/core/styles';
import NavActionButton from "./NavActionButton";
import NavActionDropdown from "./NavActionDropdown";
import NavActionMenu from "./NavActionMenu";
var useStyles = makeStyles(function (theme) {
  return {
    root: {
      fontSize: "14px"
    }
  };
});

var NavAction = function NavAction(_ref) {
  var className = _ref.className,
      _ref$variant = _ref.variant,
      variant = _ref$variant === void 0 ? "contained" : _ref$variant,
      _ref$color = _ref.color,
      color = _ref$color === void 0 ? "primary" : _ref$color,
      _ref$placement = _ref.placement,
      placement = _ref$placement === void 0 ? "bottom-start" : _ref$placement,
      primaryAction = _ref.primaryAction,
      _ref$menu = _ref.menu,
      menu = _ref$menu === void 0 ? [] : _ref$menu,
      onSelect = _ref.onSelect;

  var _useState = useState(false),
      expanded = _useState[0],
      setExpanded = _useState[1];

  var anchorRef = useRef(null);
  var classes = useStyles();

  var handleToggle = function handleToggle(event) {
    setExpanded(function (prevExpanded) {
      return !prevExpanded;
    });
  };

  var handleSelect = function handleSelect(_ref2) {
    var onClick = _ref2.onClick,
        item = _objectWithoutPropertiesLoose(_ref2, ["onClick"]);

    onClick && onClick() || onSelect && onSelect(item);
    setExpanded(false);
  };

  if (!primaryAction && menu.length) {
    primaryAction = menu[0];
  }

  if (primaryAction && primaryAction.children) {
    var _menu = primaryAction.children;
    var role = primaryAction.role;

    if (role !== "button") {
      return /*#__PURE__*/React.createElement("nav", {
        className: className || classes.root
      }, /*#__PURE__*/React.createElement(NavActionButton, _extends({}, primaryAction, {
        variant: primaryAction.variant || variant,
        color: color,
        expanded: expanded,
        forwardedRef: anchorRef,
        onToggle: handleToggle
      })), /*#__PURE__*/React.createElement(NavActionDropdown, {
        open: expanded,
        anchorEl: anchorRef.current,
        placement: placement,
        onClickAway: handleToggle
      }, /*#__PURE__*/React.createElement(NavActionMenu, {
        menu: _menu,
        onSelect: handleSelect
      })));
    }

    return /*#__PURE__*/React.createElement("nav", {
      className: className || classes.root
    }, /*#__PURE__*/React.createElement(NavActionButton, _extends({}, primaryAction, {
      variant: primaryAction.variant || variant,
      color: color,
      expanded: expanded,
      forwardedRef: anchorRef,
      onClick: function onClick() {
        return handleSelect(primaryAction);
      },
      onToggle: handleToggle
    })), /*#__PURE__*/React.createElement(NavActionDropdown, {
      open: expanded,
      anchorEl: anchorRef.current,
      placement: placement,
      onClickAway: handleToggle
    }, /*#__PURE__*/React.createElement(NavActionMenu, {
      menu: _menu,
      onSelect: handleSelect
    })));
  }

  if (primaryAction) {
    return /*#__PURE__*/React.createElement("nav", {
      className: className || classes.root
    }, /*#__PURE__*/React.createElement(NavActionButton, _extends({}, primaryAction, {
      variant: primaryAction.variant || variant,
      color: color,
      onClick: function onClick() {
        return handleSelect(primaryAction);
      }
    })));
  }

  return null;
};

export default NavAction;
