// @kit-ui/admin/es/NavAction/NavActionMenuItem.js
import React from 'react';
import MenuItem from "@material-ui/core/MenuItem";
import ListItem from "@material-ui/core/ListItem";
import { makeStyles } from '@material-ui/core/styles';
var useStyles = makeStyles(function (theme) {
  return {
    actionListItem: {
      width: "100%",
      padding: 0,
      "&[role=group] + *": {
        borderTop: "1px solid",
        borderTopColor: theme.palette.divider,
        marginTop: theme.spacing(1.5),
        paddingTop: theme.spacing(1.5)
      },
      "& + &[role=group]": {
        borderTop: "1px solid",
        borderTopColor: theme.palette.divider,
        marginTop: theme.spacing(1.5),
        paddingTop: theme.spacing(1.5),
        "& + *": {
          borderTop: "1px solid",
          borderTopColor: theme.palette.divider,
          marginTop: theme.spacing(1.5),
          paddingTop: theme.spacing(1.5)
        }
      }
    }
  };
});

var ActionMenuItem = function ActionMenuItem(_ref) {
  var className = _ref.className,
      role = _ref.role,
      children = _ref.children;
  var classes = useStyles();
  return /*#__PURE__*/React.createElement(ListItem, {
    className: className || classes.actionListItem,
    role: role
  }, children);
};

export default ActionMenuItem;
