// @kit-ui/admin/es/AppSwitcher/AppsListItem.js
import React, { Component } from 'react';
import ButtonBase from '@material-ui/core/ButtonBase';
import IconButton from '@material-ui/core/IconButton';
import Typography from '@material-ui/core/Typography';
import { makeStyles } from '@material-ui/core/styles';
var useStyles = makeStyles(function (theme) {
  var _root;

  return {
    root: (_root = {
      width: "100%",
      listStyle: "none",
      padding: 0,
      margin: 0,
      display: "flex",
      alignItems: "center",
      justifyContent: "left"
    }, _root["padding"] = theme.spacing(1.25), _root),
    figure: {
      width: function width(props) {
        return props.size;
      },
      height: function height(props) {
        return props.size;
      },
      "& svg, & img": {
        boxSizing: "inherit",
        position: "absolute",
        top: 0,
        right: 0,
        bottom: 0,
        left: 0,
        width: "100%",
        height: "100%",
        margin: "auto"
      }
    },
    label: {
      fontSize: "14px",
      fontWeight: "bold",
      margin: theme.spacing(1)
    }
  };
});

var AppsListItem = function AppsListItem(_ref) {
  var title = _ref.title,
      imageUrl = _ref.imageUrl,
      icon = _ref.icon,
      _ref$size = _ref.size,
      size = _ref$size === void 0 ? "large" : _ref$size,
      onClick = _ref.onClick;
  var sizes = {
    "small": 36,
    "medium": 48,
    "large": 64
  };
  size = sizes[size] || size || 64;
  var classes = useStyles({
    size: size
  });
  return /*#__PURE__*/React.createElement(ButtonBase, {
    className: classes.root,
    disableRipple: true,
    onClick: onClick
  }, /*#__PURE__*/React.createElement(IconButton, {
    className: classes.figure,
    component: "figure"
  }, !icon && imageUrl && /*#__PURE__*/React.createElement("img", {
    className: classes.img,
    src: imageUrl
  }), icon), /*#__PURE__*/React.createElement(Typography, {
    className: classes.label,
    noWrap: true
  }, title));
};

export default AppsListItem;
