// @kit-ui/admin/es/AppSwitcher/FilterAppsControlled.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _inheritsLoose(subClass, superClass) { subClass.prototype = Object.create(superClass.prototype); subClass.prototype.constructor = subClass; subClass.__proto__ = superClass; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

import React, { Component } from 'react';
import FilterAppsUncontrolled from "./FilterAppsUncontrolled";

var FilterAppsControlled = /*#__PURE__*/function (_Component) {
  _inheritsLoose(FilterAppsControlled, _Component);

  function FilterAppsControlled() {
    var _this;

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = _Component.call.apply(_Component, [this].concat(args)) || this;

    _defineProperty(_assertThisInitialized(_this), "state", {
      q: "",
      filteredApps: []
    });

    _defineProperty(_assertThisInitialized(_this), "componentDidUpdate", function (prevProps, prevState) {
      if (prevProps.apps !== _this.props.apps) {
        _this.filterApps();
      }

      if (prevState.q !== _this.state.q) {
        _this.filterApps();
      }
    });

    _defineProperty(_assertThisInitialized(_this), "filterApps", function () {
      var apps = _this.props.apps;
      var q = _this.state.q;
      var filteredApps = apps.filter(function (owner) {
        var title = owner.title;

        if (q && !title.toLowerCase().includes(q.toLowerCase())) {
          return false;
        }

        return true;
      });

      _this.setState({
        filteredApps: filteredApps
      });
    });

    _defineProperty(_assertThisInitialized(_this), "onSearch", function (q, event) {
      event.preventDefault();

      _this.setState({
        q: q
      });
    });

    return _this;
  }

  var _proto = FilterAppsControlled.prototype;

  _proto.componentDidMount = function componentDidMount() {
    this.filterApps();
  };

  _proto.render = function render() {
    var _this$state = this.state,
        q = _this$state.q,
        filteredApps = _this$state.filteredApps;
    return /*#__PURE__*/React.createElement(FilterAppsUncontrolled, _extends({}, this.props, {
      q: q,
      onSearch: this.onSearch,
      apps: filteredApps
    }));
  };

  return FilterAppsControlled;
}(Component);

_defineProperty(FilterAppsControlled, "defaultProps", {
  apps: [],
  layout: "list"
});

export default FilterAppsControlled;
