// @kit-ui/admin/es/AppSwitcher/OwnersList.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import React from 'react';
import OwnersListItem from "./OwnersListItem";
import { makeStyles } from '@material-ui/core/styles';
var useStyles = makeStyles(function (theme) {
  return {
    list: {
      display: "flex",
      flexDirection: "column"
    }
  };
});

var OwnersList = function OwnersList(_ref) {
  var className = _ref.className,
      _ref$owners = _ref.owners,
      owners = _ref$owners === void 0 ? [] : _ref$owners,
      children = _ref.children,
      onSelect = _ref.onSelect;
  var classes = useStyles();

  if (!owners.length) {
    return false;
  }

  return /*#__PURE__*/React.createElement("div", {
    className: className || classes.list
  }, owners && owners.length && owners.map(function (owner, index) {
    return /*#__PURE__*/React.createElement(OwnersListItem, _extends({}, owner, {
      onClick: function onClick() {
        return onSelect(owner);
      },
      key: index
    }));
  }), children);
};

export default OwnersList;
