// @kit-ui/admin/es/AppSwitcher/OwnersListFiltered.js
import React from 'react';
import FilterOwners from "./FilterOwnersControlled";
import OwnersList from "./OwnersList";
import { makeStyles } from '@material-ui/core/styles';
var useStyles = makeStyles(function (theme) {
  return {
    root: {
      borderTop: "1px solid",
      borderColor: "#ccc",
      position: "relative"
    },
    list: {
      height: theme.spacing(7 * 8)
    }
  };
});

var AppsOwnersList = function AppsOwnersList(_ref) {
  var _ref$owners = _ref.owners,
      owners = _ref$owners === void 0 ? [] : _ref$owners,
      onSelect = _ref.onSelect;
  var classes = useStyles();

  if (owners.length > 7) {
    return /*#__PURE__*/React.createElement("div", {
      className: classes.root
    }, /*#__PURE__*/React.createElement(FilterOwners, {
      className: classes.list,
      owners: owners,
      onSelect: onSelect
    }));
  }

  return /*#__PURE__*/React.createElement("div", {
    className: classes.root
  }, /*#__PURE__*/React.createElement(OwnersList, {
    owners: owners,
    onSelect: onSelect
  }));
};

export default AppsOwnersList;
