// @kit-ui/admin/lib/NavToolbar/NavToolbar.stories.js
"use strict";

exports.__esModule = true;
exports.CustomToolbar = exports.MultipleOptions = exports.HideableHidden = exports.Hideable = exports.DeletedRestorableErasable = exports.DeletedRestorable = exports.DeletedErasable = exports.Deleted = exports.EditableDeletable = exports.Editable = exports.Default = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _NavToolbar = _interopRequireDefault(require("./NavToolbar"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _default = {
  title: 'Navigation/NavToolbar',
  component: _NavToolbar.default,
  args: {},
  argTypes: {}
};
exports.default = _default;

var Template = function Template(args) {
  return /*#__PURE__*/_react.default.createElement(_NavToolbar.default, args);
};

var Default = Template.bind({});
exports.Default = Default;
Default.args = {};
var Editable = Template.bind({});
exports.Editable = Editable;
Editable.args = {
  editable: true
};
var EditableDeletable = Template.bind({});
exports.EditableDeletable = EditableDeletable;
EditableDeletable.args = {
  editable: true,
  deletable: true
};
var Deleted = Template.bind({});
exports.Deleted = Deleted;
Deleted.args = {
  editable: true,
  deletable: true,
  deleted: true
};
var DeletedErasable = Template.bind({});
exports.DeletedErasable = DeletedErasable;
DeletedErasable.args = {
  editable: true,
  deletable: true,
  deleted: true,
  erasable: true
};
var DeletedRestorable = Template.bind({});
exports.DeletedRestorable = DeletedRestorable;
DeletedRestorable.args = {
  deleted: true,
  restorable: true
};
var DeletedRestorableErasable = Template.bind({});
exports.DeletedRestorableErasable = DeletedRestorableErasable;
DeletedRestorableErasable.args = {
  deleted: true,
  restorable: true,
  erasable: true
};
var Hideable = Template.bind({});
exports.Hideable = Hideable;
Hideable.args = {
  hideable: true
};
var HideableHidden = Template.bind({});
exports.HideableHidden = HideableHidden;
HideableHidden.args = {
  hideable: true,
  hidden: true
};
var MultipleOptions = Template.bind({});
exports.MultipleOptions = MultipleOptions;
MultipleOptions.args = {
  hideable: true,
  deletable: true,
  restoreable: true
};
var CustomToolbar = Template.bind({});
exports.CustomToolbar = CustomToolbar;
CustomToolbar.args = {
  toolbar: [{
    name: "size",
    value: "small",
    options: [{
      value: "small"
    }, {
      value: "medium"
    }, {
      value: "large"
    }]
  }, {
    name: "color",
    value: "#ff0",
    options: [{
      value: "red"
    }, {
      value: "blue"
    }, {
      value: "green"
    }]
  }]
};
