// @kit-ui/admin/lib/App/AppLayout.stories.js
"use strict";

exports.__esModule = true;
exports.SubviewExpanded = exports.SearchExpanded = exports.GrowingSearchAndSidebar = exports.GrowingSearch = exports.CollapsibleSidebarAndSearch = exports.CollapsibleSidebar = exports.WithSidebar = exports.WithSearch = exports.Default = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _AppLayout = _interopRequireDefault(require("./AppLayout"));

var _AppLayout2 = _interopRequireDefault(require("./AppLayout.example"));

var _ = require("..");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var theme = (0, _.getRandomTheme)();
var _default = {
  title: 'App/AppLayout',
  component: _AppLayout.default,
  args: {
    header: {
      title: "App",
      subtitle: "Owner"
    },
    primaryAction: {
      title: "Create"
    },
    menu: [{
      title: "Menu 1",
      url: "/1"
    }, {
      title: "Menu 2",
      url: "/2"
    }],
    menuByUrl: {
      "/1": {
        count: 200
      }
    },
    theme: (0, _.getRandomTheme)()
  },
  argTypes: {}
};
exports.default = _default;

var Template = function Template(args) {
  return /*#__PURE__*/_react.default.createElement("div", {
    style: {
      minHeight: '320px'
    }
  }, /*#__PURE__*/_react.default.createElement(_AppLayout2.default, args, JSON.stringify(args)));
};

var Default = Template.bind({});
exports.Default = Default;
Default.args = {};
var WithSearch = Template.bind({});
exports.WithSearch = WithSearch;
WithSearch.args = {
  search: {
    expanded: false
  }
};
var WithSidebar = Template.bind({});
exports.WithSidebar = WithSidebar;
WithSidebar.args = {
  sidebar: {
    expanded: true
  }
};
var CollapsibleSidebar = Template.bind({});
exports.CollapsibleSidebar = CollapsibleSidebar;
CollapsibleSidebar.args = {
  sidebar: {
    collapsible: true,
    expanded: true
  }
};
var CollapsibleSidebarAndSearch = Template.bind({});
exports.CollapsibleSidebarAndSearch = CollapsibleSidebarAndSearch;
CollapsibleSidebarAndSearch.args = {
  search: {
    expanded: false
  },
  sidebar: {
    collapsible: true,
    expanded: true
  }
};
var GrowingSearch = Template.bind({});
exports.GrowingSearch = GrowingSearch;
GrowingSearch.args = {
  header: {
    title: "Growing search"
  },
  search: {
    variant: "growing",
    expanded: false
  }
};
var GrowingSearchAndSidebar = Template.bind({});
exports.GrowingSearchAndSidebar = GrowingSearchAndSidebar;
GrowingSearch.args = {
  header: {
    title: "Growing search"
  },
  sidebar: {
    collapsible: true
  },
  search: {
    variant: "growing",
    expanded: false
  }
};
var SearchExpanded = Template.bind({});
exports.SearchExpanded = SearchExpanded;
SearchExpanded.args = {
  search: {
    expanded: true
  }
};
var SubviewExpanded = Template.bind({});
exports.SubviewExpanded = SubviewExpanded;
SubviewExpanded.args = {
  subview: {
    expanded: true,
    title: "Subview",
    description: "Description"
  }
};
