// @kit-ui/admin/lib/App/AppLayout.example.js
"use strict";

exports.__esModule = true;
exports.default = exports.AppLayoutExample = void 0;

var _react = _interopRequireWildcard(require("react"));

var _AppLayout = _interopRequireDefault(require("./AppLayout"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var AppLayoutExample = function AppLayoutExample(_ref) {
  var children = _ref.children,
      props = _objectWithoutPropertiesLoose(_ref, ["children"]);

  // search
  var _useState = (0, _react.useState)(props.search),
      search = _useState[0],
      setSearch = _useState[1];

  var _useState2 = (0, _react.useState)(null),
      searchQ = _useState2[0],
      setSearchQ = _useState2[1];

  var searchChange = function searchChange(q) {
    setSearchQ(q);
  };

  var resetSearch = function resetSearch() {
    setSearchQ(null);

    if (!searchQ) {
      toggleSearch();
    }
  };

  var toggleSearch = function toggleSearch() {
    setSearch(_extends({}, search, {
      expanded: !search.expanded
    }));
  };

  var searchState = search && _extends({}, search, {
    onChange: searchChange,
    onToggle: toggleSearch,
    onReset: resetSearch,
    q: searchQ
  }); // subview


  var _useState3 = (0, _react.useState)(props.subview),
      subview = _useState3[0],
      setSubview = _useState3[1];

  var toggleSubview = function toggleSubview() {
    setSubview(_extends({}, subview, {
      expanded: !subview.expanded
    }));
  };

  var subviewState = subview && _extends({}, subview, {
    onToggle: toggleSubview
  }); // sidebar


  var _useState4 = (0, _react.useState)(props.sidebar),
      sidebar = _useState4[0],
      setSidebar = _useState4[1];

  var toggleSidebar = function toggleSidebar() {
    setSidebar(_extends({}, sidebar, {
      expanded: !sidebar.expanded
    }));
  };

  var sidebarState = props.sidebar && _extends({}, sidebar, {
    onToggle: toggleSidebar
  });

  return /*#__PURE__*/_react.default.createElement(_AppLayout.default, _extends({}, props, {
    search: searchState,
    sidebar: sidebarState,
    subview: subviewState
  }), children);
};

exports.AppLayoutExample = AppLayoutExample;
var _default = AppLayoutExample;
exports.default = _default;
