// @kit-ui/admin/lib/App/AppSidebar.stories.js
"use strict";

exports.__esModule = true;
exports.CalendarSidebar = exports.FinderSidebar = exports.Default = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _AppSidebar = _interopRequireDefault(require("./AppSidebar"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _default = {
  title: 'App/AppSidebar',
  component: _AppSidebar.default,
  args: {},
  argTypes: {}
};
exports.default = _default;

var Template = function Template(args) {
  return /*#__PURE__*/_react.default.createElement(_AppSidebar.default, args);
};

var Default = Template.bind({});
exports.Default = Default;
Default.args = {};
var FinderSidebar = Template.bind({});
exports.FinderSidebar = FinderSidebar;
FinderSidebar.args = {
  primaryAction: {
    title: "Create",
    children: [{
      icon: "document",
      title: "New dcument"
    }, {
      icon: "page",
      title: "New page"
    }]
  },
  menu: [{
    title: "Menu 1",
    url: "/1"
  }, {
    title: "Menu 2",
    url: "/2"
  }],
  menuByUrl: {
    "/1": {
      count: 200
    }
  }
};
var CalendarSidebar = Template.bind({});
exports.CalendarSidebar = CalendarSidebar;
CalendarSidebar.args = {
  primaryAction: {
    title: "New event"
  },
  calendar: {
    date: "1999-05-26"
  },
  menu: [{
    title: "Menu 1",
    url: "/1"
  }, {
    title: "Menu 2",
    url: "/2"
  }],
  menuByUrl: {
    "/1": {
      count: 200
    }
  }
};
