// @kit-ui/schema/es/SizeSettings/SizeOptions.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import React, { forwardRef, useState, useEffect, useRef } from 'react';
import PropTypes from "prop-types";
import { Dropdown } from "@kit-ui/core";
import Button from '@material-ui/core/ButtonBase';
import IconButton from "@material-ui/core/IconButton";
import Icon from "@material-ui/core/Icon";
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemText from '@material-ui/core/ListItemText';
import { makeStyles } from '@material-ui/core/styles';
var sizes = {
  "xlarge": "4:4",
  "xl": "4:4",
  "large": "4:3",
  "l": "4:3",
  "l-m": "4:2",
  "m-l": "2:4",
  "medium": "2:2",
  "m": "2:2",
  "m-s": "2:1",
  "s-m": "1:2",
  "small": "1:1",
  "s": "1:1",
  "xsmall": "0.5:0.5",
  "xs": "0.5:0.5"
};
var useStyles = makeStyles(function (theme) {
  return {
    list: {
      position: "relative",
      padding: 0,
      margin: 0,
      width: function width(props) {
        return props.cols * theme.spacing(7);
      },
      height: function height(props) {
        return props.rows * theme.spacing(7);
      }
    },
    listItem: {
      position: "absolute",
      display: "block",
      top: 0,
      left: 0,
      backgroundColor: "rgba(0,0,0,.125)",
      zIndex: function zIndex(props) {
        return props.zIndex;
      },
      outline: "1px solid white",
      width: function width(props) {
        return props.cols * theme.spacing(7);
      },
      height: function height(props) {
        return props.rows * theme.spacing(7);
      },
      "&:hover": {
        backgroundColor: "rgba(0,0,0,.5)",
        color: "white"
      },
      "&[aria-selected=true]": {
        backgroundColor: theme.palette.primary.main,
        color: "white"
      }
    },
    label: {
      position: "absolute",
      top: 'auto',
      right: 0,
      bottom: 0,
      left: 'auto',
      fontFamily: "Akkurat, sans-serif",
      textTransform: "uppercase",
      fontSize: '12px',
      padding: theme.spacing(0.5)
    }
  };
});

var SizeSettingsOptions = function SizeSettingsOptions(_ref) {
  var value = _ref.value,
      options = _ref.options,
      onChange = _ref.onChange;

  if (!value) {
    value = options[0].value;
  }

  var _onSelect = function _onSelect(value) {
    onChange && onChange(value);
  };

  var maxRows = 4;
  var maxCols = 4;

  var ListOption = function ListOption(option) {
    var label = option.label,
        value = option.value,
        selected = option.selected;
    var grid = sizes && sizes[value] && sizes[value].split(':') || [4, 4];
    var cols = grid[0];
    var rows = grid[1];
    var zIndex = maxRows * maxCols - cols * rows;
    var classes = useStyles({
      cols: cols,
      rows: rows,
      zIndex: zIndex
    });
    return /*#__PURE__*/React.createElement("li", {
      className: classes.listItem,
      "aria-selected": selected,
      onClick: function onClick() {
        return _onSelect(value);
      }
    }, /*#__PURE__*/React.createElement("b", {
      className: classes.label
    }, label || value));
  };

  var classes = useStyles({
    cols: maxCols,
    rows: maxRows
  });
  return /*#__PURE__*/React.createElement("ul", {
    className: classes.list
  }, options && options.map(function (option, index) {
    return /*#__PURE__*/React.createElement(ListOption, _extends({}, option, {
      key: index,
      selected: option.value === value
    }));
  }));
};

SizeSettingsOptions.defaultProps = {};
SizeSettingsOptions.propTypes = process.env.NODE_ENV !== "production" ? {} : {};
export default SizeSettingsOptions;
