// @kit-ui/schema/es/SizeSettings/SizeSettings.js
import React, { forwardRef, useState, useEffect, useRef } from 'react';
import PropTypes from "prop-types";
import { Dropdown } from "@kit-ui/core";
import ListSettingsButton from '../ListSettings/ListSettingsButton';
import SizeOptions from "./SizeSettingsOptions";

var SizeSettings = function SizeSettings(_ref) {
  var value = _ref.value,
      options = _ref.options,
      onChange = _ref.onChange;

  if (!value) {
    value = options[0].value;
  }

  var _useState = useState(false),
      expanded = _useState[0],
      setExpanded = _useState[1];

  var anchorRef = useRef(null);

  var _onChange = function _onChange(value) {
    onChange && onChange(value);
    setExpanded(false);
  };

  var _onToggle = function _onToggle(event) {
    setExpanded(function (prevExpanded) {
      return !prevExpanded;
    });
  };

  var _onCollapse = function _onCollapse(event) {
    setExpanded(false);
  };

  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(ListSettingsButton, {
    onClick: _onToggle,
    ref: anchorRef
  }, value), /*#__PURE__*/React.createElement(Dropdown, {
    expanded: expanded,
    anchorEl: anchorRef.current,
    onClickAway: _onCollapse
  }, /*#__PURE__*/React.createElement(SizeOptions, {
    options: options,
    onChange: _onChange
  })));
};

SizeSettings.defaultProps = {};
SizeSettings.propTypes = process.env.NODE_ENV !== "production" ? {} : {};
export default SizeSettings;
