// @kit-ui/admin/lib/Section/CollapsibleSection.stories.js
"use strict";

exports.__esModule = true;
exports.ExpandedWithButtons = exports.Default = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _CollapsibleSection = _interopRequireDefault(require("./CollapsibleSection"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _default = {
  title: 'Section/CollapsibleSection',
  component: _CollapsibleSection.default,
  args: {
    title: "Title",
    description: "description"
  },
  argTypes: {}
};
exports.default = _default;

var Template = function Template(args) {
  return /*#__PURE__*/_react.default.createElement(_CollapsibleSection.default, args, "Content");
};

var Default = Template.bind({});
exports.Default = Default;
Default.args = {};
var ExpandedWithButtons = Template.bind({});
exports.ExpandedWithButtons = ExpandedWithButtons;
ExpandedWithButtons.args = {
  expanded: true,
  buttons: [{
    title: "Click"
  }, {
    title: "Dont click",
    disabled: true
  }]
};
