// @kit-ui/admin/lib/Section/DefaultSection.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _SectionBase = _interopRequireDefault(require("./SectionBase"));

var _SectionHeader = _interopRequireDefault(require("./SectionHeader"));

var _SectionBody = _interopRequireDefault(require("./SectionBody"));

var _SectionFooter = _interopRequireDefault(require("./SectionFooter"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var DefaultSection = function DefaultSection(_ref) {
  var _ref$expanded = _ref.expanded,
      expanded = _ref$expanded === void 0 ? true : _ref$expanded,
      _ref$position = _ref.position,
      position = _ref$position === void 0 ? "relative" : _ref$position,
      collapsible = _ref.collapsible,
      title = _ref.title,
      description = _ref.description,
      buttons = _ref.buttons,
      children = _ref.children,
      props = _objectWithoutPropertiesLoose(_ref, ["expanded", "position", "collapsible", "title", "description", "buttons", "children"]);

  return /*#__PURE__*/_react.default.createElement(_SectionBase.default, {
    position: position,
    expanded: expanded
  }, /*#__PURE__*/_react.default.createElement(_SectionHeader.default, _extends({}, props, {
    title: title,
    description: description
  })), /*#__PURE__*/_react.default.createElement(_SectionBody.default, _extends({
    position: position
  }, props), children), /*#__PURE__*/_react.default.createElement(_SectionFooter.default, _extends({}, props, {
    buttons: buttons
  })));
};

DefaultSection.propTypes = process.env.NODE_ENV !== "production" ? {
  collapsible: _propTypes.default.bool,
  expanded: _propTypes.default.bool
} : {};
DefaultSection.defaultProps = {
  collapsible: false,
  expanded: true,
  padding: 2,
  spacing: 0
};
var _default = DefaultSection;
exports.default = _default;
module.exports = exports.default;
