// @kit-ui/admin/lib/Section/SectionFooter.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _styles = require("@material-ui/core/styles");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var useStyles = (0, _styles.makeStyles)(function (theme) {
  return {
    footer: {
      paddingLeft: function paddingLeft(props) {
        return theme.spacing(props.padding);
      },
      paddingRight: function paddingRight(props) {
        return theme.spacing(props.padding);
      },
      paddingBottom: function paddingBottom(props) {
        return theme.spacing(props.padding);
      }
    }
  };
});

var SectionFooter = function SectionFooter(_ref) {
  var className = _ref.className,
      _ref$padding = _ref.padding,
      padding = _ref$padding === void 0 ? 2 : _ref$padding,
      children = _ref.children;
  var classes = useStyles({
    padding: padding
  });

  if (!children) {
    return false;
  }

  return /*#__PURE__*/_react.default.createElement("footer", {
    className: className || classes.footer
  }, children);
};

var _default = SectionFooter;
exports.default = _default;
module.exports = exports.default;
