// @kit-ui/admin/lib/Section/SectionBase.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireWildcard(require("react"));

var _Paper = _interopRequireDefault(require("@material-ui/core/Paper"));

var _styles = require("@material-ui/core/styles");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var useStyles = (0, _styles.makeStyles)(function (theme) {
  return {
    section: {
      position: function position(props) {
        return props.position;
      },
      zIndex: function zIndex(props) {
        return props.zIndex;
      },
      top: function top(props) {
        return props.top;
      },
      right: function right(props) {
        return props.right;
      },
      bottom: function bottom(props) {
        return props.bottom;
      },
      left: function left(props) {
        return props.left;
      },
      width: "100%",
      backgroundColor: theme.palette.background.default,
      display: "flex",
      flexDirection: "column",
      "& + $section": {
        borderTop: "1px solid",
        borderColor: theme.palette.divider
      }
    }
  };
});

var SectionBase = function SectionBase(_ref) {
  var className = _ref.className,
      _ref$position = _ref.position,
      position = _ref$position === void 0 ? "relative" : _ref$position,
      _ref$collapsible = _ref.collapsible,
      collapsible = _ref$collapsible === void 0 ? false : _ref$collapsible,
      _ref$editable = _ref.editable,
      editable = _ref$editable === void 0 ? false : _ref$editable,
      expanded = _ref.expanded,
      children = _ref.children,
      props = _objectWithoutPropertiesLoose(_ref, ["className", "position", "collapsible", "editable", "expanded", "children"]);

  var elevation = collapsible && expanded && 4 || editable && expanded && 4 || 1;

  if (position === "absolute") {
    props.top = 0;
    props.right = 0;
    props.bottom = 0;
    props.left = 0;
  }

  var zIndex = elevation || 1;
  var classes = useStyles(_extends({
    position: position,
    zIndex: zIndex
  }, props));

  if (!elevation) {
    return /*#__PURE__*/_react.default.createElement("section", {
      className: className || classes.section,
      "aria-expanded": expanded
    }, children);
  }

  return /*#__PURE__*/_react.default.createElement(_Paper.default, {
    component: "section",
    className: className || classes.section,
    elevation: elevation,
    square: true,
    "data-elevation": elevation,
    "aria-expanded": expanded
  }, children);
};

SectionBase.defaultProps = {
  top: "auto",
  right: "auto",
  bottom: "auto",
  left: "auto"
};
var _default = SectionBase;
exports.default = _default;
module.exports = exports.default;
