// @kit-ui/admin/es/AppSearch/AppSearch.example.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import React, { useState } from "react";
import AppSearch from "./AppSearch";
export var AppSearchExample = function AppSearchExample(props) {
  // search
  var _useState = useState(props),
      search = _useState[0],
      setSearch = _useState[1];

  var _useState2 = useState(null),
      searchQ = _useState2[0],
      setSearchQ = _useState2[1];

  var searchChange = function searchChange(q) {
    setSearchQ(q);
  };

  var resetSearch = function resetSearch() {
    setSearchQ(null);

    if (!searchQ) {
      toggleSearch();
    }
  };

  var toggleSearch = function toggleSearch() {
    setSearch(_extends({}, search, {
      expanded: !search.expanded
    }));
  };

  var searchState = _extends({}, props, {
    onChange: searchChange,
    onToggle: toggleSearch,
    onReset: resetSearch,
    q: searchQ
  });

  return /*#__PURE__*/React.createElement(AppSearch, searchState);
};
export default AppSearchExample;
