// @kit-ui/schema/es/LocalizedString/LocalizedStringField.stories.js
import React from 'react';
import LocalizedStringField from "./LocalizedStringField";
import SchemaEditor from "../SchemaEditor/SchemaEditor";
export default {
  title: 'Fields/LocalizedStringField',
  component: LocalizedStringField,
  args: {
    formContext: {
      languages: ["no", "sv", "en"],
      defaultLocale: "no"
    },
    schema: {
      "type": "object",
      "properties": {
        "content": {
          "type": "localizedString"
        }
      }
    },
    uiSchema: {
      "ui:field": "pageEditor",
      "ui:fieldset": ["content"]
    }
  },
  argTypes: {}
};

var Template = function Template(args) {
  return /*#__PURE__*/React.createElement(SchemaEditor, args);
};

export var Default = Template.bind({});
Default.args = {};
export var TransformString = Template.bind({});
TransformString.args = {
  formData: {
    "content": "Transform string"
  }
};
export var CustomWidget = Template.bind({});
CustomWidget.args = {
  formData: {
    "locale:no": "Text"
  },
  uiSchema: {
    "ui:widget": "textarea",
    "ui:title": "Textarea",
    "ui:help": "Helper text"
  }
};
