// @kit-ui/admin/es/NavAction/NavActionDropdown.js
import React from 'react';
import ClickAwayListener from '@material-ui/core/ClickAwayListener';
import Popper from '@material-ui/core/Popper';
import Paper from '@material-ui/core/Paper';
import { makeStyles } from '@material-ui/core/styles';
var useStyles = makeStyles(function (theme) {
  return {
    dropdown: {
      minWidth: function minWidth(props) {
        return props.minWidth;
      }
    }
  };
});

var NavActionDropdown = function NavActionDropdown(_ref) {
  var className = _ref.className,
      _ref$minWidth = _ref.minWidth,
      minWidth = _ref$minWidth === void 0 ? 192 : _ref$minWidth,
      open = _ref.open,
      anchorEl = _ref.anchorEl,
      placement = _ref.placement,
      onClickAway = _ref.onClickAway,
      children = _ref.children;
  var classes = useStyles({
    minWidth: minWidth
  });
  return /*#__PURE__*/React.createElement(Popper, {
    open: open,
    style: {
      zIndex: 2000
    },
    role: undefined,
    transition: true,
    anchorEl: anchorEl,
    placement: placement
  }, /*#__PURE__*/React.createElement(ClickAwayListener, {
    onClickAway: onClickAway
  }, /*#__PURE__*/React.createElement(Paper, {
    className: className || classes.dropdown,
    elevation: 16,
    square: true
  }, children)));
};

export default NavActionDropdown;
