// @kit-ui/admin/es/NavAction/NavActionMenuLink.js
import React from 'react';
import ButtonBase from '@material-ui/core/ButtonBase';
import MenuItem from '@material-ui/core/MenuItem';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemText from '@material-ui/core/ListItemText';
import Icon from '@material-ui/core/Icon';
import { makeStyles } from '@material-ui/core/styles';
var useStyles = makeStyles(function (theme) {
  return {
    actionLink: {
      width: "100%",
      paddingLeft: theme.spacing(2),
      paddingRight: theme.spacing(2),
      "& .MuiTypography-root": {
        fontFamily: "Akkurat, sans-serif"
      },
      "&[role=group] + *": {
        borderTop: "1px solid",
        borderTopColor: theme.palette.divider,
        marginTop: theme.spacing(1.5),
        paddingTop: theme.spacing(1.5)
      },
      "& + &[role=group]": {
        borderTop: "1px solid",
        borderTopColor: theme.palette.divider,
        marginTop: theme.spacing(1.5),
        paddingTop: theme.spacing(1.5),
        "& + *": {
          borderTop: "1px solid",
          borderTopColor: theme.palette.divider,
          marginTop: theme.spacing(1.5),
          paddingTop: theme.spacing(1.5)
        }
      }
    },
    icon: {
      color: "rgba(0,0,0,.54)"
    },
    text: {
      fontFamily: "Akkurat, sans-serif !important",
      fontSize: "inherit",
      "& strong": {
        fontWeight: "bold"
      }
    }
  };
});

var ActionMenuLink = function ActionMenuLink(_ref) {
  var className = _ref.className,
      icon = _ref.icon,
      label = _ref.label,
      title = _ref.title,
      description = _ref.description,
      onClick = _ref.onClick;
  var classes = useStyles();

  if (!label && title) {
    label = title;
  }

  if (!label) {
    return null;
  }

  return /*#__PURE__*/React.createElement(MenuItem, {
    className: className || classes.actionLink,
    onClick: onClick
  }, icon && /*#__PURE__*/React.createElement(ListItemIcon, null, /*#__PURE__*/React.createElement(Icon, {
    className: classes.icon
  }, icon)), /*#__PURE__*/React.createElement(ListItemText, {
    className: classes.text,
    primary: label,
    secondary: description
  }));
};

export default ActionMenuLink;
