// @kit-ui/schema/es/ArrayField/__DefaultArrayItemLayout.js
import React from "react";
import { NavToolbar } from "@kit-ui/admin";
import { makeStyles } from '@material-ui/core/styles';
var useStyles = makeStyles(function (theme) {
  return {
    arrayItem: {
      width: "100%",
      display: "flex",
      flexDirection: "row",
      alignItems: "flexStart",
      borderTop: "1px solid",
      borderTopColor: theme.palette.divider,
      borderBottom: "1px solid",
      borderBottomColor: theme.palette.divider,
      padding: theme.spacing(1, 0),
      "& + $arrayItem": {
        borderTop: "none",
        borderColor: theme.palette.divider
      }
    },
    startAdornment: {
      height: theme.spacing(7),
      marginRight: theme.spacing(1),
      display: "flex",
      alignItems: "center",
      justifyContent: "center"
    },
    endAdornment: {
      height: theme.spacing(7),
      marginLeft: theme.spacing(1),
      display: "flex",
      alignItems: "center",
      justifyContent: "center"
    },
    content: {
      flexBasis: 0,
      flexGrow: 1
    },
    toolbar: {}
  };
});

var ArrayItemLayout = function ArrayItemLayout(_ref) {
  var children = _ref.children,
      startAdornment = _ref.startAdornment,
      toolbar = _ref.toolbar;
  var classes = useStyles();
  return /*#__PURE__*/React.createElement("div", {
    className: classes.arrayItem
  }, startAdornment && /*#__PURE__*/React.createElement("div", {
    className: classes.startAdornment
  }, startAdornment), /*#__PURE__*/React.createElement("div", {
    className: classes.content
  }, children), /*#__PURE__*/React.createElement("div", {
    className: classes.endAdornment
  }, /*#__PURE__*/React.createElement(NavToolbar, {
    toolbar: toolbar
  })));
};

export default ArrayItemLayout;
