// @kit-ui/schema/es/ArrayField/ArrayField.stories.js
import React from 'react';
import SchemaEditor from "../SchemaEditor/SchemaEditor";
export default {
  title: 'Fields/ArrayField',
  component: SchemaEditor,
  args: {
    schema: {},
    formData: {},
    formContext: {}
  },
  argTypes: {}
};

var Template = function Template(args) {
  return /*#__PURE__*/React.createElement(SchemaEditor, args);
};

export var ArrayOfStrings = Template.bind({});
ArrayOfStrings.args = {
  schema: {
    "type": "array",
    "items": {
      "type": "string"
    }
  },
  uiSchema: {
    "items": {
      "ui:layout": "module"
    }
  },
  formData: ["String 1", "String 2", "String 3"]
};
export var SortableStrings = Template.bind({});
SortableStrings.args = {
  schema: {
    "type": "array",
    "items": {
      "type": "string"
    }
  },
  uiSchema: {
    "ui:sortable": true
  },
  formData: ["String 1", "String 2", "String 3"]
};
export var ArrayOfObjects = Template.bind({});
ArrayOfObjects.args = {
  schema: {
    "type": "array",
    "items": {
      "type": "object",
      "properties": {
        "title": {
          "type": "string"
        },
        "description": {
          "type": "string"
        }
      }
    }
  },
  uiSchema: {},
  formData: [{
    "title": "Object 1"
  }, {
    "title": "Object 2"
  }, {
    "title": "Object 3"
  }]
};
export var SortableArrayOfObjects = Template.bind({});
SortableArrayOfObjects.args = {
  schema: {
    "type": "array",
    "items": {
      "type": "object",
      "properties": {
        "title": {
          "type": "string"
        },
        "description": {
          "type": "string"
        }
      }
    }
  },
  uiSchema: {
    "ui:sortable": true
  },
  formData: [{
    "title": "Object 1"
  }, {
    "title": "Object 2"
  }, {
    "title": "Object 3"
  }]
};
export var ModuleArray = Template.bind({});
ModuleArray.args = {
  schema: {
    "type": "array",
    "items": {
      "type": "object",
      "properties": {
        "title": {
          "type": "string"
        },
        "description": {
          "type": "string"
        }
      }
    }
  },
  uiSchema: {
    "items": {
      "ui:layout": "module",
      "ui:editable": true,
      "ui:hideable": true
    }
  },
  formData: [{
    "title": "Object 1"
  }, {
    "title": "Object 2"
  }, {
    "title": "Object 3"
  }]
};
export var SortableModuleArray = Template.bind({});
SortableModuleArray.args = {
  schema: {
    "type": "array",
    "items": {
      "type": "object",
      "properties": {
        "title": {
          "type": "string"
        },
        "description": {
          "type": "string"
        }
      }
    }
  },
  formData: [{
    "title": "Object 1"
  }, {
    "title": "Object 2"
  }, {
    "title": "Object 3"
  }],
  uiSchema: {
    "ui:sortable": true,
    "items": {
      "ui:layout": "module",
      "ui:editable": true,
      "ui:hideable": true
    }
  }
};
export var CustomButtons = Template.bind({});
CustomButtons.args = {
  schema: {
    "type": "array",
    "items": {
      "type": "object",
      "properties": {
        "title": {
          "type": "string"
        },
        "description": {
          "type": "string"
        }
      }
    }
  },
  formData: [{
    "title": "Object 1"
  }, {
    "title": "Object 2"
  }, {
    "title": "Object 3"
  }],
  uiSchema: {
    "ui:sortable": true,
    "items": {
      "ui:layout": "module",
      "ui:editable": true
    },
    "ui:buttons": ["dialog", {
      title: "Add item",
      item: {
        title: "Added by button"
      }
    }, {
      title: "Add"
    }]
  }
};
export var ArrayWithDialog = Template.bind({});
ArrayWithDialog.args = {
  schema: {
    "type": "object",
    "properties": {
      "array": {
        "type": "array",
        "items": {
          "type": "object",
          "properties": {
            "title": {
              "type": "string"
            },
            "description": {
              "type": "string"
            }
          }
        }
      }
    }
  },
  formData: {
    "array": [{
      "title": "Object 1"
    }, {
      "title": "Object 2"
    }, {
      "title": "Object 3"
    }]
  },
  uiSchema: {
    "ui:field": "pageEditor",
    "ui:fieldset": ["content"],
    "content": {
      "ui:field": "pageContent",
      "ui:fieldset": ["array"],
      "array": {
        "ui:sortable": true,
        "items": {
          "ui:layout": "module",
          "ui:editable": true
        },
        "ui:buttons": ["dialog", {
          title: "Add item",
          item: {
            title: "Added by button"
          }
        }, {
          title: "Add"
        }]
      }
    }
  }
};
