// @kit-ui/schema/es/ArrayField/SortableArrayFieldTemplate.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import React from 'react';
import { getArrayButtons, getUiToolbar, getUiOptions } from '../utils';
import { sortableContainer, sortableElement, sortableHandle } from 'react-sortable-hoc';
import IconButton from '@material-ui/core/IconButton';
import SortableIcon from '@material-ui/icons/DragHandle';
import arrayMove from "array-move";
var SortableHandle = sortableHandle(function () {
  return /*#__PURE__*/React.createElement(IconButton, null, /*#__PURE__*/React.createElement(SortableIcon, null));
});
var SortableItem = sortableElement(function (_ref) {
  var props = _ref.props,
      sortIndex = _ref.sortIndex,
      item = _objectWithoutPropertiesLoose(_ref, ["props", "sortIndex"]);

  item = _extends({}, item, {
    index: sortIndex
  });
  var _item = item,
      hasToolbar = _item.hasToolbar,
      hasMoveUp = _item.hasMoveUp,
      hasMoveDown = _item.hasMoveDown,
      hasRemove = _item.hasRemove;
  var _item2 = item,
      onAddIndexClick = _item2.onAddIndexClick,
      onDropIndexClick = _item2.onDropIndexClick,
      onReorderClick = _item2.onReorderClick;
  var itemUiSchema = item.children.props && item.children.props.uiSchema || {};

  var uiSchema = _extends({}, itemUiSchema, {
    "ui:removable": hasRemove
  });

  var itemProps = _extends({}, item.children.props, {
    formContext: props.formContext,
    hasToolbar: hasToolbar,
    hasMoveUp: false,
    hasMoveDown: false,
    hasRemove: hasRemove,
    onAddIndexClick: onAddIndexClick,
    onDropIndexClick: onDropIndexClick,
    onReorderClick: onReorderClick
  });

  var uiOptions = getUiOptions(uiSchema || {});
  var uiToolbar = getUiToolbar(_extends({}, itemProps, {
    "uiSchema": uiSchema
  }));
  var ArrayItemLayout = props.registry.fields.ArrayItemLayout;
  var arrayItem = React.cloneElement(item.children, _extends({}, itemProps, {
    uiSchema: _extends({}, uiSchema, {
      "ui:sortableHandle": /*#__PURE__*/React.createElement(SortableHandle, null),
      "ui:startAdornment": /*#__PURE__*/React.createElement(SortableHandle, null),
      "ui:toolbar": uiToolbar,
      "ui:layout": uiOptions.layout || ArrayItemLayout
    })
  }));
  return arrayItem;
});
var SortableList = sortableContainer(function (_ref2) {
  var _ref2$items = _ref2.items,
      items = _ref2$items === void 0 ? [] : _ref2$items,
      props = _objectWithoutPropertiesLoose(_ref2, ["items"]);

  var uiOptions = getUiOptions(props.uiSchema || {});
  var _props$registry$field = props.registry.fields,
      LayoutField = _props$registry$field.LayoutField,
      ArrayLayout = _props$registry$field.ArrayLayout;

  var uiSchema = _extends({}, props.uiSchema, {
    "ui:buttons": getArrayButtons(props),
    "ui:layout": uiOptions.layout || ArrayLayout
  });

  return /*#__PURE__*/React.createElement(LayoutField, _extends({}, props, {
    uiSchema: uiSchema
  }), items.map(function (item, index) {
    return /*#__PURE__*/React.createElement(SortableItem, _extends({
      key: "item-" + index,
      sortIndex: index
    }, item, {
      props: props
    }));
  }));
  return /*#__PURE__*/React.createElement("div", {
    key: "array-item-list-" + props.idSchema.$id
  }, items.map(function (item, index) {
    return /*#__PURE__*/React.createElement(SortableItem, _extends({
      key: "item-" + index,
      sortIndex: index
    }, item, {
      props: props
    }));
  }));
});

var SortableArrayFieldTemplate = function SortableArrayFieldTemplate(props) {
  var formData = props.formData,
      onChange = props.onChange;

  var _onSortEnd = function _onSortEnd(_ref3) {
    var oldIndex = _ref3.oldIndex,
        newIndex = _ref3.newIndex;
    var newFormData = arrayMove(formData, oldIndex, newIndex);
    onChange(newFormData);
  };

  return /*#__PURE__*/React.createElement(SortableList, _extends({}, props, {
    onSortEnd: _onSortEnd,
    helperClass: "sortable-helper",
    useDragHandle: true
  }));
};

export default SortableArrayFieldTemplate;
