// @kit-ui/schema/es/SelectWidget/SelectWidget.stories.js
import React from 'react';
import Schema from "../SchemaEditor/";
export default {
  title: 'Widgets/SelectWidget',
  component: Schema,
  args: {
    schema: {
      type: "string",
      enum: ["one", "two", "three"]
    },
    uiSchema: {}
  },
  argTypes: {}
};

var Template = function Template(args) {
  return /*#__PURE__*/React.createElement(Schema, args);
};

export var Default = Template.bind({});
Default.args = {};
export var WithLabel = Template.bind({});
WithLabel.args = {
  uiSchema: {
    "ui:title": "Title",
    "ui:placeholder": "Select something",
    "ui:help": "Helper text"
  }
};
export var OutlinedVariant = Template.bind({});
OutlinedVariant.args = {
  uiSchema: {
    "ui:title": "Title",
    "ui:variant": "outlined",
    "ui:help": "Helper text"
  }
};
export var StandardVariant = Template.bind({});
StandardVariant.args = {
  uiSchema: {
    "ui:title": "Title",
    "ui:variant": "standard",
    "ui:help": "Helper text"
  }
};
export var StandardNoLabel = Template.bind({});
StandardNoLabel.args = {
  uiSchema: {
    "ui:label": false,
    "ui:title": "Title",
    "ui:variant": "standard"
  }
};
