// @kit-ui/admin/es/NavUser/UserAppInfo.js
import React, { Component } from 'react';
import PropTypes from "prop-types";
import ButtonBase from '@material-ui/core/ButtonBase';
import Typography from '@material-ui/core/Typography';
import { makeStyles } from '@material-ui/core/styles';
var useStyles = makeStyles(function (theme) {
  return {
    root: {
      backgroundColor: "#eee",
      position: "relative",
      display: "flex",
      alignItems: "center",
      justifyContent: "flex-start",
      padding: "12px"
    },
    info: {
      flexBasis: "100%"
    },
    typography: {
      display: "block",
      fontFamily: "Akkurat, sans-serif",
      fontWeight: "normal",
      fontSize: "14px"
    },
    button: {
      fontFamily: "Akkurat, sans-serif",
      fontWeight: "bold",
      fontSize: "14px",
      padding: "6px 9px",
      border: "1px solid"
    }
  };
});

var UserAppInfo = function UserAppInfo(_ref) {
  var _ref$version = _ref.version,
      version = _ref$version === void 0 ? 1 : _ref$version,
      date = _ref.date,
      onLogout = _ref.onLogout;
  var classes = useStyles();
  return /*#__PURE__*/React.createElement("footer", {
    className: classes.root
  }, /*#__PURE__*/React.createElement("div", {
    className: classes.info
  }, /*#__PURE__*/React.createElement(Typography, {
    className: classes.typography
  }, /*#__PURE__*/React.createElement("strong", null, "Version"), " ", version), /*#__PURE__*/React.createElement(Typography, {
    className: classes.typography,
    datetime: date
  }, /*#__PURE__*/React.createElement("strong", null, "Date"), " ", date)), /*#__PURE__*/React.createElement(ButtonBase, {
    className: classes.button,
    onClick: onLogout
  }, "Logout"));
};

UserAppInfo.propTypes = process.env.NODE_ENV !== "production" ? {
  version: PropTypes.string,
  date: PropTypes.string,
  onLogout: PropTypes.func
} : {};
export default UserAppInfo;
