// @kit-ui/admin/es/NavUser/UserDropdown.js
import React from 'react';
import ClickAwayListener from '@material-ui/core/ClickAwayListener';
import Popper from '@material-ui/core/Popper';
import Paper from '@material-ui/core/Paper';
import { makeStyles } from '@material-ui/core/styles';
var useStyles = makeStyles(function (theme) {
  return {
    paper: {
      backgroundColor: function backgroundColor(props) {
        return props.backgroundColor;
      },
      color: "black",
      maxWidth: "320px",
      zIndex: 2
    },
    arrow: {
      position: "absolute",
      zIndex: 1,
      fontSize: "7px",
      width: "3em",
      height: "3em",
      "[x-placement*=top] &": {
        bottom: 0,
        width: 0,
        height: 0,
        borderLeft: "1em solid transparent",
        borderRight: "1em solid transparent",
        borderTop: "1em solid",
        borderTopColor: function borderTopColor(props) {
          return props.arrowColor;
        },
        marginBottom: "-1em"
      },
      "[x-placement*=bottom] &": {
        width: 0,
        height: 0,
        borderLeft: "1em solid transparent",
        borderRight: "1em solid transparent",
        borderBottom: "1em solid",
        borderBottomColor: function borderBottomColor(props) {
          return props.arrowColor;
        },
        marginTop: "-1em"
      },
      "[x-placement*=right] &": {
        left: 0,
        width: 0,
        height: 0,
        borderTop: "1em solid transparent",
        borderBottom: "1em solid transparent",
        borderRight: "1em solid",
        borderRightColor: function borderRightColor(props) {
          return props.arrowColor;
        },
        marginLeft: "-1em"
      },
      "[x-placement*=left] &": {
        right: 0,
        width: 0,
        height: 0,
        borderTop: "1em solid transparent",
        borderBottom: "1em solid transparent",
        borderLeft: "1em solid",
        marginRight: "-0.9em"
      }
    }
  };
});

var UserDropdown = function UserDropdown(_ref) {
  var children = _ref.children,
      expanded = _ref.expanded,
      onCollapse = _ref.onCollapse,
      anchorEl = _ref.anchorEl,
      _ref$placement = _ref.placement,
      placement = _ref$placement === void 0 ? "bottom-end" : _ref$placement;
  var classes = useStyles({
    backgroundColor: "white",
    arrowColor: "white"
  });
  return /*#__PURE__*/React.createElement(Popper, {
    open: expanded,
    style: {
      zIndex: 2000
    },
    role: undefined,
    transition: true,
    anchorEl: anchorEl,
    placement: placement,
    modifiers: {
      arrow: {
        enabled: true,
        element: "[data-popper-arrow]"
      }
    }
  }, /*#__PURE__*/React.createElement("div", {
    className: classes.arrow,
    "data-popper-arrow": true
  }), /*#__PURE__*/React.createElement(ClickAwayListener, {
    onClickAway: onCollapse
  }, /*#__PURE__*/React.createElement(Paper, {
    className: classes.paper,
    elevation: 24,
    square: true
  }, children)));
};

export default UserDropdown;
