// @kit-ui/admin/lib/Layout/LayoutBody.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _styles = require("@material-ui/core/styles");

var _LayoutGrid = _interopRequireDefault(require("./LayoutGrid"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var useStyles = (0, _styles.makeStyles)(function (theme) {
  return {
    root: {
      position: "relative",
      padding: function padding(props) {
        return theme.spacing(props.padding);
      },
      "& > * + *": {
        marginTop: function marginTop(props) {
          return theme.spacing(props.spacing);
        }
      }
    }
  };
});

var LayoutBody = function LayoutBody(_ref) {
  var _ref$grid = _ref.grid,
      grid = _ref$grid === void 0 ? false : _ref$grid,
      padding = _ref.padding,
      spacing = _ref.spacing,
      children = _ref.children;
  var classes = useStyles({
    padding: padding,
    spacing: spacing
  });

  if (!children) {
    return false;
  }

  return /*#__PURE__*/_react.default.createElement("div", {
    className: classes.root
  }, /*#__PURE__*/_react.default.createElement(_LayoutGrid.default, {
    grid: grid,
    padding: padding,
    spacing: spacing
  }, children));
};

var _default = LayoutBody;
exports.default = _default;
module.exports = exports.default;
