// @kit-ui/admin/lib/Layout/LayoutObject.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _styles = require("@material-ui/core/styles");

var _LayoutGrid = _interopRequireDefault(require("./LayoutGrid"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var useStyles = (0, _styles.makeStyles)(function (theme) {
  return {
    root: {
      position: "relative",
      width: "100%",
      padding: function padding(props) {
        return theme.spacing(props.padding);
      }
    }
  };
});

var LayoutObject = function LayoutObject(_ref) {
  var grid = _ref.grid,
      children = _ref.children,
      props = _objectWithoutPropertiesLoose(_ref, ["grid", "children"]);

  if (!grid) {
    return children;
  }

  return /*#__PURE__*/_react.default.createElement(_LayoutGrid.default, props, children);
};

LayoutObject.defaultProps = {
  header: true,
  footer: true,
  grid: undefined,
  padding: 0,
  spacing: 0
};
var _default = LayoutObject;
exports.default = _default;
module.exports = exports.default;
