// @kit-ui/admin/lib/Layout/LayoutTable.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _styles = require("@material-ui/core/styles");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var useStyles = (0, _styles.makeStyles)(function (theme) {
  return {
    root: {
      position: "relative",
      width: "100%",
      tableLayout: "fixed",
      padding: function padding(props) {
        return theme.spacing(props.padding);
      },
      fontFamily: "Akkurat, sans-serif",
      "& tr + tr": {
        borderTop: "1px solid",
        borderColor: theme.palette.divider
      },
      "& thead th": {
        position: "sticky",
        zIndex: 2,
        backgroundColor: "inherit",
        top: 0
      },
      "& th": {
        textAlign: "left"
      }
    }
  };
});

var LayoutTable = function LayoutTable(_ref) {
  var _ref$spacing = _ref.spacing,
      spacing = _ref$spacing === void 0 ? 1 : _ref$spacing,
      _ref$padding = _ref.padding,
      padding = _ref$padding === void 0 ? 0 : _ref$padding,
      children = _ref.children;
  var classes = useStyles({
    padding: padding
  });
  return /*#__PURE__*/_react.default.createElement("table", {
    className: classes.root,
    container: true,
    "data-spacing": spacing,
    "data-layout": "table"
  }, children);
};

var _default = LayoutTable;
exports.default = _default;
module.exports = exports.default;
