// @kit-ui/admin/lib/Layout/LayoutFieldset.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _styles = require("@material-ui/core/styles");

var _LayoutHeader = _interopRequireDefault(require("./LayoutHeader"));

var _LayoutGrid = _interopRequireDefault(require("./LayoutGrid"));

var _LayoutFooter = _interopRequireDefault(require("./LayoutFooter"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var useStyles = (0, _styles.makeStyles)(function (theme) {
  return {
    fieldset: {
      width: "100%",
      padding: function padding(props) {
        return theme.spacing(props.padding);
      }
    }
  };
});

var LayoutFieldset = function LayoutFieldset(_ref) {
  var name = _ref.name,
      _ref$grid = _ref.grid,
      grid = _ref$grid === void 0 ? false : _ref$grid,
      _ref$spacing = _ref.spacing,
      spacing = _ref$spacing === void 0 ? 1 : _ref$spacing,
      _ref$padding = _ref.padding,
      padding = _ref$padding === void 0 ? 0 : _ref$padding,
      toolbar = _ref.toolbar,
      children = _ref.children,
      props = _objectWithoutPropertiesLoose(_ref, ["name", "grid", "spacing", "padding", "toolbar", "children"]);

  var classes = useStyles({
    padding: padding,
    spacing: spacing
  });
  return /*#__PURE__*/_react.default.createElement("section", {
    className: classes.fieldset,
    "data-layout": "fieldset",
    "data-name": name
  }, /*#__PURE__*/_react.default.createElement(_LayoutHeader.default, props), /*#__PURE__*/_react.default.createElement(_LayoutGrid.default, {
    grid: grid,
    padding: padding,
    spacing: spacing
  }, children), /*#__PURE__*/_react.default.createElement(_LayoutFooter.default, props));
};

var _default = LayoutFieldset;
exports.default = _default;
module.exports = exports.default;
