// @kit-ui/admin/es/NavMenu/NavMenuItem.js
import React from 'react';
import IconButton from '@material-ui/core/IconButton';
import ToggleIcon from '@material-ui/icons/ArrowDropDown';
import Typography from '@material-ui/core/Typography';
import { makeStyles } from '@material-ui/core/styles';
var useStyles = makeStyles(function (theme) {
  return {
    root: {
      boxSizing: "border-box",
      position: "relative",
      display: "block",
      width: "100%",
      fontFamily: "Akkurat, sans-serif",
      fontSize: "14px",
      lineHeight: "24px",
      "&[role=group] + *": {
        borderTop: "1px solid",
        borderTopColor: theme.palette.divider,
        marginTop: theme.spacing(1.5),
        paddingTop: theme.spacing(1.5)
      },
      "& + &[role=group]": {
        borderTop: "1px solid",
        borderTopColor: theme.palette.divider,
        marginTop: theme.spacing(1.5),
        paddingTop: theme.spacing(1.5),
        "& + *": {
          borderTop: "1px solid",
          borderTopColor: theme.palette.divider,
          marginTop: theme.spacing(1.5),
          paddingTop: theme.spacing(1.5)
        }
      }
    },
    toggle: {
      position: "absolute",
      left: "-24px",
      padding: 0,
      margin: 0,
      opacity: 0.5
    },
    toggleCollapsed: {
      transition: ".125s ease-in-out",
      transform: "rotate(-90deg)"
    },
    toggleExpanded: {
      transition: ".125s ease-in-out",
      transform: "rotate(0deg)"
    },
    count: {
      position: "absolute",
      right: 0,
      fontFamily: "Akkurat, sans-serif",
      fontSize: "12px",
      fontStyle: "normal",
      lineHeight: "24px",
      opacity: 0.5
    }
  };
});

var MenuToggle = function MenuToggle(_ref) {
  var _ref$expanded = _ref.expanded,
      expanded = _ref$expanded === void 0 ? false : _ref$expanded,
      onClick = _ref.onClick;
  var classes = useStyles();
  var toggleIcon = expanded && classes.toggleExpanded || classes.toggleCollapsed;
  return /*#__PURE__*/React.createElement(IconButton, {
    className: classes.toggle,
    onClick: onClick,
    "aria-expanded": expanded,
    size: "small"
  }, /*#__PURE__*/React.createElement(ToggleIcon, {
    className: toggleIcon
  }));
};

var MenuItem = function MenuItem(_ref2) {
  var className = _ref2.className,
      _ref2$role = _ref2.role,
      role = _ref2$role === void 0 ? "treeitem" : _ref2$role,
      count = _ref2.count,
      _ref2$expanded = _ref2.expanded,
      expanded = _ref2$expanded === void 0 ? false : _ref2$expanded,
      onToggle = _ref2.onToggle,
      children = _ref2.children;
  var classes = useStyles();
  return /*#__PURE__*/React.createElement("li", {
    className: className || classes.root,
    role: role,
    "aria-expanded": expanded
  }, onToggle && /*#__PURE__*/React.createElement(MenuToggle, {
    expanded: expanded,
    onClick: onToggle
  }), children);
};

export default MenuItem;
