// @kit-ui/schema/lib/ObjectField/ObjectField.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _ObjectField = _interopRequireDefault(require("@rjsf/core/lib/components/fields/ObjectField"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var CustomObjectField = function CustomObjectField(props) {
  var LayoutField = props.registry.fields.LayoutField;

  var newRegistry = _extends({}, props.registry, {
    formContext: _extends({}, props.formContext, {
      parentSchema: props.schema,
      parentFormData: props.formData,
      onParentChange: props.onChange
    })
  });

  return /*#__PURE__*/_react.default.createElement(LayoutField, _extends({}, props, {
    registry: newRegistry,
    formContext: newRegistry.formContext
  }), /*#__PURE__*/_react.default.createElement(_ObjectField.default, _extends({}, props, {
    registry: newRegistry,
    formContext: newRegistry.formContext
  })));
};

var _default = CustomObjectField;
exports.default = _default;
module.exports = exports.default;
