// @kit-ui/admin/lib/AppSwitcher/AppsListItem.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireWildcard(require("react"));

var _ButtonBase = _interopRequireDefault(require("@material-ui/core/ButtonBase"));

var _IconButton = _interopRequireDefault(require("@material-ui/core/IconButton"));

var _Typography = _interopRequireDefault(require("@material-ui/core/Typography"));

var _styles = require("@material-ui/core/styles");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

var useStyles = (0, _styles.makeStyles)(function (theme) {
  var _root;

  return {
    root: (_root = {
      width: "100%",
      listStyle: "none",
      padding: 0,
      margin: 0,
      display: "flex",
      alignItems: "center",
      justifyContent: "left"
    }, _root["padding"] = theme.spacing(1.25), _root),
    figure: {
      width: function width(props) {
        return props.size;
      },
      height: function height(props) {
        return props.size;
      },
      "& svg, & img": {
        boxSizing: "inherit",
        position: "absolute",
        top: 0,
        right: 0,
        bottom: 0,
        left: 0,
        width: "100%",
        height: "100%",
        margin: "auto"
      }
    },
    label: {
      fontSize: "14px",
      fontWeight: "bold",
      margin: theme.spacing(1)
    }
  };
});

var AppsListItem = function AppsListItem(_ref) {
  var title = _ref.title,
      imageUrl = _ref.imageUrl,
      icon = _ref.icon,
      _ref$size = _ref.size,
      size = _ref$size === void 0 ? "large" : _ref$size,
      onClick = _ref.onClick;
  var sizes = {
    "small": 36,
    "medium": 48,
    "large": 64
  };
  size = sizes[size] || size || 64;
  var classes = useStyles({
    size: size
  });
  return /*#__PURE__*/_react.default.createElement(_ButtonBase.default, {
    className: classes.root,
    disableRipple: true,
    onClick: onClick
  }, /*#__PURE__*/_react.default.createElement(_IconButton.default, {
    className: classes.figure,
    component: "figure"
  }, !icon && imageUrl && /*#__PURE__*/_react.default.createElement("img", {
    className: classes.img,
    src: imageUrl
  }), icon), /*#__PURE__*/_react.default.createElement(_Typography.default, {
    className: classes.label,
    noWrap: true
  }, title));
};

var _default = AppsListItem;
exports.default = _default;
module.exports = exports.default;
