// @kit-ui/admin/lib/AppSwitcher/AppSwitcherMenu.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Paper = _interopRequireDefault(require("@material-ui/core/Paper"));

var _styles = require("@material-ui/core/styles");

var _AppsByUser = _interopRequireDefault(require("./AppsByUser"));

var _AppsByOwner = _interopRequireDefault(require("./AppsByOwner"));

var _OwnersList = _interopRequireDefault(require("./OwnersList"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var useStyles = (0, _styles.makeStyles)(function (theme) {
  return {
    menu: {
      backgroundColor: function backgroundColor(props) {
        return theme.palette.background.paper;
      },
      maxWidth: "320px",
      zIndex: 2,
      "& > * + *": {
        borderTop: "1px solid",
        borderColor: theme.palette.divider
      }
    }
  };
});

var AppSwitcherMenu = function AppSwitcherMenu(_ref) {
  var className = _ref.className,
      _ref$user = _ref.user,
      user = _ref$user === void 0 ? {} : _ref$user,
      _ref$owners = _ref.owners,
      owners = _ref$owners === void 0 ? [] : _ref$owners,
      _ref$currentOwner = _ref.currentOwner,
      currentOwner = _ref$currentOwner === void 0 ? undefined : _ref$currentOwner,
      onSelect = _ref.onSelect;
  var classes = useStyles();
  return /*#__PURE__*/_react.default.createElement(_Paper.default, {
    className: classes.menu,
    elevation: 24,
    square: true
  }, user.apps && /*#__PURE__*/_react.default.createElement(_AppsByUser.default, {
    apps: user.apps,
    onSelect: onSelect
  }), currentOwner && /*#__PURE__*/_react.default.createElement(_AppsByOwner.default, _extends({}, currentOwner, {
    onSelect: onSelect
  })), !currentOwner && owners && /*#__PURE__*/_react.default.createElement(_OwnersList.default, {
    owners: owners,
    onSelect: onSelect
  }));
};

AppSwitcherMenu.propTypes = process.env.NODE_ENV !== "production" ? {
  user: _propTypes.default.shape({
    uniqueId: _propTypes.default.string,
    apps: _propTypes.default.array
  }),
  owners: _propTypes.default.array,
  currentOwner: _propTypes.default.shape({
    uniqueId: _propTypes.default.string,
    title: _propTypes.default.string,
    apps: _propTypes.default.array
  }),
  onSelect: _propTypes.default.func
} : {};
var _default = AppSwitcherMenu;
exports.default = _default;
module.exports = exports.default;
