// @kit-ui/admin/lib/AppSwitcher/FilterOwnersUncontrolled.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireWildcard(require("react"));

var _Widget = require("../Widget");

var _OwnersList = _interopRequireDefault(require("./OwnersList"));

var _styles = require("@material-ui/core/styles");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var useStyles = (0, _styles.makeStyles)(function (theme) {
  return {
    root: {
      position: "absolute",
      width: "100%",
      top: 0,
      right: 0,
      bottom: 0,
      left: 0
    },
    search: {
      display: "flex",
      position: "relative",
      backgroundColor: "transparent",
      color: "black",
      height: "36px",
      overflow: "hidden",
      fontFamily: "Akkurat, sans-serif",
      margin: "12px"
    },
    scroller: {
      position: "absolute",
      zIndex: 1,
      top: 0,
      right: 0,
      bottom: 0,
      left: 0
    },
    results: {
      paddingTop: theme.spacing(7),
      display: "flex",
      flexDirection: "column",
      width: "100%"
    },
    nohits: {
      fontFamily: "Akkurat, sans-serif",
      fontSize: "14px",
      padding: theme.spacing(1.5)
    }
  };
});

var FilterOwnersUncontrolled = function FilterOwnersUncontrolled(_ref) {
  var className = _ref.className,
      search = _ref.search,
      q = _ref.q,
      _ref$owners = _ref.owners,
      owners = _ref$owners === void 0 ? [] : _ref$owners,
      onSearch = _ref.onSearch,
      onSelect = _ref.onSelect;
  var classes = useStyles();

  if (!owners.length) {
    return /*#__PURE__*/_react.default.createElement("div", {
      className: className || classes.root
    }, /*#__PURE__*/_react.default.createElement(_Widget.WidgetSearch, _extends({}, search, {
      value: q,
      onChange: onSearch
    })), /*#__PURE__*/_react.default.createElement(_Widget.WidgetScroller, {
      className: classes.scroller
    }, /*#__PURE__*/_react.default.createElement("div", {
      className: classes.results
    }, /*#__PURE__*/_react.default.createElement("p", {
      className: classes.nohits
    }, "Ingen treff"))));
  }

  var placeholder = search.placeholder && search.placeholder.replace('{hits}', owners.length);
  return /*#__PURE__*/_react.default.createElement("div", {
    className: className || classes.root
  }, /*#__PURE__*/_react.default.createElement(_Widget.WidgetSearch, _extends({}, search, {
    placeholder: placeholder,
    value: q,
    onChange: onSearch
  })), /*#__PURE__*/_react.default.createElement(_Widget.WidgetScroller, {
    className: classes.scroller
  }, /*#__PURE__*/_react.default.createElement(_OwnersList.default, {
    className: classes.results,
    owners: owners,
    onSelect: onSelect
  })));
};

FilterOwnersUncontrolled.defaultProps = {
  search: {
    placeholder: "Søk i {hits} eiere"
  }
};
var _default = FilterOwnersUncontrolled;
exports.default = _default;
module.exports = exports.default;
