// @kit-ui/admin/es/Media/ImageFocalpointReticle.js
import React, { useRef, useState } from 'react';
import { makeStyles } from '@material-ui/core/styles';
var useStyles = makeStyles(function (theme) {
  return {
    root: {
      position: "absolute",
      width: function width(props) {
        return theme.spacing(props.size);
      },
      height: function height(props) {
        return theme.spacing(props.size);
      },
      top: function top(props) {
        return props.top + "%" || "50%";
      },
      left: function left(props) {
        return props.left + "%" || "50%";
      },
      pointerEvents: "none"
    },
    point: {
      position: "absolute",
      display: "block",
      width: "100%",
      height: "100%",
      transform: "translate(-50%, -50%)",
      backgroundColor: "white",
      borderRadius: "50%",
      opacity: 0.5
    },
    crosshair: {
      position: "absolute",
      display: "block",
      width: "100%",
      height: "100%",
      transform: "translate(-50%, -50%)"
    },
    crosshairX: {
      display: "block",
      position: "absolute",
      left: "-15%",
      top: "50%",
      backgroundColor: "white",
      width: "130%",
      height: "1px"
    },
    crosshairY: {
      display: "block",
      position: "absolute",
      left: "50%",
      top: "-15%",
      backgroundColor: "white",
      width: "1px",
      height: "130%"
    }
  };
});

var ImageFocalpointReticle = function ImageFocalpointReticle(_ref) {
  var _ref$size = _ref.size,
      size = _ref$size === void 0 ? 8 : _ref$size,
      _ref$left = _ref.left,
      left = _ref$left === void 0 ? "50%" : _ref$left,
      _ref$top = _ref.top,
      top = _ref$top === void 0 ? "50%" : _ref$top;
  var classes = useStyles({
    left: left,
    top: top,
    size: size
  });
  return /*#__PURE__*/React.createElement("div", {
    className: classes.root
  }, /*#__PURE__*/React.createElement("div", {
    className: classes.point
  }), /*#__PURE__*/React.createElement("div", {
    className: classes.crosshair
  }, /*#__PURE__*/React.createElement("div", {
    className: classes.crosshairX
  }), /*#__PURE__*/React.createElement("div", {
    className: classes.crosshairY
  })));
};

export default ImageFocalpointReticle;
