// @kit-ui/admin/es/Media/ImageAnnotations.custom.js
import React, { useState } from 'react';
import ImageAnnotations from "./ImageAnnotations";
import ImagePreview from "./ImagePreview";
import { makeStyles } from '@material-ui/core/styles';
var useStyles = makeStyles(function (theme) {
  return {
    root: {
      position: "absolute",
      top: 0,
      right: 0,
      bottom: 0,
      left: 0
    },
    header: {
      position: "absolute",
      top: 0,
      right: 0,
      bottom: "50%",
      left: 0
    },
    footer: {
      position: "absolute",
      top: "50%",
      right: 0,
      bottom: 0,
      left: 0
    },
    image: {
      position: "absolute",
      top: 0,
      right: "50%",
      bottom: 0,
      left: 0
    },
    preview: {
      position: "absolute",
      top: 0,
      right: 0,
      bottom: 0,
      left: "50%"
    }
  };
});

var CustomMarker = function CustomMarker(_ref) {
  var index = _ref.index,
      top = _ref.top,
      left = _ref.left;
  return /*#__PURE__*/React.createElement("div", {
    style: {
      backgroundColor: "red",
      color: "white",
      padding: "1em",
      position: "absolute",
      top: top + "%",
      left: left + "%"
    }
  }, index);
};

var ExampleAnnotations = function ExampleAnnotations(_ref2) {
  var _ref2$imageUrl = _ref2.imageUrl,
      imageUrl = _ref2$imageUrl === void 0 ? undefined : _ref2$imageUrl,
      _ref2$imageAnnotation = _ref2.imageAnnotations,
      imageAnnotations = _ref2$imageAnnotation === void 0 ? [] : _ref2$imageAnnotation;
  var classes = useStyles();
  /*
   const [formData, setFormData] = useState(imageAnnotations)
   const onChange = (formData) => {
      setFormData(formData)
  }
   */

  return /*#__PURE__*/React.createElement(ImageAnnotations, {
    imageUrl: imageUrl,
    imageAnnotations: imageAnnotations,
    markerTemplate: CustomMarker
  });
};

export default ExampleAnnotations;
