// @kit-ui/admin/es/Media/MediaPlayPauseButton.js
import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import IconButton from '@material-ui/core/IconButton';
import PlayIcon from '@material-ui/icons/PlayArrow';
import PlayIconFilled from '@material-ui/icons/PlayCircleFilled';
import PauseIcon from '@material-ui/icons/Pause';
var useStyles = makeStyles(function (theme) {
  return {
    root: {
      display: "flex",
      alignItems: "center",
      justifyContent: "center"
    }
  };
});

var MediaButtonPlayPause = function MediaButtonPlayPause(_ref) {
  var className = _ref.className,
      _ref$filled = _ref.filled,
      filled = _ref$filled === void 0 ? true : _ref$filled,
      _ref$color = _ref.color,
      color = _ref$color === void 0 ? "primary" : _ref$color,
      paused = _ref.paused,
      onClick = _ref.onClick;
  var classes = useStyles({
    filled: filled
  });

  if (paused) {
    return /*#__PURE__*/React.createElement(IconButton, {
      color: color,
      className: className || classes.root,
      onClick: onClick
    }, /*#__PURE__*/React.createElement(PlayIcon, null));
  }

  return /*#__PURE__*/React.createElement(IconButton, {
    color: color,
    className: className || classes.root,
    onClick: onClick
  }, /*#__PURE__*/React.createElement(PauseIcon, null));
};

export default MediaButtonPlayPause;
