// @kit-ui/admin/es/Media/SourcePreview.js
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import React from 'react';
import PropTypes from "prop-types";
import { makeStyles } from '@material-ui/core/styles';
import { Iframe } from "@kit-ui/core";
var useStyles = makeStyles(function (theme) {
  return {
    root: {
      width: "100%",
      position: "absolute",
      top: 0,
      right: 0,
      bottom: 0,
      left: 0
    }
  };
});

var SketchfabPreview = function SketchfabPreview(_ref) {
  var sourceId = _ref.sourceId;
  var embedUrl = "https://sketchfab.com/models/" + sourceId + "/embed";
  return /*#__PURE__*/React.createElement(Iframe, {
    url: embedUrl
  });
};

var YouTubePreview = function YouTubePreview(_ref2) {
  var sourceId = _ref2.sourceId;
  var embedUrl = "https://www.youtube.com/embed/" + sourceId; // content.playerUrl 

  return /*#__PURE__*/React.createElement(Iframe, {
    url: embedUrl
  });
};

var VimeoPreview = function VimeoPreview(_ref3) {
  var sourceId = _ref3.sourceId;
  var embedUrl = "https://player.vimeo.com/video/" + sourceId;
  return /*#__PURE__*/React.createElement(Iframe, {
    url: embedUrl
  });
};

var templates = {
  "vimeo": VimeoPreview,
  "youtube": YouTubePreview,
  "sketchfab": SketchfabPreview
};
/** Preview template for media sources (vimeo, youtube, sketchfab) */

var SourcePreview = function SourcePreview(_ref4) {
  var className = _ref4.className,
      source = _ref4.source,
      sourceId = _ref4.sourceId,
      props = _objectWithoutPropertiesLoose(_ref4, ["className", "source", "sourceId"]);

  var classes = useStyles();
  var PreviewTemplate = source && templates[source];

  if (!PreviewTemplate) {
    return /*#__PURE__*/React.createElement("figure", {
      className: className || classes.root
    }, /*#__PURE__*/React.createElement("p", null, "No template for ", source));
  }

  return /*#__PURE__*/React.createElement("figure", {
    className: className || classes.root
  }, /*#__PURE__*/React.createElement(PreviewTemplate, {
    sourceId: sourceId
  }));
};

SourcePreview.defaultProps = {
  source: undefined,
  sourceId: undefined
};
SourcePreview.propTypes = process.env.NODE_ENV !== "production" ? {
  className: PropTypes.string,
  source: PropTypes.string,
  sourceId: PropTypes.string
} : {};
export default SourcePreview;
