// @kit-ui/admin/es/Media/VideoPlayer.js
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import React, { useRef, useState, useEffect } from 'react';
import PropTypes from "prop-types";
import { makeStyles } from '@material-ui/core/styles';
import VideoControls from "./VideoControls";
var useStyles = makeStyles(function (theme) {
  return {
    player: {
      backgroundColor: "black",
      color: "white",
      position: "absolute",
      top: 0,
      right: 0,
      bottom: 0,
      left: 0,
      display: "flex",
      alignItems: "center",
      justifyContent: "center"
    },
    video: {
      position: "absolute",
      top: 0,
      right: 0,
      bottom: 0,
      left: 0,
      display: "block",
      width: "100%",
      height: "auto",
      maxWidth: "100%",
      maxHeight: "100%",
      margin: "auto"
    },
    controls: {
      position: "absolute",
      top: 0,
      right: 0,
      bottom: 0,
      left: 0,
      display: "flex",
      flexDirection: "column",
      justifyContent: "space-between",
      margin: theme.spacing(1)
    }
  };
});

var VideoPlayer = function VideoPlayer(_ref) {
  var className = _ref.className,
      _ref$position = _ref.position,
      position = _ref$position === void 0 ? "absolute" : _ref$position,
      _ref$minHeight = _ref.minHeight,
      minHeight = _ref$minHeight === void 0 ? undefined : _ref$minHeight,
      _ref$maxWidth = _ref.maxWidth,
      maxWidth = _ref$maxWidth === void 0 ? undefined : _ref$maxWidth,
      _ref$maxHeight = _ref.maxHeight,
      maxHeight = _ref$maxHeight === void 0 ? undefined : _ref$maxHeight,
      _ref$format = _ref.format,
      format = _ref$format === void 0 ? undefined : _ref$format,
      title = _ref.title,
      description = _ref.description,
      _ref$imageUrl = _ref.imageUrl,
      imageUrl = _ref$imageUrl === void 0 ? undefined : _ref$imageUrl,
      _ref$videoUrl = _ref.videoUrl,
      videoUrl = _ref$videoUrl === void 0 ? undefined : _ref$videoUrl,
      _ref$videoType = _ref.videoType,
      videoType = _ref$videoType === void 0 ? undefined : _ref$videoType,
      _ref$videoSources = _ref.videoSources,
      videoSources = _ref$videoSources === void 0 ? [] : _ref$videoSources,
      _ref$resources = _ref.resources,
      resources = _ref$resources === void 0 ? {} : _ref$resources,
      props = _objectWithoutPropertiesLoose(_ref, ["className", "position", "minHeight", "maxWidth", "maxHeight", "format", "title", "description", "imageUrl", "videoUrl", "videoType", "videoSources", "resources"]);

  //    size = size === "large" && 32 || size === "medium" && 24 || size === "small" && 12
  var playerRef = useRef(null);

  var _useState = useState(null),
      buffered = _useState[0],
      setBuffered = _useState[1];

  var _useState2 = useState(0),
      currentTime = _useState2[0],
      setCurrentTime = _useState2[1];

  var _useState3 = useState(0),
      duration = _useState3[0],
      setDuration = _useState3[1];

  var _useState4 = useState(null),
      seeking = _useState4[0],
      setSeeking = _useState4[1];

  var _useState5 = useState(true),
      paused = _useState5[0],
      setPaused = _useState5[1];

  var _useState6 = useState(true),
      volume = _useState6[0],
      setVolume = _useState6[1];

  var _useState7 = useState(true),
      muted = _useState7[0],
      setMuted = _useState7[1];

  var _useState8 = useState(null),
      videoWidth = _useState8[0],
      setVideoWidth = _useState8[1];

  var _useState9 = useState(null),
      videoHeight = _useState9[0],
      setVideoHeight = _useState9[1];

  var _useState10 = useState(null),
      videoRatio = _useState10[0],
      setVideoRatio = _useState10[1]; //    const [maxWidth, setMaxWidth] = useState(null)


  var _useState11 = useState(null),
      scale = _useState11[0],
      setScale = _useState11[1];

  var _useState12 = useState(null),
      ratio = _useState12[0],
      setRatio = _useState12[1];

  var classes = useStyles();
  useEffect(function () {
    if (format && format.includes(":")) {
      var wh = format.split(":");
      setRatio(wh[1] / wh[0]);
    } else if (videoWidth && videoHeight) {
      setRatio(videoHeight / videoWidth);
    }
  }, [format, videoWidth, videoHeight]);
  useEffect(function () {
    if (!playerRef.current) {
      return;
    }

    playerRef.current.addEventListener('loadedmetadata', function (event) {
      var offsetWidth = event.target.offsetWidth;
      var videoWidth = event.target.videoWidth;
      var videoHeight = event.target.videoHeight;
      var scale = offsetWidth / videoWidth;
      var duration = event.target.duration;
      console.log('event', event);
      setVideoWidth(videoWidth);
      setVideoHeight(videoHeight);
      setVideoRatio(videoHeight / videoWidth);
      setBuffered(event.target.buffered);
      setDuration(event.target.duration);
      setCurrentTime(event.target.currentTime);
      setPaused(event.target.paused);
      setMuted(event.target.muted);
      setVolume(event.target.volume);
      props.onMetadata && props.onMetadata({
        videoWidth: videoWidth,
        videoHeight: videoHeight,
        duration: duration
      });
    });
    playerRef.current.addEventListener('canplay', function (event) {
      /*
      setBuffered(event.target.buffered)
      setDuration(event.target.duration)
      setCurrentTime(event.target.currentTime)
      setPaused(event.target.paused)
      setMuted(event.target.muted)
      setVolume(event.target.volume)
      */
    });
    playerRef.current.addEventListener('timeupdate', function (event) {
      setDuration(event.target.duration);
      setCurrentTime(event.target.currentTime);
    });
    playerRef.current.addEventListener('seeking', function (event) {
      setCurrentTime(event.target.currentTime);
      setSeeking(true);
    });
    playerRef.current.addEventListener('seeked', function (event) {
      setCurrentTime(event.target.currentTime);
      setSeeking(false);
    });
    playerRef.current.addEventListener('volumechange', function (event) {
      setVolume(event.target.volume);
    });
    playerRef.current.addEventListener('play', function (event) {
      setPaused(event.target.paused);
    });
    playerRef.current.addEventListener('pause', function (event) {
      setPaused(event.target.paused);
    });
  }, [playerRef.current]);

  var handlePlayPause = function handlePlayPause() {
    var paused = playerRef.current.paused;

    if (paused) {
      playerRef.current.play();
    } else {
      playerRef.current.pause();
    }

    setPaused(!paused);
  };

  var handleSeek = function handleSeek(value) {
    playerRef.current.currentTime = value;
  };

  var handleVolume = function handleVolume(value) {
    playerRef.current.volume = value;
  };

  var handleMute = function handleMute() {
    var mute = playerRef.current.mute;
    playerRef.current.mute = !mute;
    setMuted(!mute);
  };

  if (!videoSources.length && resources) {
    Object.values(resources).map(function (source) {
      var mimeType = source.mimeType,
          url = source.url;

      if (mimeType.startsWith('video') && url) {
        videoSources.push({
          src: url,
          type: mimeType
        });
      }
    });
  }

  if (!videoSources.length) {
    return /*#__PURE__*/React.createElement("p", null, "No video sources");
  }

  return /*#__PURE__*/React.createElement("div", {
    className: classes.player,
    "data-layout": "videoPlayer"
  }, /*#__PURE__*/React.createElement("video", {
    ref: playerRef,
    className: classes.video,
    poster: imageUrl
  }, videoSources.map(function (_ref2) {
    var src = _ref2.src,
        type = _ref2.type;
    return /*#__PURE__*/React.createElement("source", {
      src: src,
      type: type
    });
  }), "Your browser does not support the video tag."), duration && /*#__PURE__*/React.createElement(VideoControls, {
    className: classes.controls,
    title: title,
    description: description,
    paused: paused,
    onPlayPause: handlePlayPause,
    duration: duration,
    currentTime: currentTime,
    onSeek: handleSeek
  }));
};

VideoPlayer.propTypes = process.env.NODE_ENV !== "production" ? {
  className: PropTypes.string,
  untitled: PropTypes.string,
  title: PropTypes.string,
  description: PropTypes.string,

  /** Video sources */
  videoSources: PropTypes.array,

  /** Resources object */
  resources: PropTypes.node,

  /** Poster image */
  imageUrl: PropTypes.string
} : {};
export default VideoPlayer;
