// @kit-ui/admin/es/Module/ButtonSelect.js
import React from 'react';
import IconButton from '@material-ui/core/IconButton';
import SelectIcon from '@material-ui/icons/Check';
import { makeStyles } from '@material-ui/core/styles';
var useStyles = makeStyles(function (theme) {
  return {
    button: {
      "& svg": {
        width: theme.spacing(2.5),
        height: theme.spacing(2.5),
        position: "absolute",
        top: "50%",
        left: "50%",
        transform: "translate(-50%,-50%)",
        color: "currentColor"
      }
    },
    select: {
      width: theme.spacing(3),
      height: theme.spacing(3),
      borderRadius: "100%",
      border: "1px dotted",
      margin: 1,
      borderColor: "currentColor",
      opacity: .5,
      "&:hover": {
        opacity: 1
      }
    },
    selected: {
      display: "block",
      width: theme.spacing(3),
      height: theme.spacing(3),
      borderRadius: "100%",
      border: "2px solid",
      borderColor: theme.palette.primary.main,
      backgroundColor: theme.palette.primary.main,
      color: theme.palette.primary.contrastText,
      boxShadow: theme.shadows[2]
    }
  };
});

var ButtonSelectable = function ButtonSelectable(_ref) {
  var className = _ref.className,
      _ref$selected = _ref.selected,
      selected = _ref$selected === void 0 ? false : _ref$selected,
      onClick = _ref.onClick;
  var classes = useStyles();

  if (selected) {
    return /*#__PURE__*/React.createElement(IconButton, {
      className: className || classes.button,
      "aria-selected": selected,
      color: "inherit",
      "aria-label": "select",
      onClick: onClick
    }, /*#__PURE__*/React.createElement("div", {
      className: classes.selected
    }, /*#__PURE__*/React.createElement(SelectIcon, null)));
  }

  return /*#__PURE__*/React.createElement(IconButton, {
    className: className || classes.button,
    "aria-selected": selected,
    color: "inherit",
    "aria-label": "select",
    onClick: onClick
  }, /*#__PURE__*/React.createElement("div", {
    className: classes.select
  }, /*#__PURE__*/React.createElement(SelectIcon, null)));
};

export default ButtonSelectable;
