// @kit-ui/admin/es/Module/EditableModule.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import React from 'react';
import PropTypes from 'prop-types';
import ExpandedModule from "./ExpandedModule";
import ListModule from "./ListModule";
/** EditableModule, primarily used in Schema */

var EditableModule = function EditableModule(_ref) {
  var editable = _ref.editable,
      editing = _ref.editing,
      buttons = _ref.buttons,
      onSave = _ref.onSave,
      children = _ref.children,
      props = _objectWithoutPropertiesLoose(_ref, ["editable", "editing", "buttons", "onSave", "children"]);

  if (editable && editing) {
    if (!buttons) {
      buttons = [{
        title: "Save",
        onClick: onSave
      }];
    }

    return /*#__PURE__*/React.createElement(ExpandedModule, _extends({}, props, {
      buttons: buttons
    }), children);
  }

  return /*#__PURE__*/React.createElement(ListModule, _extends({}, props, {
    editable: editable
  }));
};

EditableModule.defaultProps = {
  padding: 2,
  editable: true
};
EditableModule.propTypes = process.env.NODE_ENV !== "production" ? {
  editable: PropTypes.bool,
  editing: PropTypes.bool
} : {};
export default EditableModule;
