// @kit-ui/admin/es/Module/EditableModule.stories.js
import React from 'react';
import EditableModule from './EditableModule';
import { ColorSettings } from "../";
export default {
  title: 'Module/Variant/EditableModule',
  component: EditableModule,
  args: {
    imageUrl: "https://dms-test.dimu.org/image/012uNXVpKSZo?mediaType=image/png",
    title: "Title",
    description: "description",
    metadata: ["25kb"],
    status: "publish",
    type: "document"
  },
  argTypes: {}
};

var Template = function Template(args) {
  return /*#__PURE__*/React.createElement(EditableModule, args, /*#__PURE__*/React.createElement("div", null, JSON.stringify(args, 0, 2)));
};

export var Default = Template.bind({});
Default.args = {};
export var Editing = Template.bind({});
Editing.args = {
  editing: true
};
export var EditingMenu = Template.bind({});
EditingMenu.args = {
  editing: true,
  menu: [{
    title: "Menu 1"
  }, {
    title: "Menu 2"
  }]
};
export var EditingButtons = Template.bind({});
EditingButtons.args = {
  editing: true,
  buttons: [{
    title: "Button 1"
  }, {
    title: "Button 2"
  }]
};
