// @kit-ui/admin/es/Module/ModuleHeader.js
import React, { Component } from 'react';
import { makeStyles } from '@material-ui/core/styles'; //import NavSettings from "../NavSettings/NavSettings"
//import NavToolbar from "../NavToolbar/NavToolbar"

import NavTabs from "../NavTabs/NavTabs";
import ModuleTitle from './ModuleTitle'; //import NavSettings from "../NavSettings/NavSettings"
//import NavToolbar from "../NavToolbar/NavToolbar"

var NavSettings = function NavSettings() {
  return /*#__PURE__*/React.createElement("p", null, "settings");
};

var NavToolbar = function NavToolbar() {
  return /*#__PURE__*/React.createElement("p", null, "toolbar");
};

var useStyles = makeStyles(function (theme) {
  return {
    header: {
      display: "flex",
      zIndex: 2,
      alignItems: "center",
      justifyContent: "flex-start",
      height: theme.spacing(6),
      color: "inherit",
      "& + *": {
        zIndex: 1,
        marginTop: theme.spacing(6)
      }
    },
    title: {
      flexGrow: 1,
      marginLeft: theme.spacing(1),
      marginRight: theme.spacing(1),
      color: "inherit",
      fontFamily: "Akkurat, sans-serif",
      fontSize: "16px",
      lineHeight: "24px",
      fontWeight: "bold"
    },
    tabs: {
      flexGrow: 1,
      marginLeft: theme.spacing(1),
      color: "inherit"
    },
    settings: {
      display: "flex"
    },
    toolbar: {
      display: "flex",
      color: "inherit"
    }
  };
});
/** Used in ExpandedModule, will show a title, menu or list children */

var ModuleHeader = function ModuleHeader(_ref) {
  var className = _ref.className,
      status = _ref.status,
      title = _ref.title,
      menu = _ref.menu,
      settings = _ref.settings,
      toolbar = _ref.toolbar,
      children = _ref.children;
  var classes = useStyles();

  if (children) {
    return /*#__PURE__*/React.createElement("header", {
      className: classes.header
    }, children);
  }

  return /*#__PURE__*/React.createElement("header", {
    className: className || classes.header
  }, menu && /*#__PURE__*/React.createElement(NavTabs, {
    className: classes.tabs,
    tabs: menu
  }), !menu && /*#__PURE__*/React.createElement(ModuleTitle, {
    className: classes.title,
    status: status,
    title: title
  }), settings && /*#__PURE__*/React.createElement(NavSettings, {
    className: classes.settings,
    settings: settings
  }), toolbar && /*#__PURE__*/React.createElement(NavToolbar, {
    className: classes.toolbar,
    toolbar: toolbar
  }));
};

export default ModuleHeader;
