// @kit-ui/admin/es/Module/ModuleMetadata.js
import React from 'react';
import PropTypes from "prop-types";
import Typography from '@material-ui/core/Typography';
import { makeStyles } from '@material-ui/core/styles';
var useStyles = makeStyles(function (theme) {
  return {
    root: {
      maxWidth: "100%",
      fontFamily: "Akkurat, sans-serif",
      fontSize: "14px",
      lineHeight: "24px",
      fontWeight: "normal",
      color: theme.palette.text.secondary
    }
  };
});
/** ModuleMetadata takes an array of strings and displays them on a single line */

var ModuleMetadata = function ModuleMetadata(_ref) {
  var className = _ref.className,
      _ref$metadata = _ref.metadata,
      metadata = _ref$metadata === void 0 ? [] : _ref$metadata;
  var classes = useStyles();

  if (!metadata) {
    return false;
  }

  metadata = metadata.join(' – ');
  return /*#__PURE__*/React.createElement(Typography, {
    className: className || classes.root,
    noWrap: true,
    component: "h3"
  }, metadata);
};

ModuleMetadata.propTypes = process.env.NODE_ENV !== "production" ? {
  component: PropTypes.string,
  metadata: PropTypes.array
} : {};
export default ModuleMetadata;
