// @kit-ui/admin/es/Module/ModuleDescription.js
import React from 'react';
import PropTypes from "prop-types";
import Typography from '@material-ui/core/Typography';
import { makeStyles } from '@material-ui/core/styles';
var useStyles = makeStyles(function (theme) {
  return {
    root: {
      maxWidth: "100%",
      fontFamily: "Akkurat, sans-serif",
      fontSize: "14px",
      lineHeight: "24px",
      fontWeight: "normal",
      color: function color(props) {
        return props.color || theme.palette.text.secondary;
      }
    }
  };
});

var ModuleDescription = function ModuleDescription(_ref) {
  var className = _ref.className,
      color = _ref.color,
      description = _ref.description,
      children = _ref.children;
  var classes = useStyles({
    color: color
  });

  if (!description && children) {
    description = children;
  }

  if (!description || typeof description === "undefined") {
    return false;
  }

  return /*#__PURE__*/React.createElement(Typography, {
    className: className || classes.root,
    noWrap: true,
    component: "h3"
  }, description);
};

ModuleDescription.propTypes = process.env.NODE_ENV !== "production" ? {
  description: PropTypes.string
} : {};
export default ModuleDescription;
