// @kit-ui/admin/es/Module/ModuleByline.js
import React from 'react';
import PropTypes from "prop-types";
import Typography from '@material-ui/core/Typography';
import ModuleDate from "./ModuleDate";
import { makeStyles } from '@material-ui/core/styles';
var useStyles = makeStyles(function (theme) {
  return {
    byline: {
      //        display: "flex",
      //        alignItems: "baseline",
      fontFamily: "Akkurat, sans-serif",
      fontSize: "14px",
      color: theme.palette.text.secondary
    },
    author: {
      display: "inline",
      fontSize: "12px",
      fontStyle: "normal",
      fontWeight: "normal"
    },
    datetime: {
      display: "inline",
      fontStyle: "normal",
      fontWeight: "normal"
    }
  };
});

var ModuleByline = function ModuleByline(_ref) {
  var author = _ref.author,
      datetime = _ref.datetime,
      _ref$locale = _ref.locale,
      locale = _ref$locale === void 0 ? "nb" : _ref$locale,
      _ref$format = _ref.format,
      format = _ref$format === void 0 ? "relative" : _ref$format;
  var classes = useStyles();

  if (!datetime) {
    return false;
  }

  return /*#__PURE__*/React.createElement(Typography, {
    className: classes.byline,
    noWrap: true,
    component: "h3"
  }, /*#__PURE__*/React.createElement(ModuleDate, {
    datetime: datetime,
    locale: locale,
    format: format,
    className: classes.datetime
  }), ' ', /*#__PURE__*/React.createElement("b", {
    className: classes.author
  }, "by ", author));
};

ModuleByline.propTypes = process.env.NODE_ENV !== "production" ? {
  author: PropTypes.string,
  datetime: PropTypes.string,
  locale: PropTypes.string,
  format: PropTypes.string
} : {};
export default ModuleByline;
