// @kit-ui/admin/es/Module/ModuleDate.js
import React from 'react';
import moment from 'moment';
import "moment/locale/nb";
import "moment/locale/nn";
import "moment/locale/sv";
import PropTypes from "prop-types";
import { makeStyles } from '@material-ui/core/styles';
var useStyles = makeStyles(function (theme) {
  return {
    root: {
      fontFamily: "Akkurat, sans-serif",
      fontSize: "14px",
      color: theme.palette.text.secondary
    }
  };
});
/** Module date, uses moment.js */

var ModuleDate = function ModuleDate(_ref) {
  var className = _ref.className,
      _ref$locale = _ref.locale,
      locale = _ref$locale === void 0 ? "nb" : _ref$locale,
      datetime = _ref.datetime,
      _ref$format = _ref.format,
      format = _ref$format === void 0 ? "relative" : _ref$format;
  var classes = useStyles();

  if (!datetime) {
    return false;
  } // format value


  moment.locale(locale);
  var date;

  if (format === "relative") {
    date = moment(datetime).fromNow();
  } else if (format) {
    date = moment(datetime).format(format);
  } else {
    date = moment(datetime).format();
  }

  return /*#__PURE__*/React.createElement("time", {
    dateTime: datetime,
    className: className || classes.root
  }, date);
};

ModuleDate.propTypes = process.env.NODE_ENV !== "production" ? {
  /** Locale string */
  locale: PropTypes.string,

  /** Datetime as YYYY-MM-DD */
  datetime: PropTypes.string,

  /** Moment.js format */
  format: PropTypes.string
} : {};
export default ModuleDate;
