// @kit-ui/admin/es/Module/TableModule.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import ModuleBase from "./ModuleBase";
import ModulePrefix from "./ModulePrefix";
import ModuleActionbar from "./ModuleActionbar";
import ModuleMedia from './ModuleMedia';
import ModuleTitle from './ModuleTitle';
import ModuleLabel from './ModuleLabel';
import ModuleDescription from './ModuleDescription';
import ModuleMetadata from './ModuleMetadata';
import ModuleStatus from './ModuleStatus';
import ModuleByline from './ModuleByline';
import { makeStyles } from '@material-ui/core/styles';
var useStyles = makeStyles(function (theme) {
  return {
    root: {
      userSelect: "none",
      "& > td": {
        minHeight: theme.spacing(8)
      },
      "&[data-status=trash]": {
        "& figure": {
          opacity: "0.5"
        },
        "& $content": {
          opacity: "0.5"
        }
      },
      "& h2": {
        fontSize: "16px",
        lineHeight: "24px"
      },
      "& $primaryAction": {
        opacity: .5
      },
      "& $actionbar": {
        opacity: .5
      },
      "&[aria-selected=true]": {
        backgroundColor: theme.palette.action.selected,
        "& $primaryAction": {
          opacity: 1,
          "& svg": {
            opacity: 1
          }
        }
      },
      "&:hover": {
        "& $primaryAction": {
          opacity: 1,
          "& svg": {
            opacity: 1
          }
        },
        "& $actionbar": {
          opacity: 1,
          "& button": {
            opacity: .5,
            "&:hover": {
              opacity: 1
            }
          }
        }
      },
      "&[data-role=button]": {
        "&:hover": {
          cursor: "pointer"
        },
        "& *": {//                pointerEvents: "none"
        },
        "& $primaryAction": {
          opacity: .5
          /*
          "& svg": {
              opacity: 0
          }*/

        },
        "&[aria-selected=true]": {}
      }
    },
    headerCell: {
      width: "50%",
      verticalAlign: "middle",
      padding: theme.spacing(1)
    },
    footerCell: {
      width: "50%",
      verticalAlign: "middle",
      padding: theme.spacing(1),
      position: "relative"
    },
    footer: {
      display: "flex",
      width: "100%"
    },
    actionCell: {
      width: theme.spacing(6),
      verticalAlign: "middle"
    },
    mediaCell: {
      width: theme.spacing(7),
      textAlign: "center",
      verticalAlign: "middle"
    },
    media: {
      position: "relative",
      width: theme.spacing(7),
      height: theme.spacing(7)
    },
    content: {
      width: "100%",
      display: "flex",
      flexGrow: 1,
      flexDirection: "column",
      alignItems: "flex-start",
      justifyContent: "flex-start"
    },
    metadata: {
      width: "100%",
      display: "flex",
      alignItems: "baseline",
      justifyContent: "flex-start",
      overflow: "hidden",
      textOverflow: "ellipsis",
      "& * + *": {
        marginLeft: theme.spacing(.5)
      },
      "& > *": {
        lineHeight: "24px"
      }
    },
    primaryAction: {
      display: "flex",
      alignItems: "center",
      justifyContent: "center"
    },
    actionbar: {
      position: "absolute",
      top: 0,
      right: 0,
      bottom: 0,
      left: "auto",
      display: "flex",
      alignItems: "center",
      justifyContent: "flex-start"
    }
  };
});

var TableModule = function TableModule(_ref) {
  var imageUrl = _ref.imageUrl,
      title = _ref.title,
      description = _ref.description,
      metadata = _ref.metadata,
      type = _ref.type,
      typeLabel = _ref.typeLabel,
      status = _ref.status,
      statusLabel = _ref.statusLabel,
      author = _ref.author,
      datetime = _ref.datetime,
      selected = _ref.selected,
      onClick = _ref.onClick,
      onEdit = _ref.onEdit,
      props = _objectWithoutPropertiesLoose(_ref, ["imageUrl", "title", "description", "metadata", "type", "typeLabel", "status", "statusLabel", "author", "datetime", "selected", "onClick", "onEdit"]);

  var classes = useStyles();
  var defaultImageUrl = "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mNk+P+/HgAFhAJ/wlseKgAAAABJRU5ErkJggg==";
  return /*#__PURE__*/React.createElement(ModuleBase, _extends({}, props, {
    component: "tr",
    className: classes.root,
    status: status,
    selected: selected,
    role: onClick && "button"
  }), /*#__PURE__*/React.createElement("td", {
    className: classes.actionCell,
    onClick: onClick
  }, /*#__PURE__*/React.createElement(ModulePrefix, _extends({
    className: classes.primaryAction,
    selected: selected
  }, props))), /*#__PURE__*/React.createElement("td", {
    className: classes.mediaCell,
    onClick: onClick
  }, /*#__PURE__*/React.createElement(ModuleMedia, {
    backgroundColor: "transparent",
    imageUrl: imageUrl || defaultImageUrl,
    className: classes.media
  })), /*#__PURE__*/React.createElement("td", {
    className: classes.headerCell,
    onClick: onClick
  }, /*#__PURE__*/React.createElement("header", {
    className: classes.content
  }, /*#__PURE__*/React.createElement(ModuleTitle, {
    status: status,
    title: title,
    onClick: !onClick && onEdit || undefined
  }), /*#__PURE__*/React.createElement("div", {
    className: classes.metadata
  }, /*#__PURE__*/React.createElement(ModuleLabel, {
    label: typeLabel || type
  }), /*#__PURE__*/React.createElement(ModuleDescription, {
    description: description
  }), /*#__PURE__*/React.createElement(ModuleMetadata, {
    metadata: metadata
  })))), /*#__PURE__*/React.createElement("td", {
    className: classes.footerCell,
    onClick: onClick
  }, /*#__PURE__*/React.createElement("div", {
    className: classes.content
  }, /*#__PURE__*/React.createElement(ModuleStatus, {
    component: "h2",
    status: statusLabel || status
  }), /*#__PURE__*/React.createElement("div", {
    className: classes.metadata
  }, /*#__PURE__*/React.createElement(ModuleByline, {
    author: author,
    datetime: datetime
  }))), /*#__PURE__*/React.createElement(ModuleActionbar, _extends({}, props, {
    className: classes.actionbar
  }))));
};

TableModule.propTypes = process.env.NODE_ENV !== "production" ? {
  imageUrl: PropTypes.string,
  title: PropTypes.string,
  description: PropTypes.string,
  type: PropTypes.string,
  typeLabel: PropTypes.string,
  status: PropTypes.string,
  statusLabel: PropTypes.string,
  author: PropTypes.string,
  datetime: PropTypes.string,
  onClick: PropTypes.func,
  onEdit: PropTypes.func
} : {};
export default TableModule;
