// @kit-ui/schema/lib/TextareaWidget/TextareaWidget.stories.js
"use strict";

exports.__esModule = true;
exports.StandardVariant = exports.OutlinedVariant = exports.WithLabel = exports.Default = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _SchemaEditor = _interopRequireDefault(require("../SchemaEditor/"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _default = {
  title: 'Widgets/TextareaWidget',
  component: _SchemaEditor.default,
  args: {
    schema: {
      type: "string"
    },
    uiSchema: {
      "ui:widget": "textarea"
    }
  },
  argTypes: {}
};
exports.default = _default;

var Template = function Template(args) {
  return /*#__PURE__*/_react.default.createElement(_SchemaEditor.default, args);
};

var Default = Template.bind({});
exports.Default = Default;
Default.args = {};
var WithLabel = Template.bind({});
exports.WithLabel = WithLabel;
WithLabel.args = {
  uiSchema: {
    "ui:title": "Title",
    "ui:help": "Write a title"
  }
};
var OutlinedVariant = Template.bind({});
exports.OutlinedVariant = OutlinedVariant;
OutlinedVariant.args = {
  uiSchema: {
    "ui:title": "Title",
    "ui:variant": "outlined",
    "ui:help": "Write a title"
  }
};
var StandardVariant = Template.bind({});
exports.StandardVariant = StandardVariant;
StandardVariant.args = {
  uiSchema: {
    "ui:title": "Title",
    "ui:variant": "standard",
    "ui:help": "Write a title"
  }
};
