// @kit-ui/admin/es/NavPath/PathDropdown.js
import React from 'react';
import ClickAwayListener from '@material-ui/core/ClickAwayListener';
import Popper from '@material-ui/core/Popper';
import Paper from '@material-ui/core/Paper'; //import "./SettingsDropdown.scss"

import { makeStyles } from '@material-ui/core/styles';
var useStyles = makeStyles(function (theme) {
  return {
    root: {
      marginTop: '6px',
      marginBottom: '6px'
    }
  };
});

var PathDropdown = function PathDropdown(_ref) {
  var children = _ref.children,
      expanded = _ref.expanded,
      onCollapse = _ref.onCollapse,
      anchorEl = _ref.anchorEl,
      _ref$placement = _ref.placement,
      placement = _ref$placement === void 0 ? "bottom-start" : _ref$placement;
  var classes = useStyles();
  return /*#__PURE__*/React.createElement(Popper, {
    open: expanded,
    style: {
      zIndex: 2000
    },
    role: undefined,
    transition: true,
    anchorEl: anchorEl,
    placement: placement
  }, /*#__PURE__*/React.createElement(ClickAwayListener, {
    onClickAway: onCollapse
  }, /*#__PURE__*/React.createElement(Paper, {
    elevation: 16,
    square: true,
    className: classes.root,
    "aria-expanded": expanded
  }, children)));
};

export default PathDropdown;
