// @kit-ui/admin/es/NavPath/PathButton.js
import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Icon from '@material-ui/core/Icon';
import ButtonBase from '@material-ui/core/ButtonBase';
var useStyles = makeStyles(function (theme) {
  return {
    root: {
      fontFamily: "Akkurat, sans-serif",
      fontWeight: "normal",
      fontSize: "18px",
      lineHeight: "24px"
    },
    icon: {},
    label: {}
  };
});

var PathButton = function PathButton(_ref) {
  var style = _ref.style,
      icon = _ref.icon,
      title = _ref.title,
      label = _ref.label,
      value = _ref.value,
      disabled = _ref.disabled,
      expanded = _ref.expanded,
      controls = _ref.controls,
      onToggle = _ref.onToggle,
      forwardedRef = _ref.forwardedRef;
  var classes = useStyles();
  return /*#__PURE__*/React.createElement(ButtonBase, {
    className: classes.root,
    style: style,
    ref: forwardedRef,
    disabled: disabled,
    "aria-controls": expanded ? controls : undefined,
    "aria-expanded": expanded ? 'true' : undefined,
    "aria-label": "options",
    "aria-haspopup": "menu",
    onClick: onToggle
  }, label || title || value);
};

export default PathButton;
