// @kit-ui/schema/es/Playground/DemoFrame.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import React from "react";
import { create } from "jss";
import { jssPreset, StylesProvider } from "@material-ui/core/styles";
import Frame from "react-frame-component";
/*
Adapted from https://github.com/mui-org/material-ui/blob/master/docs/src/modules/components/DemoSandboxed.js

The MIT License (MIT)

Copyright (c) 2014 Call-Em-All

Permission is hereby granted, free of charge, to any person obtaining a copy
of this software and associated documentation files (the "Software"), to deal
in the Software without restriction, including without limitation the rights
to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
copies of the Software, and to permit persons to whom the Software is
furnished to do so, subject to the following conditions:

The above copyright notice and this permission notice shall be included in all
copies or substantial portions of the Software.

THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
SOFTWARE.
*/

function DemoFrame(props) {
  var children = props.children,
      classes = props.classes,
      theme = props.theme,
      other = _objectWithoutPropertiesLoose(props, ["children", "classes", "theme"]);

  var _React$useState = React.useState({
    ready: false
  }),
      state = _React$useState[0],
      setState = _React$useState[1];

  var instanceRef = React.useRef();
  var handleRef = React.useCallback(function (ref) {
    instanceRef.current = {
      contentDocument: ref ? ref.node.contentDocument : null,
      contentWindow: ref ? ref.node.contentWindow : null
    };
  }, []);

  var onContentDidMount = function onContentDidMount() {
    setState({
      ready: true,
      jss: create({
        plugins: jssPreset().plugins,
        insertionPoint: instanceRef.current.contentWindow["demo-frame-jss"]
      }),
      sheetsManager: new Map(),
      container: instanceRef.current.contentDocument.body,
      window: function window() {
        return instanceRef.current.contentWindow;
      }
    });
  };

  if (theme === "fluent-ui") {
    // TODO: find a better way to render fluent-ui in an iframe, if we need to do so.
    var head = props.head;
    return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement("style", {
      dangerouslySetInnerHTML: {
        __html: "label { font-weight: normal; }"
      }
    }), head, children);
  }

  return /*#__PURE__*/React.createElement(Frame, _extends({
    ref: handleRef,
    contentDidMount: onContentDidMount
  }, other), /*#__PURE__*/React.createElement("div", {
    id: "demo-frame-jss"
  }), theme === "material-ui" ? state.ready ? /*#__PURE__*/React.createElement(StylesProvider, {
    jss: state.jss,
    sheetsManager: state.sheetsManager
  }, React.cloneElement(children, {
    container: state.container,
    window: state.window
  })) : null : children);
}

export default DemoFrame;
