// @kit-ui/admin/es/Media/ImageBase.js
import React, { useRef, useState, useEffect } from 'react';
import { withResizeDetector } from 'react-resize-detector';
import { makeStyles } from '@material-ui/core/styles';
import { getImageStyles } from "./utils/getImageStyles";
import SvgFilters from "../SvgFilters/SvgFilters";
var useStyles = makeStyles(function (theme) {
  return {
    imageBase: {
      width: "100%",
      position: "absolute",
      top: 0,
      right: 0,
      bottom: 0,
      left: 0,
      overflow: "hidden"
    },
    cropbox: {
      position: "absolute",
      overflow: function overflow(props) {
        return props.overflow;
      },
      top: 0,
      right: 0,
      bottom: 0,
      left: 0,
      margin: "auto"
    },
    image: {
      position: "absolute",
      opacity: 1,
      top: 0,
      right: 0,
      bottom: 0,
      left: 0,
      display: "block",
      width: "auto",
      height: "auto",
      maxWidth: "100%",
      maxHeight: "100%",
      margin: "auto"
    }
  };
});

var ImageBase = function ImageBase(_ref) {
  var className = _ref.className,
      style = _ref.style,
      imageUrl = _ref.imageUrl,
      imageCropdata = _ref.imageCropdata,
      imageFocalpoint = _ref.imageFocalpoint,
      imageFilters = _ref.imageFilters,
      _ref$objectFit = _ref.objectFit,
      objectFit = _ref$objectFit === void 0 ? "contain" : _ref$objectFit,
      _ref$position = _ref.position,
      position = _ref$position === void 0 ? "absolute" : _ref$position,
      _ref$overflow = _ref.overflow,
      overflow = _ref$overflow === void 0 ? "hidden" : _ref$overflow,
      children = _ref.children,
      width = _ref.width,
      height = _ref.height,
      onLoad = _ref.onLoad,
      onClick = _ref.onClick;
  var parentRef = useRef(null);
  var cropboxRef = useRef(null);
  var imageRef = useRef(null);
  var classes = useStyles({
    overflow: overflow
  });

  var _useState = useState(false),
      imageLoaded = _useState[0],
      setImageLoaded = _useState[1];

  var imageWidth = imageRef && imageRef.current && imageRef.current.offsetWidth;
  var imageHeight = imageRef && imageRef.current && imageRef.current.offsetHeight;
  var parentWidth = parentRef && parentRef.current && parentRef.current.offsetWidth;
  var parentHeight = parentRef && parentRef.current && parentRef.current.offsetHeight;
  var cropboxWidth = cropboxRef && cropboxRef.current && cropboxRef.current.offsetWidth;
  var cropboxHeight = cropboxRef && cropboxRef.current && cropboxRef.current.offsetHeight;

  var handleLoad = function handleLoad() {
    setImageLoaded(true);
    onLoad && onLoad({
      parentRef: parentRef,
      parentWidth: parentWidth,
      parentHeight: parentHeight,
      cropboxRef: cropboxRef,
      cropboxWidth: cropboxWidth,
      cropboxHeight: cropboxHeight,
      imageRef: imageRef,
      imageWidth: imageWidth,
      imageHeight: imageHeight
    });
  }; // get styles


  var _useState2 = useState(null),
      cropboxStyle = _useState2[0],
      setCropboxStyle = _useState2[1];

  var _useState3 = useState(null),
      imageStyle = _useState3[0],
      setImageStyle = _useState3[1];

  useEffect(function () {
    if (!imageRef.current || !parentRef.current) {
      return;
    }

    var cropStyles = getImageStyles({
      imageCropdata: imageCropdata,
      imageFocalpoint: imageFocalpoint,
      imageFilters: imageFilters,
      objectFit: objectFit,
      parentRef: parentRef,
      imageRef: imageRef
    });
    setCropboxStyle(cropStyles.cropbox);
    setImageStyle(cropStyles.image);
  }, [width, height, imageCropdata, imageFilters, imageLoaded]); // return cropped image with styles

  return /*#__PURE__*/React.createElement("div", {
    className: className || classes.imageBase,
    ref: parentRef
  }, /*#__PURE__*/React.createElement("div", {
    className: classes.cropbox,
    ref: cropboxRef,
    style: cropboxStyle,
    onClick: onClick
  }, /*#__PURE__*/React.createElement(SvgFilters, {
    cropboxRef: cropboxRef,
    imageRef: imageRef,
    imageFilters: imageFilters
  }, /*#__PURE__*/React.createElement("img", {
    src: imageUrl,
    className: classes.image,
    ref: imageRef,
    style: imageStyle,
    onLoad: handleLoad
  })), children));
};

ImageBase.defaultProps = {
  imageCropdata: {
    x: 0,
    y: 0,
    width: 100,
    height: 100
  }
};
export default withResizeDetector(ImageBase);
