// @kit-ui/admin/es/Media/ImageFilters.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import React, { useRef, useEffect, useState } from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Typography from '@material-ui/core/Typography';
import ButtonGroup from '@material-ui/core/ButtonGroup';
import ImageBase from "../Media/ImageBase";
import FilterButton from "../SvgFilters/SvgFilterButton";
import { BrightnessControls, ContrastControls, OpacityControls, BlurControls, SaturationControls, DuotoneControls, VignetteControls, ColorMatrixControls } from "../SvgFilters/";
import { getImagePointEvent } from "./utils/getImagePoint";
var controls = {
  "brightness": BrightnessControls,
  "contrast": ContrastControls,
  "vignette": VignetteControls,
  "blur": BlurControls,
  "opacity": OpacityControls,
  "saturation": SaturationControls,
  "duotone": DuotoneControls,
  "colorMatrix": ColorMatrixControls
};
var useStyles = makeStyles(function (theme) {
  return {
    root: {
      position: "absolute",
      top: 0,
      right: 0,
      bottom: 0,
      left: 0
    },
    preview: {
      position: "absolute",
      top: 0,
      right: 0,
      bottom: 0,
      left: 0,
      marginBottom: theme.spacing(8)
    },
    controls: {
      position: "absolute",
      top: "auto",
      right: 0,
      bottom: 0,
      left: 0,
      height: theme.spacing(4),
      margin: theme.spacing(1, 2),
      display: "flex",
      alignItems: "center",
      justifyContent: "center",
      "& + *": {
        marginTop: theme.spacing(1),
        marginBottom: theme.spacing(7)
      }
    },
    footer: {
      position: "absolute",
      top: "auto",
      right: 0,
      left: 0,
      bottom: 0,
      height: theme.spacing(8),
      overflowY: "visible"
    },
    header: {
      position: "absolute",
      top: "auto",
      right: 0,
      left: 0,
      bottom: theme.spacing(14),
      display: "flex",
      alignItems: "center",
      justifyContent: "center"
    },
    title: {
      fontFamily: "Akkurat, sans-serif",
      backgroundColor: theme.palette.primary.main,
      color: theme.palette.primary.contrastText,
      padding: theme.spacing(.5, 1.5)
    },
    buttongroup: {
      display: "flex",
      alignItems: "center",
      justifyContent: "flex-start",
      flexWrap: "nowrap",
      paddingLeft: "50%",
      paddingRight: theme.spacing(1),
      transition: ".125s ease-out",
      overflowX: "scroll",
      marginLeft: function marginLeft(props) {
        return theme.spacing(7 * props.index * -1) - theme.spacing(7 / 2);
      }
    },
    button: {
      marginLeft: theme.spacing(.5),
      marginRight: theme.spacing(.5)
    }
  };
});

var ImageFilters = function ImageFilters(_ref) {
  var className = _ref.className,
      imageUrl = _ref.imageUrl,
      _ref$imageCropdata = _ref.imageCropdata,
      imageCropdata = _ref$imageCropdata === void 0 ? {} : _ref$imageCropdata,
      _ref$imageFilters = _ref.imageFilters,
      imageFilters = _ref$imageFilters === void 0 ? {} : _ref$imageFilters,
      _ref$filterOptions = _ref.filterOptions,
      filterOptions = _ref$filterOptions === void 0 ? [] : _ref$filterOptions,
      onChange = _ref.onChange,
      props = _objectWithoutPropertiesLoose(_ref, ["className", "imageUrl", "imageCropdata", "imageFilters", "filterOptions", "onChange"]);

  var _useState = useState(0),
      index = _useState[0],
      setIndex = _useState[1];

  var _useState2 = useState(filterOptions[0].name),
      selected = _useState2[0],
      setSelected = _useState2[1];

  var classes = useStyles({
    index: index
  });

  var currentFilter = selected && _extends({}, imageFilters[selected], {
    name: selected
  });

  var changeFocalpoint = function changeFocalpoint(event) {
    var _extends2;

    var focalpoint;

    if (event.shiftKey) {
      focalpoint = {
        x: 0,
        y: 0,
        top: 50,
        left: 50
      };
    } else {
      focalpoint = getImagePointEvent(event);
    }

    currentFilter && currentFilter.focalpoint && onChange && onChange(_extends({}, imageFilters, (_extends2 = {}, _extends2[selected] = _extends({}, imageFilters[selected], focalpoint), _extends2)));
  };

  var selectFilter = function selectFilter(_ref2, event) {
    var name = _ref2.name,
        index = _ref2.index;
    setSelected(name);
    setIndex(index);

    if (event.shiftKey) {
      changeFilter({
        name: name,
        undefined: undefined
      });
    }
  };

  var changeFilter = function changeFilter(_ref3) {
    var _extends3;

    var name = _ref3.name,
        filter = _ref3.filter;
    onChange && onChange(_extends({}, imageFilters, (_extends3 = {}, _extends3[name] = filter, _extends3)));
  };

  var FilterNav = function FilterNav() {
    return /*#__PURE__*/React.createElement(ButtonGroup, {
      className: classes.buttongroup
    }, filterOptions.map(function (option, index) {
      var name = option.name;

      var filter = _extends({}, imageFilters[name], {
        index: index,
        name: name
      });

      return /*#__PURE__*/React.createElement(FilterButton, _extends({}, option, filter, {
        selected: selected === name,
        key: index,
        index: index,
        name: name,
        onClick: function onClick(event) {
          return selectFilter(filter, event);
        }
      }));
    }));
  };

  var FilterControls = function FilterControls(_ref4) {
    var name = _ref4.name;
    var ControlsTemplate = controls && controls[name];
    var filter = imageFilters && imageFilters[name];

    if (ControlsTemplate) {
      return /*#__PURE__*/React.createElement(ControlsTemplate, _extends({}, filter, {
        className: classes.controls,
        onChange: function onChange(filter) {
          return changeFilter({
            name: name,
            filter: filter
          });
        }
      }));
    }

    return false;
  };

  return /*#__PURE__*/React.createElement("div", {
    className: className || classes.root
  }, /*#__PURE__*/React.createElement("div", {
    className: classes.preview
  }, /*#__PURE__*/React.createElement(FilterControls, {
    name: selected
  }), /*#__PURE__*/React.createElement(ImageBase, {
    imageCropdata: imageCropdata,
    imageFilters: imageFilters,
    imageUrl: imageUrl,
    onClick: changeFocalpoint
  })), /*#__PURE__*/React.createElement("header", {
    className: classes.header
  }, /*#__PURE__*/React.createElement(Typography, {
    className: classes.title
  }, currentFilter && currentFilter.name)), /*#__PURE__*/React.createElement("footer", {
    className: classes.footer
  }, /*#__PURE__*/React.createElement(FilterNav, null)));
};

ImageFilters.defaultProps = {
  imageFilters: {
    brightness: {},
    contrast: {},
    vignette: {}
  },
  filterOptions: [{
    name: "brightness",
    icon: "wb_sunny"
  }, {
    name: "contrast",
    icon: "tonality"
  }, {
    name: "blur",
    icon: "blur_on"
  }, {
    name: "saturation",
    icon: "filter_tilt_shift"
  }, {
    name: "opacity",
    icon: "opacity"
  }, {
    name: "vignette",
    icon: "filter_vintage"
  }]
};
export default ImageFilters;
