// @kit-ui/admin/es/Media/ImageFocalpoint.example.js
import React, { useState } from 'react';
import ImageFocalpoint from "./ImageFocalpointBase";
import { makeStyles } from '@material-ui/core/styles';
var useStyles = makeStyles(function (theme) {
  return {
    root: {
      position: "absolute",
      top: 0,
      right: 0,
      bottom: 0,
      left: 0
    },
    data: {
      pointerEvents: "none",
      position: "absolute",
      top: 0,
      right: 0,
      bottom: 0,
      left: "50%",
      textAlign: "right"
    }
  };
});

var ExampleFocalpoint = function ExampleFocalpoint(_ref) {
  var _ref$imageUrl = _ref.imageUrl,
      imageUrl = _ref$imageUrl === void 0 ? undefined : _ref$imageUrl;
  var classes = useStyles();

  var _useState = useState(undefined),
      formData = _useState[0],
      setFormData = _useState[1];

  var _onChange = function onChange(formData) {
    setFormData(formData);
  };

  return /*#__PURE__*/React.createElement("div", {
    className: classes.root
  }, /*#__PURE__*/React.createElement(ImageFocalpoint, {
    imageUrl: imageUrl,
    imageFocalpoint: formData,
    onChange: function onChange(formData) {
      return _onChange(formData);
    }
  }), /*#__PURE__*/React.createElement("data", {
    className: classes.data
  }, JSON.stringify(formData)));
};

export default ExampleFocalpoint;
