// @kit-ui/admin/es/Media/ImageAnnotationsMarker.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import React, { useState, useEffect } from 'react'; // import "./ImageAnnotationsMarker.scss"

import Draggable, { DraggableCore } from 'react-draggable';
import { Transform } from 'stream';
import { makeStyles } from '@material-ui/core/styles';
import { getImagePointEvent } from "./utils/getImagePoint";
var useStyles = makeStyles(function (theme) {
  var _marker;

  return {
    root: {
      position: "absolute",
      zIndex: function zIndex(props) {
        return props.zIndex;
      },
      top: function top(props) {
        return props.top + "%" || "50%";
      },
      left: function left(props) {
        return props.left + "%" || "50%";
      }
    },
    draghandle: {
      position: "absolute",
      //        top: 0,
      //        left: 0,
      display: "block",
      backgroundColor: "blue",
      width: theme.spacing(4) + "px",
      height: theme.spacing(4) + "px",
      transform: "translate(-50%, -50%)",
      borderRadius: "50%",
      opacity: 0.5,
      "&:hover": {
        cursor: function cursor(props) {
          return props.cursor || "move";
        }
      }
    },
    marker: (_marker = {
      width: function width(props) {
        return theme.spacing(props.size);
      },
      height: function height(props) {
        return theme.spacing(props.size);
      },
      top: "0",
      left: "0",
      position: "absolute",
      display: "block",
      transform: "translate(-50%, -50%)",
      backgroundColor: "white",
      borderRadius: "50%",
      opacity: 0.5
    }, _marker["display"] = "flex", _marker.alignItems = "center", _marker.justifyContent = "center", _marker.pointerEvents = "none", _marker),
    label: {
      fontFamily: "Akkurat Mono, monspace",
      fontSize: "14px",
      color: "black"
    }
  };
});

var ImageAnnotationsMarker = function ImageAnnotationsMarker(_ref) {
  var index = _ref.index,
      _ref$top = _ref.top,
      top = _ref$top === void 0 ? 50 : _ref$top,
      _ref$left = _ref.left,
      left = _ref$left === void 0 ? 50 : _ref$left,
      _ref$size = _ref.size,
      size = _ref$size === void 0 ? 8 : _ref$size,
      onStart = _ref.onStart,
      onDrag = _ref.onDrag,
      onStop = _ref.onStop;

  var _useState = useState({
    top: top,
    left: left
  }),
      coords = _useState[0],
      setCoords = _useState[1];

  var _useState2 = useState("grab"),
      cursor = _useState2[0],
      setCursor = _useState2[1];

  var handleStart = function handleStart(event) {
    setCursor('grabbing');
    onStart && onStart(event);
  };

  var handleDrag = function handleDrag(event) {
    var marker = getImagePointEvent(event);
    onDrag && onDrag(event);
    setCoords(marker);
  };

  var handleStop = function handleStop(event) {
    setCursor('grab');
    onStop && onStop(event);
  };

  var classes = useStyles(_extends({
    size: size,
    zIndex: index,
    cursor: cursor
  }, coords));
  return /*#__PURE__*/React.createElement(DraggableCore, {
    bounds: "parent",
    handle: ".draghandle",
    onStart: handleStart,
    onDrag: handleDrag,
    onStop: handleStop
  }, /*#__PURE__*/React.createElement("div", {
    className: classes.root
  }, /*#__PURE__*/React.createElement("div", {
    className: classes.draghandle + " draghandle"
  }), /*#__PURE__*/React.createElement("div", {
    className: classes.marker
  }, /*#__PURE__*/React.createElement("div", {
    className: classes.label
  }, index + 1))));
};

export default ImageAnnotationsMarker;
