// @kit-ui/admin/es/Media/ImageAnnotations.example.js
import React, { useState } from 'react';
import ImageAnnotations from "./ImageAnnotations";
import ImagePreview from "./ImagePreview";
import { makeStyles } from '@material-ui/core/styles';
var useStyles = makeStyles(function (theme) {
  return {
    root: {
      position: "absolute",
      top: 0,
      right: 0,
      bottom: 0,
      left: 0
    },
    header: {
      position: "absolute",
      top: 0,
      right: 0,
      bottom: "50%",
      left: 0
    },
    footer: {
      position: "absolute",
      top: "50%",
      right: 0,
      bottom: 0,
      left: 0
    },
    image: {
      position: "absolute",
      top: 0,
      right: "50%",
      bottom: 0,
      left: 0
    },
    preview: {
      position: "absolute",
      top: 0,
      right: 0,
      bottom: 0,
      left: "50%"
    }
  };
});

var ExampleAnnotations = function ExampleAnnotations(_ref) {
  var _ref$imageUrl = _ref.imageUrl,
      imageUrl = _ref$imageUrl === void 0 ? undefined : _ref$imageUrl,
      _ref$imageAnnotations = _ref.imageAnnotations,
      imageAnnotations = _ref$imageAnnotations === void 0 ? [] : _ref$imageAnnotations;
  var classes = useStyles();

  var _useState = useState(imageAnnotations),
      formData = _useState[0],
      setFormData = _useState[1];

  var _onChange = function onChange(formData) {
    setFormData(formData);
  };

  return /*#__PURE__*/React.createElement("div", {
    className: classes.root
  }, /*#__PURE__*/React.createElement("header", {
    className: classes.header
  }, /*#__PURE__*/React.createElement("div", {
    className: classes.image
  }, /*#__PURE__*/React.createElement(ImageAnnotations, {
    imageUrl: imageUrl,
    imageAnnotations: formData,
    onChange: function onChange(formData) {
      return _onChange(formData);
    }
  })), /*#__PURE__*/React.createElement("div", {
    className: classes.preview
  }, /*#__PURE__*/React.createElement(ImagePreview, {
    imageUrl: imageUrl,
    imageAnnotations: formData
  }))), /*#__PURE__*/React.createElement("footer", {
    className: classes.footer
  }, JSON.stringify(formData)));
};

export default ExampleAnnotations;
