// @kit-ui/admin/es/Media/ImagePreview.js
import React, { useRef, useEffect, useState } from 'react';
import { withResizeDetector } from 'react-resize-detector';
import { makeStyles } from '@material-ui/core/styles';
import ImageBase from "./ImageBase";
var useStyles = makeStyles(function (theme) {
  return {
    figure: {
      position: "absolute",
      top: 0,
      right: 0,
      bottom: 0,
      left: 0,
      margin: 0
    }
  };
});

var ImagePreview = function ImagePreview(_ref) {
  var className = _ref.className,
      _ref$overflow = _ref.overflow,
      overflow = _ref$overflow === void 0 ? "hidden" : _ref$overflow,
      width = _ref.width,
      height = _ref.height,
      imageUrl = _ref.imageUrl,
      _ref$imageCropdata = _ref.imageCropdata,
      imageCropdata = _ref$imageCropdata === void 0 ? undefined : _ref$imageCropdata,
      _ref$imageFilters = _ref.imageFilters,
      imageFilters = _ref$imageFilters === void 0 ? undefined : _ref$imageFilters;
  var classes = useStyles();
  return /*#__PURE__*/React.createElement("figure", {
    className: className || classes.figure
  }, /*#__PURE__*/React.createElement(ImageBase, {
    imageUrl: imageUrl,
    imageCropdata: imageCropdata,
    imageFilters: imageFilters
  }));
};

export default withResizeDetector(ImagePreview);
