// @kit-ui/admin/es/Media/ImageAnnotationsMarkerDefault.js
import React, { useRef, useState, useEffect } from 'react'; // import "./ImageAnnotationsMarker.scss"

import Draggable, { DraggableCore } from 'react-draggable';
import { makeStyles } from '@material-ui/core/styles';
import { getImagePointEvent } from "./utils/getImagePoint";
var useStyles = makeStyles(function (theme) {
  var _marker;

  return {
    root: {
      position: "absolute",
      zIndex: function zIndex(props) {
        return props.zIndex;
      } //        top: props => { return props.top + "%" || "50%"},
      //        left: props => { return props.left + "%" || "50%"},

    },
    draghandle: {
      position: "absolute",
      display: "block",
      width: function width(props) {
        return theme.spacing(props.size);
      },
      height: function height(props) {
        return theme.spacing(props.size);
      },
      transform: "translate(-50%, -50%)",
      borderRadius: "50%",
      "&:hover": {
        cursor: function cursor(props) {
          return props.cursor || "move";
        }
      }
    },
    marker: (_marker = {
      width: function width(props) {
        return theme.spacing(props.size);
      },
      height: function height(props) {
        return theme.spacing(props.size);
      },
      backgroundColor: theme.palette.primary.main,
      color: theme.palette.primary.contrastText,
      top: "0",
      left: "0",
      position: "absolute",
      display: "block",
      transform: "translate(-50%, -50%)",
      borderRadius: "50%"
    }, _marker["display"] = "flex", _marker.alignItems = "center", _marker.justifyContent = "center", _marker.pointerEvents = "none", _marker),
    label: {
      fontFamily: "Akkurat, sans-serif",
      fontSize: "14px"
    }
  };
});

var ImageAnnotationsMarker = function ImageAnnotationsMarker(_ref) {
  var index = _ref.index,
      position = _ref.position,
      _ref$size = _ref.size,
      size = _ref$size === void 0 ? 4 : _ref$size,
      onStart = _ref.onStart,
      onDrag = _ref.onDrag,
      onStop = _ref.onStop;

  var _useState = useState("grab"),
      cursor = _useState[0],
      setCursor = _useState[1];

  var handleStart = function handleStart(event, data) {
    setCursor('grabbing');
    onStart && onStart(event, data);
  };

  var handleDrag = function handleDrag(event, data) {
    onDrag && onDrag(event, data);
  };

  var handleStop = function handleStop(event, data) {
    setCursor('grab');
    onStop && onStop(event, data);
  };

  var classes = useStyles({
    size: size,
    zIndex: index,
    cursor: cursor
  });

  if (!position) {
    return false;
  }

  return /*#__PURE__*/React.createElement(Draggable, {
    bounds: "parent",
    handle: ".draghandle",
    position: position,
    onStart: handleStart,
    onDrag: handleDrag,
    onStop: handleStop
  }, /*#__PURE__*/React.createElement("div", {
    className: classes.root
  }, /*#__PURE__*/React.createElement("div", {
    className: classes.draghandle + " draghandle"
  }), /*#__PURE__*/React.createElement("div", {
    className: classes.marker
  }, /*#__PURE__*/React.createElement("div", {
    className: classes.label
  }, index + 1))));
};

export default ImageAnnotationsMarker;
