// @kit-ui/admin/es/Media/MediaMetadata.js
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
var useStyles = makeStyles(function (theme) {
  return {
    metadata: {
      listStyle: "none",
      display: "flex",
      flexDirection: "column",
      fontFamily: "Akkurat, sans-serif",
      fontSize: "16px",
      lineHeight: "24px",
      "& > li": {
        padding: theme.spacing(1, 0)
      },
      "& > * + *": {
        borderTop: "1px solid",
        borderColor: theme.palette.divider
      },
      "& b": {
        fontWeight: "bold"
      }
    }
  };
});

var MediaMetadata = function MediaMetadata(_ref) {
  var filename = _ref.filename,
      mimeType = _ref.mimeType,
      uniqueId = _ref.uniqueId,
      sourceId = _ref.sourceId,
      dimensions = _ref.dimensions,
      aspectRatio = _ref.aspectRatio,
      duration = _ref.duration,
      size = _ref.size,
      props = _objectWithoutPropertiesLoose(_ref, ["filename", "mimeType", "uniqueId", "sourceId", "dimensions", "aspectRatio", "duration", "size"]);

  var classes = useStyles();
  return /*#__PURE__*/React.createElement("ul", {
    className: classes.metadata
  }, mimeType && /*#__PURE__*/React.createElement("li", null, /*#__PURE__*/React.createElement("b", null, "Format:"), " ", mimeType), uniqueId && /*#__PURE__*/React.createElement("li", null, /*#__PURE__*/React.createElement("b", null, "UniqueId:"), " ", uniqueId), sourceId && /*#__PURE__*/React.createElement("li", null, /*#__PURE__*/React.createElement("b", null, "SourceId:"), " ", sourceId), dimensions && /*#__PURE__*/React.createElement("li", null, /*#__PURE__*/React.createElement("b", null, "Dimensions:"), " ", dimensions), aspectRatio && /*#__PURE__*/React.createElement("li", null, /*#__PURE__*/React.createElement("b", null, "Format:"), " ", aspectRatio), duration && /*#__PURE__*/React.createElement("li", null, /*#__PURE__*/React.createElement("b", null, "Duration:"), " ", duration), size && /*#__PURE__*/React.createElement("li", null, /*#__PURE__*/React.createElement("b", null, "Size:"), " ", size));
};

export default MediaMetadata;
