// @kit-ui/admin/es/Media/MediaFormat.js
import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
var useStyles = makeStyles(function (theme) {
  return {
    root: {
      position: "relative",
      width: "100%",
      paddingBottom: function paddingBottom(props) {
        return props.ratio && props.ratio * 100 + "%";
      },
      margin: "auto"
    }
  };
});

var MediaFormat = function MediaFormat(_ref) {
  var _ref$format = _ref.format,
      format = _ref$format === void 0 ? undefined : _ref$format,
      _ref$ratio = _ref.ratio,
      ratio = _ref$ratio === void 0 ? undefined : _ref$ratio,
      children = _ref.children;

  if (format && format.includes(':')) {
    var size = format.split(":");
    var w = size[0];
    var h = size[1];
    ratio = h / w;
  }

  var classes = useStyles({
    ratio: ratio
  });

  if (ratio) {
    return /*#__PURE__*/React.createElement("div", {
      className: classes.root,
      "data-format": format,
      "data-ratio": ratio
    }, children);
  }

  return children;
};

export default MediaFormat;
