// @kit-ui/admin/es/Media/MediaVolumeButton.js
import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import IconButton from '@material-ui/core/IconButton';
import VolumeOffIcon from '@material-ui/icons/VolumeMute';
import VolumeDownIcon from '@material-ui/icons/VolumeDown';
import VolumeUpIcon from '@material-ui/icons/VolumeUp';
var useStyles = makeStyles(function (theme) {
  return {
    root: {
      display: "flex",
      alignItems: "center",
      justifyContent: "center"
    }
  };
});

var MediaButtonPlayPause = function MediaButtonPlayPause(_ref) {
  var className = _ref.className,
      muted = _ref.muted,
      onClick = _ref.onClick;
  var classes = useStyles();

  if (muted) {
    return /*#__PURE__*/React.createElement(IconButton, {
      className: className || classes.root,
      onClick: onClick
    }, /*#__PURE__*/React.createElement(VolumeOffIcon, null));
  }

  return /*#__PURE__*/React.createElement(IconButton, {
    className: className || classes.root,
    onClick: onClick
  }, /*#__PURE__*/React.createElement(VolumeUpIcon, null));
};

export default MediaButtonPlayPause;
