// @kit-ui/admin/es/Media/MediaPreview.stories.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import React from 'react';
import MediaPreview from './MediaPreview';
import image from "../data/media/image";
import video from "../data/media/video";
import audio from "../data/media/audio";
export default {
  title: 'Media/MediaPreview',
  component: MediaPreview,
  args: {},
  argTypes: {}
};

var Template = function Template(args) {
  return /*#__PURE__*/React.createElement(MediaPreview, args);
};

export var Default = Template.bind({});
Default.args = {};
export var MediatypeImage = Template.bind({});
MediatypeImage.args = _extends({}, image);
export var MediatypeVideo = Template.bind({});
MediatypeVideo.args = _extends({}, video);
export var MediatypeAudio = Template.bind({});
MediatypeAudio.args = _extends({}, audio);
export var SourceYoutube = Template.bind({});
SourceYoutube.args = {
  source: "youtube",
  sourceId: "RnTbc9wyl8s"
};
export var SourceVimeo = Template.bind({});
SourceVimeo.args = {
  source: "vimeo",
  sourceId: "304002147"
};
export var SourceSketchfab = Template.bind({});
SourceSketchfab.args = {
  source: "sketchfab",
  sourceId: "adb409f636e64140a2704aba5a8417cf"
};
