// @kit-ui/admin/es/Media/VideoControls.js
import React from 'react';
import PropTypes from "prop-types";
import { makeStyles } from '@material-ui/core/styles';
import Typography from '@material-ui/core/Typography';
import MediaPlayPauseButton from "./MediaPlayPauseButton";
import MediaVolumeControls from "./MediaVolumeControls";
import MediaTimeControls from "./MediaTimeControls";
var useStyles = makeStyles(function (theme) {
  return {
    root: {
      width: "100%",
      zindex: 2,
      display: "flex",
      flexDirection: "column",
      justifyContent: "space-between",
      backgroundColor: "black",
      color: "white"
    },
    header: {
      display: "flex",
      width: "100%",
      alignItems: "center"
    },
    footer: {
      margin: theme.spacing(0, 1)
    },
    playPauseButton: {
      width: theme.spacing(6),
      height: theme.spacing(6),
      backgroundColor: theme.palette.primary.main,
      color: theme.palette.primary.contrastText,
      margin: theme.spacing(1),
      "&:hover": {
        backgroundColor: theme.palette.primary.dark
      }
    },
    info: {
      flexGrow: 1,
      display: "flex",
      flexDirection: "column",
      margin: theme.spacing(0, 1)
    },
    title: {
      fontFamily: "Akkurat, sans-serif",
      fontWeight: "bold",
      color: "white"
    },
    description: {
      fontFamily: "Akkurat, sans-serif",
      color: "rgba(255, 255, 255, 0.7)"
    }
  };
});

var VideoControls = function VideoControls(_ref) {
  var className = _ref.className,
      title = _ref.title,
      untitled = _ref.untitled,
      description = _ref.description,
      paused = _ref.paused,
      buffered = _ref.buffered,
      _ref$duration = _ref.duration,
      duration = _ref$duration === void 0 ? 0 : _ref$duration,
      _ref$currentTime = _ref.currentTime,
      currentTime = _ref$currentTime === void 0 ? 0 : _ref$currentTime,
      onPlayPause = _ref.onPlayPause,
      onSeek = _ref.onSeek,
      muted = _ref.muted,
      volume = _ref.volume,
      onMute = _ref.onMute,
      onVolume = _ref.onVolume;
  var classes = useStyles();
  return /*#__PURE__*/React.createElement("nav", {
    className: className || classes.root
  }, /*#__PURE__*/React.createElement("header", {
    className: classes.header
  }, /*#__PURE__*/React.createElement(MediaPlayPauseButton, {
    className: classes.playPauseButton,
    paused: paused,
    onClick: onPlayPause
  }), /*#__PURE__*/React.createElement("div", {
    className: classes.info
  }, /*#__PURE__*/React.createElement(Typography, {
    component: "h2",
    className: classes.title
  }, title || untitled), description && /*#__PURE__*/React.createElement(Typography, {
    component: "h3",
    className: classes.description
  }, description))), /*#__PURE__*/React.createElement("footer", {
    className: classes.footer
  }, /*#__PURE__*/React.createElement(MediaTimeControls, {
    duration: duration,
    currentTime: currentTime,
    onChange: onSeek
  })));
};

VideoControls.propTypes = process.env.NODE_ENV !== "production" ? {
  className: PropTypes.string,
  untitled: PropTypes.string,
  title: PropTypes.string,
  description: PropTypes.string,

  /** Paused */
  paused: PropTypes.bool,

  /** Duration in seconds */
  duration: PropTypes.number,

  /** Current time in seconds */
  currentTime: PropTypes.number,

  /** Play/Pause */
  onPlayPause: PropTypes.func,

  /** Seek */
  onSeek: PropTypes.func
} : {};
export default VideoControls;
