// @kit-ui/schema/es/ListSettings/ListSettings.js
import React, { useState, useRef } from 'react';
import PropTypes from "prop-types";
import { Dropdown } from "@kit-ui/core";
import ListSettingsButton from './ListSettingsButton';
import ListSettingsOptions from "./ListSettingsOptions";

var ListSettings = function ListSettings(_ref) {
  var value = _ref.value,
      options = _ref.options,
      onChange = _ref.onChange;

  if (!value) {
    value = options[0].value;
  }

  var label;
  options.map(function (option) {
    if (option.value === value) {
      label = option.label;
    }
  });

  var _useState = useState(false),
      expanded = _useState[0],
      setExpanded = _useState[1];

  var anchorRef = useRef(null);

  var _onChange = function _onChange(value) {
    onChange && onChange(value);
    setExpanded(false);
  };

  var _onToggle = function _onToggle(event) {
    setExpanded(function (prevExpanded) {
      return !prevExpanded;
    });
  };

  var _onCollapse = function _onCollapse(event) {
    setExpanded(false);
  };

  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(ListSettingsButton, {
    onClick: _onToggle,
    ref: anchorRef
  }, label || value), /*#__PURE__*/React.createElement(Dropdown, {
    expanded: expanded,
    anchorEl: anchorRef.current,
    onClickAway: _onCollapse
  }, /*#__PURE__*/React.createElement(ListSettingsOptions, {
    options: options,
    value: value,
    onChange: _onChange
  })));
};

ListSettings.defaultProps = {};
ListSettings.propTypes = process.env.NODE_ENV !== "production" ? {} : {};
export default ListSettings;
