// @kit-ui/admin/es/NavPath/Path.js
import React from 'react';
import Color from 'color';
import SeparatorIcon from '@material-ui/icons/NavigateNext';
import Breadcrumbs from '@material-ui/core/Breadcrumbs';
import { makeStyles, withStyles } from '@material-ui/core/styles';

var getColor = function getColor(color) {
  color = Color(color);

  if (color.isDark()) {
    return color.hex();
  } else {
    return 'black';
  }
};

var useStyles = makeStyles(function (theme) {
  return {
    root: {
      //        fontSize: "18px",
      //        lineHeight: "24px",
      listStyle: "none",
      color: getColor(theme.palette.primary.main),
      '& li[aria-hidden]': {
        marginLeft: theme.spacing(0.5),
        marginRight: theme.spacing(0.5)
      },
      "& li": {}
    }
  };
});

var Path = function Path(_ref) {
  var className = _ref.className,
      children = _ref.children;
  var classes = useStyles();
  return /*#__PURE__*/React.createElement(Breadcrumbs, {
    className: className || classes.root,
    separator: /*#__PURE__*/React.createElement(SeparatorIcon, null)
  }, children);
};

export default Path;
