// @kit-ui/schema/lib/AutocompleteWidget/AutocompleteWidget.stories.js
"use strict";

exports.__esModule = true;
exports.WithValue = exports.WithLabel = exports.Default = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _SchemaEditor = _interopRequireDefault(require("../SchemaEditor/"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _default = {
  title: 'Widgets/AutocompleteWidget (BETA)',
  component: _SchemaEditor.default,
  args: {
    schema: {
      type: "string",
      enum: ["one", "two", "three"],
      default: "two"
    },
    uiSchema: {
      "ui:widget": "autocomplete"
    },
    formData: undefined
  },
  argTypes: {}
};
exports.default = _default;

var Template = function Template(args) {
  return /*#__PURE__*/_react.default.createElement(_SchemaEditor.default, args);
};

var Default = Template.bind({});
exports.Default = Default;
Default.args = {};
var WithLabel = Template.bind({});
exports.WithLabel = WithLabel;
WithLabel.args = {
  uiSchema: {
    "ui:title": "Title",
    "ui:placeholder": "Select something",
    "ui:help": "Help text"
  }
};
var WithValue = Template.bind({});
exports.WithValue = WithValue;
WithValue.args = {
  formData: "three"
};
