// @kit-ui/admin/es/App/AppLayout.stories.js
import React from 'react';
import AppLayout from './AppLayout';
import AppLayoutExample from './AppLayout.example';
import { getRandomTheme } from "..";
var theme = getRandomTheme();
export default {
  title: 'App/AppLayout',
  component: AppLayout,
  args: {
    header: {
      title: "App",
      subtitle: "Owner"
    },
    primaryAction: {
      title: "Create"
    },
    menu: [{
      title: "Menu 1",
      url: "/1"
    }, {
      title: "Menu 2",
      url: "/2"
    }],
    menuByUrl: {
      "/1": {
        count: 200
      }
    },
    theme: getRandomTheme()
  },
  argTypes: {}
};

var Template = function Template(args) {
  return /*#__PURE__*/React.createElement("div", {
    style: {
      minHeight: '320px'
    }
  }, /*#__PURE__*/React.createElement(AppLayoutExample, args, JSON.stringify(args)));
};

export var Default = Template.bind({});
Default.args = {};
export var WithSearch = Template.bind({});
WithSearch.args = {
  search: {
    expanded: false
  }
};
export var WithSidebar = Template.bind({});
WithSidebar.args = {
  sidebar: {
    expanded: true
  }
};
export var CollapsibleSidebar = Template.bind({});
CollapsibleSidebar.args = {
  sidebar: {
    collapsible: true,
    expanded: true
  }
};
export var CollapsibleSidebarAndSearch = Template.bind({});
CollapsibleSidebarAndSearch.args = {
  search: {
    expanded: false
  },
  sidebar: {
    collapsible: true,
    expanded: true
  }
};
export var GrowingSearch = Template.bind({});
GrowingSearch.args = {
  header: {
    title: "Growing search"
  },
  search: {
    variant: "growing",
    expanded: false
  }
};
export var GrowingSearchAndSidebar = Template.bind({});
GrowingSearch.args = {
  header: {
    title: "Growing search"
  },
  sidebar: {
    collapsible: true
  },
  search: {
    variant: "growing",
    expanded: false
  }
};
export var SearchExpanded = Template.bind({});
SearchExpanded.args = {
  search: {
    expanded: true
  }
};
export var SubviewExpanded = Template.bind({});
SubviewExpanded.args = {
  subview: {
    expanded: true,
    title: "Subview",
    description: "Description"
  }
};
