// @kit-ui/admin/lib/Media/ImageAnnotations.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _Typography = _interopRequireDefault(require("@material-ui/core/Typography"));

var _styles = require("@material-ui/core/styles");

var _ImageAnnotationsBase = _interopRequireDefault(require("./ImageAnnotationsBase"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var useStyles = (0, _styles.makeStyles)(function (theme) {
  return {
    root: {
      //        backgroundColor: "#333",
      //        color: "white",
      position: "absolute",
      top: 0,
      right: 0,
      bottom: 0,
      left: 0
    },
    preview: {
      position: "absolute",
      top: 0,
      right: 0,
      bottom: 0,
      left: 0,
      //        marginTop: theme.spacing(7),
      marginBottom: theme.spacing(6)
    },
    footer: {
      position: "absolute",
      top: "auto",
      right: 0,
      left: 0,
      bottom: 0,
      paddingTop: theme.spacing(1),
      height: theme.spacing(5)
    },
    description: {
      fontFamily: "Akkurat, sans-serif",
      fontSize: "12px",
      textAlign: "center"
    }
  };
});

var ImageAnnotations = function ImageAnnotations(_ref) {
  var className = _ref.className,
      imageUrl = _ref.imageUrl,
      _ref$imageAnnotations = _ref.imageAnnotations,
      imageAnnotations = _ref$imageAnnotations === void 0 ? [] : _ref$imageAnnotations,
      imageFocalpoint = _ref.imageFocalpoint,
      imageCropdata = _ref.imageCropdata,
      imageFilters = _ref.imageFilters,
      markerTemplate = _ref.markerTemplate,
      onChange = _ref.onChange,
      props = _objectWithoutPropertiesLoose(_ref, ["className", "imageUrl", "imageAnnotations", "imageFocalpoint", "imageCropdata", "imageFilters", "markerTemplate", "onChange"]);

  var classes = useStyles();

  var handleChange = function handleChange(imageAnnotations) {
    onChange && onChange(imageAnnotations);
  };

  return /*#__PURE__*/_react.default.createElement("div", {
    className: className || classes.root
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: classes.preview
  }, /*#__PURE__*/_react.default.createElement(_ImageAnnotationsBase.default, {
    imageUrl: imageUrl,
    imageAnnotations: imageAnnotations,
    imageCropdata: imageCropdata,
    imageFilters: imageFilters,
    imageFocalpoint: imageFocalpoint,
    markerTemplate: markerTemplate,
    onChange: handleChange
  })), /*#__PURE__*/_react.default.createElement("footer", {
    className: classes.footer
  }, /*#__PURE__*/_react.default.createElement(_Typography.default, {
    nowrap: true,
    className: classes.description
  }, "Klikk for \xE5 legge til en mark\xF8r")));
};

var _default = ImageAnnotations;
exports.default = _default;
module.exports = exports.default;
