// @kit-ui/admin/lib/Media/MediaFormat.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _styles = require("@material-ui/core/styles");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var useStyles = (0, _styles.makeStyles)(function (theme) {
  return {
    root: {
      position: "relative",
      width: "100%",
      paddingBottom: function paddingBottom(props) {
        return props.ratio && props.ratio * 100 + "%";
      },
      margin: "auto"
    }
  };
});

var MediaFormat = function MediaFormat(_ref) {
  var _ref$format = _ref.format,
      format = _ref$format === void 0 ? undefined : _ref$format,
      _ref$ratio = _ref.ratio,
      ratio = _ref$ratio === void 0 ? undefined : _ref$ratio,
      children = _ref.children;

  if (format && format.includes(':')) {
    var size = format.split(":");
    var w = size[0];
    var h = size[1];
    ratio = h / w;
  }

  var classes = useStyles({
    ratio: ratio
  });

  if (ratio) {
    return /*#__PURE__*/_react.default.createElement("div", {
      className: classes.root,
      "data-format": format,
      "data-ratio": ratio
    }, children);
  }

  return children;
};

var _default = MediaFormat;
exports.default = _default;
module.exports = exports.default;
