// @kit-ui/admin/lib/Media/MediaPreview.stories.js
"use strict";

exports.__esModule = true;
exports.SourceSketchfab = exports.SourceVimeo = exports.SourceYoutube = exports.MediatypeAudio = exports.MediatypeVideo = exports.MediatypeImage = exports.Default = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _MediaPreview = _interopRequireDefault(require("./MediaPreview"));

var _image = _interopRequireDefault(require("../data/media/image"));

var _video = _interopRequireDefault(require("../data/media/video"));

var _audio = _interopRequireDefault(require("../data/media/audio"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var _default = {
  title: 'Media/MediaPreview',
  component: _MediaPreview.default,
  args: {},
  argTypes: {}
};
exports.default = _default;

var Template = function Template(args) {
  return /*#__PURE__*/_react.default.createElement(_MediaPreview.default, args);
};

var Default = Template.bind({});
exports.Default = Default;
Default.args = {};
var MediatypeImage = Template.bind({});
exports.MediatypeImage = MediatypeImage;
MediatypeImage.args = _extends({}, _image.default);
var MediatypeVideo = Template.bind({});
exports.MediatypeVideo = MediatypeVideo;
MediatypeVideo.args = _extends({}, _video.default);
var MediatypeAudio = Template.bind({});
exports.MediatypeAudio = MediatypeAudio;
MediatypeAudio.args = _extends({}, _audio.default);
var SourceYoutube = Template.bind({});
exports.SourceYoutube = SourceYoutube;
SourceYoutube.args = {
  source: "youtube",
  sourceId: "RnTbc9wyl8s"
};
var SourceVimeo = Template.bind({});
exports.SourceVimeo = SourceVimeo;
SourceVimeo.args = {
  source: "vimeo",
  sourceId: "304002147"
};
var SourceSketchfab = Template.bind({});
exports.SourceSketchfab = SourceSketchfab;
SourceSketchfab.args = {
  source: "sketchfab",
  sourceId: "adb409f636e64140a2704aba5a8417cf"
};
