// @kit-ui/admin/lib/Media/MediaVolumeButton.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _styles = require("@material-ui/core/styles");

var _IconButton = _interopRequireDefault(require("@material-ui/core/IconButton"));

var _VolumeMute = _interopRequireDefault(require("@material-ui/icons/VolumeMute"));

var _VolumeDown = _interopRequireDefault(require("@material-ui/icons/VolumeDown"));

var _VolumeUp = _interopRequireDefault(require("@material-ui/icons/VolumeUp"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var useStyles = (0, _styles.makeStyles)(function (theme) {
  return {
    root: {
      display: "flex",
      alignItems: "center",
      justifyContent: "center"
    }
  };
});

var MediaButtonPlayPause = function MediaButtonPlayPause(_ref) {
  var className = _ref.className,
      muted = _ref.muted,
      onClick = _ref.onClick;
  var classes = useStyles();

  if (muted) {
    return /*#__PURE__*/_react.default.createElement(_IconButton.default, {
      className: className || classes.root,
      onClick: onClick
    }, /*#__PURE__*/_react.default.createElement(_VolumeMute.default, null));
  }

  return /*#__PURE__*/_react.default.createElement(_IconButton.default, {
    className: className || classes.root,
    onClick: onClick
  }, /*#__PURE__*/_react.default.createElement(_VolumeUp.default, null));
};

var _default = MediaButtonPlayPause;
exports.default = _default;
module.exports = exports.default;
