// @kit-ui/admin/lib/Media/MediaVolumeControls.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _styles = require("@material-ui/core/styles");

var _MediaVolumeButton = _interopRequireDefault(require("./MediaVolumeButton"));

var _MediaVolumeSlider = _interopRequireDefault(require("./MediaVolumeSlider"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var useStyles = (0, _styles.makeStyles)(function (theme) {
  return {
    root: {
      display: "flex",
      alignItems: "center"
    }
  };
});

var AudioControls = function AudioControls(_ref) {
  var className = _ref.className,
      muted = _ref.muted,
      volume = _ref.volume,
      onClick = _ref.onClick,
      onChange = _ref.onChange;
  var classes = useStyles();
  return /*#__PURE__*/_react.default.createElement("div", {
    className: className || classes.root
  }, /*#__PURE__*/_react.default.createElement(_MediaVolumeButton.default, {
    muted: muted,
    onClick: onClick
  }), /*#__PURE__*/_react.default.createElement(_MediaVolumeSlider.default, {
    muted: muted,
    min: 0,
    max: 1,
    value: volume,
    onChange: onChange
  }));
};

var _default = AudioControls;
exports.default = _default;
module.exports = exports.default;
