// @kit-ui/admin/lib/Module/ButtonSelect.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _IconButton = _interopRequireDefault(require("@material-ui/core/IconButton"));

var _Check = _interopRequireDefault(require("@material-ui/icons/Check"));

var _styles = require("@material-ui/core/styles");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var useStyles = (0, _styles.makeStyles)(function (theme) {
  return {
    button: {
      "& svg": {
        width: theme.spacing(2.5),
        height: theme.spacing(2.5),
        position: "absolute",
        top: "50%",
        left: "50%",
        transform: "translate(-50%,-50%)",
        color: "currentColor"
      }
    },
    select: {
      width: theme.spacing(3),
      height: theme.spacing(3),
      borderRadius: "100%",
      border: "1px dotted",
      margin: 1,
      borderColor: "currentColor",
      opacity: .5,
      "&:hover": {
        opacity: 1
      }
    },
    selected: {
      display: "block",
      width: theme.spacing(3),
      height: theme.spacing(3),
      borderRadius: "100%",
      border: "2px solid",
      borderColor: theme.palette.primary.main,
      backgroundColor: theme.palette.primary.main,
      color: theme.palette.primary.contrastText,
      boxShadow: theme.shadows[2]
    }
  };
});

var ButtonSelectable = function ButtonSelectable(_ref) {
  var className = _ref.className,
      _ref$selected = _ref.selected,
      selected = _ref$selected === void 0 ? false : _ref$selected,
      onClick = _ref.onClick;
  var classes = useStyles();

  if (selected) {
    return /*#__PURE__*/_react.default.createElement(_IconButton.default, {
      className: className || classes.button,
      "aria-selected": selected,
      color: "inherit",
      "aria-label": "select",
      onClick: onClick
    }, /*#__PURE__*/_react.default.createElement("div", {
      className: classes.selected
    }, /*#__PURE__*/_react.default.createElement(_Check.default, null)));
  }

  return /*#__PURE__*/_react.default.createElement(_IconButton.default, {
    className: className || classes.button,
    "aria-selected": selected,
    color: "inherit",
    "aria-label": "select",
    onClick: onClick
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: classes.select
  }, /*#__PURE__*/_react.default.createElement(_Check.default, null)));
};

var _default = ButtonSelectable;
exports.default = _default;
module.exports = exports.default;
