// @kit-ui/admin/lib/Module/ModuleFooter.stories.js
"use strict";

exports.__esModule = true;
exports.WithButtons = exports.WithChildren = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _ModuleFooter = _interopRequireDefault(require("./ModuleFooter"));

var _ModuleButtons = _interopRequireDefault(require("./ModuleButtons"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _default = {
  title: 'Module/ModuleFooter',
  component: _ModuleFooter.default,
  args: {},
  argTypes: {}
};
exports.default = _default;

var Template = function Template(args) {
  return /*#__PURE__*/_react.default.createElement(_ModuleFooter.default, args);
};

var WithChildren = Template.bind({});
exports.WithChildren = WithChildren;
WithChildren.args = {
  children: "Children"
};
var WithButtons = Template.bind({});
exports.WithButtons = WithButtons;
WithButtons.args = {
  children: /*#__PURE__*/_react.default.createElement(_ModuleButtons.default, {
    buttons: [{
      title: "Button 1"
    }, {
      title: "Button 2"
    }]
  })
};
