// @kit-ui/admin/lib/Module/ModuleDate.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _moment = _interopRequireDefault(require("moment"));

require("moment/locale/nb");

require("moment/locale/nn");

require("moment/locale/sv");

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styles = require("@material-ui/core/styles");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var useStyles = (0, _styles.makeStyles)(function (theme) {
  return {
    root: {
      fontFamily: "Akkurat, sans-serif",
      fontSize: "14px",
      color: theme.palette.text.secondary
    }
  };
});
/** Module date, uses moment.js */

var ModuleDate = function ModuleDate(_ref) {
  var className = _ref.className,
      _ref$locale = _ref.locale,
      locale = _ref$locale === void 0 ? "nb" : _ref$locale,
      datetime = _ref.datetime,
      _ref$format = _ref.format,
      format = _ref$format === void 0 ? "relative" : _ref$format;
  var classes = useStyles();

  if (!datetime) {
    return false;
  } // format value


  _moment.default.locale(locale);

  var date;

  if (format === "relative") {
    date = (0, _moment.default)(datetime).fromNow();
  } else if (format) {
    date = (0, _moment.default)(datetime).format(format);
  } else {
    date = (0, _moment.default)(datetime).format();
  }

  return /*#__PURE__*/_react.default.createElement("time", {
    dateTime: datetime,
    className: className || classes.root
  }, date);
};

ModuleDate.propTypes = process.env.NODE_ENV !== "production" ? {
  /** Locale string */
  locale: _propTypes.default.string,

  /** Datetime as YYYY-MM-DD */
  datetime: _propTypes.default.string,

  /** Moment.js format */
  format: _propTypes.default.string
} : {};
var _default = ModuleDate;
exports.default = _default;
module.exports = exports.default;
