// @kit-ui/admin/lib/Module/ModuleStatus.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Typography = _interopRequireDefault(require("@material-ui/core/Typography"));

var _styles = require("@material-ui/core/styles");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var useStyles = (0, _styles.makeStyles)(function (theme) {
  return {
    label: {
      fontFamily: "Akkurat, sans-serif",
      fontSize: "14px",
      lineHeight: "24px",
      fontWeight: "bold",
      color: theme.palette.text.secondary
    }
  };
});
/** ModuleStatus, draft, publish, trash, erased */

var ModuleStatus = function ModuleStatus(_ref) {
  var _ref$component = _ref.component,
      component = _ref$component === void 0 ? "h3" : _ref$component,
      status = _ref.status,
      children = _ref.children;
  var classes = useStyles();

  if (!status && children) {
    status = children;
  }

  if (!status) {
    return false;
  }

  return /*#__PURE__*/_react.default.createElement(_Typography.default, {
    className: classes.label,
    noWrap: true,
    component: component
  }, status);
};

ModuleStatus.propTypes = process.env.NODE_ENV !== "production" ? {
  status: _propTypes.default.string
} : {};
var _default = ModuleStatus;
exports.default = _default;
module.exports = exports.default;
