// @kit-ui/admin/lib/Module/ModuleLabel.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _color = _interopRequireDefault(require("color"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Typography = _interopRequireDefault(require("@material-ui/core/Typography"));

var _styles = require("@material-ui/core/styles");

var _utils = require("../utils");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var useStyles = (0, _styles.makeStyles)(function (theme) {
  return {
    root: {
      display: "inline-block",
      fontSize: "12px"
    },
    label: {
      fontFamily: "Akkurat, sans-serif",
      fontSize: "inherit",
      fontWeight: "bold",
      padding: ".125em .75em",
      border: "1px solid",
      backgroundColor: function backgroundColor(props) {
        return props.color || "transparent";
      },
      borderColor: function borderColor(props) {
        return props.color || theme.palette.divider;
      },
      color: function color(props) {
        return props.color && (0, _utils.getContrastColor)(props.color) || theme.palette.text.secondary;
      },
      borderRadius: "1.25em"
    }
  };
});

var ModuleLabel = function ModuleLabel(_ref) {
  var className = _ref.className,
      _ref$component = _ref.component,
      component = _ref$component === void 0 ? "h4" : _ref$component,
      label = _ref.label,
      color = _ref.color,
      children = _ref.children;
  var classes = useStyles({
    color: color
  });

  if (!label && children) {
    label = children;
  }

  if (!label) {
    return false;
  }

  return /*#__PURE__*/_react.default.createElement("span", {
    className: className || classes.root
  }, /*#__PURE__*/_react.default.createElement(_Typography.default, {
    className: classes.label,
    noWrap: true,
    component: component
  }, label));
};

ModuleLabel.propTypes = process.env.NODE_ENV !== "production" ? {
  component: _propTypes.default.string,
  label: _propTypes.default.string,
  color: _propTypes.default.string
} : {};
ModuleLabel.defaultProps = {
  label: undefined,
  color: undefined
};
var _default = ModuleLabel;
exports.default = _default;
module.exports = exports.default;
