// @kit-ui/admin/lib/Module/Module.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _EditableModule = _interopRequireDefault(require("./EditableModule"));

var _ExpandedModule = _interopRequireDefault(require("./ExpandedModule"));

var _ListModule = _interopRequireDefault(require("./ListModule"));

var _TableModule = _interopRequireDefault(require("./TableModule"));

var _GridModule = _interopRequireDefault(require("./GridModule"));

var _CardModule = _interopRequireDefault(require("./CardModule"));

var _ref2;

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var templates = {
  "list": _ListModule.default,
  "table": _TableModule.default,
  "grid": _GridModule.default,
  "card": _CardModule.default
};

var Module = function Module(_ref) {
  var variant = _ref.variant,
      expanded = _ref.expanded,
      editable = _ref.editable,
      editing = _ref.editing,
      children = _ref.children,
      props = _objectWithoutPropertiesLoose(_ref, ["variant", "expanded", "editable", "editing", "children"]);

  if (editable && editing) {
    return /*#__PURE__*/_react.default.createElement(_EditableModule.default, _extends({}, props, {
      editing: true
    }), children);
  }

  if (expanded) {
    return /*#__PURE__*/_react.default.createElement(_ExpandedModule.default, _extends({}, props, {
      editable: editable,
      editing: editing
    }), children);
  }

  var ModuleTemplate = templates && templates[variant] || _ListModule.default;
  return /*#__PURE__*/_react.default.createElement(ModuleTemplate, _extends({}, props, {
    editable: editable,
    editing: editing
  }));
};

Module.defaultProps = {
  variant: "list",
  expanded: false,
  editable: false,
  editing: false
};
Module.propTypes = process.env.NODE_ENV !== "production" ? (_ref2 = {
  variant: _propTypes.default.oneOf(["list", "table", "grid", "card"]),
  expanded: _propTypes.default.bool,
  editable: _propTypes.default.bool,
  editing: _propTypes.default.bool,
  title: _propTypes.default.string,
  description: _propTypes.default.string,
  label: _propTypes.default.any,
  status: _propTypes.default.string,
  datetime: _propTypes.default.string,
  author: _propTypes.default.string
}, _ref2["status"] = _propTypes.default.string, _ref2) : {};
var _default = Module;
exports.default = _default;
module.exports = exports.default;
