// @kit-ui/admin/lib/PreviewCode/PreviewCode.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _server = require("react-dom/server");

var _reactElementToJsxString = _interopRequireDefault(require("react-element-to-jsx-string"));

var _pretty = _interopRequireDefault(require("pretty"));

var _reactSyntaxHighlighter = require("react-syntax-highlighter");

var _PreviewCodeRender = _interopRequireDefault(require("./PreviewCodeRender"));

var _styles = require("@material-ui/core/styles");

var _PreviewCode = _interopRequireDefault(require("./PreviewCode.themeDark"));

var _PreviewCode2 = _interopRequireDefault(require("./PreviewCode.themeLight"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var themes = {
  "light": _PreviewCode2.default,
  "dark": _PreviewCode.default
};

function isJSON(something) {
  try {
    JSON.parse(JSON.stringify(something));
    return true;
  } catch (e) {
    return false;
  }
}

function isClassComponent(component) {
  return typeof component === 'function' && !!component.prototype.isReactComponent ? true : false;
}

function isFunctionComponent(component) {
  return typeof component === 'function' && String(component).includes('return React.createElement') ? true : false;
}

function isReactComponent(component) {
  return isClassComponent(component) || isFunctionComponent(component) ? true : false;
}

function isElement(element) {
  return _react.default.isValidElement(element);
}

function isDOMTypeElement(element) {
  return isElement(element) && typeof element.type === 'string';
}

function isCompositeTypeElement(element) {
  return isElement(element) && typeof element.type === 'function';
}

var preTag = function preTag(_ref) {
  var className = _ref.className,
      theme = _ref.theme,
      language = _ref.language,
      children = _ref.children,
      props = _objectWithoutPropertiesLoose(_ref, ["className", "theme", "language", "children"]);

  return /*#__PURE__*/_react.default.createElement("pre", {
    className: className,
    "data-theme": theme,
    "data-language": language
  }, children);
};

var PreviewCode = function PreviewCode(_ref2) {
  var _ref2$theme = _ref2.theme,
      theme = _ref2$theme === void 0 ? "dark" : _ref2$theme,
      language = _ref2.language,
      data = _ref2.data,
      children = _ref2.children;
  var value = data || children;

  if (isJSON(value)) {
    language = "json";
    value = JSON.stringify(value, 0, 2);
  } else if (typeof value === "function") {
    language = "javascript";
    value = value.toString();
  } else if (isCompositeTypeElement(value)) {
    language = "jsx";
    value = (0, _reactElementToJsxString.default)(value);
  } else if (isDOMTypeElement(value)) {
    value = (0, _pretty.default)((0, _server.renderToStaticMarkup)(value));
  }

  if (!value || typeof value !== "string") {
    return false;
  }

  value = value.trim();
  return /*#__PURE__*/_react.default.createElement(_styles.ThemeProvider, {
    theme: themes[theme] || _PreviewCode2.default
  }, /*#__PURE__*/_react.default.createElement(_reactSyntaxHighlighter.Prism, {
    "data-theme": theme,
    "data-language": language,
    language: language,
    style: false,
    wrapLines: true,
    useInlineStyles: false,
    PreTag: function PreTag(props) {
      return (0, _PreviewCodeRender.default)(_extends({}, props, {
        theme: theme,
        language: language
      }));
    }
  }, value));
};

PreviewCode.propTypes = process.env.NODE_ENV !== "production" ? {
  language: _propTypes.default.string,
  theme: _propTypes.default.string,
  data: _propTypes.default.oneOfType([_propTypes.default.string, _propTypes.default.object, _propTypes.default.func])
} : {};
var _default = PreviewCode;
exports.default = _default;
module.exports = exports.default;
