// @kit-ui/schema/lib/ArrayField/ArrayField.stories.js
"use strict";

exports.__esModule = true;
exports.ArrayWithDialog = exports.CustomButtons = exports.SortableModuleArray = exports.ModuleArray = exports.SortableArrayOfObjects = exports.ArrayOfObjects = exports.SortableStrings = exports.ArrayOfStrings = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _SchemaEditor = _interopRequireDefault(require("../SchemaEditor/SchemaEditor"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _default = {
  title: 'Fields/ArrayField',
  component: _SchemaEditor.default,
  args: {
    schema: {},
    formData: {},
    formContext: {}
  },
  argTypes: {}
};
exports.default = _default;

var Template = function Template(args) {
  return /*#__PURE__*/_react.default.createElement(_SchemaEditor.default, args);
};

var ArrayOfStrings = Template.bind({});
exports.ArrayOfStrings = ArrayOfStrings;
ArrayOfStrings.args = {
  schema: {
    "type": "array",
    "items": {
      "type": "string"
    }
  },
  uiSchema: {
    "items": {
      "ui:layout": "module"
    }
  },
  formData: ["String 1", "String 2", "String 3"]
};
var SortableStrings = Template.bind({});
exports.SortableStrings = SortableStrings;
SortableStrings.args = {
  schema: {
    "type": "array",
    "items": {
      "type": "string"
    }
  },
  uiSchema: {
    "ui:sortable": true
  },
  formData: ["String 1", "String 2", "String 3"]
};
var ArrayOfObjects = Template.bind({});
exports.ArrayOfObjects = ArrayOfObjects;
ArrayOfObjects.args = {
  schema: {
    "type": "array",
    "items": {
      "type": "object",
      "properties": {
        "title": {
          "type": "string"
        },
        "description": {
          "type": "string"
        }
      }
    }
  },
  uiSchema: {},
  formData: [{
    "title": "Object 1"
  }, {
    "title": "Object 2"
  }, {
    "title": "Object 3"
  }]
};
var SortableArrayOfObjects = Template.bind({});
exports.SortableArrayOfObjects = SortableArrayOfObjects;
SortableArrayOfObjects.args = {
  schema: {
    "type": "array",
    "items": {
      "type": "object",
      "properties": {
        "title": {
          "type": "string"
        },
        "description": {
          "type": "string"
        }
      }
    }
  },
  uiSchema: {
    "ui:sortable": true
  },
  formData: [{
    "title": "Object 1"
  }, {
    "title": "Object 2"
  }, {
    "title": "Object 3"
  }]
};
var ModuleArray = Template.bind({});
exports.ModuleArray = ModuleArray;
ModuleArray.args = {
  schema: {
    "type": "array",
    "items": {
      "type": "object",
      "properties": {
        "title": {
          "type": "string"
        },
        "description": {
          "type": "string"
        }
      }
    }
  },
  uiSchema: {
    "items": {
      "ui:layout": "module",
      "ui:editable": true,
      "ui:hideable": true
    }
  },
  formData: [{
    "title": "Object 1"
  }, {
    "title": "Object 2"
  }, {
    "title": "Object 3"
  }]
};
var SortableModuleArray = Template.bind({});
exports.SortableModuleArray = SortableModuleArray;
SortableModuleArray.args = {
  schema: {
    "type": "array",
    "items": {
      "type": "object",
      "properties": {
        "title": {
          "type": "string"
        },
        "description": {
          "type": "string"
        }
      }
    }
  },
  formData: [{
    "title": "Object 1"
  }, {
    "title": "Object 2"
  }, {
    "title": "Object 3"
  }],
  uiSchema: {
    "ui:sortable": true,
    "items": {
      "ui:layout": "module",
      "ui:editable": true,
      "ui:hideable": true
    }
  }
};
var CustomButtons = Template.bind({});
exports.CustomButtons = CustomButtons;
CustomButtons.args = {
  schema: {
    "type": "array",
    "items": {
      "type": "object",
      "properties": {
        "title": {
          "type": "string"
        },
        "description": {
          "type": "string"
        }
      }
    }
  },
  formData: [{
    "title": "Object 1"
  }, {
    "title": "Object 2"
  }, {
    "title": "Object 3"
  }],
  uiSchema: {
    "ui:sortable": true,
    "items": {
      "ui:layout": "module",
      "ui:editable": true
    },
    "ui:buttons": ["dialog", {
      title: "Add item",
      item: {
        title: "Added by button"
      }
    }, {
      title: "Add"
    }]
  }
};
var ArrayWithDialog = Template.bind({});
exports.ArrayWithDialog = ArrayWithDialog;
ArrayWithDialog.args = {
  schema: {
    "type": "object",
    "properties": {
      "array": {
        "type": "array",
        "items": {
          "type": "object",
          "properties": {
            "title": {
              "type": "string"
            },
            "description": {
              "type": "string"
            }
          }
        }
      }
    }
  },
  formData: {
    "array": [{
      "title": "Object 1"
    }, {
      "title": "Object 2"
    }, {
      "title": "Object 3"
    }]
  },
  uiSchema: {
    "ui:field": "pageEditor",
    "ui:fieldset": ["content"],
    "content": {
      "ui:field": "pageContent",
      "ui:fieldset": ["array"],
      "array": {
        "ui:sortable": true,
        "items": {
          "ui:layout": "module",
          "ui:editable": true
        },
        "ui:buttons": ["dialog", {
          title: "Add item",
          item: {
            title: "Added by button"
          }
        }, {
          title: "Add"
        }]
      }
    }
  }
};
