// @kit-ui/schema/es/__SchemaFieldset/SchemaFieldset.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import React from 'react';
import { retrieveSchema, getDefaultFormState, getUiOptions } from "../utils";

var SchemaFieldset = function SchemaFieldset(props) {
  var id = props.id,
      idPrefix = props.idPrefix,
      schema = props.schema,
      uiSchema = props.uiSchema,
      formData = props.formData,
      children = props.children;
  var uiOptions = getUiOptions(props);
  var fieldset = uiOptions && uiOptions.fieldset;

  if (!fieldset) {
    return children;
  }

  var renderFieldset = function renderFieldset(name, index) {
    var fieldSchema = schema && schema.properties && schema.properties[name] && retrieveSchema(schema.properties[name], props.registry.definitions, formData[name]);
    var fieldFormData = fieldSchema && formData && formData[name] && getDefaultFormState(fieldSchema, formData[name]);
    var fieldUiSchema = uiSchema && uiSchema[name] || {};
    var fieldUiOptions = fieldUiSchema && getUiOptions(fieldUiSchema);
    var SchemaField = props.registry.fields.SchemaField;

    if (fieldSchema) {
      return /*#__PURE__*/React.createElement(SchemaField, _extends({}, props, {
        key: index,
        idPrefix: id,
        name: name,
        schema: fieldSchema,
        uiSchema: fieldUiSchema,
        formData: fieldFormData,
        onChange: function onChange(value) {
          var _extends2;

          return props.onChange(_extends({}, formData, (_extends2 = {}, _extends2[name] = value, _extends2)));
        }
      }));
    }

    if (fieldUiSchema && name) {
      return /*#__PURE__*/React.createElement(SchemaField, _extends({}, props, {
        key: index,
        idPrefix: id,
        name: name,
        schema: schema,
        uiSchema: fieldUiSchema,
        formData: formData
      }));
      return /*#__PURE__*/React.createElement("p", null, "fieldset:", name);
    }

    return /*#__PURE__*/React.createElement("p", null, "fieldset:", name);
  };

  return /*#__PURE__*/React.createElement(React.Fragment, null, fieldset && fieldset.map(renderFieldset));
};

export default SchemaFieldset;
