// @kit-ui/schema/es/Media/MediaField.stories.js
import React from 'react';
import MediaField from "./MediaField";
import Schema from "../SchemaEditor";
import imageData from "../data/media/image/formData";
import videoData from "../data/media/video/formData";
import audioData from "../data/media/audio/formData";
export default {
  title: 'Fields/MediaField',
  component: MediaField,
  args: {},
  argTypes: {}
};

var Template = function Template(args) {
  return /*#__PURE__*/React.createElement(Schema, args);
};

export var EmptyMedia = Template.bind({});
EmptyMedia.args = {
  schema: {
    type: "media"
  }
};
export var MediaWithDialog = Template.bind({});
MediaWithDialog.args = {
  schema: {
    type: "object",
    "properties": {
      "media": {
        type: "media",
        mediaType: "image"
      }
    }
  },
  uiSchema: {
    "ui:field": "pageEditor",
    "ui:fieldset": ["content"],
    "content": {
      "ui:field": "pageContent",
      "ui:fieldset": ["media"],
      "media": {
        "ui:dialog": true
      }
    }
  }
};
export var EmptyImage = Template.bind({});
EmptyImage.args = {
  schema: {
    type: "media",
    mediaType: "image"
  }
};
export var Image = Template.bind({});
Image.args = {
  schema: {
    type: "media",
    mediaType: "image"
  },
  formData: {
    mediaId: 1,
    media: imageData
  }
};
export var ImageModule = Template.bind({});
ImageModule.args = {
  schema: {
    type: "media",
    mediaType: "image"
  },
  uiSchema: {
    "ui:layout": "module",
    "ui:editable": true
  },
  formData: {
    mediaId: 1,
    media: imageData
  }
};
export var MediaArray = Template.bind({});
MediaArray.args = {
  schema: {
    type: "array",
    items: {
      type: "media",
      mediaType: "image",
      cropdata: true
    }
  },
  uiSchema: {
    "items": {
      "ui:layout": "module",
      "ui:editable": true
    }
  },
  formData: [{
    mediaId: 1,
    media: imageData
  }, {
    mediaId: 2,
    media: videoData
  }, {
    mediaId: 3,
    media: audioData
  }]
};
export var MediaArrayWithDialog = Template.bind({});
MediaArrayWithDialog.args = {
  schema: {
    "type": "object",
    "properties": {
      "array": {
        "type": "array",
        "items": {
          "type": "media",
          "mediaType": "image"
        }
      }
    }
  },
  uiSchema: {
    "ui:field": "pageEditor",
    "ui:fieldset": ["content"],
    "content": {
      "ui:field": "pageContent",
      "ui:fieldset": ["array"],
      "array": {
        "ui:sortable": true,
        "ui:dialog": true,
        "items": {
          "ui:layout": "module",
          "ui:editable": true
        }
      }
    }
  },
  formData: {
    array: [{
      mediaId: 1,
      media: imageData
    }, {
      mediaId: 2,
      media: videoData
    }, {
      mediaId: 3,
      media: audioData
    }]
  }
};
export var ImageCropdata = Template.bind({});
ImageCropdata.args = {
  schema: {
    type: "media",
    mediaType: "image",
    cropdata: true
  },
  formData: {
    mediaId: 1,
    media: imageData
  }
};
export var ImageFocalpoint = Template.bind({});
ImageFocalpoint.args = {
  schema: {
    type: "media",
    mediaType: "image",
    focalpoint: true
  },
  formData: {
    mediaId: 1,
    media: imageData
  }
};
export var ImageAnnotations = Template.bind({});
ImageAnnotations.args = {
  schema: {
    type: "media",
    mediaType: "image",
    annotations: true
  },
  formData: {
    mediaId: 1,
    media: imageData
  }
};
export var ImageFilters = Template.bind({});
ImageFilters.args = {
  schema: {
    type: "media",
    mediaType: "image",
    filters: ["brightness"]
  },
  formData: {
    mediaId: 1,
    media: imageData
  }
};
export var EmptyVideo = Template.bind({});
EmptyVideo.args = {
  schema: {
    type: "media",
    mediaType: "video"
  }
};
export var Video = Template.bind({});
Video.args = {
  schema: {
    type: "media",
    mediaType: "video"
  },
  formData: {
    mediaId: 1,
    media: videoData
  }
};
export var EmptyAudio = Template.bind({});
EmptyAudio.args = {
  schema: {
    type: "media",
    mediaType: "audio"
  }
};
export var Audio = Template.bind({});
Audio.args = {
  schema: {
    type: "media",
    mediaType: "audio"
  },
  formData: {
    mediaId: 1,
    media: audioData
  }
};
