// @kit-ui/schema/es/Media/MediaLayout.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import React from 'react';
import PropTypes from "prop-types";
import { MediaBase, MediaUpload, NavSettings, NavToolbar, NavTabs } from "@kit-ui/admin";
import MediaButtons from "./MediaButtons";
import { makeStyles } from '@material-ui/core/styles';
var useStyles = makeStyles(function (theme) {
  return {
    media: {
      position: function position(props) {
        return props.align && "absolute" || props.position || "relative";
      },
      top: function top(props) {
        return props.align && 0;
      },
      left: function left(props) {
        return props.align && props.align === "left" && 0;
      },
      right: function right(props) {
        return props.align && props.align === "right" && 0;
      },
      backgroundColor: function backgroundColor(props) {
        return props.backgroundColor;
      },
      color: function color(props) {
        return props.color;
      },
      "& .MuiFilledInput-root": {
        backgroundColor: function backgroundColor(props) {
          return props.color && props.color === "white" && "rgba(255,255,255,.5)";
        },
        "&.Mui-focused": {
          backgroundColor: function backgroundColor(props) {
            return props.color && props.color === "white" && "rgba(255,255,255,.9)";
          }
        }
      },
      "& + *": {
        marginLeft: function marginLeft(props) {
          if (props.align === "left") return props.minWidth + theme.spacing(props.spacing);
        },
        marginRight: function marginRight(props) {
          if (props.align === "right") return props.minWidth + theme.spacing(props.spacing);
        },
        marginBottom: function marginBottom(props) {
          return props.align && theme.spacing(props.spacing);
        },
        minHeight: function minHeight(props) {
          return props.align && props.minHeight;
        },
        width: function width(props) {
          return props.align && "auto !important";
        }
      },
      minWidth: function minWidth(props) {
        return props.minWidth;
      },
      minHeight: function minHeight(props) {
        return props.minHeight;
      },
      maxWidth: function maxWidth(props) {
        return props.maxWidth || props.minWidth || "100%";
      },
      maxHeight: function maxHeight(props) {
        return props.maxHeight;
      },
      margin: "auto"
    },
    header: {
      position: "absolute",
      top: 0,
      right: 0,
      bottom: "auto",
      left: 0,
      display: "flex",
      zIndex: 2,
      alignItems: "center",
      justifyContent: "flex-end",
      height: theme.spacing(6),
      marginLeft: theme.spacing(1),
      //        marginRight: theme.spacing(1),
      color: "inherit",
      "& + [role=tablist]": {
        position: "absolute",
        top: 0,
        right: 0,
        bottom: 0,
        left: 0
      },
      "& + [role=tablist] > [role=tab] > *": {
        zIndex: 1,
        marginTop: theme.spacing(6)
      },
      "& + [role=tablist] > [role=tab][data-name=media] > *": {
        zIndex: 1,
        marginTop: 0
      }
    },
    tabs: {
      flexGrow: 1,
      marginLeft: theme.spacing(1),
      color: "white"
    },
    settings: {
      display: "flex"
    },
    toolbar: {
      display: "flex",
      color: "white",
      "& > button": {
        color: "inherit"
      }
    }
  };
});

var MediaLayout = function MediaLayout(_ref) {
  var mediaId = _ref.mediaId,
      media = _ref.media,
      minWidth = _ref.minWidth,
      minHeight = _ref.minHeight,
      backgroundColor = _ref.backgroundColor,
      color = _ref.color,
      upload = _ref.upload,
      uploadProgress = _ref.uploadProgress,
      menu = _ref.menu,
      _ref$buttons = _ref.buttons,
      buttons = _ref$buttons === void 0 ? [] : _ref$buttons,
      children = _ref.children,
      props = _objectWithoutPropertiesLoose(_ref, ["mediaId", "media", "minWidth", "minHeight", "backgroundColor", "color", "upload", "uploadProgress", "menu", "buttons", "children"]);

  var mediaType = media && media.mediaType;

  if (mediaType === "image") {
    backgroundColor = "#333";
    color = "white";
  } else if (mediaType === "video") {
    backgroundColor = "black";
    color = "white";
  } else if (mediaType === "audio") {
    backgroundColor = "white";
  } else {
    backgroundColor = "#666";
  }

  var classes = useStyles(_extends({}, props, {
    minWidth: minWidth,
    minHeight: minHeight,
    backgroundColor: backgroundColor,
    color: color
  }));

  var MediaHeader = function MediaHeader(_ref2) {
    var menu = _ref2.menu,
        toolbar = _ref2.toolbar,
        settings = _ref2.settings;
    toolbar = toolbar && toolbar.map(function (item) {
      if (item.name === "edit") {
        return false;
      }

      return item;
    });
    return /*#__PURE__*/React.createElement("header", {
      className: classes.header
    }, menu && /*#__PURE__*/React.createElement(NavTabs, {
      className: classes.tabs,
      tabs: menu
    }), settings && /*#__PURE__*/React.createElement(NavSettings, {
      className: classes.settings,
      settings: settings
    }), toolbar && /*#__PURE__*/React.createElement(NavToolbar, {
      className: classes.toolbar,
      toolbar: toolbar
    }));
  };

  if (uploadProgress && uploadProgress < 100) {
    return /*#__PURE__*/React.createElement("div", {
      className: classes.media
    }, /*#__PURE__*/React.createElement(MediaHeader, props), /*#__PURE__*/React.createElement(MediaUpload, _extends({}, props, {
      mediaType: mediaType,
      uploadProgress: uploadProgress
    })));
  }

  if (!mediaId && buttons) {
    return /*#__PURE__*/React.createElement("div", {
      className: classes.media
    }, /*#__PURE__*/React.createElement(MediaButtons, {
      buttons: buttons
    }));
  }

  return /*#__PURE__*/React.createElement("div", {
    className: classes.media
  }, /*#__PURE__*/React.createElement(MediaHeader, _extends({}, props, {
    menu: menu
  })), children);
};

MediaLayout.propTypes = process.env.NODE_ENV !== "production" ? {} : {};
MediaLayout.defaultProps = {
  minWidth: 0,
  minHeight: 256
};
export default MediaLayout;
