// @kit-ui/schema/lib/ArrayItemLayout/ArrayItemLayout.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _admin = require("@kit-ui/admin");

var _styles = require("@material-ui/core/styles");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var useStyles = (0, _styles.makeStyles)(function (theme) {
  return {
    arrayItem: {
      width: "100%",
      display: "flex",
      flexDirection: "row",
      alignItems: "flexStart",
      borderTop: "1px solid",
      borderTopColor: theme.palette.divider,
      borderBottom: "1px solid",
      borderBottomColor: theme.palette.divider,
      padding: theme.spacing(1, 0),
      "& + $arrayItem": {
        borderTop: "none",
        borderColor: theme.palette.divider
      }
    },
    startAdornment: {
      height: theme.spacing(7),
      marginRight: theme.spacing(1),
      display: "flex",
      alignItems: "center",
      justifyContent: "center"
    },
    endAdornment: {
      height: theme.spacing(7),
      marginLeft: theme.spacing(1),
      display: "flex",
      alignItems: "center",
      justifyContent: "center"
    },
    content: {
      flexBasis: 0,
      flexGrow: 1
    },
    toolbar: {}
  };
});

var ArrayItemLayout = function ArrayItemLayout(_ref) {
  var children = _ref.children,
      startAdornment = _ref.startAdornment,
      toolbar = _ref.toolbar;
  var classes = useStyles();
  return /*#__PURE__*/_react.default.createElement("div", {
    className: classes.arrayItem
  }, startAdornment && /*#__PURE__*/_react.default.createElement("div", {
    className: classes.startAdornment
  }, startAdornment), /*#__PURE__*/_react.default.createElement("div", {
    className: classes.content
  }, children), /*#__PURE__*/_react.default.createElement("div", {
    className: classes.endAdornment
  }, /*#__PURE__*/_react.default.createElement(_admin.NavToolbar, {
    toolbar: toolbar
  })));
};

var _default = ArrayItemLayout;
exports.default = _default;
module.exports = exports.default;
