// @kit-ui/admin/es/utils/getEvents.js
import _ from 'lodash';
import moment from 'moment';
export function getEventsByHour(datetime, events) {
  datetime = moment(datetime).format();
  var date = {
    date: moment(datetime).format('YYYY-MM-DD'),
    datetime: datetime,
    events: []
  };

  if (events && events.byHour && events.byHour[datetime]) {
    date.events = events.byHour[datetime].map(function (id) {
      return events.byId[id];
    });
  }

  return date;
}
export function getEventsByDate(datetime, events) {
  datetime = moment(datetime).format('YYYY-MM-DD');
  var date = {
    date: moment(datetime).format('YYYY-MM-DD'),
    datetime: datetime,
    events: []
  };

  if (events && events.byDate && events.byDate[datetime]) {
    date.events = events.byDate[datetime].map(function (id) {
      return events.byId[id];
    });
  }

  return date;
}
export function getEvents(_ref) {
  var items = _ref.items,
      dtStart = _ref.dtStart,
      dtEnd = _ref.dtEnd;
  var dates = [];
  var hours = [];
  var byId = {};
  var byDate = {};
  var byHour = {};

  var sortedEvents = _.sortBy(items, 'dtStart', 'dtEnd');

  for (var i = 0; i < sortedEvents.length; i++) {
    var item = sortedEvents[i];
    var id = item['uniqueId'] || i;
    var date = moment(item['dtStart']).format('YYYY-MM-DD');
    var hour = moment(item['dtStart']);
    hour = moment(hour).set('minute', 0);
    hour = moment(hour).set('second', 0);
    hour = moment(hour).format();
    byId[id] = item;

    if (byDate[date]) {
      byDate[date].push(id);
    } else {
      byDate[date] = [id];
      dates.push(date);
    }

    if (byHour[hour]) {
      byHour[hour].push(id);
    } else {
      byHour[hour] = [id];
      hours.push(hour);
    }
  }

  var events = {
    dates: dates,
    hours: hours,
    byId: byId,
    byDate: byDate,
    byHour: byHour
  };
  return events;
}
