// @kit-ui/admin/es/utils/getSchedule.js
import _ from 'lodash';
import moment from 'moment';
import { getEvents, getEventsByDate } from "..";
export function getSchedule(_ref) {
  var date = _ref.date,
      _ref$items = _ref.items,
      items = _ref$items === void 0 ? [] : _ref$items,
      months = _ref.months;

  if (!months) {
    months = 3;
  }

  var dt = moment(date).format('YYYY-MM-DD');
  var dtStart = moment(dt).format('YYYY-MM-DD');
  var dtEnd = moment(dt).add(months, 'months').format('YYYY-MM-DD');
  var events = {
    dates: []
  };

  if (items) {
    events = getEvents({
      items: items,
      dtStart: dtStart,
      dtEnd: dtEnd
    });
  }

  var rows = [];

  for (var i = 0; i < events.dates.length; i++) {
    var _dt = events.dates[i];
    var datetime = moment(_dt);
    datetime = moment(datetime).set('hour', 0);
    datetime = moment(datetime).set('minute', 0);
    datetime = moment(datetime).set('second', 0);

    var _date = getEventsByDate(datetime, events);

    rows.push(_date);
  }

  var schedule = {
    dt: dt,
    dtStart: dtStart,
    dtEnd: dtEnd,
    rows: rows,
    events: events
  };
  return schedule;
}
export default getSchedule;
