// @kit-ui/schema/lib/LayoutField/LayoutField.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _utils = require("../utils");

var _SchemaFieldset = _interopRequireDefault(require("../SchemaFieldset/SchemaFieldset"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var DefaultLayout = function DefaultLayout(_ref) {
  var children = _ref.children;
  return /*#__PURE__*/_react.default.createElement("div", null, children);
};

var LayoutField = function LayoutField(props) {
  var id = props.id,
      idPrefix = props.idPrefix,
      idSchema = props.idSchema,
      schema = props.schema,
      uiSchema = props.uiSchema,
      formData = props.formData,
      formContext = props.formContext,
      children = props.children,
      registry = props.registry;
  var uiLayout = (0, _utils.getUiLayout)(props);
  var uiMenu = uiLayout.nav && (0, _utils.getUiMenu)(props);
  var layout = uiLayout && uiLayout.layout;
  var fieldset = uiLayout && uiLayout.fieldset;

  var getLayoutTemplate = function getLayoutTemplate(layout) {
    if (typeof layout === "function") {
      return layout;
    }
    /* Layout field = layout+Layout */


    var fields = registry.fields;
    var fieldLayout = layout + "Layout";

    if (typeof layout === "string" && fields[fieldLayout]) {
      return fields[fieldLayout];
    }

    return DefaultLayout;
  };

  var LayoutTemplate = layout && getLayoutTemplate(layout);

  if (!LayoutTemplate) {
    return /*#__PURE__*/_react.default.createElement("p", null, "Missing layout: ", layout);
  }

  var uiPreview = (0, _utils.getUiPreview)(props);
  var uiContext = (0, _utils.getUiContext)(props);
  var uiActions = (0, _utils.getUiActions)(props);
  var uiSettings = (0, _utils.getUiSettings)(props);
  var uiToolbar = (0, _utils.getUiToolbar)(props);

  if (fieldset && fieldset.length) {
    return /*#__PURE__*/_react.default.createElement(LayoutTemplate, _extends({}, uiContext, uiActions, uiLayout, uiPreview, {
      menu: uiMenu,
      toolbar: uiToolbar,
      settings: uiSettings
    }), /*#__PURE__*/_react.default.createElement(_SchemaFieldset.default, _extends({}, props, {
      fieldset: fieldset
    })));
  }

  return /*#__PURE__*/_react.default.createElement(LayoutTemplate, _extends({}, uiLayout, uiPreview, uiActions, {
    menu: uiMenu,
    settings: uiSettings
  }), children);
};

var _default = LayoutField;
exports.default = _default;
module.exports = exports.default;
