// @kit-ui/admin/es/Layout/LayoutFooter.js
import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import LayoutButtons from "./LayoutButtons";
var useStyles = makeStyles(function (theme) {
  return {
    root: {
      marginTop: theme.spacing(2)
    }
  };
});

var LayoutFooter = function LayoutFooter(_ref) {
  var className = _ref.className,
      _ref$padding = _ref.padding,
      padding = _ref$padding === void 0 ? 1 : _ref$padding,
      _ref$buttons = _ref.buttons,
      buttons = _ref$buttons === void 0 ? [] : _ref$buttons,
      buttonGroupProps = _ref.buttonGroupProps,
      buttonProps = _ref.buttonProps;
  var classes = useStyles({
    padding: padding
  });

  if (!buttons.length) {
    return false;
  }

  return /*#__PURE__*/React.createElement("footer", {
    className: className || classes.root
  }, buttons && /*#__PURE__*/React.createElement(LayoutButtons, {
    buttons: buttons,
    buttonGroupProps: buttonGroupProps,
    buttonProps: buttonProps
  }));
};

export default LayoutFooter;
