// @kit-ui/schema/es/CustomLayout/CustomLayout.stories.js
import React from 'react';
import Schema from "../SchemaEditor/";
import CustomLayout from "./CustomLayout";
export default {
  title: 'Layouts/CustomLayouts/CustomLayout',
  component: Schema,
  args: {
    schema: {
      type: "object",
      properties: {
        size: {
          "type": "string",
          "enum": ["s", "m", "l"]
        }
      }
    },
    uiSchema: {
      "ui:layout": CustomLayout,
      "ui:options": {
        "editable": true
      },
      "ui:settings": ["size"],
      "size": {
        "ui:widget": "list"
      }
    },
    formData: {}
  },
  argTypes: {}
};

var Template = function Template(args) {
  return /*#__PURE__*/React.createElement(Schema, args);
};

export var Default = Template.bind({});
Default.args = {};
export var XsmallXlarge = Template.bind({});
XsmallXlarge.args = {
  schema: {
    type: "object",
    properties: {
      size: {
        "type": "string",
        "enum": ["xs", "s", "s-m", "m-s", "m", "m-l", "l-m", "l", "xl"]
      }
    }
  }
};
