// @kit-ui/admin/lib/NavMenu/NavMenuList.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _styles = require("@material-ui/core/styles");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var useStyles = (0, _styles.makeStyles)(function (theme) {
  return {
    root: {
      boxSizing: "border-box",
      listStyle: "none",
      padding: 0,
      margin: 0,
      "&[role=tree]": {
        marginLeft: theme.spacing(3)
      },
      "& > li + li": {
        marginTop: theme.spacing(.5)
      },
      "& > li > ul > li + li": {
        marginTop: 0
      },
      "&[role=group] > li + li": {
        marginTop: theme.spacing(.5)
      }
    }
  };
});

var MenuList = function MenuList(_ref) {
  var className = _ref.className,
      _ref$role = _ref.role,
      role = _ref$role === void 0 ? "tree" : _ref$role,
      children = _ref.children;
  var classes = useStyles();
  return /*#__PURE__*/_react.default.createElement("ul", {
    className: className || classes.root,
    role: role
  }, children);
};

var _default = MenuList;
exports.default = _default;
module.exports = exports.default;
