// @kit-ui/admin/lib/App/AppDrawer.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _styles = require("@material-ui/core/styles");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var useStyles = (0, _styles.makeStyles)(function (theme) {
  var _ariaExpandedTrue, _ariaExpandedTrue2;

  return {
    root: {
      position: "absolute",
      backgroundColor: theme.palette.background.default,
      zIndex: 1,
      top: 0,
      right: "auto",
      left: 0,
      bottom: 0,
      width: function width(props) {
        return props.width;
      },
      marginLeft: function marginLeft(props) {
        return "-" + props.width + "px";
      },
      transition: ".125s ease-out",
      "&[aria-expanded=true]": (_ariaExpandedTrue = {}, _ariaExpandedTrue[theme.breakpoints.up('sm')] = {
        marginLeft: 0
      }, _ariaExpandedTrue),
      "&[aria-expanded=true] + *": (_ariaExpandedTrue2 = {}, _ariaExpandedTrue2[theme.breakpoints.up('sm')] = {
        marginLeft: function marginLeft(props) {
          return props.width;
        }
      }, _ariaExpandedTrue2)
    }
  };
});

var AppDrawer = function AppDrawer(_ref) {
  var _ref$expanded = _ref.expanded,
      expanded = _ref$expanded === void 0 ? true : _ref$expanded,
      _ref$width = _ref.width,
      width = _ref$width === void 0 ? 224 : _ref$width,
      children = _ref.children;
  var classes = useStyles({
    width: width
  });
  return /*#__PURE__*/_react.default.createElement("aside", {
    className: classes.root,
    "aria-expanded": expanded
  }, children);
};

var _default = AppDrawer;
exports.default = _default;
module.exports = exports.default;
