// @kit-ui/admin/lib/App/AppLayout.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _ = require("../");

var _styles = require("@material-ui/core/styles");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var useStyles = (0, _styles.makeStyles)(function (theme) {
  var _ariaExpandedTrue, _ariaExpandedTrue2;

  return {
    app: {
      position: "absolute",
      backgroundColor: theme.palette.background.default,
      zIndex: 1,
      top: 0,
      right: 0,
      left: 0,
      bottom: 0,
      overflow: "hidden",
      display: "flex",
      flexDirection: "column"
    },
    header: {
      position: "absolute",
      height: theme.spacing(8),
      overflow: "hidden",
      marginTop: theme.spacing(-8),
      transition: ".125s ease-out",
      "&[aria-expanded=true]": {
        marginTop: 0,
        "& + *": {
          marginTop: theme.spacing(8)
        }
      }
    },
    body: {
      position: "relative",
      transition: ".125s ease-out",
      display: "flex",
      flexBasis: 0,
      flexGrow: 1
    },
    sidebar: {
      width: function width(props) {
        return props.sidebar.width;
      },
      marginLeft: function marginLeft(props) {
        return props.sidebar.width * -1;
      },
      transition: ".125s ease-out",
      "&[aria-expanded=true]": (_ariaExpandedTrue = {}, _ariaExpandedTrue[theme.breakpoints.up('sm')] = {
        marginLeft: 0
      }, _ariaExpandedTrue),
      "&[aria-expanded=true] + *": (_ariaExpandedTrue2 = {}, _ariaExpandedTrue2[theme.breakpoints.up('sm')] = {//               marginLeft: props => { return props.sidebar.width },
      }, _ariaExpandedTrue2)
    },
    content: {
      flexGrow: 1,
      display: "flex",
      flexDirection: "column",
      position: "relative"
    },
    navAction: {
      position: "relative",
      zIndex: 2,
      margin: theme.spacing(3)
    },
    navCalendar: {
      margin: theme.spacing(2, .5)
    },
    navMenu: {
      margin: theme.spacing(2)
    }
  };
});
/** AppLayout is a wrapper for all apps using AppHeader */

var AppLayout = function AppLayout(_ref) {
  var _ref$app = _ref.app,
      app = _ref$app === void 0 ? {} : _ref$app,
      _ref$theme = _ref.theme,
      theme = _ref$theme === void 0 ? undefined : _ref$theme,
      _ref$header = _ref.header,
      header = _ref$header === void 0 ? {} : _ref$header,
      _ref$sidebar = _ref.sidebar,
      sidebar = _ref$sidebar === void 0 ? {
    width: 224
  } : _ref$sidebar,
      _ref$search = _ref.search,
      search = _ref$search === void 0 ? undefined : _ref$search,
      _ref$subview = _ref.subview,
      subview = _ref$subview === void 0 ? undefined : _ref$subview,
      _ref$calendar = _ref.calendar,
      calendar = _ref$calendar === void 0 ? undefined : _ref$calendar,
      _ref$primaryAction = _ref.primaryAction,
      primaryAction = _ref$primaryAction === void 0 ? undefined : _ref$primaryAction,
      menu = _ref.menu,
      _ref$menuByUrl = _ref.menuByUrl,
      menuByUrl = _ref$menuByUrl === void 0 ? {} : _ref$menuByUrl,
      _ref$currentUrl = _ref.currentUrl,
      currentUrl = _ref$currentUrl === void 0 ? undefined : _ref$currentUrl,
      onSelect = _ref.onSelect,
      children = _ref.children;

  if (search && app.search) {
    search = _extends({}, app.search, search);
  }

  header = _extends({
    title: app.title,
    subtitle: app.subtitle,
    subview: subview,
    search: search
  }, header);
  sidebar = _extends({}, sidebar, {
    width: 256
  });
  var classes = useStyles({
    header: header,
    sidebar: sidebar
  });
  return /*#__PURE__*/_react.default.createElement(_.AppBase, {
    theme: theme,
    className: classes.app
  }, /*#__PURE__*/_react.default.createElement(_.AppHeader, _extends({}, header, {
    className: classes.header
  })), /*#__PURE__*/_react.default.createElement(_.AppBody, {
    className: classes.body
  }, /*#__PURE__*/_react.default.createElement(_.AppSidebar, _extends({}, sidebar, {
    className: classes.sidebar
  }), primaryAction && /*#__PURE__*/_react.default.createElement(_.NavAction, {
    className: classes.navAction,
    primaryAction: primaryAction,
    onSelect: onSelect
  }), calendar && /*#__PURE__*/_react.default.createElement(_.NavCalendar, _extends({
    className: classes.navCalendar
  }, calendar, {
    onChange: function onChange(date) {
      return onSelect({
        type: "calendar",
        date: date
      });
    }
  })), menu && /*#__PURE__*/_react.default.createElement(_.NavMenuControlled, {
    className: classes.navMenu,
    menu: menu,
    menuByUrl: menuByUrl,
    currentUrl: currentUrl,
    onSelect: onSelect
  })), /*#__PURE__*/_react.default.createElement(_.AppContent, {
    className: classes.content
  }, children)));
};

AppLayout.defaultProps = {
  app: {},
  header: {
    expanded: true
  }
};
AppLayout.propTypes = process.env.NODE_ENV !== "production" ? {
  /** App properties */
  app: _propTypes.default.shape({
    uniqueId: _propTypes.default.string,
    title: _propTypes.default.string
  }),

  /** Header properties */
  header: _propTypes.default.shape({
    expanded: _propTypes.default.bool
  }),

  /** Search properties */
  search: _propTypes.default.shape({
    expanded: _propTypes.default.bool
  }),

  /** Subview properties */
  subview: _propTypes.default.shape({
    expanded: _propTypes.default.bool
  })
} : {};
var _default = AppLayout;
exports.default = _default;
module.exports = exports.default;
