// @kit-ui/admin/lib/App/AppSidebar.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styles = require("@material-ui/core/styles");

var _NavAction = _interopRequireDefault(require("../NavAction/NavAction"));

var _NavCalendar = _interopRequireDefault(require("../NavCalendar/NavCalendar"));

var _NavMenu = _interopRequireDefault(require("../NavMenu/NavMenu"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var useStyles = (0, _styles.makeStyles)(function (theme) {
  return {
    sidebar: {
      position: "relative"
    },
    action: {
      position: "relative",
      zIndex: 2,
      margin: theme.spacing(3)
    }
  };
});

var AppSidebar = function AppSidebar(_ref) {
  var className = _ref.className,
      _ref$expanded = _ref.expanded,
      expanded = _ref$expanded === void 0 ? false : _ref$expanded,
      _ref$width = _ref.width,
      width = _ref$width === void 0 ? 224 : _ref$width,
      calendar = _ref.calendar,
      menu = _ref.menu,
      _ref$menuByUrl = _ref.menuByUrl,
      menuByUrl = _ref$menuByUrl === void 0 ? {} : _ref$menuByUrl,
      primaryAction = _ref.primaryAction,
      onSelect = _ref.onSelect,
      children = _ref.children;
  var classes = useStyles({
    width: width
  });
  return /*#__PURE__*/_react.default.createElement("aside", {
    className: className || classes.sidebar,
    "aria-expanded": expanded
  }, primaryAction && /*#__PURE__*/_react.default.createElement(_NavAction.default, {
    className: classes.action,
    primaryAction: primaryAction,
    onSelect: onSelect
  }), calendar && /*#__PURE__*/_react.default.createElement(_NavCalendar.default, _extends({}, calendar, {
    onSelect: onSelect
  })), menu && /*#__PURE__*/_react.default.createElement(_NavMenu.default, {
    menu: menu,
    menuByUrl: menuByUrl,
    onSelect: onSelect
  }), children);
};

AppSidebar.propTypes = process.env.NODE_ENV !== "production" ? {
  className: _propTypes.default.string,
  expanded: _propTypes.default.bool,
  calendar: _propTypes.default.shape({
    date: _propTypes.default.string
  }),
  menu: _propTypes.default.array
} : {};
AppSidebar.defaultProps = {
  calendar: undefined
};
var _default = AppSidebar;
exports.default = _default;
module.exports = exports.default;
