// @kit-ui/admin/es/AppSwitcher/AppsList.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import React, { Component } from 'react';
import AppsListItem from "./AppsListItem";
import { makeStyles } from '@material-ui/core/styles';
var useStyles = makeStyles(function (theme) {
  return {
    root: {
      display: "flex",
      flexDirection: "column"
    }
  };
});

var AppsList = function AppsList(_ref) {
  var className = _ref.className,
      _ref$apps = _ref.apps,
      apps = _ref$apps === void 0 ? undefined : _ref$apps,
      size = _ref.size,
      children = _ref.children,
      onSelect = _ref.onSelect;
  var classes = useStyles();
  return /*#__PURE__*/React.createElement("section", {
    className: className || classes.root
  }, apps && apps.length && apps.map(function (app) {
    return /*#__PURE__*/React.createElement(AppsListItem, _extends({}, app, {
      size: size,
      onClick: function onClick() {
        return onSelect(app);
      }
    }));
  }), children);
};

AppsList.defaultProps = {
  size: "small"
};
export default AppsList;
