// @kit-ui/admin/es/AppSwitcher/AppsGridItem.js
import React, { Component } from 'react';
import ButtonBase from '@material-ui/core/ButtonBase';
import IconButton from '@material-ui/core/IconButton';
import Typography from '@material-ui/core/Typography';
import { makeStyles } from '@material-ui/core/styles';
var useStyles = makeStyles(function (theme) {
  return {
    root: {
      display: "flex",
      flexDirection: "column",
      backgroundColor: "transparent",
      color: "inherit"
    },
    figure: {
      width: function width(props) {
        return props.size;
      },
      height: function height(props) {
        return props.size;
      },
      "& svg, & img": {
        position: "absolute",
        top: 0,
        right: 0,
        bottom: 0,
        left: 0,
        width: function width(props) {
          return props.size;
        },
        height: function height(props) {
          return props.size;
        },
        margin: "auto"
      }
    },

    /*
    img: {
        position: "absolute",
        top: 0,
        right: 0,
        bottom: 0,
        left: 0,
        maxWidth: "64px",
        maxHeight: "64px",
        margin: "auto"
    },
    */
    label: {
      display: "block",
      fontFamily: "Akkurat, sans-serif",
      fontSize: "12px",
      marginTop: "-1em"
    }
  };
});

var AppsGridItem = function AppsGridItem(_ref) {
  var title = _ref.title,
      imageUrl = _ref.imageUrl,
      icon = _ref.icon,
      _ref$size = _ref.size,
      size = _ref$size === void 0 ? "large" : _ref$size,
      onClick = _ref.onClick;
  var sizes = {
    "small": 36,
    "medium": 48,
    "large": 64
  };
  size = sizes[size] || size || 64;
  var classes = useStyles({
    size: size
  });
  return /*#__PURE__*/React.createElement(ButtonBase, {
    className: classes.root,
    disableRipple: true,
    onClick: onClick
  }, /*#__PURE__*/React.createElement(IconButton, {
    className: classes.figure,
    component: "figure"
  }, !icon && imageUrl && /*#__PURE__*/React.createElement("img", {
    className: classes.img,
    src: imageUrl
  }), icon), /*#__PURE__*/React.createElement(Typography, {
    className: classes.label,
    noWrap: true
  }, title));
};

export default AppsGridItem;
