// @kit-ui/admin/es/AppSwitcher/AppsByOwner.js
import React from 'react';
import PropTypes from "prop-types";
import AppsGrid from './AppsGrid';
import ButtonBase from '@material-ui/core/ButtonBase';
import Typography from '@material-ui/core/Typography';
import { makeStyles } from '@material-ui/core/styles';

var ButtonBack = function ButtonBack(_ref) {
  var className = _ref.className,
      onClick = _ref.onClick;
  return /*#__PURE__*/React.createElement(ButtonBase, {
    className: className,
    onClick: onClick
  }, "Endre");
};

var useStyles = makeStyles(function (theme) {
  return {
    root: {
      display: "block",
      position: "relative"
    },
    header: {
      display: "flex",
      borderColor: "#ccc",
      alignItems: "center",
      justifyContent: "center",
      paddingTop: "18px"
    },
    title: {
      fontFamily: "Akkurat, sans-serif",
      fontWeight: "bold",
      lineHeight: "24px"
    },
    toggleButton: {
      fontFamily: "Akkurat, sans-serif",
      fontSize: "14px",
      fontWeight: "normal",
      margin: "0 6px"
    }
  };
});

var AppsByOwner = function AppsByOwner(_ref2) {
  var _ref2$title = _ref2.title,
      title = _ref2$title === void 0 ? "Owner" : _ref2$title,
      onCollapse = _ref2.onCollapse,
      apps = _ref2.apps,
      onSelect = _ref2.onSelect;
  var classes = useStyles();
  return /*#__PURE__*/React.createElement("div", {
    className: classes.root
  }, /*#__PURE__*/React.createElement("header", {
    className: classes.header
  }, /*#__PURE__*/React.createElement(Typography, {
    className: classes.title
  }, title), /*#__PURE__*/React.createElement(ButtonBack, {
    className: classes.toggleButton,
    onClick: onCollapse
  })), /*#__PURE__*/React.createElement(AppsGrid, {
    apps: apps,
    onSelect: onSelect
  }));
};

AppsByOwner.propTypes = process.env.NODE_ENV !== "production" ? {
  owners: PropTypes.array,
  currentOwner: PropTypes.shape({
    uniqueId: PropTypes.string,
    title: PropTypes.string,
    apps: PropTypes.array
  })
} : {};
export default AppsByOwner;
