// @kit-ui/admin/es/AppSwitcher/AppsGrid.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import React, { Component } from 'react';
import AppsGridItem from "./AppsGridItem";
import { makeStyles } from '@material-ui/core/styles';
var useStyles = makeStyles(function (theme) {
  return {
    root: {
      display: "flex",
      flexWrap: "wrap",
      justifyContent: "center",
      padding: theme.spacing(3.5)
    },
    grid: {
      display: "flex",
      flexWrap: "wrap",
      margin: "auto"
    }
  };
});

var AppsGrid = function AppsGrid(_ref) {
  var className = _ref.className,
      _ref$apps = _ref.apps,
      apps = _ref$apps === void 0 ? undefined : _ref$apps,
      _ref$size = _ref.size,
      size = _ref$size === void 0 ? "large" : _ref$size,
      children = _ref.children,
      onSelect = _ref.onSelect;
  var classes = useStyles();
  return /*#__PURE__*/React.createElement("div", {
    className: className || classes.root
  }, /*#__PURE__*/React.createElement("div", {
    className: classes.grid
  }, apps && apps.length && apps.map(function (app) {
    return /*#__PURE__*/React.createElement(AppsGridItem, _extends({}, app, {
      size: size,
      onClick: function onClick() {
        return onSelect(app);
      }
    }));
  }), children));
};

AppsGrid.defaultProps = {
  size: "large"
};
export default AppsGrid;
