// @kit-ui/admin/es/AppSwitcher/FilterOwnersControlled.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _inheritsLoose(subClass, superClass) { subClass.prototype = Object.create(superClass.prototype); subClass.prototype.constructor = subClass; subClass.__proto__ = superClass; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

import React, { Component } from 'react';
import FilterOwnersUncontrolled from "./FilterOwnersUncontrolled";

var FilterOwnersControlled = /*#__PURE__*/function (_Component) {
  _inheritsLoose(FilterOwnersControlled, _Component);

  function FilterOwnersControlled() {
    var _this;

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = _Component.call.apply(_Component, [this].concat(args)) || this;

    _defineProperty(_assertThisInitialized(_this), "state", {
      q: "",
      filteredOwners: []
    });

    _defineProperty(_assertThisInitialized(_this), "componentDidUpdate", function (prevProps, prevState) {
      if (prevProps.owners !== _this.props.owners) {
        _this.filterOwners();
      }

      if (prevState.q !== _this.state.q) {
        _this.filterOwners();
      }
    });

    _defineProperty(_assertThisInitialized(_this), "filterOwners", function () {
      var owners = _this.props.owners;
      var q = _this.state.q;
      var filteredOwners = owners.filter(function (owner) {
        var title = owner.title;

        if (q && !title.toLowerCase().includes(q.toLowerCase())) {
          return false;
        }

        return true;
      });

      _this.setState({
        filteredOwners: filteredOwners
      });
    });

    _defineProperty(_assertThisInitialized(_this), "onSearch", function (q, event) {
      event.preventDefault();

      _this.setState({
        q: q
      });
    });

    return _this;
  }

  var _proto = FilterOwnersControlled.prototype;

  _proto.componentDidMount = function componentDidMount() {
    this.filterOwners();
  };

  _proto.render = function render() {
    var _this$state = this.state,
        q = _this$state.q,
        filteredOwners = _this$state.filteredOwners;
    return /*#__PURE__*/React.createElement(FilterOwnersUncontrolled, _extends({}, this.props, {
      q: q,
      onSearch: this.onSearch,
      owners: filteredOwners
    }));
  };

  return FilterOwnersControlled;
}(Component);

_defineProperty(FilterOwnersControlled, "defaultProps", {
  owners: []
});

export default FilterOwnersControlled;
