// @kit-ui/admin/es/AppSwitcher/FilterOwnersUncontrolled.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import React, { useRef } from 'react';
import { WidgetSearch, WidgetScroller } from "../Widget";
import OwnersList from "./OwnersList";
import { makeStyles } from '@material-ui/core/styles';
var useStyles = makeStyles(function (theme) {
  return {
    root: {
      position: "absolute",
      width: "100%",
      top: 0,
      right: 0,
      bottom: 0,
      left: 0
    },
    search: {
      display: "flex",
      position: "relative",
      backgroundColor: "transparent",
      color: "black",
      height: "36px",
      overflow: "hidden",
      fontFamily: "Akkurat, sans-serif",
      margin: "12px"
    },
    scroller: {
      position: "absolute",
      zIndex: 1,
      top: 0,
      right: 0,
      bottom: 0,
      left: 0
    },
    results: {
      paddingTop: theme.spacing(7),
      display: "flex",
      flexDirection: "column",
      width: "100%"
    },
    nohits: {
      fontFamily: "Akkurat, sans-serif",
      fontSize: "14px",
      padding: theme.spacing(1.5)
    }
  };
});

var FilterOwnersUncontrolled = function FilterOwnersUncontrolled(_ref) {
  var className = _ref.className,
      search = _ref.search,
      q = _ref.q,
      _ref$owners = _ref.owners,
      owners = _ref$owners === void 0 ? [] : _ref$owners,
      onSearch = _ref.onSearch,
      onSelect = _ref.onSelect;
  var classes = useStyles();

  if (!owners.length) {
    return /*#__PURE__*/React.createElement("div", {
      className: className || classes.root
    }, /*#__PURE__*/React.createElement(WidgetSearch, _extends({}, search, {
      value: q,
      onChange: onSearch
    })), /*#__PURE__*/React.createElement(WidgetScroller, {
      className: classes.scroller
    }, /*#__PURE__*/React.createElement("div", {
      className: classes.results
    }, /*#__PURE__*/React.createElement("p", {
      className: classes.nohits
    }, "Ingen treff"))));
  }

  var placeholder = search.placeholder && search.placeholder.replace('{hits}', owners.length);
  return /*#__PURE__*/React.createElement("div", {
    className: className || classes.root
  }, /*#__PURE__*/React.createElement(WidgetSearch, _extends({}, search, {
    placeholder: placeholder,
    value: q,
    onChange: onSearch
  })), /*#__PURE__*/React.createElement(WidgetScroller, {
    className: classes.scroller
  }, /*#__PURE__*/React.createElement(OwnersList, {
    className: classes.results,
    owners: owners,
    onSelect: onSelect
  })));
};

FilterOwnersUncontrolled.defaultProps = {
  search: {
    placeholder: "Søk i {hits} eiere"
  }
};
export default FilterOwnersUncontrolled;
