// @kit-ui/core/lib/Iframe/IframeChildren.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _reactFrameComponent = _interopRequireDefault(require("react-frame-component"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _inheritsLoose(subClass, superClass) { subClass.prototype = Object.create(superClass.prototype); subClass.prototype.constructor = subClass; subClass.__proto__ = superClass; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var IframeChildren = /*#__PURE__*/function (_Component) {
  _inheritsLoose(IframeChildren, _Component);

  function IframeChildren() {
    var _this;

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = _Component.call.apply(_Component, [this].concat(args)) || this;

    _defineProperty(_assertThisInitialized(_this), "state", {});

    _defineProperty(_assertThisInitialized(_this), "componentDidUpdate", function (prevProps) {
      if (_this.props.children !== prevProps.children) {
        _this.getHead();
      }
    });

    _defineProperty(_assertThisInitialized(_this), "iframeDidMount", function () {
      _this.getHead();
    });

    _defineProperty(_assertThisInitialized(_this), "getHead", function () {
      var stylesheets = [],
          styles = [];
      var linkTags = document.head.querySelectorAll('[rel="stylesheet"]');
      var styleTags = document.head.getElementsByTagName("style");

      for (var i = 0; i < linkTags.length; i++) {
        var styleheet = linkTags[i].getAttribute('href');
        stylesheets.push(styleheet);
      }

      for (var _i = 0; _i < styleTags.length; _i++) {
        var style = styleTags[_i].innerHTML;
        styles.push(style);
      }

      var head = [];

      if (_this.props.style) {
        head.push( /*#__PURE__*/_react.default.createElement("style", null, _this.props.style));
      } else {
        stylesheets.forEach(function (stylesheet) {
          head.push( /*#__PURE__*/_react.default.createElement("link", {
            type: "text/css",
            rel: "stylesheet",
            href: stylesheet
          }));
        });
        styles.forEach(function (style) {
          head.push( /*#__PURE__*/_react.default.createElement("style", null, style));
        });
      }

      _this.setState({
        head: head
      });
    });

    return _this;
  }

  var _proto = IframeChildren.prototype;

  _proto.componentDidMount = function componentDidMount() {
    this.getHead();
  };

  _proto.render = function render() {
    var className = this.props.className;
    var head = this.state.head;

    if (!head) {
      return false;
    }

    var template = '<!DOCTYPE html><html><head></head><body id="preview-iframe"></body></html>';
    return /*#__PURE__*/_react.default.createElement(_reactFrameComponent.default, {
      className: className,
      initialContent: template,
      head: head,
      mountTarget: "#preview-iframe",
      contentDidMount: this.iframeDidMount
    }, this.props.children);
  };

  return IframeChildren;
}(_react.Component);

_defineProperty(IframeChildren, "defaultProps", {});

IframeChildren.propTypes = process.env.NODE_ENV !== "production" ? {
  className: _propTypes.default.string,
  children: _propTypes.default.node
} : {};
var _default = IframeChildren;
exports.default = _default;
module.exports = exports.default;
