// @kit-ui/core/lib/Iframe/Iframe.stories.js
"use strict";

exports.__esModule = true;
exports.UrlWithPostmessage = exports.Url = exports.Children = exports.Default = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _Iframe = _interopRequireDefault(require("./Iframe"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _default = {
  title: 'Iframe',
  component: _Iframe.default,
  argTypes: {}
};
exports.default = _default;

var Template = function Template(args) {
  return /*#__PURE__*/_react.default.createElement(_Iframe.default, args);
};

var Default = Template.bind({});
exports.Default = Default;
Default.args = {};
var Children = Template.bind({});
exports.Children = Children;
Children.args = {
  children: "children"
};
var Url = Template.bind({});
exports.Url = Url;
Url.args = {
  url: "//digitaltmuseum.no/search/"
};
var UrlWithPostmessage = Template.bind({});
exports.UrlWithPostmessage = UrlWithPostmessage;
UrlWithPostmessage.args = {
  url: "//digitaltmuseum.no/search/",
  postMessage: {
    q: "ølbolle"
  }
};
