// @kit-ui/schema/lib/LocalizedString/LocalizedStringSettings.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _core = require("@kit-ui/core");

var _utils = require("../utils");

var _IconButton = _interopRequireDefault(require("@material-ui/core/IconButton"));

var _styles = require("@material-ui/core/styles");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var useStyles = (0, _styles.makeStyles)(function (theme) {
  return {
    button: {},
    icon: {
      width: 20,
      height: 20,
      //        borderRadius: "50%",
      border: "1px solid",
      borderColor: theme.palette.divider,
      display: "flex",
      justifyContent: "center",
      alignItems: "center" //        boxShadow: theme.shadows[1]

    },
    label: {
      fontFamily: "Akkurat, sans-serif",
      fontSize: 11,
      textTransform: "uppercase",
      fontWeight: "bold",
      lineHeight: '1',
      textAlign: "center"
    }
  };
});

var LocalizedStringSettings = function LocalizedStringSettings(props) {
  var formContext = props.formContext;
  var languages = formContext.languages || ["en"];
  var defaultLocale = formContext.defaultLocale || languages[0];
  var currentLocale = formContext.currentLocale || defaultLocale;

  var _useState = (0, _react.useState)(false),
      expanded = _useState[0],
      setExpanded = _useState[1];

  var anchorRef = (0, _react.useRef)(null);

  var _onToggle = function _onToggle(event) {
    setExpanded(function (prevExpanded) {
      return !prevExpanded;
    });
  };

  var _onCollapse = function _onCollapse(event) {
    setExpanded(false);
  };

  var classes = useStyles();
  var ObjectField = props.registry.fields.ObjectField;
  var uiSchema = {
    "ui:options": {
      "padding": 2,
      "spacing": 2
    }
  };
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_IconButton.default, {
    className: classes.button,
    onClick: _onToggle,
    ref: anchorRef
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: classes.icon
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: classes.label
  }, currentLocale))), /*#__PURE__*/_react.default.createElement(_core.Dropdown, {
    expanded: expanded,
    anchorEl: anchorRef.current,
    onClickAway: _onCollapse
  }, /*#__PURE__*/_react.default.createElement(ObjectField, _extends({}, props, {
    uiSchema: uiSchema
  }))));
};

LocalizedStringSettings.defaultProps = {};
LocalizedStringSettings.propTypes = process.env.NODE_ENV !== "production" ? {} : {};
var _default = LocalizedStringSettings;
exports.default = _default;
module.exports = exports.default;
