// @kit-ui/schema/lib/LocalizedString/LocalizedStringField.stories.js
"use strict";

exports.__esModule = true;
exports.CustomWidget = exports.TransformString = exports.Default = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _LocalizedStringField = _interopRequireDefault(require("./LocalizedStringField"));

var _SchemaEditor = _interopRequireDefault(require("../SchemaEditor/SchemaEditor"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _default = {
  title: 'Fields/LocalizedStringField',
  component: _LocalizedStringField.default,
  args: {
    formContext: {
      languages: ["no", "sv", "en"],
      defaultLocale: "no"
    },
    schema: {
      "type": "object",
      "properties": {
        "content": {
          "type": "localizedString"
        }
      }
    },
    uiSchema: {
      "ui:field": "pageEditor",
      "ui:fieldset": ["content"]
    }
  },
  argTypes: {}
};
exports.default = _default;

var Template = function Template(args) {
  return /*#__PURE__*/_react.default.createElement(_SchemaEditor.default, args);
};

var Default = Template.bind({});
exports.Default = Default;
Default.args = {};
var TransformString = Template.bind({});
exports.TransformString = TransformString;
TransformString.args = {
  formData: {
    "content": "Transform string"
  }
};
var CustomWidget = Template.bind({});
exports.CustomWidget = CustomWidget;
CustomWidget.args = {
  formData: {
    "locale:no": "Text"
  },
  uiSchema: {
    "ui:widget": "textarea",
    "ui:title": "Textarea",
    "ui:help": "Helper text"
  }
};
