// @kit-ui/admin/lib/Editor/EditorFooter.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _styles = require("@material-ui/core/styles");

var _AppBar = _interopRequireDefault(require("@material-ui/core/AppBar"));

var _Toolbar = _interopRequireDefault(require("@material-ui/core/Toolbar"));

var _EditorStatus = _interopRequireDefault(require("./EditorStatus"));

var _EditorAction = _interopRequireDefault(require("./EditorAction"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var useStyles = (0, _styles.makeStyles)(function (theme) {
  var _root;

  return {
    root: (_root = {
      position: "absolute",
      zIndex: 3,
      top: "auto",
      right: 0,
      bottom: 0,
      left: 0,
      //        backgroundColor: theme.palette.primary.main,
      //        color: theme.palette.primary.contrastText,
      display: "flex",
      alignItems: "center",
      justifyContent: "space-between",
      height: theme.spacing(7),
      fontSize: "16px"
    }, _root[theme.breakpoints.up('sm')] = {
      display: "none"
    }, _root),
    status: {
      display: "flex",
      alignItems: "center",
      height: theme.spacing(5),
      minWidth: theme.spacing(14),
      justifyContent: "center",
      marginTop: theme.spacing(1),
      marginBottom: theme.spacing(1),
      paddingRight: theme.spacing(3),
      paddingLeft: theme.spacing(3),
      fontSize: "14px",
      opacity: 0.5
    },
    action: {
      borderLeft: "1px solid",
      borderColor: "rgba(255,255,255,0.25)",
      "& > *": {
        height: theme.spacing(7)
      },
      "& button": {
        paddingLeft: theme.spacing(1),
        paddingRight: theme.spacing(1),
        height: theme.spacing(7),
        minWidth: theme.spacing(7)
      }
    }
  };
});

var EditorFooter = function EditorFooter(_ref) {
  var className = _ref.className,
      id = _ref.id,
      _ref$elevation = _ref.elevation,
      elevation = _ref$elevation === void 0 ? 6 : _ref$elevation,
      props = _objectWithoutPropertiesLoose(_ref, ["className", "id", "elevation"]);

  var classes = useStyles();
  return /*#__PURE__*/_react.default.createElement(_AppBar.default, {
    component: "footer",
    elevation: elevation,
    square: true,
    className: className || classes.root
  }, /*#__PURE__*/_react.default.createElement(_Toolbar.default, null, /*#__PURE__*/_react.default.createElement(_EditorStatus.default, _extends({}, props, {
    className: classes.status
  })), /*#__PURE__*/_react.default.createElement(_EditorAction.default, _extends({}, props, {
    className: classes.action
  }))));
};

var _default = EditorFooter;
exports.default = _default;
module.exports = exports.default;
