// @kit-ui/admin/lib/Editor/EditorHeader.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styles = require("@material-ui/core/styles");

var _EditorPath = _interopRequireDefault(require("./EditorPath"));

var _EditorLocale = _interopRequireDefault(require("./EditorLocale"));

var _EditorStatus = _interopRequireDefault(require("./EditorStatus"));

var _EditorAction = _interopRequireDefault(require("./EditorAction"));

var _EditorToogle = _interopRequireDefault(require("./EditorToogle"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var useStyles = (0, _styles.makeStyles)(function (theme) {
  var _trailing;

  return {
    root: {
      backgroundColor: theme.palette.background.paper,
      display: "flex",
      alignItems: "center",
      justifyContent: "space-between",
      flexWrap: "nowrap",
      maxWidth: "100%",
      width: "100%",
      minHeight: theme.spacing(8),
      "& + *": {
        marginTop: theme.spacing(8)
      }
    },
    leading: {
      display: "flex",
      alignItems: "center",
      justifyContent: "flex-start",
      flexWrap: "nowrap",
      maxWidth: "100%"
    },
    path: {
      display: "flex",
      alignItems: "center",
      justifyContent: "flex-start",
      flexWrap: "nowrap"
    },
    locale: {
      marginLeft: theme.spacing(1)
    },
    trailing: (_trailing = {
      display: "flex",
      alignItems: "center",
      justifyContent: "center",
      marginLeft: theme.spacing(1),
      marginRight: theme.spacing(1)
    }, _trailing[theme.breakpoints.down('xs')] = {
      display: "none"
    }, _trailing),
    status: {
      fontSize: "14px",
      margin: theme.spacing(1.5)
    },
    action: {
      fontSize: "14px"
    }
  };
});

var EditorHeader = function EditorHeader(_ref) {
  var collapsible = _ref.collapsible,
      expanded = _ref.expanded,
      onToggle = _ref.onToggle,
      onSelect = _ref.onSelect,
      className = _ref.className,
      props = _objectWithoutPropertiesLoose(_ref, ["collapsible", "expanded", "onToggle", "onSelect", "className"]);

  var classes = useStyles();
  return /*#__PURE__*/_react.default.createElement("header", {
    className: className || classes.root
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: classes.leading
  }, /*#__PURE__*/_react.default.createElement(_EditorPath.default, _extends({}, props, {
    className: classes.path,
    onSelect: onSelect
  })), /*#__PURE__*/_react.default.createElement(_EditorLocale.default, _extends({}, props, {
    className: classes.locale
  })), collapsible && /*#__PURE__*/_react.default.createElement(_EditorToogle.default, {
    expanded: expanded,
    onClick: onToggle
  })), /*#__PURE__*/_react.default.createElement("div", {
    className: classes.trailing
  }, /*#__PURE__*/_react.default.createElement(_EditorStatus.default, _extends({}, props, {
    className: classes.status
  })), /*#__PURE__*/_react.default.createElement(_EditorAction.default, _extends({}, props, {
    className: classes.action
  }))));
};

EditorHeader.defaultProps = {
  untitled: "Untitled"
};
EditorHeader.propTypes = process.env.NODE_ENV !== "production" ? {
  status: _propTypes.default.string,
  statusLabel: _propTypes.default.string,
  statusDate: _propTypes.default.string,
  statusCode: _propTypes.default.string,
  statusMessage: _propTypes.default.string,
  untitled: _propTypes.default.string,
  title: _propTypes.default.string,
  parents: _propTypes.default.array,
  languages: _propTypes.default.array,
  currentLocale: _propTypes.default.string,
  onLocale: _propTypes.default.func,
  primaryAction: _propTypes.default.object
} : {};
var _default = EditorHeader;
exports.default = _default;
module.exports = exports.default;
