// @kit-ui/admin/lib/Editor/EditorContent.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _styles = require("@material-ui/core/styles");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var useStyles = (0, _styles.makeStyles)(function (theme) {
  return {
    root: {
      position: "absolute",
      zIndex: 2,
      top: 0,
      right: 0,
      left: 0,
      bottom: 0,
      transition: ".125s ease-out",
      overflowY: "scroll"
    }
  };
});

var EditorContent = function EditorContent(_ref) {
  var className = _ref.className,
      _ref$expanded = _ref.expanded,
      expanded = _ref$expanded === void 0 ? false : _ref$expanded,
      children = _ref.children;
  var classes = useStyles();
  return /*#__PURE__*/_react.default.createElement("div", {
    className: className || classes.root,
    "aria-expanded": expanded
  }, children);
};

var _default = EditorContent;
exports.default = _default;
module.exports = exports.default;
