// @kit-ui/admin/lib/Editor/EditorOverlay.stories.js
"use strict";

exports.__esModule = true;
exports.Collapsed = exports.Expanded = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _EditorBase = _interopRequireDefault(require("./EditorBase"));

var _EditorHeader = _interopRequireDefault(require("./EditorHeader"));

var _EditorBody = _interopRequireDefault(require("./EditorBody"));

var _EditorSection = _interopRequireDefault(require("./EditorSection"));

var _EditorOverlay = _interopRequireDefault(require("./EditorOverlay"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _default = {
  title: 'Editor/EditorOverlay',
  component: _EditorOverlay.default,
  args: {
    header: {
      title: "Overlay"
    },
    overlay: {
      expanded: true
    }
  },
  argTypes: {}
};
exports.default = _default;

var Template = function Template(args) {
  return /*#__PURE__*/_react.default.createElement(_EditorBase.default, null, /*#__PURE__*/_react.default.createElement(_EditorHeader.default, args.header), /*#__PURE__*/_react.default.createElement(_EditorBody.default, null, "Content"), /*#__PURE__*/_react.default.createElement(_EditorOverlay.default, args.overlay, "Overlay"));
};

var Expanded = Template.bind({});
exports.Expanded = Expanded;
Expanded.args = {
  header: {
    title: "Overlay expanded"
  },
  overlay: {
    expanded: true
  }
};
var Collapsed = Template.bind({});
exports.Collapsed = Collapsed;
Collapsed.args = {
  header: {
    title: "Overlay collapsed"
  },
  overlay: {
    expanded: false
  }
};
