// @kit-ui/admin/lib/Editor/EditorStatus.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _moment = _interopRequireDefault(require("moment"));

require("moment/locale/nb");

require("moment/locale/nn");

require("moment/locale/sv");

var _styles = require("@material-ui/core/styles");

var _Typography = _interopRequireDefault(require("@material-ui/core/Typography"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var useStyles = (0, _styles.makeStyles)(function (theme) {
  return {
    root: {
      display: "flex",
      alignItems: "center",
      justifyContent: "center",
      fontFamily: "Akkurat, sans-serif",
      fontSize: "12px",
      margin: theme.spacing(1)
    },
    status: {
      display: "inline",
      fontWeight: "bold",
      "&[data-status=error]": {
        color: theme.palette.error.main
      }
    },
    message: {
      display: "inline",
      marginLeft: theme.spacing(1),
      "&[data-status=error]": {
        color: theme.palette.error.main
      }
    },
    datetime: {
      display: "inline",
      marginLeft: theme.spacing(1)
    }
  };
});
var statusLabels = {
  "new": "Ny",
  "draft": "Utkast",
  "publish": "Publisert",
  "error": "Feil"
};

var StatusLabel = function StatusLabel(_ref) {
  var className = _ref.className,
      _ref$locale = _ref.locale,
      locale = _ref$locale === void 0 ? "nb" : _ref$locale,
      status = _ref.status,
      children = _ref.children;
  var classes = useStyles();
  var statusLabel = children || statusLabels[status] || status;
  return /*#__PURE__*/_react.default.createElement("span", {
    className: className || classes.status,
    "data-status": status
  }, statusLabel);
};

var StatusDate = function StatusDate(_ref2) {
  var className = _ref2.className,
      _ref2$locale = _ref2.locale,
      locale = _ref2$locale === void 0 ? "nb" : _ref2$locale,
      _ref2$format = _ref2.format,
      format = _ref2$format === void 0 ? "relative" : _ref2$format,
      datetime = _ref2.datetime,
      children = _ref2.children;
  var classes = useStyles();
  datetime = children || datetime;

  if (!datetime) {
    return false;
  } // format value


  _moment.default.locale(locale);

  var date;

  if (format === "relative") {
    date = _moment.default.utc(datetime).local().fromNow();
  } else if (format) {
    date = _moment.default.utc(datetime).local().format(format);
  }

  return /*#__PURE__*/_react.default.createElement("time", {
    datetime: datetime,
    className: className || classes.datetime
  }, date);
};

var StatusMessage = function StatusMessage(_ref3) {
  var className = _ref3.className,
      status = _ref3.status,
      code = _ref3.code,
      children = _ref3.children;
  var classes = useStyles();

  if (!children) {
    return false;
  }

  return /*#__PURE__*/_react.default.createElement("span", {
    "data-status": status,
    "data-code": code,
    className: className || classes.message
  }, children);
};

var EditorStatus = function EditorStatus(_ref4) {
  var className = _ref4.className,
      status = _ref4.status,
      statusLabel = _ref4.statusLabel,
      statusDate = _ref4.statusDate,
      statusCode = _ref4.statusCode,
      statusMessage = _ref4.statusMessage;
  var classes = useStyles();
  return /*#__PURE__*/_react.default.createElement(_Typography.default, {
    className: className || classes.root,
    noWrap: true
  }, /*#__PURE__*/_react.default.createElement(StatusLabel, {
    className: classes.status,
    status: status
  }, statusLabel), /*#__PURE__*/_react.default.createElement(StatusMessage, {
    className: classes.message,
    status: status,
    code: statusCode
  }, statusMessage), !statusMessage && /*#__PURE__*/_react.default.createElement(StatusDate, {
    className: classes.datetime
  }, statusDate));
};

var _default = EditorStatus;
exports.default = _default;
module.exports = exports.default;
