// @kit-ui/admin/lib/Editor/EditorTheme.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _color = _interopRequireDefault(require("color"));

var _styles = require("@material-ui/core/styles");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var createTheme = function createTheme(_ref) {
  var color = _ref.color;
  var backgroundColor = (0, _color.default)(color);

  if (backgroundColor.isDark()) {
    return (0, _styles.createMuiTheme)({
      palette: {
        type: "dark",
        background: {
          paper: backgroundColor.hex()
        }
      }
    });
  }

  return (0, _styles.createMuiTheme)({
    palette: {
      background: {
        paper: backgroundColor.hex()
      }
    }
  });
};

var EditorTheme = function EditorTheme(_ref2) {
  var children = _ref2.children,
      _ref2$color = _ref2.color,
      color = _ref2$color === void 0 ? "white" : _ref2$color;
  var theme;

  if (color === "dark") {
    theme = (0, _styles.createMuiTheme)({
      palette: {
        type: "dark"
      }
    });
  } else if (color) {
    theme = createTheme({
      color: color
    });
  }

  return /*#__PURE__*/_react.default.createElement(_styles.ThemeProvider, {
    theme: theme
  }, children);
};

var _default = EditorTheme;
exports.default = _default;
module.exports = exports.default;
