// @kit-ui/admin/lib/Editor/EditorPreview.stories.js
"use strict";

exports.__esModule = true;
exports.ExpandedSidebar = exports.Default = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _EditorBase = _interopRequireDefault(require("./EditorBase"));

var _EditorHeader = _interopRequireDefault(require("./EditorHeader"));

var _EditorSection = _interopRequireDefault(require("./EditorSection"));

var _EditorSidebar = _interopRequireDefault(require("./EditorSidebar"));

var _EditorPreview = _interopRequireDefault(require("./EditorPreview"));

var _EditorContent = _interopRequireDefault(require("./EditorContent"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _default = {
  title: 'Editor/EditorPreview',
  component: _EditorPreview.default,
  args: {
    header: {
      title: "Preview"
    },
    sidebar: {
      expanded: false
    }
  },
  argTypes: {}
};
exports.default = _default;

var Template = function Template(args) {
  return /*#__PURE__*/_react.default.createElement(_EditorBase.default, null, /*#__PURE__*/_react.default.createElement(_EditorHeader.default, args.header), /*#__PURE__*/_react.default.createElement(_EditorSection.default, null, /*#__PURE__*/_react.default.createElement(_EditorSidebar.default, args.sidebar, "Sidebar"), /*#__PURE__*/_react.default.createElement(_EditorSection.default, null, /*#__PURE__*/_react.default.createElement(_EditorPreview.default, null, "Preview"), /*#__PURE__*/_react.default.createElement(_EditorContent.default, null, "Content"))));
};

var Default = Template.bind({});
exports.Default = Default;
Default.args = {};
var ExpandedSidebar = Template.bind({});
exports.ExpandedSidebar = ExpandedSidebar;
ExpandedSidebar.args = {
  header: {
    title: "Sidebar expanded"
  },
  sidebar: {
    expanded: true
  }
};
