// @kit-ui/core/es/Button/PrimaryButton.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import React from 'react';
import PropTypes from 'prop-types';
import { makeStyles } from '@material-ui/core/styles';
import ButtonBase from '@material-ui/core/ButtonBase';
var useStyles = makeStyles(function (theme) {
  return {
    button: {
      fontFamily: "Akkurat, sans-serif",
      fontSize: "14px",
      fontWeight: "bold",
      textTransform: "none",
      letterSpacing: 0,
      border: "1px solid",
      borderRadius: 1,
      minWidth: theme.spacing(12),
      minHeight: theme.spacing(7)
    }
  };
});

var PrimaryButton = function PrimaryButton(_ref) {
  var children = _ref.children,
      props = _objectWithoutPropertiesLoose(_ref, ["children"]);

  var classes = useStyles();
  return /*#__PURE__*/React.createElement(ButtonBase, _extends({}, props, {
    className: classes.button
  }), children);
};

export default PrimaryButton;
