// @kit-ui/admin/lib/AppHeader/AppHeader.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _AppBar = _interopRequireDefault(require("@material-ui/core/AppBar"));

var _AppSearch = _interopRequireDefault(require("../AppSearch/AppSearch"));

var _AppHeaderIdentifier = _interopRequireDefault(require("./AppHeaderIdentifier"));

var _AppHeaderSubview = _interopRequireDefault(require("./AppHeaderSubview"));

var _styles = require("@material-ui/core/styles");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var useStyles = (0, _styles.makeStyles)(function (theme) {
  var _search;

  return {
    header: {
      position: "relative",
      width: "100%",
      height: theme.spacing(8),
      overflow: "hidden"
    },
    identifer: {
      position: "relative",
      width: "100%",
      height: theme.spacing(8),
      marginTop: theme.spacing(-8),
      transition: ".125s ease-out",
      "&[aria-expanded=true]": {
        marginTop: 0
      }
    },
    subview: {
      backgroundColor: theme.palette.background.default,
      color: theme.palette.text.primary,
      position: "relative",
      width: "100%",
      height: theme.spacing(8),
      marginTop: theme.spacing(-8),
      transition: ".125s ease-out",
      "&[aria-expanded=true]": {
        marginTop: 0
      }
    },
    search: (_search = {
      position: "relative",
      height: theme.spacing(8),
      transition: ".125s ease-out",
      display: "flex"
    }, _search["position"] = "relative", _search.backgroundColor = "white", _search.color = theme.palette.text.primary, _search.width = "100%", _search["height"] = theme.spacing(8), _search.overflow = "hidden", _search.fontFamily = "Akkurat, sans-serif", _search["&[aria-expanded=true]"] = {
      marginTop: theme.spacing(-8)
    }, _search)
  };
});

var AppHeader = function AppHeader(_ref) {
  var className = _ref.className,
      _ref$expanded = _ref.expanded,
      expanded = _ref$expanded === void 0 ? true : _ref$expanded,
      _ref$elevation = _ref.elevation,
      elevation = _ref$elevation === void 0 ? 2 : _ref$elevation,
      _ref$color = _ref.color,
      color = _ref$color === void 0 ? "primary" : _ref$color,
      title = _ref.title,
      subtitle = _ref.subtitle,
      menu = _ref.menu,
      search = _ref.search,
      subview = _ref.subview,
      children = _ref.children;
  var classes = useStyles({
    color: color,
    search: search
  }); //    const classesSearch = search && search.variant === "grow" && classes.growingSearch || classes.search
  //    const justifyContent = search && search.variant === "grow" && "flex-start" || "center"

  return /*#__PURE__*/_react.default.createElement(_AppBar.default, {
    position: "absolute",
    className: className || classes.header,
    component: "header",
    "aria-expanded": expanded,
    elevation: expanded && elevation
  }, subview && /*#__PURE__*/_react.default.createElement(_AppHeaderSubview.default, _extends({}, subview, {
    className: classes.subview,
    expanded: subview && subview.expanded
  })), /*#__PURE__*/_react.default.createElement(_AppHeaderIdentifier.default, {
    className: classes.identifer,
    menu: menu,
    search: search,
    title: title,
    subtitle: subtitle,
    expanded: search && search.expanded && false || true
  }), search && /*#__PURE__*/_react.default.createElement(_AppSearch.default, _extends({}, search, {
    className: classes.search
  })), children);
};

AppHeader.defaultProps = {
  expanded: true,
  search: undefined,
  subview: undefined
};
AppHeader.propTypes = process.env.NODE_ENV !== "production" ? {
  expanded: _propTypes.default.bool,
  elevation: _propTypes.default.number,
  color: _propTypes.default.string,
  title: _propTypes.default.string,
  subtitle: _propTypes.default.string,
  search: _propTypes.default.shape({
    expanded: _propTypes.default.bool,
    placeholder: _propTypes.default.string
  }),
  subview: _propTypes.default.shape({
    expanded: _propTypes.default.bool,
    title: _propTypes.default.string,
    subtitle: _propTypes.default.string
  })
} : {};
var _default = AppHeader;
exports.default = _default;
module.exports = exports.default;
