// @kit-ui/admin/lib/AppHeader/AppHeaderIdentifier.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Typography = _interopRequireDefault(require("@material-ui/core/Typography"));

var _IconButton = _interopRequireDefault(require("@material-ui/core/IconButton"));

var _Search = _interopRequireDefault(require("@material-ui/icons/Search"));

var _Menu = _interopRequireDefault(require("@material-ui/icons/Menu"));

var _styles = require("@material-ui/core/styles");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var useStyles = (0, _styles.makeStyles)(function (theme) {
  var _divider, _subtitle;

  return {
    root: {
      backgroundColor: theme.palette.primary.main,
      color: theme.palette.primary.contrastText
    },
    section: {
      height: theme.spacing(8),
      marginLeft: theme.spacing(1),
      marginRight: theme.spacing(1),
      display: "flex",
      alignItems: "center",
      justifyContent: "space-between"
    },
    primaryToggle: {
      width: theme.spacing(6)
    },
    secondaryToggle: {
      width: theme.spacing(6)
    },
    button: {
      color: "inherit"
    },
    content: {
      display: "flex",
      alignItems: "center",
      flexGrow: 1,
      justifyContent: function justifyContent(props) {
        return props.justifyContent;
      },
      height: theme.spacing(8),
      margin: theme.spacing(1)
    },
    title: {
      fontFamily: "Akkurat, sans-serif",
      fontSize: "18px",
      fontWeight: "bold",
      color: function color(props) {
        return theme.palette.header && theme.palette.header.title || theme.palette.primary.contrastText;
      }
    },
    divider: (_divider = {
      backgroundColor: theme.palette.primary.contrastText,
      width: "1px",
      height: theme.spacing(3),
      opacity: 0.85,
      margin: theme.spacing(1.5)
    }, _divider[theme.breakpoints.down('xs')] = {
      display: "none"
    }, _divider),
    subtitle: (_subtitle = {
      fontFamily: "Akkurat, sans-serif",
      fontSize: "18px",
      fontWeight: "normal",
      opacity: 0.85
    }, _subtitle[theme.breakpoints.down('xs')] = {
      display: "none"
    }, _subtitle)
  };
});

var AppMenuToggle = function AppMenuToggle(_ref) {
  var menu = _ref.menu,
      onToggle = _ref.onToggle;
  var classes = useStyles();

  if (!menu) {
    return false;
  }

  return /*#__PURE__*/_react.default.createElement(_IconButton.default, {
    className: classes.button,
    onClick: onToggle
  }, /*#__PURE__*/_react.default.createElement(_Menu.default, null));
};

var AppSearchToggle = function AppSearchToggle(_ref2) {
  var search = _ref2.search,
      onToggle = _ref2.onToggle;
  var classes = useStyles();

  if (!search || search.variant === "growing") {
    return false;
  }

  return /*#__PURE__*/_react.default.createElement(_IconButton.default, {
    className: classes.button,
    onClick: onToggle
  }, /*#__PURE__*/_react.default.createElement(_Search.default, null));
};

var AppIdentifier = function AppIdentifier(_ref3) {
  var className = _ref3.className,
      expanded = _ref3.expanded,
      menu = _ref3.menu,
      search = _ref3.search,
      title = _ref3.title,
      subtitle = _ref3.subtitle,
      _ref3$justifyContent = _ref3.justifyContent,
      justifyContent = _ref3$justifyContent === void 0 ? "center" : _ref3$justifyContent;
  var classes = useStyles({
    search: search,
    justifyContent: justifyContent
  });

  var toggleSearch = function toggleSearch() {
    search && search.onToggle && search.onToggle();
  };

  var toggleMenu = function toggleMenu() {
    menu && menu.onToggle && menu.onToggle();
  };

  return /*#__PURE__*/_react.default.createElement("div", {
    className: className || classes.root,
    "aria-expanded": expanded
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: classes.section
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: classes.primaryToggle
  }, menu && /*#__PURE__*/_react.default.createElement(AppMenuToggle, {
    menu: menu,
    onToggle: toggleMenu
  }), !menu && /*#__PURE__*/_react.default.createElement(AppSearchToggle, {
    search: search,
    onToggle: toggleSearch
  })), /*#__PURE__*/_react.default.createElement("div", {
    className: classes.content
  }, /*#__PURE__*/_react.default.createElement(_Typography.default, {
    component: "h1",
    className: classes.title,
    nowrap: true
  }, title), title && subtitle && /*#__PURE__*/_react.default.createElement("div", {
    className: classes.divider
  }), /*#__PURE__*/_react.default.createElement(_Typography.default, {
    component: "h2",
    className: classes.subtitle,
    nowrap: true
  }, subtitle)), /*#__PURE__*/_react.default.createElement("div", {
    className: classes.secondaryToggle
  }, menu && /*#__PURE__*/_react.default.createElement(AppSearchToggle, {
    search: search,
    onToggle: toggleSearch
  }))));
};

AppIdentifier.propTypes = process.env.NODE_ENV !== "production" ? {
  title: _propTypes.default.string,
  subtitle: _propTypes.default.string
} : {};
var _default = AppIdentifier;
exports.default = _default;
module.exports = exports.default;
