// @kit-ui/admin/es/App/utils/createAppTheme.js
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import Color from 'color';
import { createMuiTheme } from '@material-ui/core/styles';
import overrides from "./theme.overrides";

var getColor = function getColor(color) {
  if (color === void 0) {
    color = "#666";
  }

  return Color(color).hex();
};

var getPalette = function getPalette(_ref) {
  var primaryColor = _ref.primaryColor,
      secondaryColor = _ref.secondaryColor;
  return {
    primary: {
      main: getColor(primaryColor)
    },
    secondary: {
      main: getColor(secondaryColor || primaryColor)
    }
  };
};

export var createAppTheme = function createAppTheme(_ref2) {
  var palette = _ref2.palette,
      theme = _objectWithoutPropertiesLoose(_ref2, ["palette"]);

  return createMuiTheme({
    typography: {
      fontFamily: "Akkurat, sans-serif !important"
    },
    palette: palette || getPalette(theme),
    overrides: overrides
  });
};
export default createAppTheme;
