// @kit-ui/admin/lib/NavAction/NavActionDropdown.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _ClickAwayListener = _interopRequireDefault(require("@material-ui/core/ClickAwayListener"));

var _Popper = _interopRequireDefault(require("@material-ui/core/Popper"));

var _Paper = _interopRequireDefault(require("@material-ui/core/Paper"));

var _styles = require("@material-ui/core/styles");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var useStyles = (0, _styles.makeStyles)(function (theme) {
  return {
    dropdown: {
      minWidth: function minWidth(props) {
        return props.minWidth;
      }
    }
  };
});

var NavActionDropdown = function NavActionDropdown(_ref) {
  var className = _ref.className,
      _ref$minWidth = _ref.minWidth,
      minWidth = _ref$minWidth === void 0 ? 192 : _ref$minWidth,
      open = _ref.open,
      anchorEl = _ref.anchorEl,
      placement = _ref.placement,
      onClickAway = _ref.onClickAway,
      children = _ref.children;
  var classes = useStyles({
    minWidth: minWidth
  });
  return /*#__PURE__*/_react.default.createElement(_Popper.default, {
    open: open,
    style: {
      zIndex: 2000
    },
    role: undefined,
    transition: true,
    anchorEl: anchorEl,
    placement: placement
  }, /*#__PURE__*/_react.default.createElement(_ClickAwayListener.default, {
    onClickAway: onClickAway
  }, /*#__PURE__*/_react.default.createElement(_Paper.default, {
    className: className || classes.dropdown,
    elevation: 16,
    square: true
  }, children)));
};

var _default = NavActionDropdown;
exports.default = _default;
module.exports = exports.default;
