// @kit-ui/admin/lib/NavAction/NavActionMenuLink.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _ButtonBase = _interopRequireDefault(require("@material-ui/core/ButtonBase"));

var _MenuItem = _interopRequireDefault(require("@material-ui/core/MenuItem"));

var _ListItemIcon = _interopRequireDefault(require("@material-ui/core/ListItemIcon"));

var _ListItemText = _interopRequireDefault(require("@material-ui/core/ListItemText"));

var _Icon = _interopRequireDefault(require("@material-ui/core/Icon"));

var _styles = require("@material-ui/core/styles");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var useStyles = (0, _styles.makeStyles)(function (theme) {
  return {
    actionLink: {
      width: "100%",
      paddingLeft: theme.spacing(2),
      paddingRight: theme.spacing(2),
      "& .MuiTypography-root": {
        fontFamily: "Akkurat, sans-serif"
      },
      "&[role=group] + *": {
        borderTop: "1px solid",
        borderTopColor: theme.palette.divider,
        marginTop: theme.spacing(1.5),
        paddingTop: theme.spacing(1.5)
      },
      "& + &[role=group]": {
        borderTop: "1px solid",
        borderTopColor: theme.palette.divider,
        marginTop: theme.spacing(1.5),
        paddingTop: theme.spacing(1.5),
        "& + *": {
          borderTop: "1px solid",
          borderTopColor: theme.palette.divider,
          marginTop: theme.spacing(1.5),
          paddingTop: theme.spacing(1.5)
        }
      }
    },
    icon: {
      color: "rgba(0,0,0,.54)"
    },
    text: {
      fontFamily: "Akkurat, sans-serif !important",
      fontSize: "inherit",
      "& strong": {
        fontWeight: "bold"
      }
    }
  };
});

var ActionMenuLink = function ActionMenuLink(_ref) {
  var className = _ref.className,
      icon = _ref.icon,
      label = _ref.label,
      title = _ref.title,
      description = _ref.description,
      onClick = _ref.onClick;
  var classes = useStyles();

  if (!label && title) {
    label = title;
  }

  if (!label) {
    return null;
  }

  return /*#__PURE__*/_react.default.createElement(_MenuItem.default, {
    className: className || classes.actionLink,
    onClick: onClick
  }, icon && /*#__PURE__*/_react.default.createElement(_ListItemIcon.default, null, /*#__PURE__*/_react.default.createElement(_Icon.default, {
    className: classes.icon
  }, icon)), /*#__PURE__*/_react.default.createElement(_ListItemText.default, {
    className: classes.text,
    primary: label,
    secondary: description
  }));
};

var _default = ActionMenuLink;
exports.default = _default;
module.exports = exports.default;
