// @kit-ui/admin/es/AppHeader/AppHeaderSubview.js
import React from 'react';
import PropTypes from 'prop-types';
import Typography from '@material-ui/core/Typography';
import IconButton from '@material-ui/core/IconButton';
import BackIcon from '@material-ui/icons/ArrowBack';
import { makeStyles } from '@material-ui/core/styles';
var useStyles = makeStyles(function (theme) {
  return {
    subview: {
      backgroundColor: theme.palette.background.default,
      color: "black"
    },
    section: {
      height: theme.spacing(8),
      display: "flex",
      alignItems: "center",
      justifyContent: "flex-start",
      paddingLeft: theme.spacing(1),
      paddingRight: theme.spacing(1)
    },
    button: {
      width: theme.spacing(6),
      height: theme.spacing(6),
      color: "inherit",
      "& + $content": {
        paddingRight: theme.spacing(8)
      }
    },
    content: {
      display: "flex",
      alignItems: "center",
      justifyContent: "center"
    },
    title: {
      fontFamily: "Akkurat, sans-serif",
      fontSize: "18px",
      fontWeight: "bold",
      marginLeft: theme.spacing(1)
    },
    description: {
      fontFamily: "Akkurat, sans-serif",
      fontSize: "18px",
      fontWeight: "normal",
      marginLeft: "0.25em"
    }
  };
});

var AppSubview = function AppSubview(_ref) {
  var className = _ref.className,
      expanded = _ref.expanded,
      title = _ref.title,
      description = _ref.description,
      onToggle = _ref.onToggle;
  var classes = useStyles();
  return /*#__PURE__*/React.createElement("div", {
    className: className || classes.subview,
    "aria-expanded": expanded
  }, /*#__PURE__*/React.createElement("div", {
    className: classes.section
  }, /*#__PURE__*/React.createElement(IconButton, {
    className: classes.button,
    onClick: onToggle
  }, /*#__PURE__*/React.createElement(BackIcon, null)), /*#__PURE__*/React.createElement("div", {
    className: classes.content
  }, /*#__PURE__*/React.createElement(Typography, {
    component: "h1",
    className: classes.title
  }, title && description && title + ":" || title), /*#__PURE__*/React.createElement(Typography, {
    component: "h2",
    className: classes.description
  }, description))));
};

AppSubview.propTypes = process.env.NODE_ENV !== "production" ? {
  className: PropTypes.string,
  expanded: PropTypes.bool,
  title: PropTypes.string,
  description: PropTypes.string,
  onBack: PropTypes.func
} : {};
export default AppSubview;
