// @kit-ui/admin/es/AppHeader/AppHeaderIdentifier.js
import React from 'react';
import PropTypes from 'prop-types';
import Typography from '@material-ui/core/Typography';
import IconButton from '@material-ui/core/IconButton';
import SearchIcon from '@material-ui/icons/Search';
import MenuIcon from '@material-ui/icons/Menu';
import { makeStyles } from '@material-ui/core/styles';
var useStyles = makeStyles(function (theme) {
  var _divider, _subtitle;

  return {
    root: {
      backgroundColor: theme.palette.primary.main,
      color: theme.palette.primary.contrastText
    },
    section: {
      height: theme.spacing(8),
      marginLeft: theme.spacing(1),
      marginRight: theme.spacing(1),
      display: "flex",
      alignItems: "center",
      justifyContent: "space-between"
    },
    primaryToggle: {
      width: theme.spacing(6)
    },
    secondaryToggle: {
      width: theme.spacing(6)
    },
    button: {
      color: "inherit"
    },
    content: {
      display: "flex",
      alignItems: "center",
      flexGrow: 1,
      justifyContent: function justifyContent(props) {
        return props.justifyContent;
      },
      height: theme.spacing(8),
      margin: theme.spacing(1)
    },
    title: {
      fontFamily: "Akkurat, sans-serif",
      fontSize: "18px",
      fontWeight: "bold",
      color: function color(props) {
        return theme.palette.header && theme.palette.header.title || theme.palette.primary.contrastText;
      }
    },
    divider: (_divider = {
      backgroundColor: theme.palette.primary.contrastText,
      width: "1px",
      height: theme.spacing(3),
      opacity: 0.85,
      margin: theme.spacing(1.5)
    }, _divider[theme.breakpoints.down('xs')] = {
      display: "none"
    }, _divider),
    subtitle: (_subtitle = {
      fontFamily: "Akkurat, sans-serif",
      fontSize: "18px",
      fontWeight: "normal",
      opacity: 0.85
    }, _subtitle[theme.breakpoints.down('xs')] = {
      display: "none"
    }, _subtitle)
  };
});

var AppMenuToggle = function AppMenuToggle(_ref) {
  var menu = _ref.menu,
      onToggle = _ref.onToggle;
  var classes = useStyles();

  if (!menu) {
    return false;
  }

  return /*#__PURE__*/React.createElement(IconButton, {
    className: classes.button,
    onClick: onToggle
  }, /*#__PURE__*/React.createElement(MenuIcon, null));
};

var AppSearchToggle = function AppSearchToggle(_ref2) {
  var search = _ref2.search,
      onToggle = _ref2.onToggle;
  var classes = useStyles();

  if (!search || search.variant === "growing") {
    return false;
  }

  return /*#__PURE__*/React.createElement(IconButton, {
    className: classes.button,
    onClick: onToggle
  }, /*#__PURE__*/React.createElement(SearchIcon, null));
};

var AppIdentifier = function AppIdentifier(_ref3) {
  var className = _ref3.className,
      expanded = _ref3.expanded,
      menu = _ref3.menu,
      search = _ref3.search,
      title = _ref3.title,
      subtitle = _ref3.subtitle,
      _ref3$justifyContent = _ref3.justifyContent,
      justifyContent = _ref3$justifyContent === void 0 ? "center" : _ref3$justifyContent;
  var classes = useStyles({
    search: search,
    justifyContent: justifyContent
  });

  var toggleSearch = function toggleSearch() {
    search && search.onToggle && search.onToggle();
  };

  var toggleMenu = function toggleMenu() {
    menu && menu.onToggle && menu.onToggle();
  };

  return /*#__PURE__*/React.createElement("div", {
    className: className || classes.root,
    "aria-expanded": expanded
  }, /*#__PURE__*/React.createElement("div", {
    className: classes.section
  }, /*#__PURE__*/React.createElement("div", {
    className: classes.primaryToggle
  }, menu && /*#__PURE__*/React.createElement(AppMenuToggle, {
    menu: menu,
    onToggle: toggleMenu
  }), !menu && /*#__PURE__*/React.createElement(AppSearchToggle, {
    search: search,
    onToggle: toggleSearch
  })), /*#__PURE__*/React.createElement("div", {
    className: classes.content
  }, /*#__PURE__*/React.createElement(Typography, {
    component: "h1",
    className: classes.title,
    nowrap: true
  }, title), title && subtitle && /*#__PURE__*/React.createElement("div", {
    className: classes.divider
  }), /*#__PURE__*/React.createElement(Typography, {
    component: "h2",
    className: classes.subtitle,
    nowrap: true
  }, subtitle)), /*#__PURE__*/React.createElement("div", {
    className: classes.secondaryToggle
  }, menu && /*#__PURE__*/React.createElement(AppSearchToggle, {
    search: search,
    onToggle: toggleSearch
  }))));
};

AppIdentifier.propTypes = process.env.NODE_ENV !== "production" ? {
  title: PropTypes.string,
  subtitle: PropTypes.string
} : {};
export default AppIdentifier;
