// @kit-ui/admin/es/AppHeader/AppHeader.stories.js
import React from 'react';
import AppHeader from './AppHeader';
import AppHeaderExample from './AppHeader.example';
export default {
  title: 'App/AppHeader',
  component: AppHeader,
  args: {
    title: "App",
    subtitle: "Owner"
  },
  argTypes: {}
};

var Template = function Template(args) {
  return /*#__PURE__*/React.createElement(AppHeaderExample, args);
};

export var Default = Template.bind({});
Default.args = {};
export var WithMenu = Template.bind({});
WithMenu.args = {
  menu: {
    expanded: false
  }
};
export var MenuAndSearch = Template.bind({});
MenuAndSearch.args = {
  menu: {
    expanded: false
  },
  search: {
    expanded: false
  }
};
export var SubviewExpanded = Template.bind({});
SubviewExpanded.args = {
  subview: {
    expanded: true,
    title: "Subview",
    description: "5 actions"
  }
};
export var Search = Template.bind({});
Search.args = {
  search: {
    expanded: false,
    placeholder: "Search app"
  }
};
export var SearchExpanded = Template.bind({});
SearchExpanded.args = {
  search: {
    expanded: true,
    placeholder: "Search app"
  }
};
export var GrowingSearch = Template.bind({});
GrowingSearch.args = {
  search: {
    variant: "growing",
    expanded: false,
    placeholder: "Search app"
  }
};
