// @kit-ui/schema/lib/Schema/SchemaBase.stories.js
"use strict";

exports.__esModule = true;
exports.KpArticle = exports.Section = exports.Article = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _SchemaBase = _interopRequireDefault(require("./SchemaBase"));

var _article = _interopRequireDefault(require("../data/article"));

var _section = _interopRequireDefault(require("../data/article/section"));

var _article2 = _interopRequireDefault(require("../data/kp/article"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var _default = {
  title: 'Schema/SchemaBase',
  component: _SchemaBase.default,
  args: {},
  argTypes: {}
};
exports.default = _default;

var Template = function Template(args) {
  return /*#__PURE__*/_react.default.createElement(_SchemaBase.default, args);
};

var Article = Template.bind({});
exports.Article = Article;
Article.args = _extends({}, _article.default);
var Section = Template.bind({});
exports.Section = Section;
Section.args = _extends({}, _section.default);
var KpArticle = Template.bind({});
exports.KpArticle = KpArticle;
KpArticle.args = _extends({}, _article2.default);
