// @kit-ui/admin/lib/Widget/SearchWidget.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireWildcard(require("react"));

var _ = require("./");

var _styles = require("@material-ui/core/styles");

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

var useStyles = (0, _styles.makeStyles)(function (theme) {
  return {
    root: {
      position: "absolute",
      width: "100%",
      top: 0,
      right: 0,
      bottom: 0,
      left: 0
    },
    search: {
      display: "flex",
      position: "relative",
      backgroundColor: "transparent",
      color: "black",
      height: "36px",
      overflow: "hidden",
      fontFamily: "Akkurat, sans-serif",
      margin: "12px"
    },
    scroller: {
      position: "absolute",
      zIndex: 1,
      top: 0,
      right: 0,
      bottom: 0,
      left: 0
    },
    results: {
      paddingTop: theme.spacing(7),
      display: "block",
      width: "100%"
    },
    nohits: {
      fontFamily: "Akkurat, sans-serif",
      fontSize: "14px",
      padding: theme.spacing(1.5)
    }
  };
});

var SearchWidget = function SearchWidget(_ref) {
  var className = _ref.className,
      placeholder = _ref.placeholder,
      q = _ref.q,
      onSearch = _ref.onSearch,
      children = _ref.children;
  var classes = useStyles();
  return /*#__PURE__*/_react.default.createElement("div", {
    className: className || classes.root
  }, /*#__PURE__*/_react.default.createElement(_.WidgetSearch, {
    placeholder: placeholder,
    value: q,
    onChange: onSearch
  }), /*#__PURE__*/_react.default.createElement(_.WidgetScroller, {
    className: classes.scroller
  }, children));
};

SearchWidget.defaultProps = {
  placeholder: "Search"
};
var _default = SearchWidget;
exports.default = _default;
module.exports = exports.default;
