// @kit-ui/admin/lib/Widget/WidgetMedia.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireWildcard(require("react"));

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _inheritsLoose(subClass, superClass) { subClass.prototype = Object.create(superClass.prototype); subClass.prototype.constructor = subClass; subClass.__proto__ = superClass; }

var WidgetMedia = /*#__PURE__*/function (_Component) {
  _inheritsLoose(WidgetMedia, _Component);

  function WidgetMedia() {
    return _Component.apply(this, arguments) || this;
  }

  var _proto = WidgetMedia.prototype;

  _proto.render = function render() {
    var _this$props = this.props,
        mediaType = _this$props.mediaType,
        backgroundImage = _this$props.backgroundImage,
        imageUrl = _this$props.imageUrl;

    if (backgroundImage) {
      return /*#__PURE__*/_react.default.createElement("figure", {
        className: "admin-widget__media",
        style: {
          backgroundImage: "url(" + backgroundImage + ")"
        }
      });
    }

    if (imageUrl) {
      return /*#__PURE__*/_react.default.createElement("figure", {
        className: "admin-widget__media"
      }, /*#__PURE__*/_react.default.createElement("img", {
        src: imageUrl
      }));
    }

    return false;
  };

  return WidgetMedia;
}(_react.Component);

var _default = WidgetMedia;
exports.default = _default;
module.exports = exports.default;
