// @kit-ui/admin/lib/Widget/WidgetResults.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _styles = require("@material-ui/core/styles");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var useStyles = (0, _styles.makeStyles)(function (theme) {
  return {
    root: {
      display: "block",
      overflowY: "scroll",
      maxHeight: function maxHeight(props) {
        return props.maxHeight;
      }
    }
  };
});

var WidgetResults = function WidgetResults(_ref) {
  var className = _ref.className,
      children = _ref.children,
      props = _objectWithoutPropertiesLoose(_ref, ["className", "children"]);

  var classes = useStyles(props);
  return /*#__PURE__*/_react.default.createElement("section", {
    className: className || classes.root
  }, children);
};

var _default = WidgetResults;
exports.default = _default;
module.exports = exports.default;
