// @kit-ui/admin/es/NavTabs/README.md.js
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _inheritsLoose(subClass, superClass) { subClass.prototype = Object.create(superClass.prototype); subClass.prototype.constructor = subClass; subClass.__proto__ = superClass; }

import React from 'react';
import { Playground, LayoutWindow as Window, LayoutCode as Code } from '@frontend-components/styleguide/lib/components/';
import { Props, Methods } from './../../.styleguide/utils.js';
import { MDXTag } from '@mdx-js/tag';
import { NavTabs } from './';
var layoutProps = {};

var MDXContent = /*#__PURE__*/function (_React$Component) {
  _inheritsLoose(MDXContent, _React$Component);

  function MDXContent(props) {
    var _this;

    _this = _React$Component.call(this, props) || this;
    _this.layout = null;
    return _this;
  }

  var _proto = MDXContent.prototype;

  _proto.render = function render() {
    var _this$props = this.props,
        components = _this$props.components,
        props = _objectWithoutPropertiesLoose(_this$props, ["components"]);

    return /*#__PURE__*/React.createElement(MDXTag, {
      name: "wrapper",
      components: components
    }, /*#__PURE__*/React.createElement(MDXTag, {
      name: "p",
      components: components
    }, "Tab navigation."), /*#__PURE__*/React.createElement(Playground, null, /*#__PURE__*/React.createElement(NavTabs, {
      tabs: [{
        title: 'Section 1',
        url: '#',
        selected: true
      }, {
        title: 'Section 2a',
        url: '/section/'
      }, {
        title: 'Section 2b',
        url: '/section/'
      }]
    })));
  };

  return MDXContent;
}(React.Component);

export { MDXContent as default };
