// @kit-ui/admin/lib/PreviewJSON/PreviewJSON.stories.js
"use strict";

exports.__esModule = true;
exports.DarkTheme = exports.Default = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _PreviewJSON = _interopRequireDefault(require("./PreviewJSON"));

var _glossary;

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _default = {
  title: 'Preview/PreviewJSON',
  component: _PreviewJSON.default,
  args: {
    data: {
      "glossary": (_glossary = {
        "title": "example glossary",
        "string": "A string",
        "boolean": true
      }, _glossary["boolean"] = false, _glossary["number"] = 5, _glossary["GlossDiv"] = {
        "title": "S",
        "GlossList": {
          "GlossEntry": {
            "ID": "SGML",
            "SortAs": "SGML",
            "GlossTerm": "Standard Generalized Markup Language",
            "Acronym": "SGML",
            "Abbrev": "ISO 8879:1986",
            "GlossDef": {
              "para": "A meta-markup language, used to create markup languages such as DocBook.",
              "GlossSeeAlso": ["GML", "XML"]
            },
            "GlossSee": "markup"
          }
        }
      }, _glossary)
    }
  },
  argTypes: {}
};
exports.default = _default;

var Template = function Template(args) {
  return /*#__PURE__*/_react.default.createElement(_PreviewJSON.default, args);
};

var Default = Template.bind({});
exports.Default = Default;
Default.args = {};
var DarkTheme = Template.bind({});
exports.DarkTheme = DarkTheme;
DarkTheme.args = {
  theme: "dark"
};
