// @kit-ui/admin/lib/Section/EditableSection.stories.js
"use strict";

exports.__esModule = true;
exports.ExpandedWithButtons = exports.RemovableHideable = exports.Removable = exports.Default = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _EditableSection = _interopRequireDefault(require("./EditableSection"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _default = {
  title: 'Section/EditableSection',
  component: _EditableSection.default,
  args: {
    title: "Title",
    description: "description"
  },
  argTypes: {}
};
exports.default = _default;

var Template = function Template(args) {
  return /*#__PURE__*/_react.default.createElement(_EditableSection.default, args, "Content");
};

var Default = Template.bind({});
exports.Default = Default;
Default.args = {};
var Removable = Template.bind({});
exports.Removable = Removable;
Removable.args = {
  removable: true
};
var RemovableHideable = Template.bind({});
exports.RemovableHideable = RemovableHideable;
RemovableHideable.args = {
  removable: true,
  hideable: true
};
var ExpandedWithButtons = Template.bind({});
exports.ExpandedWithButtons = ExpandedWithButtons;
ExpandedWithButtons.args = {
  expanded: true,
  buttons: [{
    title: "Click"
  }, {
    title: "Dont click",
    disabled: true
  }]
};
