// @kit-ui/admin/lib/Section/SectionLayout.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Collapse = _interopRequireDefault(require("@material-ui/core/Collapse"));

var _ = require("./");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var SectionLayout = function SectionLayout(_ref) {
  var position = _ref.position,
      collapsible = _ref.collapsible,
      expanded = _ref.expanded,
      editable = _ref.editable,
      title = _ref.title,
      description = _ref.description,
      buttons = _ref.buttons,
      children = _ref.children,
      props = _objectWithoutPropertiesLoose(_ref, ["position", "collapsible", "expanded", "editable", "title", "description", "buttons", "children"]);

  return /*#__PURE__*/_react.default.createElement(_.SectionBase, {
    position: position,
    collapsible: collapsible || editable,
    expanded: expanded
  }, /*#__PURE__*/_react.default.createElement(_.SectionHeader, _extends({}, props, {
    title: title,
    description: description,
    collapsible: collapsible,
    editable: editable,
    expanded: expanded
  })), /*#__PURE__*/_react.default.createElement(_Collapse.default, {
    in: expanded,
    timeout: "auto",
    unmountOnExit: true
  }, /*#__PURE__*/_react.default.createElement(_.SectionBody, _extends({
    position: position
  }, props), children), expanded && /*#__PURE__*/_react.default.createElement(_.SectionFooter, null, /*#__PURE__*/_react.default.createElement(_.SectionButtons, {
    buttons: buttons
  }))));
};

SectionLayout.propTypes = process.env.NODE_ENV !== "production" ? {
  expanded: _propTypes.default.bool,
  onToggle: _propTypes.default.func
} : {};
SectionLayout.defaultProps = {
  position: "relative",
  editable: false,
  collapsible: false,
  expanded: true
};
var _default = SectionLayout;
exports.default = _default;
module.exports = exports.default;
