// @kit-ui/admin/lib/Section/SectionHeader.stories.js
"use strict";

exports.__esModule = true;
exports.WithSettings = exports.LongDescription = exports.Default = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _SectionHeader = _interopRequireDefault(require("./SectionHeader"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _default = {
  title: 'Section/SectionHeader',
  component: _SectionHeader.default,
  args: {
    title: "Title",
    description: "description"
  },
  argTypes: {}
};
exports.default = _default;

var Template = function Template(args) {
  return /*#__PURE__*/_react.default.createElement(_SectionHeader.default, args);
};

var Default = Template.bind({});
exports.Default = Default;
Default.args = {};
var LongDescription = Template.bind({});
exports.LongDescription = LongDescription;
LongDescription.args = {
  title: "title",
  description: "Long description should be cut lorem ipsum dolor sit amet consecteteur adipiscing elit",
  settings: [{
    template: "color"
  }]
};
var WithSettings = Template.bind({});
exports.WithSettings = WithSettings;
WithSettings.args = {
  settings: [{
    template: "color"
  }]
};
