// @kit-ui/admin/lib/Section/SectionButtons.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _ButtonGroup = _interopRequireDefault(require("@material-ui/core/ButtonGroup"));

var _Button = _interopRequireDefault(require("@material-ui/core/Button"));

var _Icon = _interopRequireDefault(require("@material-ui/core/Icon"));

var _styles = require("@material-ui/core/styles");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var useStyles = (0, _styles.makeStyles)(function (theme) {
  return {
    buttongroup: {
      width: "100%",
      overflowX: "scroll",
      borderRadius: 0,
      "& > * + *": {
        marginLeft: theme.spacing(1)
      }
    },
    button: {
      fontFamily: "Akkurat, sans-serif",
      fontSize: "14px",
      fontWeight: "bold",
      textTransform: "none",
      letterSpacing: 0,
      borderRadius: 1,
      minWidth: theme.spacing(16),
      minHeight: theme.spacing(7)
    },
    icon: {
      marginRight: theme.spacing(1),
      "& + $label": {
        marginRight: theme.spacing(1)
      }
    },
    label: {}
  };
});

var SectionButtons = function SectionButtons(_ref) {
  var className = _ref.className,
      _ref$buttons = _ref.buttons,
      buttons = _ref$buttons === void 0 ? [] : _ref$buttons;
  var classes = useStyles();

  if (!buttons.length) {
    return null;
  }

  var SectionButton = function SectionButton(_ref2) {
    var icon = _ref2.icon,
        title = _ref2.title,
        _ref2$color = _ref2.color,
        color = _ref2$color === void 0 ? "primary" : _ref2$color,
        _ref2$variant = _ref2.variant,
        variant = _ref2$variant === void 0 ? "outlined" : _ref2$variant,
        _ref2$size = _ref2.size,
        size = _ref2$size === void 0 ? "large" : _ref2$size,
        props = _objectWithoutPropertiesLoose(_ref2, ["icon", "title", "color", "variant", "size"]);

    return /*#__PURE__*/_react.default.createElement(_Button.default, _extends({}, props, {
      type: "button",
      "data-size": size,
      "data-variant": variant,
      disableElevation: true,
      color: color,
      variant: variant,
      size: size,
      className: classes.button
    }), icon && /*#__PURE__*/_react.default.createElement(_Icon.default, {
      className: classes.icon
    }, icon), /*#__PURE__*/_react.default.createElement("b", {
      className: classes.label
    }, title));
  };

  return /*#__PURE__*/_react.default.createElement(_ButtonGroup.default, {
    color: "primary",
    className: className || classes.buttongroup
  }, buttons && buttons.map(function (button, index) {
    return /*#__PURE__*/_react.default.createElement(SectionButton, _extends({}, button, {
      key: index
    }));
  }));
};

SectionButtons.defaultProps = {};
SectionButtons.propTypes = process.env.NODE_ENV !== "production" ? {
  buttons: _propTypes.default.array
} : {};
var _default = SectionButtons;
exports.default = _default;
module.exports = exports.default;
