// @kit-ui/admin/lib/Section/Section.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _ = require("./");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var Section = function Section(_ref) {
  var collapsible = _ref.collapsible,
      editable = _ref.editable,
      props = _objectWithoutPropertiesLoose(_ref, ["collapsible", "editable"]);

  if (editable) {
    return /*#__PURE__*/_react.default.createElement(_.EditableSection, props);
  }

  if (collapsible) {
    return /*#__PURE__*/_react.default.createElement(_.CollapsibleSection, props);
  }

  return /*#__PURE__*/_react.default.createElement(_.DefaultSection, props);
};

Section.propTypes = process.env.NODE_ENV !== "production" ? {
  expanded: _propTypes.default.bool,
  onToggle: _propTypes.default.func
} : {};
Section.defaultProps = {
  position: "relative",
  editable: false,
  collapsible: false,
  expanded: true
};
var _default = Section;
exports.default = _default;
module.exports = exports.default;
