// @kit-ui/admin/lib/NavPath/PathDropdown.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _ClickAwayListener = _interopRequireDefault(require("@material-ui/core/ClickAwayListener"));

var _Popper = _interopRequireDefault(require("@material-ui/core/Popper"));

var _Paper = _interopRequireDefault(require("@material-ui/core/Paper"));

var _styles = require("@material-ui/core/styles");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

//import "./SettingsDropdown.scss"
var useStyles = (0, _styles.makeStyles)(function (theme) {
  return {
    root: {
      marginTop: '6px',
      marginBottom: '6px'
    }
  };
});

var PathDropdown = function PathDropdown(_ref) {
  var children = _ref.children,
      expanded = _ref.expanded,
      onCollapse = _ref.onCollapse,
      anchorEl = _ref.anchorEl,
      _ref$placement = _ref.placement,
      placement = _ref$placement === void 0 ? "bottom-start" : _ref$placement;
  var classes = useStyles();
  return /*#__PURE__*/_react.default.createElement(_Popper.default, {
    open: expanded,
    style: {
      zIndex: 2000
    },
    role: undefined,
    transition: true,
    anchorEl: anchorEl,
    placement: placement
  }, /*#__PURE__*/_react.default.createElement(_ClickAwayListener.default, {
    onClickAway: onCollapse
  }, /*#__PURE__*/_react.default.createElement(_Paper.default, {
    elevation: 16,
    square: true,
    className: classes.root,
    "aria-expanded": expanded
  }, children)));
};

var _default = PathDropdown;
exports.default = _default;
module.exports = exports.default;
