// @kit-ui/schema/lib/CheckboxesWidget/CheckboxesWidget.stories.js
"use strict";

exports.__esModule = true;
exports.Inline = exports.Default = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _SchemaEditor = _interopRequireDefault(require("../SchemaEditor/"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _default = {
  title: 'Widgets/Checkboxes',
  component: _SchemaEditor.default,
  args: {
    schema: {
      "type": "array",
      "items": {
        "type": "string",
        "enum": ["foo", "bar", "fuzz", "qux"]
      },
      "uniqueItems": true
    },
    uiSchema: {
      "ui:widget": "checkboxes"
    }
  },
  argTypes: {}
};
exports.default = _default;

var Template = function Template(args) {
  return /*#__PURE__*/_react.default.createElement(_SchemaEditor.default, args);
};

var Default = Template.bind({});
exports.Default = Default;
Default.args = {};
var Inline = Template.bind({});
exports.Inline = Inline;
Inline.args = {
  uiSchema: {
    "ui:title": "Title",
    "ui:help": "Help text",
    "ui:inline": true
  }
};
