// @kit-ui/admin/lib/Calendar/Calendar.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _moment = _interopRequireDefault(require("moment"));

require("moment/locale/nb");

require("moment/locale/nn");

require("moment/locale/sv");

var _styles = require("@material-ui/core/styles");

var _IconButton = _interopRequireDefault(require("@material-ui/core/IconButton"));

var _utils = require("../utils");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var useStyles = (0, _styles.makeStyles)(function (theme) {
  var _ref;

  return _ref = {
    table: {
      display: "flex",
      flexDirection: "column",
      alignItems: "flex-start",
      "& > *": {
        flexBasis: 0,
        flexGrow: 1
      },
      "& > thead": {
        display: "block",
        width: "100%",
        "& th": {
          textAlign: "left",
          width: "100%"
        }
      },
      "& > tbody": {
        display: "block",
        width: "100%",
        "& td": {
          borderTop: "1px solid",
          borderColor: theme.palette.divider,
          textAlign: "left",
          width: "100%"
        }
      },
      "& tr": {
        display: "flex",
        alignItems: "flex-start",
        "& > *": {
          flexBasis: 0,
          flexGrow: 1
        }
      }
    },
    date: {
      display: "flex",
      flexDirection: "column",
      alignItems: "flex-start",
      justifyContent: "flex-start"
    },
    weekday: {
      fontFamily: "Akkurat, sans-serif",
      fontSize: '12px',
      lineHeight: 1
    }
  }, _ref["date"] = {
    //        width: theme.spacing(4.5),
    //        height: theme.spacing(4.5),
    margin: theme.spacing(.25),
    fontFamily: "Akkurat, sans-serif",
    fontSize: '12px',
    lineHeight: 1,
    "&[data-today=true]": {
      border: "1px solid",
      borderColor: theme.palette.primary.main
    },
    "&[data-selected=true]": {
      backgroundColor: theme.palette.primary.main,
      color: theme.palette.primary.contrastText
    },
    "&[data-range=true]": {
      borderRadius: 0,
      width: theme.spacing(5)
    },
    "&[data-range-start=true]": {
      borderTopRightRadius: 0,
      borderBottomRightRadius: 0,
      width: theme.spacing(4.75),
      marginRight: 0
    },
    "&[data-range-end=true]": {
      borderTopLeftRadius: 0,
      borderBottomLeftRadius: 0,
      width: theme.spacing(4.75),
      marginLeft: 0
    }
  }, _ref;
});

var CalendarWeekday = function CalendarWeekday(_ref2) {
  var date = _ref2.date;
  var classes = useStyles();
  var weekday = (0, _moment.default)(date).format('dddd');
  return /*#__PURE__*/_react.default.createElement("div", {
    className: classes.weekday
  }, weekday);
};

var CalendarEvent = function CalendarEvent(_ref3) {
  var dtStart = _ref3.dtStart,
      title = _ref3.title;
  var classes = useStyles();
  var hour = (0, _moment.default)(dtStart).format('HH:mm');
  return /*#__PURE__*/_react.default.createElement("div", null, /*#__PURE__*/_react.default.createElement("time", {
    className: classes.timeLabel
  }, hour), title);
};

var CalendarEvents = function CalendarEvents(_ref4) {
  var _ref4$events = _ref4.events,
      events = _ref4$events === void 0 ? [] : _ref4$events;
  var classes = useStyles();
  return /*#__PURE__*/_react.default.createElement("div", null, events && events.map(function (event) {
    return /*#__PURE__*/_react.default.createElement(CalendarEvent, event);
  }));
};

var CalendarDate = function CalendarDate(_ref5) {
  var today = _ref5.today,
      startDate = _ref5.startDate,
      endDate = _ref5.endDate,
      date = _ref5.date,
      _ref5$events = _ref5.events,
      events = _ref5$events === void 0 ? [] : _ref5$events,
      onChange = _ref5.onChange;
  var classes = useStyles();
  var isRangeStart = startDate !== endDate && (0, _moment.default)(date).isSame(startDate, 'day');
  var isRangeEnd = startDate !== endDate && (0, _moment.default)(date).isSame(endDate, 'day');
  var isRange = (0, _moment.default)(date).isAfter(startDate, 'day') && (0, _moment.default)(date).isBefore(endDate, 'day');
  var isSelected = (0, _moment.default)(date).isSameOrAfter(startDate, 'day') && (0, _moment.default)(date).isSameOrBefore(endDate, 'day');
  var isToday = (0, _moment.default)(date).isSame(today, 'day');
  return /*#__PURE__*/_react.default.createElement("div", {
    className: classes.date
  }, /*#__PURE__*/_react.default.createElement(_IconButton.default, {
    selected: isSelected,
    onClick: function onClick(event) {
      return onChange(date, event);
    },
    "data-selected": isSelected,
    "data-range": isRange,
    "data-range-start": isRangeStart,
    "data-range-end": isRangeEnd,
    "data-today": isToday,
    className: classes.day
  }, (0, _moment.default)(date).format('D')), /*#__PURE__*/_react.default.createElement(CalendarEvents, {
    events: events
  }));
};

var Calendar = function Calendar(_ref6) {
  var _ref6$format = _ref6.format,
      format = _ref6$format === void 0 ? "YYYY-MM-DD" : _ref6$format,
      today = _ref6.today,
      date = _ref6.date,
      startDate = _ref6.startDate,
      endDate = _ref6.endDate,
      items = _ref6.items,
      onChange = _ref6.onChange;
  today = today || (0, _moment.default)().format(format);
  date = date || today;
  startDate = startDate || date;
  endDate = endDate || startDate;

  var _getCalendar = (0, _utils.getCalendar)({
    date: startDate,
    items: items
  }),
      rows = _getCalendar.rows;

  var classes = useStyles();
  return /*#__PURE__*/_react.default.createElement("table", {
    className: classes.table
  }, /*#__PURE__*/_react.default.createElement("thead", null, /*#__PURE__*/_react.default.createElement("tr", null, rows[0].map(function (col) {
    return /*#__PURE__*/_react.default.createElement("th", {
      className: classes.col
    }, /*#__PURE__*/_react.default.createElement(CalendarWeekday, col));
  }))), /*#__PURE__*/_react.default.createElement("tbody", null, rows.map(function (row, r) {
    return /*#__PURE__*/_react.default.createElement("tr", {
      key: r
    }, row.map(function (col) {
      return /*#__PURE__*/_react.default.createElement("td", null, /*#__PURE__*/_react.default.createElement(CalendarDate, _extends({}, col, {
        today: today,
        startDate: startDate,
        endDate: endDate,
        onChange: onChange
      })));
    }));
  })));
};

var _default = Calendar;
exports.default = _default;
module.exports = exports.default;
