// @kit-ui/admin/lib/Calendar/Schedule.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _moment = _interopRequireDefault(require("moment"));

var _utils = require("../utils/");

var _styles = require("@material-ui/core/styles");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var useStyles = (0, _styles.makeStyles)(function (theme) {
  return {
    schedule: {
      display: "flex",
      flexDirection: "row",
      alignItems: "flex-start",
      paddingLeft: theme.spacing(7),
      "& > *": {
        flexBasis: 0,
        flexGrow: 1
      }
    },
    date: {
      display: "flex",
      flexDirection: "column",
      alignItems: "flex-start",
      "& $hourLabel": {
        display: "none"
      },
      "&:first-child $hourLabel": {
        display: "block"
      }
    },
    dateLabel: {
      position: "sticky",
      zIndex: 2,
      top: 0,
      padding: theme.spacing(1, 0)
    }
  };
});

var ScheduleDate = function ScheduleDate(_ref) {
  var datetime = _ref.datetime,
      _ref$events = _ref.events,
      events = _ref$events === void 0 ? [] : _ref$events;
  var date = (0, _moment.default)(datetime).format('LL');
  var classes = useStyles();
  return /*#__PURE__*/_react.default.createElement("div", {
    className: classes.date
  }, /*#__PURE__*/_react.default.createElement("date", {
    className: classes.dateLabel
  }, date), /*#__PURE__*/_react.default.createElement(ScheduleEvents, {
    events: events
  }));
};

var ScheduleEvent = function ScheduleEvent(_ref2) {
  var dtStart = _ref2.dtStart,
      title = _ref2.title;
  var classes = useStyles();
  var hour = (0, _moment.default)(dtStart).format('HH:mm');
  return /*#__PURE__*/_react.default.createElement("div", null, /*#__PURE__*/_react.default.createElement("time", {
    className: classes.timeLabel
  }, hour), title);
};

var ScheduleEvents = function ScheduleEvents(_ref3) {
  var _ref3$events = _ref3.events,
      events = _ref3$events === void 0 ? [] : _ref3$events;
  var classes = useStyles();
  return /*#__PURE__*/_react.default.createElement("div", null, events && events.map(function (event) {
    return /*#__PURE__*/_react.default.createElement(ScheduleEvent, event);
  }));
};

var Schedule = function Schedule(_ref4) {
  var date = _ref4.date,
      _ref4$items = _ref4.items,
      items = _ref4$items === void 0 ? [] : _ref4$items,
      _ref4$days = _ref4.days,
      days = _ref4$days === void 0 ? 4 : _ref4$days,
      _ref4$timeFrom = _ref4.timeFrom,
      timeFrom = _ref4$timeFrom === void 0 ? 8 : _ref4$timeFrom,
      _ref4$timeTo = _ref4.timeTo,
      timeTo = _ref4$timeTo === void 0 ? 23 : _ref4$timeTo;
  var timetable = (0, _utils.getSchedule)({
    date: date,
    items: items,
    days: days,
    timeFrom: timeFrom,
    timeTo: timeTo
  });
  var rows = timetable.rows,
      events = timetable.events;
  var classes = useStyles();
  return /*#__PURE__*/_react.default.createElement("div", {
    className: classes.schedule
  }, rows.map(function (row) {
    return /*#__PURE__*/_react.default.createElement(ScheduleDate, row);
  }));
};

var _default = Schedule;
exports.default = _default;
module.exports = exports.default;
