// @kit-ui/schema/es/License/SelectLicenseWidget.js
import React from "react";
import SelectLicenseField from "./SelectLicenseField";

var SelectLicenseWidget = function SelectLicenseWidget(props) {
  var id = props.id,
      options = props.options,
      value = props.value,
      required = props.required,
      disabled = props.disabled,
      readonly = props.readonly,
      multiple = props.multiple,
      autofocus = props.autofocus,
      onChange = props.onChange,
      onBlur = props.onBlur,
      onFocus = props.onFocus,
      placeholder = props.placeholder;
  var emptyValue = multiple ? [] : "";
  var label = options.title || props.title || props.label;
  var helperText = options.help || undefined;

  var handleChange = function handleChange(value, event) {
    onChange(value);
  };

  return /*#__PURE__*/React.createElement(SelectLicenseField, {
    id: id,
    placeholder: placeholder,
    multiple: multiple,
    label: label,
    helperText: helperText,
    value: typeof value === "undefined" ? emptyValue : value,
    required: required,
    disabled: disabled || readonly,
    autoFocus: autofocus,
    onChange: handleChange
  });
};

export default SelectLicenseWidget;
