// @kit-ui/admin/es/Media/utils/getAnnotateImageField.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import { getUiOptions } from "../../Schema/utils";
var defaultProps = {
  defaultSchema: {
    "type": "object",
    "properties": {
      "mediaId": {
        "type": "string",
        "readonly": true
      },
      "media": {
        "type": "object",
        "properties": {
          "mediaType": {
            "type": "string",
            "default": "image",
            "readonly": true
          },
          "uniqueId": {
            "type": "string",
            "readonly": true
          }
        }
      },
      "imageAnnotations": {
        "type": "array",
        "items": {
          "type": "object",
          "properties": {
            "top": {
              "type": "number",
              "readonly": true
            },
            "left": {
              "type": "number",
              "readonly": true
            },
            "x": {
              "type": "number",
              "readonly": true
            },
            "y": {
              "type": "number",
              "readonly": true
            },
            "radius": {
              "type": "number",
              "minimum": 1,
              "maximum": 100,
              "default": 1
            },
            "title": {
              "type": "string"
            },
            "description": {
              "type": "string"
            }
          }
        }
      }
    }
  }
};
export var getAnnotateImageSchema = function getAnnotateImageSchema(props) {
  var schema = props.schema,
      defaultSchema = props.defaultSchema;
  return defaultSchema;
  return _extends({}, defaultSchema, {
    properties: _extends({}, defaultSchema.properties, schema.properties)
  });
};
export function getAnnotateImageField(props) {
  props = _extends({}, defaultProps, props);
  var schema = getAnnotateImageSchema(props);
  return _extends({}, props, {
    schema: schema,
    uiSchema: _extends({}, props.uiSchema, {
      "ui:field": "annotateImage"
    })
  });
}
