// @kit-ui/admin/es/Media/utils/getImageCropdataStylesX.js
export var getImageCropdataStyles = function getImageCropdataStyles(_ref) {
  var imageCropdata = _ref.imageCropdata,
      parent = _ref.parent,
      image = _ref.image;
  var parentW = parent.offsetWidth;
  var parentH = parent.offsetHeight;
  var parentRatio = parentW / parentH;
  var imageW = image.naturalWidth;
  var imageH = image.naturalHeight;
  var imageRatio = imageW / imageH;
  var cropW = imageCropdata && imageCropdata.width || imageW;
  var cropH = imageCropdata && imageCropdata.height || imageH;
  var cropX = imageCropdata && imageCropdata.x || 0;
  var cropY = imageCropdata && imageCropdata.y || 0;
  var cropRatio = cropW && cropH && cropW / cropH || imageRatio; // horiz or vert

  var cropboxStyle = {};

  if (cropRatio > parentRatio) {
    cropboxStyle = {
      width: "100%",
      height: parentW / cropW * cropH + "px"
    };
  } else {
    cropboxStyle = {
      width: parentH / cropH * cropW + "px",
      height: "100%"
    };
  } // image


  var imageWidth = imageW / cropW * 100;
  var imageHeight = imageH / cropH * 100;
  var imageX = cropX / imageW * 100 * -1;
  var imageY = cropY / imageH * 100 * -1;
  var imageTransform = 'translateX(' + imageX + '%) translateY(' + imageY + '%)';
  var imageStyle = {
    position: 'absolute',
    top: 0,
    right: 'auto',
    bottom: 'auto',
    left: 0,
    width: imageWidth + "%",
    height: 'auto',
    maxWidth: 'none',
    maxHeight: 'none',
    transformOrigin: 'top left',
    transform: imageTransform
  };
  return {
    cropbox: cropboxStyle,
    image: imageStyle
  };
};
export default getImageCropdataStyles;
