// @kit-ui/admin/es/Editor/EditorOverlay.stories.js
import React from 'react';
import EditorBase from './EditorBase';
import EditorHeader from './EditorHeader';
import EditorBody from "./EditorBody";
import EditorSection from './EditorSection';
import EditorOverlay from './EditorOverlay';
export default {
  title: 'Editor/EditorOverlay',
  component: EditorOverlay,
  args: {
    header: {
      title: "Overlay"
    },
    overlay: {
      expanded: true
    }
  },
  argTypes: {}
};

var Template = function Template(args) {
  return /*#__PURE__*/React.createElement(EditorBase, null, /*#__PURE__*/React.createElement(EditorHeader, args.header), /*#__PURE__*/React.createElement(EditorBody, null, "Content"), /*#__PURE__*/React.createElement(EditorOverlay, args.overlay, "Overlay"));
};

export var Expanded = Template.bind({});
Expanded.args = {
  header: {
    title: "Overlay expanded"
  },
  overlay: {
    expanded: true
  }
};
export var Collapsed = Template.bind({});
Collapsed.args = {
  header: {
    title: "Overlay collapsed"
  },
  overlay: {
    expanded: false
  }
};
