// @kit-ui/admin/es/Editor/EditorTheme.js
import React from "react";
import Color from 'color';
import { ThemeProvider, createMuiTheme, getContrastText } from '@material-ui/core/styles';

var createTheme = function createTheme(_ref) {
  var color = _ref.color;
  var backgroundColor = Color(color);

  if (backgroundColor.isDark()) {
    return createMuiTheme({
      palette: {
        type: "dark",
        background: {
          paper: backgroundColor.hex()
        }
      }
    });
  }

  return createMuiTheme({
    palette: {
      background: {
        paper: backgroundColor.hex()
      }
    }
  });
};

var EditorTheme = function EditorTheme(_ref2) {
  var children = _ref2.children,
      _ref2$color = _ref2.color,
      color = _ref2$color === void 0 ? "white" : _ref2$color;
  var theme;

  if (color === "dark") {
    theme = createMuiTheme({
      palette: {
        type: "dark"
      }
    });
  } else if (color) {
    theme = createTheme({
      color: color
    });
  }

  return /*#__PURE__*/React.createElement(ThemeProvider, {
    theme: theme
  }, children);
};

export default EditorTheme;
