// @kit-ui/admin/es/Editor/EditorAction.js
import React, { useState, useRef, forwardRef } from 'react';
import { makeStyles } from '@material-ui/core/styles';
import ClickAwayListener from '@material-ui/core/ClickAwayListener';
import Popper from '@material-ui/core/Popper';
import Paper from '@material-ui/core/Paper';
import DropDownIcon from '@material-ui/icons/ArrowDropDown';
import MenuList from '@material-ui/core/MenuList';
import MenuItem from '@material-ui/core/MenuItem';
import ButtonGroup from '@material-ui/core/ButtonGroup';
import Button from '@material-ui/core/Button';
import ButtonBase from '@material-ui/core/ButtonBase';
import EditIcon from '@material-ui/icons/Edit';
import ViewIcon from '@material-ui/icons/RemoveRedEye';
import NavAction from "../NavAction/NavAction";
var useStyles = makeStyles(function (theme) {
  var _iconButton;

  return {
    root: {
      fontSize: "14px",
      marginLeft: theme.spacing(1),
      marginRight: theme.spacing(1)
    },
    label: {
      fontFamily: "Akkurat, sans-serif",
      fontSize: "inherit",
      fontWeight: "bold",
      marginLeft: theme.spacing(2),
      marginRight: theme.spacing(2)
    },
    primaryButton: {
      borderRadius: 0,
      minWidth: theme.spacing(7),
      paddingLeft: theme.spacing(2),
      paddingRight: theme.spacing(2),
      height: theme.spacing(5),
      display: "flex",
      alignItems: "center",
      justifyContent: "center",
      fontWeight: "bold"
    },
    primaryToggle: {
      borderRadius: 0,
      width: theme.spacing(5),
      height: theme.spacing(5)
    },
    iconButton: (_iconButton = {
      fontSize: "inherit",
      minWidth: theme.spacing(5),
      minHeight: theme.spacing(5),
      border: "1px solid",
      borderColor: "rgba(0,0,0,.25)"
    }, _iconButton[theme.breakpoints.down('xs')] = {
      border: "none"
    }, _iconButton),
    icon: {
      fontSize: "18px"
    }
  };
});

var PrimaryButton = function PrimaryButton(_ref) {
  var disabled = _ref.disabled,
      label = _ref.label,
      onClick = _ref.onClick;
  var classes = useStyles();
  return /*#__PURE__*/React.createElement(Button, {
    variant: "contained",
    color: "primary",
    className: classes.primaryButton,
    disabled: disabled,
    onClick: onClick
  }, label);
};

var PrimaryComboButton = function PrimaryComboButton(_ref2) {
  var forwardedRef = _ref2.forwardedRef,
      disabled = _ref2.disabled,
      expanded = _ref2.expanded,
      controls = _ref2.controls,
      label = _ref2.label,
      onClick = _ref2.onClick,
      onToggle = _ref2.onToggle;
  var classes = useStyles();
  return /*#__PURE__*/React.createElement(ButtonGroup, {
    "aria-label": "action",
    color: "primary",
    ref: forwardedRef
  }, /*#__PURE__*/React.createElement(Button, {
    variant: "contained",
    className: classes.primaryButton,
    disabled: disabled,
    onClick: onClick
  }, label), /*#__PURE__*/React.createElement(Button, {
    variant: "contained",
    className: classes.primaryToggle,
    size: "small",
    "aria-controls": expanded ? controls : undefined,
    "aria-expanded": expanded ? 'true' : undefined,
    "aria-label": "options",
    "aria-haspopup": "menu",
    onClick: onToggle
  }, /*#__PURE__*/React.createElement(DropDownIcon, null)));
};

var PrimaryIconButton = function PrimaryIconButton(_ref3) {
  var onClick = _ref3.onClick,
      children = _ref3.children;
  var classes = useStyles();
  return /*#__PURE__*/React.createElement(ButtonBase, {
    className: classes.iconButton,
    onClick: onClick
  }, children);
};

var EditorAction = function EditorAction(_ref4) {
  var className = _ref4.className,
      primaryAction = _ref4.primaryAction;

  var _useState = useState(false),
      expanded = _useState[0],
      setExpanded = _useState[1];

  var anchorRef = useRef(null);
  var classes = useStyles();

  if (!primaryAction) {
    return false;
  }

  var handleToggle = function handleToggle(event) {
    setExpanded(function (prevExpanded) {
      return !prevExpanded;
    });
  };

  var handleSelect = function handleSelect(_ref5) {
    var onClick = _ref5.onClick;
    onClick && onClick();
    setExpanded(false);
  }; //    const primaryAction = action && action[0]


  if (primaryAction && primaryAction.type === "view") {
    return /*#__PURE__*/React.createElement("nav", {
      className: className || classes.root
    }, /*#__PURE__*/React.createElement(PrimaryIconButton, primaryAction, /*#__PURE__*/React.createElement(ViewIcon, {
      className: classes.icon
    })));
  } else if (primaryAction && primaryAction.type === "edit") {
    return /*#__PURE__*/React.createElement("nav", {
      className: className || classes.root
    }, /*#__PURE__*/React.createElement(PrimaryIconButton, primaryAction, /*#__PURE__*/React.createElement(EditIcon, {
      className: classes.icon
    })));
  } // NavAction


  if (primaryAction) {
    return /*#__PURE__*/React.createElement(NavAction, {
      className: className || classes.root,
      menu: [primaryAction],
      placement: "bottom-end"
    });
  }
  /*
   if (primaryAction && primaryAction.children) {
       return (
          <nav className={className || classes.root}>
              <PrimaryComboButton {...primaryAction} expanded={expanded} forwardedRef={anchorRef} onToggle={handleToggle} />
              <Popper open={expanded} style={{zIndex: 2000}}
                      role={undefined} transition anchorEl={anchorRef.current} placement="bottom-start">
                      <ClickAwayListener onClickAway={handleToggle}>
                          <div>
                          <Paper elevation={1} square={true} aria-expanded={expanded}>
                              <MenuList>
                                  {primaryAction.children.map((item, index) => (
                                      <MenuItem {...item} key={index} onClick={() => handleSelect(item)}>
                                          { item.label || item.title }
                                      </MenuItem>
                                  ))}
                              </MenuList>
                          </Paper>
                          </div>
                      </ClickAwayListener>
              </Popper>
          </nav>
      )
      
  }
   if (primaryAction) {
      return (
          <nav className={className || classes.root}>
              <PrimaryButton {...primaryAction} />
          </nav>
      )
  }
   */

};

export default EditorAction;
