// @kit-ui/admin/es/Editor/EditorHeader.stories.js
import React from 'react';
import EditorHeader from './EditorHeader';
export default {
  title: 'Components/EditorHeader',
  component: EditorHeader,
  args: {},
  argTypes: {}
};

var Template = function Template(args) {
  return /*#__PURE__*/React.createElement(EditorHeader, args);
};

export var Default = Template.bind({});
Default.args = {
  title: "Document",
  parents: [{
    "title": "Grandparent"
  }, {
    "title": "Parent"
  }],
  primaryAction: {
    title: "Save"
  },
  status: "draft",
  statusDate: "1999-05-26"
};
export var ParentsUntitled = Template.bind({});
ParentsUntitled.args = {
  untitled: "No title",
  parents: [{
    "title": "Grandparent"
  }, {
    "title": "Parent"
  }]
};
export var ParentsTitle = Template.bind({});
ParentsTitle.args = {
  untitled: "No title",
  title: "Document title",
  parents: [{
    "title": "Grandparent"
  }, {
    "title": "Parent"
  }]
};
export var Collapsible = Template.bind({});
Collapsible.args = {
  untitled: "No title",
  title: "Document title",
  collapsible: true,
  parents: [{
    "title": "Grandparent"
  }, {
    "title": "Parent"
  }]
};
export var CollapsibleExpanded = Template.bind({});
CollapsibleExpanded.args = {
  untitled: "No title",
  title: "Document title",
  collapsible: true,
  expanded: true,
  parents: [{
    "title": "Grandparent"
  }, {
    "title": "Parent"
  }]
};
