// @kit-ui/admin/lib/Media/utils/getImageCropdataStyles.js
"use strict";

exports.__esModule = true;
exports.default = exports.getImageCropdataStyles = void 0;

var getImageCropdataStyles = function getImageCropdataStyles(_ref) {
  var imageCropdata = _ref.imageCropdata,
      _ref$objectFit = _ref.objectFit,
      objectFit = _ref$objectFit === void 0 ? "contain" : _ref$objectFit,
      parent = _ref.parent,
      image = _ref.image;
  var parentWidth = parent.offsetWidth;
  var parentHeight = parent.offsetHeight;
  var parentRatio = parentWidth / parentHeight;
  var imageWidth = image.naturalWidth;
  var imageHeight = image.naturalHeight;
  var imageRatio = imageWidth / imageHeight;
  var cropX = imageCropdata && imageCropdata.x * imageWidth / 100 || 0;
  var cropY = imageCropdata && imageCropdata.y * imageHeight / 100 || 0;
  var cropWidth = imageCropdata && imageCropdata.width * imageWidth / 100 || imageWidth;
  var cropHeight = imageCropdata && imageCropdata.height * imageHeight / 100 || imageHeight;
  var cropRatio = cropWidth / cropHeight; // horiz or vert

  var cropboxStyle = {};

  if (cropRatio > parentRatio) {
    cropboxStyle = {
      width: "100%",
      height: parentWidth / cropWidth * cropHeight + "px"
    };
  } else {
    cropboxStyle = {
      width: parentHeight / cropHeight * cropWidth + "px",
      height: "100%"
    };
  } // image


  var cropImageWidth = imageWidth / cropWidth * 100;
  var cropImageHeight = imageHeight / cropHeight * 100;
  var cropImageX = cropX / imageWidth * 100 * -1;
  var cropImageY = cropY / imageHeight * 100 * -1;
  var imageTransform = 'translateX(' + cropImageX + '%) translateY(' + cropImageY + '%)';
  var imageStyle = {
    position: 'absolute',
    top: 0,
    right: 'auto',
    bottom: 'auto',
    left: 0,
    width: cropImageWidth + "%",
    height: 'auto',
    maxWidth: 'none',
    maxHeight: 'none',
    transformOrigin: 'top left',
    transform: imageTransform
  };
  return {
    cropbox: cropboxStyle,
    image: imageStyle
  };
};

exports.getImageCropdataStyles = getImageCropdataStyles;
var _default = getImageCropdataStyles;
exports.default = _default;
