// @kit-ui/schema/lib/ArrayLayout/ArrayHeader.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _Typography = _interopRequireDefault(require("@material-ui/core/Typography"));

var _ButtonBase = _interopRequireDefault(require("@material-ui/core/ButtonBase"));

var _styles = require("@material-ui/core/styles");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var useStyles = (0, _styles.makeStyles)(function (theme) {
  return {
    header: {
      margin: theme.spacing(2, 0),
      display: "flex"
    },
    title: {
      fontFamily: "Akkurat, sans-serif",
      fontWeight: "bold"
    },
    description: {
      fontFamily: "Akkurat, sans-serif"
    }
  };
});

var ArrayButton = function ArrayButton(_ref) {
  var title = _ref.title,
      button = _objectWithoutPropertiesLoose(_ref, ["title"]);

  var classes = useStyles();
  return /*#__PURE__*/_react.default.createElement(_ButtonBase.default, _extends({
    className: classes.button
  }, button), title);
};

var ArrayLayout = function ArrayLayout(_ref2) {
  var _ref2$title = _ref2.title,
      title = _ref2$title === void 0 ? "Array" : _ref2$title,
      description = _ref2.description,
      _ref2$buttons = _ref2.buttons,
      buttons = _ref2$buttons === void 0 ? [] : _ref2$buttons,
      children = _ref2.children;
  var classes = useStyles();
  return /*#__PURE__*/_react.default.createElement("header", {
    className: classes.header
  }, /*#__PURE__*/_react.default.createElement(_Typography.default, {
    className: classes.title
  }, title), /*#__PURE__*/_react.default.createElement(_Typography.default, {
    className: classes.description
  }, description));
};

var _default = ArrayLayout;
exports.default = _default;
module.exports = exports.default;
