// @kit-ui/admin/es/NavUser/UserInfo.js
import React, { Component } from 'react';
import PropTypes from "prop-types";
import Typography from '@material-ui/core/Typography';
import { UserIcon } from "@kit-ui/icons";
import { makeStyles } from '@material-ui/core/styles';
var useStyles = makeStyles(function (theme) {
  return {
    root: {
      position: "relative",
      display: "flex",
      alignItems: "center",
      justifyContent: "flex-start",
      width: "320px",
      padding: "6px"
    },
    figure: {
      position: "relative",
      display: "block",
      width: "72px",
      height: "72px",
      margin: "6px"
    },
    header: {
      margin: "6px"
    },
    icon: {
      width: "72px",
      height: "72px"
    },
    name: {
      fontFamily: "Akkurat, sans-serif",
      fontWeight: "bold",
      fontSize: "16px"
    },
    email: {
      fontFamily: "Akkurat, sans-serif",
      fontWeight: "normal",
      fontSize: "14px"
    }
  };
});

var UserInfo = function UserInfo(_ref) {
  var imageUrl = _ref.imageUrl,
      name = _ref.name,
      email = _ref.email,
      initials = _ref.initials;
  var classes = useStyles();
  return /*#__PURE__*/React.createElement("div", {
    className: classes.root
  }, /*#__PURE__*/React.createElement("figure", {
    className: classes.figure
  }, /*#__PURE__*/React.createElement(UserIcon, {
    className: classes.icon,
    name: name,
    initials: initials,
    imageUrl: imageUrl
  })), /*#__PURE__*/React.createElement("header", {
    className: classes.header
  }, /*#__PURE__*/React.createElement(Typography, {
    className: classes.name,
    nowrap: true
  }, name), /*#__PURE__*/React.createElement(Typography, {
    className: classes.email,
    nowrap: true
  }, email)));
};

UserInfo.propTypes = process.env.NODE_ENV !== "production" ? {
  imageUrl: PropTypes.string,
  name: PropTypes.string,
  email: PropTypes.string
} : {};
export default UserInfo;
