// @kit-ui/admin/es/Layout/Grid.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import { getUiOptions } from '../Schema/utils';
import Grid from '@material-ui/core/Grid';
var useStyles = makeStyles(function (theme) {
  return {
    root: {
      padding: function padding(props) {
        return theme.spacing(props.padding);
      }
    }
  };
});

var LayoutGridItem = function LayoutGridItem(_ref) {
  var uiSchema = _ref.uiSchema,
      children = _ref.children,
      props = _objectWithoutPropertiesLoose(_ref, ["uiSchema", "children"]);

  var uiOptions = uiSchema && getUiOptions(uiSchema);
  var gridCols = props.gridCols || uiOptions && uiOptions.gridCols;

  if (gridCols && typeof gridCols === 'number' && gridCols !== 0) {
    return /*#__PURE__*/React.createElement(Grid, {
      item: true,
      xs: gridCols
    }, children);
  }

  if (gridCols && typeof gridCols === "object") {
    return /*#__PURE__*/React.createElement(Grid, _extends({
      item: true
    }, gridCols), children);
  }

  return /*#__PURE__*/React.createElement(Grid, {
    item: true,
    xs: 12
  }, children);
};

var LayoutGrid = function LayoutGrid(_ref2) {
  var className = _ref2.className,
      _ref2$spacing = _ref2.spacing,
      spacing = _ref2$spacing === void 0 ? 1 : _ref2$spacing,
      _ref2$padding = _ref2.padding,
      padding = _ref2$padding === void 0 ? 0 : _ref2$padding,
      alignItems = _ref2.alignItems,
      alignContent = _ref2.alignContent,
      justify = _ref2.justify,
      children = _ref2.children;
  var classes = useStyles({
    padding: padding
  });

  if (!children.length) {//        children = [children];
  }

  var items = React.Children.toArray(children);
  return /*#__PURE__*/React.createElement(Grid, {
    className: className || classes.root,
    container: true,
    alignItems: alignItems,
    alignContent: alignContent,
    justify: justify,
    spacing: spacing,
    "data-layout": "grid"
  }, items.map(function (child, index) {
    return /*#__PURE__*/React.createElement(LayoutGridItem, _extends({}, child.props, {
      key: index
    }), child);
  }));
};

export default LayoutGrid;
