// @kit-ui/admin/es/Layout/LayoutBody.js
import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import LayoutGrid from "./LayoutGrid";
var useStyles = makeStyles(function (theme) {
  return {
    root: {
      position: "relative",
      padding: function padding(props) {
        return theme.spacing(props.padding);
      },
      "& > * + *": {
        marginTop: function marginTop(props) {
          return theme.spacing(props.spacing);
        }
      }
    }
  };
});

var LayoutBody = function LayoutBody(_ref) {
  var _ref$grid = _ref.grid,
      grid = _ref$grid === void 0 ? false : _ref$grid,
      padding = _ref.padding,
      spacing = _ref.spacing,
      children = _ref.children;
  var classes = useStyles({
    padding: padding,
    spacing: spacing
  });

  if (!children) {
    return false;
  }

  return /*#__PURE__*/React.createElement("div", {
    className: classes.root
  }, /*#__PURE__*/React.createElement(LayoutGrid, {
    grid: grid,
    padding: padding,
    spacing: spacing
  }, children));
};

export default LayoutBody;
