// @kit-ui/admin/es/Layout/LayoutFormat.js
import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
var useStyles = makeStyles(function (theme) {
  return {
    format: {
      position: "relative",
      width: "100%",
      minHeight: function minHeight(props) {
        return props.minHeight;
      },
      paddingBottom: function paddingBottom(props) {
        return props.ratio * 100 + "%";
      }
    }
  };
});

var LayoutFormat = function LayoutFormat(_ref) {
  var format = _ref.format,
      minHeight = _ref.minHeight,
      children = _ref.children;
  var ratio;

  if (format) {
    var size = format.split(":");
    var w = size[0];
    var h = size[1];
    ratio = h / w;
  }

  var classes = useStyles({
    ratio: ratio,
    minHeight: minHeight
  });

  if (ratio || minHeight) {
    return /*#__PURE__*/React.createElement("div", {
      className: classes.format,
      "data-minHeight": minHeight,
      "data-ratio": ratio,
      "data-format": format
    }, children);
  }

  return children;
};

export default LayoutFormat;
