// @kit-ui/admin/es/Layout/LayoutObject.js
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import LayoutGrid from "./LayoutGrid";
var useStyles = makeStyles(function (theme) {
  return {
    root: {
      position: "relative",
      width: "100%",
      padding: function padding(props) {
        return theme.spacing(props.padding);
      }
    }
  };
});

var LayoutObject = function LayoutObject(_ref) {
  var grid = _ref.grid,
      children = _ref.children,
      props = _objectWithoutPropertiesLoose(_ref, ["grid", "children"]);

  if (!grid) {
    return children;
  }

  return /*#__PURE__*/React.createElement(LayoutGrid, props, children);
};

LayoutObject.defaultProps = {
  header: true,
  footer: true,
  grid: undefined,
  padding: 0,
  spacing: 0
};
export default LayoutObject;
