// @kit-ui/admin/es/Layout/LayoutHeader.js
import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Typography from '@material-ui/core/Typography';
var useStyles = makeStyles(function (theme) {
  return {
    root: {
      display: "flex",
      alignItems: "center",
      marginTop: theme.spacing(2),
      marginBottom: theme.spacing(1)
    },
    title: {
      fontFamily: "Akkurat, sans-serif",
      fontSize: "14px",
      lineHeight: "24px",
      fontWeight: "bold"
    },
    description: {
      marginLeft: theme.spacing(1),
      fontFamily: "Akkurat, sans-serif",
      fontSize: "14px",
      lineHeight: "24px"
    }
  };
});

var LayoutHeader = function LayoutHeader(_ref) {
  var className = _ref.className,
      _ref$padding = _ref.padding,
      padding = _ref$padding === void 0 ? 1 : _ref$padding,
      title = _ref.title,
      description = _ref.description;
  var classes = useStyles({
    padding: padding
  });

  if (!title) {
    return false;
  }

  return /*#__PURE__*/React.createElement("header", {
    className: className || classes.root
  }, /*#__PURE__*/React.createElement(Typography, {
    noWrap: true,
    component: "h2",
    className: classes.title
  }, title), /*#__PURE__*/React.createElement(Typography, {
    noWrap: true,
    component: "h3",
    className: classes.description
  }, description));
};

export default LayoutHeader;
