// @kit-ui/admin/lib/NavTabs/NavTabs.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireWildcard(require("react"));

var _styles = require("@material-ui/core/styles");

var _Tabs = _interopRequireDefault(require("@material-ui/core/Tabs"));

var _Tab = _interopRequireDefault(require("@material-ui/core/Tab"));

var _ButtonTab = _interopRequireDefault(require("./ButtonTab"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

//import "./NavTabs.scss";
var useStyles = (0, _styles.makeStyles)({});
var StyledTabs = (0, _styles.withStyles)({
  root: {
    padding: 0,
    margin: 0,
    minHeight: "36px"
  },
  indicator: {
    backgroundColor: '#666'
  }
})(_Tabs.default);
var StyledTab = (0, _styles.withStyles)(function (theme) {
  return {
    root: {
      textTransform: 'none',
      minWidth: 0,
      minHeight: "36px",
      paddingLeft: 0,
      paddingRight: 0,
      marginLeft: 0,
      marginRight: theme.spacing(2),
      fontFamily: "Akkurat, sans-serif",
      fontSize: "14px",
      fontWeight: "bold",
      letterSpacing: 0,
      color: "inherit",
      '&:hover': {
        color: "inherit",
        opacity: 1
      },
      '&$selected': {
        opacity: 1
      },
      '&:focus': {
        opacity: 1
      }
    },
    selected: {}
  };
})(function (props) {
  return /*#__PURE__*/_react.default.createElement(_Tab.default, _extends({
    disableRipple: true
  }, props));
});

var NavTabs = function NavTabs(_ref) {
  var className = _ref.className,
      _ref$tabs = _ref.tabs,
      tabs = _ref$tabs === void 0 ? [] : _ref$tabs;
  var classes = useStyles();

  var _React$useState = _react.default.useState(0),
      value = _React$useState[0],
      setValue = _React$useState[1];

  var onTabIndex = function onTabIndex(event, index) {
    setValue(index);

    if (tabs[index].onClick) {
      tabs[index].onClick();
    }
  };

  return /*#__PURE__*/_react.default.createElement("nav", {
    className: className || classes.root
  }, /*#__PURE__*/_react.default.createElement(StyledTabs, {
    value: value,
    onChange: onTabIndex,
    indicatorColor: "primary",
    textColor: "inherit"
  }, tabs && tabs.map(function (tab, index) {
    return /*#__PURE__*/_react.default.createElement(StyledTab, {
      label: tab.label || tab.title,
      value: index,
      key: index
    });
  })));
};

var _default = NavTabs;
exports.default = _default;
module.exports = exports.default;
