// @kit-ui/admin/lib/NavTabs/README.md.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _components = require("@frontend-components/styleguide/lib/components/");

var _utils = require("./../../.styleguide/utils.js");

var _tag = require("@mdx-js/tag");

var _ = require("./");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _inheritsLoose(subClass, superClass) { subClass.prototype = Object.create(superClass.prototype); subClass.prototype.constructor = subClass; subClass.__proto__ = superClass; }

var layoutProps = {};

var MDXContent = /*#__PURE__*/function (_React$Component) {
  _inheritsLoose(MDXContent, _React$Component);

  function MDXContent(props) {
    var _this;

    _this = _React$Component.call(this, props) || this;
    _this.layout = null;
    return _this;
  }

  var _proto = MDXContent.prototype;

  _proto.render = function render() {
    var _this$props = this.props,
        components = _this$props.components,
        props = _objectWithoutPropertiesLoose(_this$props, ["components"]);

    return /*#__PURE__*/_react.default.createElement(_tag.MDXTag, {
      name: "wrapper",
      components: components
    }, /*#__PURE__*/_react.default.createElement(_tag.MDXTag, {
      name: "p",
      components: components
    }, "Tab navigation."), /*#__PURE__*/_react.default.createElement(_components.Playground, null, /*#__PURE__*/_react.default.createElement(_.NavTabs, {
      tabs: [{
        title: 'Section 1',
        url: '#',
        selected: true
      }, {
        title: 'Section 2a',
        url: '/section/'
      }, {
        title: 'Section 2b',
        url: '/section/'
      }]
    })));
  };

  return MDXContent;
}(_react.default.Component);

exports.default = MDXContent;
module.exports = exports.default;
