// @kit-ui/admin/lib/AppSwitcher/AppSwitcherMenu.stories.js
"use strict";

exports.__esModule = true;
exports.OwnerSelected = exports.OwnersList = exports.UserApps = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _AppSwitcherMenu = _interopRequireDefault(require("./AppSwitcherMenu"));

var _ = require("..");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _default = {
  title: 'Components/AppSwitcher/AppSwitcherMenu',
  component: _AppSwitcherMenu.default,
  args: {
    user: (0, _.getRandomUser)()
  },
  argTypes: {}
};
exports.default = _default;

var Template = function Template(args) {
  return /*#__PURE__*/_react.default.createElement(_AppSwitcherMenu.default, args);
};

var UserApps = Template.bind({});
exports.UserApps = UserApps;
UserApps.args = {};
var OwnersList = Template.bind({});
exports.OwnersList = OwnersList;
OwnersList.args = {
  owners: (0, _.getRandomOwners)(20)
};
var OwnerSelected = Template.bind({});
exports.OwnerSelected = OwnerSelected;
OwnerSelected.args = {
  currentOwner: (0, _.getRandomOwner)()
};
