// @kit-ui/admin/lib/AppSwitcher/AppsByOwner.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _AppsGrid = _interopRequireDefault(require("./AppsGrid"));

var _ButtonBase = _interopRequireDefault(require("@material-ui/core/ButtonBase"));

var _Typography = _interopRequireDefault(require("@material-ui/core/Typography"));

var _styles = require("@material-ui/core/styles");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var ButtonBack = function ButtonBack(_ref) {
  var className = _ref.className,
      onClick = _ref.onClick;
  return /*#__PURE__*/_react.default.createElement(_ButtonBase.default, {
    className: className,
    onClick: onClick
  }, "Endre");
};

var useStyles = (0, _styles.makeStyles)(function (theme) {
  return {
    root: {
      display: "block",
      position: "relative"
    },
    header: {
      display: "flex",
      borderColor: "#ccc",
      alignItems: "center",
      justifyContent: "center",
      paddingTop: "18px"
    },
    title: {
      fontFamily: "Akkurat, sans-serif",
      fontWeight: "bold",
      lineHeight: "24px"
    },
    toggleButton: {
      fontFamily: "Akkurat, sans-serif",
      fontSize: "14px",
      fontWeight: "normal",
      margin: "0 6px"
    }
  };
});

var AppsByOwner = function AppsByOwner(_ref2) {
  var _ref2$title = _ref2.title,
      title = _ref2$title === void 0 ? "Owner" : _ref2$title,
      onCollapse = _ref2.onCollapse,
      apps = _ref2.apps,
      onSelect = _ref2.onSelect;
  var classes = useStyles();
  return /*#__PURE__*/_react.default.createElement("div", {
    className: classes.root
  }, /*#__PURE__*/_react.default.createElement("header", {
    className: classes.header
  }, /*#__PURE__*/_react.default.createElement(_Typography.default, {
    className: classes.title
  }, title), /*#__PURE__*/_react.default.createElement(ButtonBack, {
    className: classes.toggleButton,
    onClick: onCollapse
  })), /*#__PURE__*/_react.default.createElement(_AppsGrid.default, {
    apps: apps,
    onSelect: onSelect
  }));
};

AppsByOwner.propTypes = process.env.NODE_ENV !== "production" ? {
  owners: _propTypes.default.array,
  currentOwner: _propTypes.default.shape({
    uniqueId: _propTypes.default.string,
    title: _propTypes.default.string,
    apps: _propTypes.default.array
  })
} : {};
var _default = AppsByOwner;
exports.default = _default;
module.exports = exports.default;
