// @kit-ui/admin/lib/AppSwitcher/FilterOwnersControlled.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireWildcard(require("react"));

var _FilterOwnersUncontrolled = _interopRequireDefault(require("./FilterOwnersUncontrolled"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _inheritsLoose(subClass, superClass) { subClass.prototype = Object.create(superClass.prototype); subClass.prototype.constructor = subClass; subClass.__proto__ = superClass; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var FilterOwnersControlled = /*#__PURE__*/function (_Component) {
  _inheritsLoose(FilterOwnersControlled, _Component);

  function FilterOwnersControlled() {
    var _this;

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = _Component.call.apply(_Component, [this].concat(args)) || this;

    _defineProperty(_assertThisInitialized(_this), "state", {
      q: "",
      filteredOwners: []
    });

    _defineProperty(_assertThisInitialized(_this), "componentDidUpdate", function (prevProps, prevState) {
      if (prevProps.owners !== _this.props.owners) {
        _this.filterOwners();
      }

      if (prevState.q !== _this.state.q) {
        _this.filterOwners();
      }
    });

    _defineProperty(_assertThisInitialized(_this), "filterOwners", function () {
      var owners = _this.props.owners;
      var q = _this.state.q;
      var filteredOwners = owners.filter(function (owner) {
        var title = owner.title;

        if (q && !title.toLowerCase().includes(q.toLowerCase())) {
          return false;
        }

        return true;
      });

      _this.setState({
        filteredOwners: filteredOwners
      });
    });

    _defineProperty(_assertThisInitialized(_this), "onSearch", function (q, event) {
      event.preventDefault();

      _this.setState({
        q: q
      });
    });

    return _this;
  }

  var _proto = FilterOwnersControlled.prototype;

  _proto.componentDidMount = function componentDidMount() {
    this.filterOwners();
  };

  _proto.render = function render() {
    var _this$state = this.state,
        q = _this$state.q,
        filteredOwners = _this$state.filteredOwners;
    return /*#__PURE__*/_react.default.createElement(_FilterOwnersUncontrolled.default, _extends({}, this.props, {
      q: q,
      onSearch: this.onSearch,
      owners: filteredOwners
    }));
  };

  return FilterOwnersControlled;
}(_react.Component);

_defineProperty(FilterOwnersControlled, "defaultProps", {
  owners: []
});

var _default = FilterOwnersControlled;
exports.default = _default;
module.exports = exports.default;
