// @kit-ui/admin/lib/AppSwitcher/NavAppsButton.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _IconButton = _interopRequireDefault(require("@material-ui/core/IconButton"));

var _Apps = _interopRequireDefault(require("@material-ui/icons/Apps"));

var _styles = require("@material-ui/core/styles");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var useStyles = (0, _styles.makeStyles)(function (theme) {
  return {
    root: {
      backgroundColor: "transparent",
      color: "inherit",
      width: "48px",
      height: "48px"
    }
  };
});

var ButtonLogin = function ButtonLogin(_ref) {
  var onClick = _ref.onClick,
      forwardedRef = _ref.forwardedRef;
  var classes = useStyles();
  return /*#__PURE__*/_react.default.createElement(_IconButton.default, {
    className: classes.root,
    onClick: onClick,
    ref: forwardedRef
  }, /*#__PURE__*/_react.default.createElement(_Apps.default, null));
};

var _default = ButtonLogin;
exports.default = _default;
module.exports = exports.default;
