// @kit-ui/admin/lib/AppSwitcher/OwnersListFiltered.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _FilterOwnersControlled = _interopRequireDefault(require("./FilterOwnersControlled"));

var _OwnersList = _interopRequireDefault(require("./OwnersList"));

var _styles = require("@material-ui/core/styles");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var useStyles = (0, _styles.makeStyles)(function (theme) {
  return {
    root: {
      borderTop: "1px solid",
      borderColor: "#ccc",
      position: "relative"
    },
    list: {
      height: theme.spacing(7 * 8)
    }
  };
});

var AppsOwnersList = function AppsOwnersList(_ref) {
  var _ref$owners = _ref.owners,
      owners = _ref$owners === void 0 ? [] : _ref$owners,
      onSelect = _ref.onSelect;
  var classes = useStyles();

  if (owners.length > 7) {
    return /*#__PURE__*/_react.default.createElement("div", {
      className: classes.root
    }, /*#__PURE__*/_react.default.createElement(_FilterOwnersControlled.default, {
      className: classes.list,
      owners: owners,
      onSelect: onSelect
    }));
  }

  return /*#__PURE__*/_react.default.createElement("div", {
    className: classes.root
  }, /*#__PURE__*/_react.default.createElement(_OwnersList.default, {
    owners: owners,
    onSelect: onSelect
  }));
};

var _default = AppsOwnersList;
exports.default = _default;
module.exports = exports.default;
