// @kit-ui/schema/lib/CustomFields/RRulesField/EventField.stories.js
"use strict";

exports.__esModule = true;
exports.CustomRRule = exports.EveryFebruary6 = exports.EveryWednesday = exports.Default = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _SchemaEditor = _interopRequireDefault(require("../../SchemaEditor"));

var _EventField = _interopRequireDefault(require("./EventField.model"));

var _EventField2 = _interopRequireDefault(require("./EventField"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var _default = {
  title: 'Fields/CustomFields/EventField',
  component: _SchemaEditor.default,
  args: _extends({}, _EventField.default, {
    uiSchema: {
      "ui:field": _EventField2.default
    }
  }),
  argTypes: {}
};
exports.default = _default;

var Template = function Template(args) {
  return /*#__PURE__*/_react.default.createElement(_SchemaEditor.default, args);
};

var Default = Template.bind({});
exports.Default = Default;
Default.args = {};
var EveryWednesday = Template.bind({});
exports.EveryWednesday = EveryWednesday;
EveryWednesday.args = {
  formData: {
    rrule: "FREQ=WEEKLY;BYDAY=WE"
  }
};
var EveryFebruary6 = Template.bind({});
exports.EveryFebruary6 = EveryFebruary6;
EveryFebruary6.args = {
  formData: {
    date: "1958-02-06",
    title: "Munich anniversary",
    allDay: true,
    rrule: "FREQ=YEARLY;BYMONTH=2;BYMONTHDAY=6"
  }
};
var CustomRRule = Template.bind({});
exports.CustomRRule = CustomRRule;
CustomRRule.args = {
  formData: {
    date: "1958-02-06",
    title: "Munich anniversary",
    allDay: true,
    rrule: "custom",
    rrules: {
      "FREQ": "WEEKLY"
    }
  }
};
