// @kit-ui/schema/lib/CustomFields/RRulesField/RRulesField.stories.js
"use strict";

exports.__esModule = true;
exports.WeeklyOnWeekends = exports.WeeklyOnFridays = exports.MonthlyOnDay5 = exports.MonthlyOnFirstSunday = exports.YearlyOnFirstMondayOfFebruary = exports.YearlyOnJanuary2nd = exports.Default = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _SchemaEditor = _interopRequireDefault(require("../../SchemaEditor"));

var _RRulesField = _interopRequireDefault(require("./RRulesField"));

var _RRulesField2 = _interopRequireDefault(require("./RRulesField.model"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var _default = {
  title: 'Fields/CustomFields/RRulesField',
  component: _SchemaEditor.default,
  args: _extends({}, _RRulesField2.default),
  argTypes: {}
};
exports.default = _default;

var Template = function Template(args) {
  return /*#__PURE__*/_react.default.createElement(_SchemaEditor.default, args);
};

var Default = Template.bind({});
exports.Default = Default;
Default.args = {};
var YearlyOnJanuary2nd = Template.bind({});
exports.YearlyOnJanuary2nd = YearlyOnJanuary2nd;
YearlyOnJanuary2nd.args = {
  schema: {},
  uiSchema: {},
  formData: {
    "FREQ": "YEARLY",
    "BYMONTH": ["1"],
    "BYMONTHDAY": ["2"]
  }
};
var YearlyOnFirstMondayOfFebruary = Template.bind({});
exports.YearlyOnFirstMondayOfFebruary = YearlyOnFirstMondayOfFebruary;
YearlyOnFirstMondayOfFebruary.args = {
  schema: {},
  uiSchema: {},
  formData: {
    "FREQ": "YEARLY",
    "BYDAY": ["MO"],
    "BYSETPOS": [1],
    "BYMONTH": [2]
  }
};
var MonthlyOnFirstSunday = Template.bind({});
exports.MonthlyOnFirstSunday = MonthlyOnFirstSunday;
MonthlyOnFirstSunday.args = {
  schema: {},
  uiSchema: {},
  formData: {
    "FREQ": "MONTHLY",
    "BYSETPOS": [1],
    "BYDAY": ["SU"],
    "INTERVAL": 2
  }
};
var MonthlyOnDay5 = Template.bind({});
exports.MonthlyOnDay5 = MonthlyOnDay5;
MonthlyOnDay5.args = {
  schema: {},
  uiSchema: {},
  formData: {
    "FREQ": "MONTHLY",
    "BYMONTHDAY": [5],
    "INTERVAL": 2
  }
};
var WeeklyOnFridays = Template.bind({});
exports.WeeklyOnFridays = WeeklyOnFridays;
WeeklyOnFridays.args = {
  schema: {},
  uiSchema: {},
  formData: {
    "FREQ": "WEEKLY",
    "BYDAY": ["FR"],
    "INTERVAL": 2
  }
};
var WeeklyOnWeekends = Template.bind({});
exports.WeeklyOnWeekends = WeeklyOnWeekends;
WeeklyOnWeekends.args = {
  schema: {},
  uiSchema: {},
  formData: {
    "FREQ": "WEEKLY",
    "BYDAY": ["SA", "SU"],
    "INTERVAL": 2
  }
};
