// @kit-ui/schema/lib/CustomFields/RRulesField/RRulesLayout.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _styles = require("@material-ui/core/styles");

var _propTypes = _interopRequireDefault(require("prop-types"));

var _FormLabel = _interopRequireDefault(require("@material-ui/core/FormLabel"));

var _FormGroup = _interopRequireDefault(require("@material-ui/core/FormGroup"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var useStyles = (0, _styles.makeStyles)(function (theme) {
  return {
    fieldset: {
      display: "flex",
      flexDirection: "column",
      border: "1px solid",
      borderColor: theme.palette.divider,
      padding: function padding(props) {
        return theme.spacing(props.padding);
      },
      margin: function margin(props) {
        return theme.spacing(props.spacing * -1 / 2);
      },
      "& > *": {
        margin: function margin(props) {
          return theme.spacing(props.spacing * 1 / 2);
        }
      }
    },
    header: {}
  };
});

var RRulesLayout = function RRulesLayout(_ref) {
  var title = _ref.title,
      children = _ref.children,
      props = _objectWithoutPropertiesLoose(_ref, ["title", "children"]);

  var classes = useStyles(props);

  if (!children) {
    return false;
  }

  return /*#__PURE__*/_react.default.createElement("div", {
    className: classes.fieldset
  }, title && /*#__PURE__*/_react.default.createElement(_FormLabel.default, {
    className: classes.header
  }, title), /*#__PURE__*/_react.default.createElement(_FormGroup.default, {
    row: true
  }, children));
};

RRulesLayout.propTypes = process.env.NODE_ENV !== "production" ? {
  expanded: _propTypes.default.bool,
  onToggle: _propTypes.default.func
} : {};
RRulesLayout.defaultProps = {
  position: "relative",
  padding: 2,
  collapsible: false,
  expanded: true
};
var _default = RRulesLayout;
exports.default = _default;
module.exports = exports.default;
