// @kit-ui/core/es/Iframe/IframeChildren.js
function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _inheritsLoose(subClass, superClass) { subClass.prototype = Object.create(superClass.prototype); subClass.prototype.constructor = subClass; subClass.__proto__ = superClass; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Frame from 'react-frame-component';

var IframeChildren = /*#__PURE__*/function (_Component) {
  _inheritsLoose(IframeChildren, _Component);

  function IframeChildren() {
    var _this;

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = _Component.call.apply(_Component, [this].concat(args)) || this;

    _defineProperty(_assertThisInitialized(_this), "state", {});

    _defineProperty(_assertThisInitialized(_this), "componentDidUpdate", function (prevProps) {
      if (_this.props.children !== prevProps.children) {
        _this.getHead();
      }
    });

    _defineProperty(_assertThisInitialized(_this), "iframeDidMount", function () {
      _this.getHead();
    });

    _defineProperty(_assertThisInitialized(_this), "getHead", function () {
      var stylesheets = [],
          styles = [];
      var linkTags = document.head.querySelectorAll('[rel="stylesheet"]');
      var styleTags = document.head.getElementsByTagName("style");

      for (var i = 0; i < linkTags.length; i++) {
        var styleheet = linkTags[i].getAttribute('href');
        stylesheets.push(styleheet);
      }

      for (var _i = 0; _i < styleTags.length; _i++) {
        var style = styleTags[_i].innerHTML;
        styles.push(style);
      }

      var head = [];

      if (_this.props.style) {
        head.push( /*#__PURE__*/React.createElement("style", null, _this.props.style));
      } else {
        stylesheets.forEach(function (stylesheet) {
          head.push( /*#__PURE__*/React.createElement("link", {
            type: "text/css",
            rel: "stylesheet",
            href: stylesheet
          }));
        });
        styles.forEach(function (style) {
          head.push( /*#__PURE__*/React.createElement("style", null, style));
        });
      }

      _this.setState({
        head: head
      });
    });

    return _this;
  }

  var _proto = IframeChildren.prototype;

  _proto.componentDidMount = function componentDidMount() {
    this.getHead();
  };

  _proto.render = function render() {
    var className = this.props.className;
    var head = this.state.head;

    if (!head) {
      return false;
    }

    var template = '<!DOCTYPE html><html><head></head><body id="preview-iframe"></body></html>';
    return /*#__PURE__*/React.createElement(Frame, {
      className: className,
      initialContent: template,
      head: head,
      mountTarget: "#preview-iframe",
      contentDidMount: this.iframeDidMount
    }, this.props.children);
  };

  return IframeChildren;
}(Component);

_defineProperty(IframeChildren, "defaultProps", {});

IframeChildren.propTypes = process.env.NODE_ENV !== "production" ? {
  className: PropTypes.string,
  children: PropTypes.node
} : {};
export default IframeChildren;
