// @kit-ui/schema/es/Layout/FlexItemLayout.js
import React from "react";
import { makeStyles } from '@material-ui/core/styles';
import Grid from '@material-ui/core/Grid';
var useStyles = makeStyles(function (theme) {
  return {
    fieldset: {
      padding: function padding(props) {
        return theme.spacing(props.padding);
      }
    },
    nav: {
      "& > [aria-hidden=true]": {
        display: "none"
      }
    },
    flex: {
      display: "flex",
      flexDirection: "column",
      margin: function margin(props) {
        return theme.spacing(props.spacing * -1 / 2);
      },
      "&[role=tablist]": {
        "& > [role=tab][aria-hidden=true]": {
          display: "none"
        }
      }
    },
    flexItem: {
      margin: function margin(props) {
        return theme.spacing(props.spacing * 1 / 2);
      }
    }
  };
});

var FlexLayout = function FlexLayout(_ref) {
  var menu = _ref.menu,
      padding = _ref.padding,
      spacing = _ref.spacing,
      children = _ref.children;
  var classes = useStyles();
  return /*#__PURE__*/React.createElement("div", {
    className: classes.fieldset
  }, /*#__PURE__*/React.createElement("div", {
    className: classes.flex,
    role: menu && "tablist"
  }, children));
};

export default FlexLayout;
