// @kit-ui/schema/es/Layout/ModuleLayout.stories.js
import React from 'react';
import ModuleLayout from './ModuleLayout';
import SchemaEditor from "../SchemaEditor/SchemaEditor";
export default {
  title: 'Layouts/ModuleLayout',
  component: ModuleLayout,
  args: {
    schema: {
      type: "object",
      properties: {
        title: {
          type: "string"
        },
        description: {
          type: "string"
        },
        size: {
          "type": "string",
          "enum": ["s", "m", "l"]
        }
      }
    },
    uiSchema: {
      "ui:layout": ModuleLayout
    },
    formData: {
      title: "My title",
      description: "My description"
    }
  },
  argTypes: {}
};

var Template = function Template(args) {
  return /*#__PURE__*/React.createElement(SchemaEditor, args);
};

export var Default = Template.bind({});
Default.args = {};
export var Editable = Template.bind({});
Editable.args = {
  uiSchema: {
    "ui:editable": true
  }
};
export var EditableWithSettings = Template.bind({});
EditableWithSettings.args = {
  uiSchema: {
    "ui:editable": true,
    "ui:options": {
      "spacing": 2,
      "grid": true
    },
    "ui:settings": ["size"]
  }
};
export var EditableWithTabs = Template.bind({});
EditableWithTabs.args = {
  uiSchema: {
    "ui:editable": true,
    "ui:fieldset": ["tab1", "tab2"],
    "tab1": {
      "ui:spacing": 2,
      "ui:fieldset": ["title", "description"]
    },
    "tab2": {
      "ui:spacing": 2,
      "ui:fieldset": ["title", "size"]
    },
    "ui:options": {
      "grid": true,
      "nav": "tabs"
    },
    "ui:settings": ["size"]
  }
};
