// @kit-ui/schema/es/Layout/SectionLayout.stories.js
import React from 'react';
import SchemaEditor from "../SchemaEditor/SchemaEditor";
import SectionLayout from './SectionLayout';
export default {
  title: 'Layouts/SectionLayout',
  component: SchemaEditor,
  args: {
    schema: {
      type: "object",
      properties: {
        title: {
          type: "string"
        },
        description: {
          type: "string"
        }
      }
    },
    uiSchema: {
      "ui:layout": "section"
    },
    formData: {
      title: "Title",
      description: "description"
    }
  },
  argTypes: {}
};

var Template = function Template(args) {
  return /*#__PURE__*/React.createElement(SchemaEditor, args);
};

export var Default = Template.bind({});
Default.args = {};
export var DefaultWithIcon = Template.bind({});
DefaultWithIcon.args = {
  uiSchema: {
    "ui:icon": "people"
  }
};
export var Collapsible = Template.bind({});
Collapsible.args = {
  uiSchema: {
    "ui:options": {
      collapsible: true,
      expanded: false,
      buttons: [{
        title: "Click"
      }, {
        title: "Dont click",
        disabled: true
      }]
    }
  }
};
export var CollapsibleExpanded = Template.bind({});
CollapsibleExpanded.args = {
  uiSchema: {
    "ui:options": {
      collapsible: true,
      expanded: true,
      buttons: [{
        title: "Click"
      }, {
        title: "Dont click",
        disabled: true
      }]
    }
  }
};
export var Editable = Template.bind({});
Editable.args = {
  uiSchema: {
    "ui:options": {
      editable: true,
      expanded: false,
      buttons: [{
        title: "Click"
      }, {
        title: "Dont click",
        disabled: true
      }]
    }
  }
};
export var EditableExpanded = Template.bind({});
EditableExpanded.args = {
  uiSchema: {
    "ui:options": {
      editable: true,
      expanded: true,
      buttons: [{
        title: "Click"
      }, {
        title: "Dont click",
        disabled: true
      }]
    }
  }
};
