// @kit-ui/schema/es/Layout/SectionLayout.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import React from 'react';
import PropTypes from "prop-types";
import Collapse from '@material-ui/core/Collapse';
import { SectionBase, SectionHeader, SectionBody, SectionFooter, SectionButtons } from "@kit-ui/admin";

var SectionLayout = function SectionLayout(_ref) {
  var position = _ref.position,
      collapsible = _ref.collapsible,
      expanded = _ref.expanded,
      title = _ref.title,
      description = _ref.description,
      buttons = _ref.buttons,
      children = _ref.children,
      props = _objectWithoutPropertiesLoose(_ref, ["position", "collapsible", "expanded", "title", "description", "buttons", "children"]);

  return /*#__PURE__*/React.createElement(SectionBase, {
    position: position,
    collapsible: collapsible,
    expanded: expanded
  }, /*#__PURE__*/React.createElement(SectionHeader, _extends({}, props, {
    title: title,
    description: description,
    collapsible: collapsible,
    expanded: expanded
  })), /*#__PURE__*/React.createElement(Collapse, {
    in: expanded,
    timeout: "auto",
    unmountOnExit: true
  }, /*#__PURE__*/React.createElement(SectionBody, _extends({
    position: position
  }, props), children), expanded && /*#__PURE__*/React.createElement(SectionFooter, null, /*#__PURE__*/React.createElement(SectionButtons, {
    buttons: buttons
  }))));
};

SectionLayout.propTypes = process.env.NODE_ENV !== "production" ? {
  expanded: PropTypes.bool,
  onToggle: PropTypes.func
} : {};
SectionLayout.defaultProps = {
  position: "relative",
  collapsible: false,
  expanded: true
};
export default SectionLayout;
