// @kit-ui/admin/es/Finder/FinderSection.js
import React, { Component } from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Typography from '@material-ui/core/Typography';
var useStyles = makeStyles(function (theme) {
  return {
    section: {
      //        position: "relative",
      //        marginBottom: props => { return theme.spacing(props.spacing)},
      "& > header": {
        position: "sticky",
        zIndex: 2,
        top: 0
      }
    },
    header: {
      display: "flex",
      width: "100%",
      alignItems: "baseline"
    },
    title: {
      display: "inline",
      fontSize: "14px",
      fontWeight: "bold"
    },
    description: {
      display: "inline",
      fontSize: "14px",
      fontWeight: "normal"
    }
  };
});

var FinderSection = function FinderSection(_ref) {
  var _ref$expanded = _ref.expanded,
      expanded = _ref$expanded === void 0 ? true : _ref$expanded,
      _ref$spacing = _ref.spacing,
      spacing = _ref$spacing === void 0 ? 2 : _ref$spacing,
      title = _ref.title,
      description = _ref.description,
      _ref$action = _ref.action,
      action = _ref$action === void 0 ? [] : _ref$action,
      children = _ref.children;
  var classes = useStyles({
    spacing: spacing
  });

  var renderHeader = function renderHeader() {
    if (title) {
      return /*#__PURE__*/React.createElement("header", {
        className: classes.header
      }, /*#__PURE__*/React.createElement("div", {
        className: classes.leading
      }, /*#__PURE__*/React.createElement(Typography, {
        className: classes.title,
        nowrap: true
      }, title), description && /*#__PURE__*/React.createElement(Typography, {
        className: classes.description
      }, description)));
    }
  };

  return /*#__PURE__*/React.createElement("div", {
    className: classes.section,
    "aria-expanded": expanded
  }, renderHeader(), expanded && children);
};

export default FinderSection;
