// @kit-ui/admin/es/Finder/FinderSidebar.js
import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
var useStyles = makeStyles(function (theme) {
  var _ariaExpandedTrue, _ariaExpandedTrue2;

  return {
    sidebar: {
      position: "absolute",
      zIndex: 1,
      top: 0,
      right: "auto",
      left: 0,
      bottom: 0,
      width: function width(props) {
        return props.width;
      },
      marginLeft: function marginLeft(props) {
        return "-" + props.width + "px";
      },
      transition: ".125s ease-out",
      "& + *": {
        transition: ".125s ease-out"
      },
      "&[aria-expanded=true]": (_ariaExpandedTrue = {}, _ariaExpandedTrue[theme.breakpoints.up('sm')] = {
        marginLeft: 0
      }, _ariaExpandedTrue),
      "&[aria-expanded=true] + *": (_ariaExpandedTrue2 = {}, _ariaExpandedTrue2[theme.breakpoints.up('sm')] = {
        marginLeft: function marginLeft(props) {
          return props.width;
        }
      }, _ariaExpandedTrue2)
    }
  };
});

var FinderSidebar = function FinderSidebar(_ref) {
  var className = _ref.className,
      children = _ref.children,
      _ref$expanded = _ref.expanded,
      expanded = _ref$expanded === void 0 ? false : _ref$expanded,
      _ref$width = _ref.width,
      width = _ref$width === void 0 ? 240 : _ref$width;
  var classes = useStyles({
    width: width
  });
  return /*#__PURE__*/React.createElement("aside", {
    className: className || classes.sidebar,
    "aria-expanded": expanded
  }, children);
};

export default FinderSidebar;
