// @kit-ui/admin/es/Finder/FinderBase.js
import React, { Component } from 'react';
import { makeStyles } from '@material-ui/core/styles';
var useStyles = makeStyles(function (theme) {
  return {
    root: {
      backgroundColor: function backgroundColor(props) {
        return props.backgroundColor || theme.palette.background.default;
      },
      position: "absolute",
      top: 0,
      right: 0,
      bottom: 0,
      left: 0,
      "& + $root": {
        left: "100%",
        width: "50%"
      },
      "&[aria-expanded=false]": {
        right: "50%"
      },
      "&[aria-expanded=false] + $root": {
        left: "50%"
      }
    }
  };
});

var FinderBase = function FinderBase(_ref) {
  var backgroundColor = _ref.backgroundColor,
      _ref$expanded = _ref.expanded,
      expanded = _ref$expanded === void 0 ? true : _ref$expanded,
      children = _ref.children;
  var classes = useStyles({
    backgroundColor: backgroundColor
  });
  return /*#__PURE__*/React.createElement("div", {
    className: classes.root,
    "aria-expanded": expanded
  }, children);
};

export default FinderBase;
