// @kit-ui/admin/es/Finder/FinderLayout.js
import React from 'react';
import PropTypes from 'prop-types';
import FinderBase from "./FinderBase";
import FinderHeader from "./FinderHeader";
import FinderToolbar from "./FinderToolbar";
import FinderBody from "./FinderBody";
import NavPath from "../NavPath/NavPath";
/** FinderLayout represents a view within an app. Normally this would be wrapped in AppLayout */

var FinderLayout = function FinderLayout(_ref) {
  var backgroundColor = _ref.backgroundColor,
      expanded = _ref.expanded,
      title = _ref.title,
      parents = _ref.parents,
      filters = _ref.filters,
      action = _ref.action,
      views = _ref.views,
      search = _ref.search,
      children = _ref.children,
      onSelect = _ref.onSelect;
  return /*#__PURE__*/React.createElement(FinderBase, {
    backgroundColor: backgroundColor,
    expanded: expanded
  }, /*#__PURE__*/React.createElement(FinderHeader, {
    backgroundColor: backgroundColor
  }, /*#__PURE__*/React.createElement(NavPath, {
    parents: parents,
    title: title,
    filters: filters,
    search: search,
    onSelect: onSelect
  }), action && /*#__PURE__*/React.createElement(FinderToolbar, {
    buttons: action
  })), /*#__PURE__*/React.createElement(FinderBody, null, children));
};

FinderLayout.propTypes = process.env.NODE_ENV !== "production" ? {
  backgroundColor: PropTypes.string,
  expanded: PropTypes.bool,
  title: PropTypes.string,

  /** Array of parents */
  parents: PropTypes.array,
  filters: PropTypes.array,
  action: PropTypes.array,
  views: PropTypes.array,
  search: PropTypes.object,

  /** Finder content */
  children: PropTypes.node
} : {};
export default FinderLayout;
