// @kit-ui/admin/es/Finder/FinderHeader.js
import React from 'react';
import Color from 'color';
import { makeStyles } from '@material-ui/core/styles';

var getBackgroundImage = function getBackgroundImage(backgroundColor) {
  var color = Color(backgroundColor);
  var c1 = color.alpha(0).rgb();
  var c2 = color.alpha(.75).rgb();
  var c3 = color.alpha(1).rgb();
  return "linear-gradient(to bottom, " + c1 + ") 0%, " + c2 + " 75%, " + c3 + " 100%)";
};

var useStyles = makeStyles(function (theme) {
  return {
    root: {
      position: "absolute",
      zIndex: 2,
      top: 0,
      right: 0,
      bottom: "auto",
      left: 0,
      backgroundColor: function backgroundColor(props) {
        return props.backgroundColor || theme.palette.background.default;
      },
      backgroundImage: function backgroundImage(props) {
        return getBackgroundImage(props.backgroundColor || theme.palette.background.default);
      },
      height: theme.spacing(8),
      paddingLeft: theme.spacing(2),
      paddingRight: theme.spacing(2),
      display: "flex",
      alignItems: "center",
      justifyContent: "space-between",
      "& + *": {
        marginTop: theme.spacing(6)
      }
    }
  };
});

var FinderHeader = function FinderHeader(_ref) {
  var backgroundColor = _ref.backgroundColor,
      className = _ref.className,
      children = _ref.children;
  var classes = useStyles({
    backgroundColor: backgroundColor
  });
  return /*#__PURE__*/React.createElement("header", {
    className: className || classes.root
  }, children);
};

export default FinderHeader;
