// @kit-ui/admin/es/Finder/README.md.js
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _inheritsLoose(subClass, superClass) { subClass.prototype = Object.create(superClass.prototype); subClass.prototype.constructor = subClass; subClass.__proto__ = superClass; }

import React from 'react';
import { Playground, LayoutWindow as Window, LayoutCode as Code } from '@frontend-components/styleguide/lib/components/';
import { Props, Methods } from './../../.styleguide/utils.js';
import { MDXTag } from '@mdx-js/tag';
import { LayoutAdmin, Admin, AdminHeader, AdminBody, AdminSidebar, AdminContent, AdminAction } from '../';
import { Finder, FinderModule, FinderHeader, FinderTitle, FinderBody, LayoutUpload, NavPath, NavAction, FinderLayout } from '../';
import Schema from "../Schema/SchemaWithState";
import { appInbox as app } from "../../resources/data";
var layoutProps = {};

var MDXContent = /*#__PURE__*/function (_React$Component) {
  _inheritsLoose(MDXContent, _React$Component);

  function MDXContent(props) {
    var _this;

    _this = _React$Component.call(this, props) || this;
    _this.layout = null;
    return _this;
  }

  var _proto = MDXContent.prototype;

  _proto.render = function render() {
    var _this$props = this.props,
        components = _this$props.components,
        props = _objectWithoutPropertiesLoose(_this$props, ["components"]);

    return /*#__PURE__*/React.createElement(MDXTag, {
      name: "wrapper",
      components: components
    }, /*#__PURE__*/React.createElement(MDXTag, {
      name: "h1",
      components: components
    }, "Finder"), /*#__PURE__*/React.createElement(MDXTag, {
      name: "h2",
      components: components
    }, "FinderLayout + FinderModule"), /*#__PURE__*/React.createElement(MDXTag, {
      name: "p",
      components: components
    }, "FinderLayout with parents and FinderModule as children:"), /*#__PURE__*/React.createElement(Playground, null, /*#__PURE__*/React.createElement(FinderLayout, {
      parents: [{
        "title": "App"
      }, {
        "title": "Parent"
      }, {
        "title": "Page"
      }]
    }, /*#__PURE__*/React.createElement(FinderModule, {
      title: "Document 1",
      description: "description",
      label: "Document"
    }), /*#__PURE__*/React.createElement(FinderModule, {
      title: "Document 2",
      description: "description",
      label: "Document"
    }), /*#__PURE__*/React.createElement(FinderModule, {
      title: "Document 3",
      description: "description",
      label: "Document"
    }))), /*#__PURE__*/React.createElement(MDXTag, {
      name: "h2",
      components: components
    }, "FinderLayout with grid + FinderModule"), /*#__PURE__*/React.createElement(MDXTag, {
      name: "p",
      components: components
    }, "FinderLayout with parents and FinderModule as children:"), /*#__PURE__*/React.createElement(Playground, null, /*#__PURE__*/React.createElement(FinderLayout, {
      layout: "grid",
      parents: [{
        "title": "App"
      }, {
        "title": "Parent"
      }, {
        "title": "Page"
      }]
    }, /*#__PURE__*/React.createElement(FinderModule, {
      title: "Document 1",
      description: "description",
      label: "Document",
      moduleLayout: "grid"
    }), /*#__PURE__*/React.createElement(FinderModule, {
      title: "Document 2",
      description: "description",
      label: "Document",
      moduleLayout: "grid"
    }), /*#__PURE__*/React.createElement(FinderModule, {
      title: "Document 3",
      description: "description",
      label: "Document",
      moduleLayout: "grid"
    }), /*#__PURE__*/React.createElement(FinderModule, {
      title: "Document 1",
      description: "description",
      label: "Document",
      moduleLayout: "grid"
    }), /*#__PURE__*/React.createElement(FinderModule, {
      title: "Document 2",
      description: "description",
      label: "Document",
      moduleLayout: "grid"
    }), /*#__PURE__*/React.createElement(FinderModule, {
      title: "Document 3",
      description: "description",
      label: "Document",
      moduleLayout: "grid"
    }), /*#__PURE__*/React.createElement(FinderModule, {
      title: "Document 1",
      description: "description",
      label: "Document",
      moduleLayout: "grid"
    }), /*#__PURE__*/React.createElement(FinderModule, {
      title: "Document 2",
      description: "description",
      label: "Document",
      moduleLayout: "grid"
    }), /*#__PURE__*/React.createElement(FinderModule, {
      title: "Document 3",
      description: "description",
      label: "Document",
      moduleLayout: "grid"
    }))), /*#__PURE__*/React.createElement(MDXTag, {
      name: "h2",
      components: components
    }, "Finder app"), /*#__PURE__*/React.createElement(MDXTag, {
      name: "p",
      components: components
    }, "FinderLayout within LayoutAdmin, makes up a finder view:"), /*#__PURE__*/React.createElement(Playground, null, /*#__PURE__*/React.createElement(LayoutAdmin, {
      app: app
    }, /*#__PURE__*/React.createElement(FinderLayout, app, /*#__PURE__*/React.createElement(FinderModule, {
      title: "Document 1",
      description: "description",
      label: "Document"
    }), /*#__PURE__*/React.createElement(FinderModule, {
      title: "Document 2",
      description: "description",
      label: "Document"
    }), /*#__PURE__*/React.createElement(FinderModule, {
      title: "Document 3",
      description: "description",
      label: "Document"
    })))), /*#__PURE__*/React.createElement(MDXTag, {
      name: "h2",
      components: components
    }, "Schema example"), /*#__PURE__*/React.createElement(MDXTag, {
      name: "p",
      components: components
    }, "Finder layout, schema example."), /*#__PURE__*/React.createElement(Playground, null, /*#__PURE__*/React.createElement(Schema, {
      schema: {
        "type": "object",
        "properties": {
          "default": {
            "type": "object",
            "properties": {
              "text": {
                "type": "string"
              },
              "switch": {
                "type": "boolean"
              }
            }
          }
        }
      },
      uiSchema: {
        "default": {
          "ui:layout": "finder"
        }
      }
    })), /*#__PURE__*/React.createElement(MDXTag, {
      name: "h2",
      components: components
    }, "Static examples"), /*#__PURE__*/React.createElement(MDXTag, {
      name: "p",
      components: components
    }, "Finder header + body."), /*#__PURE__*/React.createElement(Playground, null, /*#__PURE__*/React.createElement(Finder, null, /*#__PURE__*/React.createElement(FinderHeader, null, ".finder__header"), /*#__PURE__*/React.createElement(FinderBody, null, ".finder__body"))), /*#__PURE__*/React.createElement(MDXTag, {
      name: "p",
      components: components
    }, "Finder header with title and action."), /*#__PURE__*/React.createElement(Playground, null, /*#__PURE__*/React.createElement(Finder, null, /*#__PURE__*/React.createElement(FinderHeader, null, /*#__PURE__*/React.createElement(FinderTitle, {
      title: "Title"
    }), /*#__PURE__*/React.createElement(NavAction, {
      button: {
        title: "Submit"
      }
    })), /*#__PURE__*/React.createElement(FinderBody, null, ".finder__body"))), /*#__PURE__*/React.createElement(MDXTag, {
      name: "p",
      components: components
    }, "Finder with path."), /*#__PURE__*/React.createElement(Playground, null, /*#__PURE__*/React.createElement(Finder, null, /*#__PURE__*/React.createElement(FinderHeader, null, /*#__PURE__*/React.createElement(NavPath, {
      parents: [{
        title: "Grandparent"
      }, {
        title: "Parent"
      }]
    })), /*#__PURE__*/React.createElement(FinderBody, null, ".finder__body"))), /*#__PURE__*/React.createElement(MDXTag, {
      name: "p",
      components: components
    }, "LayoutAdmin + FinderLayout."), /*#__PURE__*/React.createElement(Playground, null, /*#__PURE__*/React.createElement(Admin, null, /*#__PURE__*/React.createElement(AdminHeader, null, ".admin__header"), /*#__PURE__*/React.createElement(AdminBody, null, /*#__PURE__*/React.createElement(AdminSidebar, null, ".admin__sidebar"), /*#__PURE__*/React.createElement(AdminContent, null, /*#__PURE__*/React.createElement(Finder, null, /*#__PURE__*/React.createElement(FinderHeader, null, ".finder__header"), /*#__PURE__*/React.createElement(FinderBody, null, ".finder__body")))))), /*#__PURE__*/React.createElement(MDXTag, {
      name: "p",
      components: components
    }, "LayoutAdmin + Finder content + action."), /*#__PURE__*/React.createElement(Playground, null, /*#__PURE__*/React.createElement(Admin, null, /*#__PURE__*/React.createElement(AdminHeader, null, ".admin__header"), /*#__PURE__*/React.createElement(AdminBody, null, /*#__PURE__*/React.createElement(AdminSidebar, {
      expanded: false
    }, ".admin__sidebar"), /*#__PURE__*/React.createElement(AdminContent, {
      expanded: false
    }, /*#__PURE__*/React.createElement(FinderLayout, {
      title: "Content"
    }, ".admin__content")), /*#__PURE__*/React.createElement(AdminAction, {
      expanded: true
    }, /*#__PURE__*/React.createElement(FinderLayout, {
      title: "Action"
    }, ".admin__action"))))), /*#__PURE__*/React.createElement(MDXTag, {
      name: "p",
      components: components
    }, "LayoutAdmin + Finder + Upload + Action."), /*#__PURE__*/React.createElement(Playground, null, /*#__PURE__*/React.createElement(Admin, null, /*#__PURE__*/React.createElement(AdminHeader, null, ".admin__header"), /*#__PURE__*/React.createElement(AdminBody, null, /*#__PURE__*/React.createElement(AdminSidebar, {
      expanded: false
    }, ".admin__sidebar"), /*#__PURE__*/React.createElement(AdminContent, {
      expanded: false
    }, /*#__PURE__*/React.createElement(LayoutUpload, null)), /*#__PURE__*/React.createElement(AdminAction, {
      expanded: true
    }, ".admin__action")))));
  };

  return MDXContent;
}(React.Component);

export { MDXContent as default };
