// @kit-ui/admin/lib/Finder/FinderApp.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _AppLayout = _interopRequireDefault(require("../App/AppLayout"));

var _FinderLayout = _interopRequireDefault(require("../Finder/FinderLayout"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var FinderApp = function FinderApp(_ref) {
  var app = _ref.app,
      primaryAction = _ref.primaryAction,
      _ref$menu = _ref.menu,
      menu = _ref$menu === void 0 ? [] : _ref$menu,
      _ref$menuByUrl = _ref.menuByUrl,
      menuByUrl = _ref$menuByUrl === void 0 ? {} : _ref$menuByUrl,
      _ref$calendar = _ref.calendar,
      calendar = _ref$calendar === void 0 ? {} : _ref$calendar,
      children = _ref.children;
  var sidebar = {
    expanded: true,
    primaryAction: primaryAction,
    calendar: calendar,
    menu: menu,
    menuByUrl: menuByUrl
  };
  return /*#__PURE__*/_react.default.createElement(_AppLayout.default, {
    app: app,
    sidebar: sidebar
  }, /*#__PURE__*/_react.default.createElement(_FinderLayout.default, null, children));
};

var _default = FinderApp;
exports.default = _default;
module.exports = exports.default;
