// @kit-ui/admin/lib/Finder/FinderBody.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireWildcard(require("react"));

var _styles = require("@material-ui/core/styles");

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

var useStyles = (0, _styles.makeStyles)(function (theme) {
  return {
    root: {
      position: "absolute",
      zIndex: 1,
      width: "100%",
      top: 0,
      right: 0,
      bottom: 0,
      left: 0,
      overflowY: "scroll",
      padding: function padding(props) {
        return theme.spacing(props.padding);
      }
    },
    body: {
      display: "block",
      width: "100%",
      paddingTop: theme.spacing(8),
      paddingRight: theme.spacing(2),
      paddingBottom: theme.spacing(2),
      paddingLeft: theme.spacing(2)
    }
  };
});

var FinderBody = function FinderBody(_ref) {
  var _ref$padding = _ref.padding,
      padding = _ref$padding === void 0 ? 2 : _ref$padding,
      children = _ref.children;
  var classes = useStyles({
    padding: padding
  });
  return /*#__PURE__*/_react.default.createElement("div", null, children);
};

var _default = FinderBody;
exports.default = _default;
module.exports = exports.default;
