// @kit-ui/admin/lib/Finder/FinderToolbar.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireWildcard(require("react"));

var _ButtonBase = _interopRequireDefault(require("@material-ui/core/ButtonBase"));

var _Icon = _interopRequireDefault(require("@material-ui/core/Icon"));

var _styles = require("@material-ui/core/styles");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var useStyles = (0, _styles.makeStyles)(function (theme) {
  return {
    root: {
      backgroundColor: "inherit",
      display: "flex",
      margin: theme.spacing(-0.5)
    },
    button: {
      padding: theme.spacing(0.5)
    },
    icon: {
      fontSize: "18px",
      marginRight: theme.spacing(1)
    },
    label: {
      fontFamily: "Akkurat, sans-serif",
      fontSize: "14px",
      fontWeight: "bold"
    }
  };
});

var ToolbarButton = function ToolbarButton(_ref) {
  var icon = _ref.icon,
      label = _ref.label,
      title = _ref.title,
      onClick = _ref.onClick;
  var classes = useStyles();
  return /*#__PURE__*/_react.default.createElement(_ButtonBase.default, {
    className: classes.button,
    onClick: onClick
  }, icon && /*#__PURE__*/_react.default.createElement("i", {
    className: classes.icon
  }, /*#__PURE__*/_react.default.createElement(_Icon.default, null, icon)), /*#__PURE__*/_react.default.createElement("b", {
    className: classes.label
  }, label || title));
};

var FinderToolbar = function FinderToolbar(_ref2) {
  var _ref2$buttons = _ref2.buttons,
      buttons = _ref2$buttons === void 0 ? [] : _ref2$buttons;
  var classes = useStyles();

  if (!buttons.length) {
    return null;
  }

  return /*#__PURE__*/_react.default.createElement("nav", {
    className: classes.root
  }, buttons.map(function (button, index) {
    return /*#__PURE__*/_react.default.createElement(ToolbarButton, _extends({}, button, {
      key: index
    }));
  }));
};

var _default = FinderToolbar;
exports.default = _default;
module.exports = exports.default;
