// @kit-ui/schema/lib/TextWidget/TextWidget.stories.js
"use strict";

exports.__esModule = true;
exports.StandardVariant = exports.OutlinedVariant = exports.WithSuffix = exports.WithPrefix = exports.WithLabel = exports.Default = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _SchemaEditor = _interopRequireDefault(require("../SchemaEditor/"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _default = {
  title: 'Widgets/TextWidget',
  component: _SchemaEditor.default,
  args: {
    schema: {
      type: "string"
    },
    uiSchema: {},
    formData: "Text"
  },
  argTypes: {}
};
exports.default = _default;

var Template = function Template(args) {
  return /*#__PURE__*/_react.default.createElement(_SchemaEditor.default, args);
};

var Default = Template.bind({});
exports.Default = Default;
Default.args = {};
var WithLabel = Template.bind({});
exports.WithLabel = WithLabel;
WithLabel.args = {
  uiSchema: {
    "ui:title": "Title",
    "ui:help": "Write a title"
  }
};
var WithPrefix = Template.bind({});
exports.WithPrefix = WithPrefix;
WithPrefix.args = {
  uiSchema: {
    "ui:title": "Title",
    "ui:help": "Write a title",
    "ui:prefix": "Kr."
  }
};
var WithSuffix = Template.bind({});
exports.WithSuffix = WithSuffix;
WithSuffix.args = {
  uiSchema: {
    "ui:title": "Title",
    "ui:help": "Write a title",
    "ui:suffix": "Suffix"
  }
};
var OutlinedVariant = Template.bind({});
exports.OutlinedVariant = OutlinedVariant;
OutlinedVariant.args = {
  uiSchema: {
    "ui:title": "Title",
    "ui:variant": "outlined",
    "ui:help": "Write a title"
  }
};
var StandardVariant = Template.bind({});
exports.StandardVariant = StandardVariant;
StandardVariant.args = {
  uiSchema: {
    "ui:title": "Title",
    "ui:variant": "standard",
    "ui:help": "Write a title"
  }
};
