// @kit-ui/schema/lib/SchemaEditor/SchemaEditor.stories.js
"use strict";

exports.__esModule = true;
exports.MediaSketchfab = exports.MediaVimeo = exports.MediaYoutube = exports.MediaAudio = exports.MediaVideo = exports.MediaImage = exports.KpArticle = exports.Section = exports.Article = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _SchemaEditor = _interopRequireDefault(require("./SchemaEditor"));

var _article = _interopRequireDefault(require("../data/article"));

var _section = _interopRequireDefault(require("../data/article/section"));

var _article2 = _interopRequireDefault(require("../data/kp/article"));

var _formData = _interopRequireDefault(require("../data/media/image/formData"));

var _schemasByName = _interopRequireDefault(require("../schemasByName"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var _default = {
  title: 'Schema/SchemaEditor',
  component: _SchemaEditor.default,
  args: {},
  argTypes: {}
};
exports.default = _default;

var Template = function Template(args) {
  return /*#__PURE__*/_react.default.createElement(_SchemaEditor.default, args);
};

var Article = Template.bind({});
exports.Article = Article;
Article.args = _extends({}, _article.default);
var Section = Template.bind({});
exports.Section = Section;
Section.args = _extends({}, _section.default);
var KpArticle = Template.bind({});
exports.KpArticle = KpArticle;
KpArticle.args = _extends({}, _article2.default);
var MediaImage = Template.bind({});
exports.MediaImage = MediaImage;
MediaImage.args = _extends({}, _schemasByName.default["media/image"], {
  formData: _formData.default
});
var MediaVideo = Template.bind({});
exports.MediaVideo = MediaVideo;
MediaVideo.args = _extends({}, _schemasByName.default["media/video"]);
var MediaAudio = Template.bind({});
exports.MediaAudio = MediaAudio;
MediaAudio.args = _extends({}, _schemasByName.default["media/audio"]);
var MediaYoutube = Template.bind({});
exports.MediaYoutube = MediaYoutube;
MediaYoutube.args = _extends({}, _schemasByName.default["media/video/youtube"]);
var MediaVimeo = Template.bind({});
exports.MediaVimeo = MediaVimeo;
MediaVimeo.args = _extends({}, _schemasByName.default["media/video/vimeo"]);
var MediaSketchfab = Template.bind({});
exports.MediaSketchfab = MediaSketchfab;
MediaSketchfab.args = _extends({}, _schemasByName.default["media/application/sketchfab"]);
