// @kit-ui/schema/lib/SchemaEditor/SchemaDialog.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _utils = require("../utils/");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

/** SchemaDialog example */
var SchemaDialog = function SchemaDialog(_ref) {
  var _ref$schema = _ref.schema,
      schema = _ref$schema === void 0 ? {} : _ref$schema,
      _ref$uiSchema = _ref.uiSchema,
      uiSchema = _ref$uiSchema === void 0 ? {} : _ref$uiSchema,
      onChange = _ref.onChange,
      onClose = _ref.onClose,
      props = _objectWithoutPropertiesLoose(_ref, ["schema", "uiSchema", "onChange", "onClose"]);

  var _useState = (0, _react.useState)(props.formData),
      formData = _useState[0],
      setFormData = _useState[1];

  (0, _react.useEffect)(function () {
    setFormData(props.formData);
  }, [props.formData]);
  var itemSchema;

  if (schema.type && schema.type === "array") {
    itemSchema = schema.items;
  } else {
    itemSchema = schema;
  }

  var dialogType;

  if (itemSchema && itemSchema.properties && itemSchema.properties.mediaId) {
    dialogType = "media";
  }

  if (itemSchema && itemSchema.properties && itemSchema.properties.referenceId) {
    dialogType = "reference";
  }

  var _onChange = function _onChange(formData) {
    var newFormData = (0, _utils.getDefaultFormState)(schema, formData);
    setFormData(newFormData);
    onChange(newFormData);
  };

  var _onAdd = function _onAdd() {
    var item;

    if (dialogType === "media") {
      item = (0, _utils.getDefaultFormState)(itemSchema, {
        mediaId: 1,
        media: {
          imageUrl: "https://akamai.vgc.no/drfront/images/2020/10/28/w=1080;h=720;564623.jpg"
        }
      });
    } else if (itemSchema.type === "object") {
      item = (0, _utils.getDefaultFormState)(itemSchema, {
        title: "Dialog addded"
      });
    } else {
      item = "Dialog string";
    }

    var newFormData;

    if (schema.type === "array") {
      newFormData = [].concat(formData, item);
    } else {
      newFormData = item;
    }

    _onChange(newFormData);
  };

  return /*#__PURE__*/_react.default.createElement("div", null, "Example dialog.", /*#__PURE__*/_react.default.createElement("button", {
    type: "button",
    onClick: _onAdd
  }, "Add dummy data"), /*#__PURE__*/_react.default.createElement("button", {
    type: "button",
    onClick: onClose
  }, "Close"), /*#__PURE__*/_react.default.createElement("ul", null, /*#__PURE__*/_react.default.createElement("li", null, "Dialog type: ", dialogType), /*#__PURE__*/_react.default.createElement("li", null, "ItemSchema: ", JSON.stringify(itemSchema))));
};

SchemaDialog.propTypes = process.env.NODE_ENV !== "production" ? {
  /** onClose */
  onClose: _propTypes.default.func
} : {};
var _default = SchemaDialog;
exports.default = _default;
module.exports = exports.default;
