// @kit-ui/schema/lib/SchemaEditor/SchemaEditor.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _SchemaBase = _interopRequireDefault(require("../Schema/SchemaBase"));

var _SchemaDialog = _interopRequireDefault(require("./SchemaDialog"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var SchemaEditor = function SchemaEditor(_ref) {
  var schema = _ref.schema,
      uiSchema = _ref.uiSchema,
      props = _objectWithoutPropertiesLoose(_ref, ["schema", "uiSchema"]);

  var _useState = (0, _react.useState)(props.formData || {}),
      formData = _useState[0],
      setFormData = _useState[1];

  var _onSubmit = function _onSubmit(_ref2) {
    var formData = _ref2.formData;
    console.log('onSubmit', formData);
  };

  var _onChange = function _onChange(_ref3) {
    var formData = _ref3.formData;
    console.log('onChange', formData);
    setFormData(formData);
  };
  /*
   const [languages, setLanguages] = useState(["en"])
  const [currentLocale, setCurrentLocale] = useState(null)
   const _onLocale = (locale) => {
      console.log('onLocale', locale)
      setCurrentLocale(locale)
  }
   */


  var _useState2 = (0, _react.useState)(null),
      currentId = _useState2[0],
      setCurrentId = _useState2[1];

  var _onToggle = function _onToggle(_ref4) {
    var id = _ref4.id,
        props = _objectWithoutPropertiesLoose(_ref4, ["id"]);

    console.log('onToggle', id);
    var parentId = id.split('_');
    parentId.pop();
    parentId = parentId.join('_');
    var expandId = undefined;

    if (id && currentId && currentId.startsWith(id)) {
      expandId = parentId;
    } else {
      expandId = id;
    }

    setCurrentId(expandId);
  };

  var _onExpand = function _onExpand(_ref5) {
    var id = _ref5.id,
        props = _objectWithoutPropertiesLoose(_ref5, ["id"]);

    console.log('onExpand', id);
    setCurrentId(id);
  };

  var _onCollapse = function _onCollapse(_ref6) {
    var id = _ref6.id,
        props = _objectWithoutPropertiesLoose(_ref6, ["id"]);

    console.log('onCollapse', id);

    _onToggle({
      id: id
    });
  };

  var _useState3 = (0, _react.useState)({}),
      sidebar = _useState3[0],
      setSidebar = _useState3[1];

  var _useState4 = (0, _react.useState)({}),
      overlay = _useState4[0],
      setOverlay = _useState4[1];

  var _useState5 = (0, _react.useState)({}),
      dialog = _useState5[0],
      setDialog = _useState5[1];

  var _onDialog = function _onDialog(props) {
    console.log('onDialog', props);

    if (dialog && dialog.id) {
      setDialog({});
      setSidebar({});
      setOverlay({});
    } else {
      setDialog(_extends({}, props, {
        expanded: true
      }));

      if (currentId && currentId.startsWith("root_settings")) {
        setOverlay(_extends({}, props, {
          expanded: true
        }));
      } else {
        setSidebar(_extends({}, props, {
          expanded: true
        }));
      }
    }
  };

  var formContext = _extends({
    //        languages: languages,
    //        currentLocale: currentLocale,
    //        onLocale: _onLocale,

    /*
    
            dialog: {
                ...dialog,
                template: SchemaDialog,
                onClose: () => _onDialog({})
            },
    */
    sidebar: _extends({}, sidebar, {
      template: _SchemaDialog.default,
      onClose: function onClose() {
        return _onDialog({});
      }
    }),
    overlay: _extends({}, overlay, {
      template: _SchemaDialog.default,
      onClose: function onClose() {
        return _onDialog({});
      }
    }),
    onDialog: _onDialog,
    currentId: currentId,
    onToggle: _onToggle,
    onExpand: _onExpand,
    onCollapse: _onCollapse,
    onEdit: _onExpand,
    onSave: _onCollapse,
    onEditReference: _onDialog,
    onEditMedia: _onDialog,
    onSubmit: function onSubmit(status) {
      return _onSubmit && _onSubmit({
        formData: _extends({}, formData, {
          status: status || formData.status || undefined
        })
      });
    }
  }, props.formContext);

  return /*#__PURE__*/_react.default.createElement(_SchemaBase.default, {
    schema: schema,
    uiSchema: uiSchema,
    formData: formData,
    formContext: formContext,
    onChange: _onChange,
    onSubmit: _onSubmit
  });
};

SchemaEditor.propTypes = process.env.NODE_ENV !== "production" ? {
  /** Schema */
  schema: _propTypes.default.node,

  /** uiSchema */
  uiSchema: _propTypes.default.node,

  /** formData */
  formData: _propTypes.default.node
} : {};
var _default = SchemaEditor;
exports.default = _default;
module.exports = exports.default;
