// @kit-ui/admin/es/utils/getCalendar.js
import moment from 'moment';
import "moment/locale/nb";
import "moment/locale/nn";
import "moment/locale/sv";
import { getEvents, getEventsByDate } from "..";
export var getCalendar = function getCalendar(_ref) {
  var date = _ref.date,
      _ref$items = _ref.items,
      items = _ref$items === void 0 ? [] : _ref$items;
  var dt = moment(date).format('YYYY-MM-DD');
  var dtStart = moment(dt).startOf('month').startOf('isoWeek').format('YYYY-MM-DD');
  var dtEnd = moment(dt).endOf('month').endOf('isoWeek').format('YYYY-MM-DD');
  var weeks = moment(dtEnd).diff(dtStart, 'week');
  var events = [];

  if (items.length) {
    events = getEvents({
      items: items,
      dtStart: dtStart,
      dtEnd: dtEnd
    });
  }

  var rows = [];

  for (var w = 0; w <= weeks; w++) {
    var datetime = moment(dtStart).add(w, 'weeks');
    datetime = moment(datetime).set('hour', 0);
    datetime = moment(datetime).set('minute', 0);
    datetime = moment(datetime).set('second', 0);
    var cols = [];
    var week = datetime;

    for (var d = 0; d <= 6; d++) {
      datetime = moment(week).add(d, 'days');

      var _date = getEventsByDate(datetime, events);

      cols.push(_date);
    }

    rows.push(cols);
  }

  return {
    dt: dt,
    dtStart: dtStart,
    dtEnd: dtEnd,
    rows: rows,
    events: events
  };
};
export default getCalendar;
