// @kit-ui/admin/es/utils/getRandomOwner.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import React from "react";
import { OwnerIcon } from "@kit-ui/icons";
import { getRandomArrayItem, getRandomTheme } from "./";
import defaultOwner from "../data/owner";
var owners = [{
  name: "val",
  title: "Valdresmusea"
}, {
  name: "byg",
  title: "Bygdøy Kongsgård"
}, {
  name: "nmm",
  title: "Norsk Maritimt Museum"
}, {
  name: "kyst",
  title: "Kystmyseet"
}, {
  name: "nkim",
  title: "Nordenfjeldske Kunstindustrimuseum"
}, {
  name: "kvi",
  title: "Kvinnemuseet"
}, {
  name: "fet",
  title: "Fetsund Lenser"
}];
var ownerImages = ["https://dms-cf-09.dimu.org/image/032yiVGVkzHu", "https://dms-cf-09.dimu.org/image/032yiziWuETr"];
export var getOwnerIcon = function getOwnerIcon(_ref) {
  var name = _ref.name,
      title = _ref.title,
      theme = _ref.theme;
  var color = theme && theme.primaryColor;
  var label = name || title;
  var text = label && label.substr(0, 3).toUpperCase();
  return /*#__PURE__*/React.createElement(OwnerIcon, {
    color: color,
    text: text
  });
};
export var getOwner = function getOwner(owner) {
  if (!owner) {
    owner = defaultOwner;
  }

  return _extends({}, owner, {
    icon: getOwnerIcon(owner)
  });
};
export var getRandomOwner = function getRandomOwner() {
  var _getRandomArrayItem = getRandomArrayItem(owners),
      name = _getRandomArrayItem.name,
      title = _getRandomArrayItem.title;

  var imageUrl = getRandomArrayItem(ownerImages);

  var owner = _extends({}, defaultOwner, {
    name: name,
    title: title,
    imageUrl: imageUrl,
    theme: getRandomTheme()
  });

  return getOwner(owner);
};
export var getRandomOwners = function getRandomOwners(_ref2) {
  var _ref2$max = _ref2.max,
      max = _ref2$max === void 0 ? 20 : _ref2$max;
  var owners = [];

  for (var i = 0; i < max; i++) {
    var owner = getRandomOwner();
    owner.uniqueid = "owner-" + i;
    owners.push(owner);
  }

  return owners;
};
export default getRandomOwner;
