// @kit-ui/admin/es/utils/getTimetable.js
import _ from 'lodash';
import moment from 'moment';
import { getEvents, getEventsByHour } from "..";
export var getTimetable = function getTimetable(_ref) {
  var date = _ref.date,
      days = _ref.days,
      _ref$items = _ref.items,
      items = _ref$items === void 0 ? [] : _ref$items,
      _ref$timeFrom = _ref.timeFrom,
      timeFrom = _ref$timeFrom === void 0 ? 0 : _ref$timeFrom,
      _ref$timeTo = _ref.timeTo,
      timeTo = _ref$timeTo === void 0 ? 23 : _ref$timeTo;

  if (!days) {
    days = 1;
  }

  var dt = moment(date).format('YYYY-MM-DD');
  var dtStart = moment(dt).format('YYYY-MM-DD');
  var dtEnd = moment(dt).add(days, 'days').format('YYYY-MM-DD');
  var events = {};

  if (items.length) {
    events = getEvents({
      items: items,
      dtStart: dtStart,
      dtEnd: dtEnd
    });
  }

  var cols = [];

  for (var d = 0; d < days; d++) {
    var datetime = moment(dtStart).add(d, 'days');
    datetime = moment(datetime).set('hour', 0);
    datetime = moment(datetime).set('minute', 0);
    datetime = moment(datetime).set('second', 0);
    var rows = [];

    for (var h = timeFrom; h <= timeTo; h++) {
      datetime = moment(datetime).set('hour', h);
      datetime = moment(datetime).set('minute', 0);
      datetime = moment(datetime).set('second', 0);
      var hour = getEventsByHour(datetime, events);
      rows.push(hour);
    }

    cols.push(rows);
  }

  var timetable = {
    dt: dt,
    dtStart: dtStart,
    dtEnd: dtEnd,
    cols: cols,
    events: events
  };
  return timetable;
};
