// @kit-ui/admin/lib/App/utils/createAppTheme.js
"use strict";

exports.__esModule = true;
exports.default = exports.createAppTheme = void 0;

var _color = _interopRequireDefault(require("color"));

var _styles = require("@material-ui/core/styles");

var _theme = _interopRequireDefault(require("./theme.overrides"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var getColor = function getColor(color) {
  if (color === void 0) {
    color = "#666";
  }

  return (0, _color.default)(color).hex();
};

var getPalette = function getPalette(_ref) {
  var primaryColor = _ref.primaryColor,
      secondaryColor = _ref.secondaryColor;
  return {
    primary: {
      main: getColor(primaryColor)
    },
    secondary: {
      main: getColor(secondaryColor || primaryColor)
    }
  };
};

var createAppTheme = function createAppTheme(_ref2) {
  var palette = _ref2.palette,
      theme = _objectWithoutPropertiesLoose(_ref2, ["palette"]);

  return (0, _styles.createMuiTheme)({
    typography: {
      fontFamily: "Akkurat, sans-serif !important"
    },
    palette: palette || getPalette(theme),
    overrides: _theme.default
  });
};

exports.createAppTheme = createAppTheme;
var _default = createAppTheme;
exports.default = _default;
