// @kit-ui/admin/lib/NavCalendar/NavCalendar.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _styles = require("@material-ui/core/styles");

var _IconButton = _interopRequireDefault(require("@material-ui/core/IconButton"));

var _moment = _interopRequireDefault(require("moment"));

var _utils = require("../utils/");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var useStyles = (0, _styles.makeStyles)(function (theme) {
  return {
    root: {},
    table: {
      width: "100%",
      display: "flex",
      flexDirection: "column",
      alignItems: "flex-start",
      "& thead": {
        display: "flex",
        flexDirection: "column",
        alignItems: "flex-start"
      },
      "& tbody": {
        display: "flex",
        flexDirection: "column",
        alignItems: "flex-start"
      },
      "& tr": {
        display: "flex",
        alignItems: "flex-start",
        width: "100%",
        "& > *": {
          width: theme.spacing(4)
        }
      }
    },
    weekday: {
      display: "flex",
      width: theme.spacing(4),
      height: theme.spacing(4),
      alignItems: "center",
      justifyContent: "center",
      fontFamily: "Akkurat, sans-serif",
      fontSize: '12px',
      lineHeight: 1
    },
    date: {
      width: theme.spacing(4),
      height: theme.spacing(4),
      margin: 0,
      fontFamily: "Akkurat, sans-serif",
      fontSize: '12px',
      color: theme.palette.text.disabled,
      lineHeight: 1,
      "&[data-month=true]": {
        color: theme.palette.text.primary
      },
      "&[data-today=true]": {
        border: "1px solid",
        borderColor: theme.palette.primary.main
      },
      "&[data-selected=true]": {
        backgroundColor: theme.palette.secondary.main,
        color: theme.palette.secondary.contrastText
      }
    }
  };
});

var NavCalendarWeekday = function NavCalendarWeekday(_ref) {
  var datetime = _ref.datetime;
  var weekday = (0, _moment.default)(datetime).format('dd').substr(0, 1).toUpperCase();
  var classes = useStyles();
  return /*#__PURE__*/_react.default.createElement("div", {
    className: classes.weekday
  }, weekday);
};

var NavCalendarDate = function NavCalendarDate(_ref2) {
  var datetime = _ref2.datetime,
      date = _ref2.date,
      monthStart = _ref2.monthStart,
      monthEnd = _ref2.monthEnd,
      onChange = _ref2.onChange;
  var classes = useStyles();
  var selected = date === datetime;
  var dtMonth = (0, _moment.default)(datetime).isSameOrAfter(monthStart, 'day') && (0, _moment.default)(datetime).isSameOrBefore(monthEnd, 'day');
  return /*#__PURE__*/_react.default.createElement(_IconButton.default, {
    className: classes.date,
    onClick: function onClick(event) {
      return onChange && onChange(datetime, event);
    },
    "data-selected": selected,
    "data-month": dtMonth
  }, (0, _moment.default)(datetime).format('D'));
};

var NavCalendar = function NavCalendar(_ref3) {
  var className = _ref3.className,
      _ref3$format = _ref3.format,
      format = _ref3$format === void 0 ? "YYYY-MM-DD" : _ref3$format,
      today = _ref3.today,
      date = _ref3.date,
      startDate = _ref3.startDate,
      endDate = _ref3.endDate,
      onChange = _ref3.onChange;
  var classes = useStyles();
  today = today && (0, _moment.default)(today, format, true).isValid() && date || (0, _moment.default)().format(format);
  date = date && (0, _moment.default)(date, format, true).isValid() && date || today;
  startDate = startDate && (0, _moment.default)(startDate, format, true).isValid() && startDate || date;
  endDate = endDate && (0, _moment.default)(endDate, format, true).isValid() && endDate || startDate;
  var monthStart = (0, _moment.default)(date).startOf('month').format('YYYY-MM-DD');
  var monthEnd = (0, _moment.default)(date).endOf('month').format('YYYY-MM-DD');

  var _getCalendar = (0, _utils.getCalendar)({
    date: startDate
  }),
      rows = _getCalendar.rows;

  return /*#__PURE__*/_react.default.createElement("table", {
    className: className || classes.table
  }, /*#__PURE__*/_react.default.createElement("thead", null, /*#__PURE__*/_react.default.createElement("tr", null, rows[0].map(function (col, c) {
    return /*#__PURE__*/_react.default.createElement("th", {
      key: c
    }, /*#__PURE__*/_react.default.createElement(NavCalendarWeekday, col));
  }))), /*#__PURE__*/_react.default.createElement("tbody", null, rows.map(function (row, r) {
    return /*#__PURE__*/_react.default.createElement("tr", {
      key: r
    }, row.map(function (col, c) {
      return /*#__PURE__*/_react.default.createElement("td", {
        key: c
      }, /*#__PURE__*/_react.default.createElement(NavCalendarDate, _extends({}, col, {
        date: date,
        today: today,
        monthStart: monthStart,
        monthEnd: monthEnd,
        onChange: onChange
      })));
    }));
  })));
};

var _default = NavCalendar;
exports.default = _default;
module.exports = exports.default;
