// @kit-ui/admin/lib/Calendar/CalendarPicker.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _moment = _interopRequireDefault(require("moment"));

require("moment/locale/nb");

require("moment/locale/nn");

require("moment/locale/sv");

var _utils = require("../utils/");

var _styles = require("@material-ui/core/styles");

var _IconButton = _interopRequireDefault(require("@material-ui/core/IconButton"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var useStyles = (0, _styles.makeStyles)(function (theme) {
  return {
    table: {
      display: "flex",
      flexDirection: "column",
      alignItems: "flex-start"
    },
    thead: {
      display: "flex",
      flexDirection: "column",
      alignItems: "flex-start"
    },
    tbody: {
      display: "flex",
      flexDirection: "column",
      alignItems: "flex-start"
    },
    tableRow: {
      display: "flex",
      alignItems: "flex-start"
    },
    tableCol: {
      width: theme.spacing(5),
      height: theme.spacing(5)
    },
    dateCol: {
      display: "flex",
      alignItems: "center",
      justifyContent: "center",
      width: theme.spacing(5),
      height: theme.spacing(5)
    },
    weekday: {
      display: "flex",
      alignItems: "center",
      justifyContent: "center",
      width: theme.spacing(5),
      height: theme.spacing(5),
      fontFamily: "Akkurat, sans-serif",
      fontSize: '12px',
      lineHeight: 1
    },
    day: {
      width: theme.spacing(4.5),
      height: theme.spacing(4.5),
      margin: theme.spacing(.25),
      fontFamily: "Akkurat, sans-serif",
      fontSize: '12px',
      lineHeight: 1,
      "&[data-today=true]": {
        border: "1px solid",
        borderColor: theme.palette.primary.main
      },
      "&[data-selected=true]": {
        backgroundColor: theme.palette.primary.main,
        color: theme.palette.primary.contrastText
      },
      "&[data-range=true]": {
        borderRadius: 0,
        width: theme.spacing(5)
      },
      "&[data-range-start=true]": {
        borderTopRightRadius: 0,
        borderBottomRightRadius: 0,
        width: theme.spacing(4.75),
        marginRight: 0
      },
      "&[data-range-end=true]": {
        borderTopLeftRadius: 0,
        borderBottomLeftRadius: 0,
        width: theme.spacing(4.75),
        marginLeft: 0
      }
    }
  };
});

var CalendarPickerWeekday = function CalendarPickerWeekday(_ref) {
  var date = _ref.date;
  var classes = useStyles();
  var weekday = (0, _moment.default)(date).format('dd').substr(0, 1).toUpperCase();
  return /*#__PURE__*/_react.default.createElement("th", {
    className: classes.tableCol
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: classes.weekday
  }, weekday));
};

var CalendarPickerDate = function CalendarPickerDate(_ref2) {
  var today = _ref2.today,
      startDate = _ref2.startDate,
      endDate = _ref2.endDate,
      date = _ref2.date,
      onChange = _ref2.onChange;
  var classes = useStyles();
  var isRangeStart = startDate !== endDate && (0, _moment.default)(date).isSame(startDate, 'day');
  var isRangeEnd = startDate !== endDate && (0, _moment.default)(date).isSame(endDate, 'day');
  var isRange = (0, _moment.default)(date).isAfter(startDate, 'day') && (0, _moment.default)(date).isBefore(endDate, 'day');
  var isSelected = (0, _moment.default)(date).isSameOrAfter(startDate, 'day') && (0, _moment.default)(date).isSameOrBefore(endDate, 'day');
  var isToday = (0, _moment.default)(date).isSame(today, 'day');
  /*
  if (!isMonth) {
      return ""
  }
  */

  return /*#__PURE__*/_react.default.createElement("td", {
    className: classes.dateCol
  }, /*#__PURE__*/_react.default.createElement(_IconButton.default, {
    selected: isSelected,
    onClick: function onClick(event) {
      return onChange(date, event);
    },
    "data-selected": isSelected,
    "data-range": isRange,
    "data-range-start": isRangeStart,
    "data-range-end": isRangeEnd,
    "data-today": isToday,
    className: classes.day
  }, (0, _moment.default)(date).format('D')));
};

var CalendarPicker = function CalendarPicker(_ref3) {
  var _ref3$format = _ref3.format,
      format = _ref3$format === void 0 ? "YYYY-MM-DD" : _ref3$format,
      today = _ref3.today,
      date = _ref3.date,
      startDate = _ref3.startDate,
      endDate = _ref3.endDate,
      onChange = _ref3.onChange;
  today = today && (0, _moment.default)(today, format, true).isValid() && date || (0, _moment.default)().format(format);
  date = date && (0, _moment.default)(date, format, true).isValid() && date || today;
  startDate = startDate && (0, _moment.default)(startDate, format, true).isValid() && startDate || date;
  endDate = endDate && (0, _moment.default)(endDate, format, true).isValid() && endDate || startDate;

  var _getCalendar = (0, _utils.getCalendar)({
    date: startDate
  }),
      rows = _getCalendar.rows;

  var classes = useStyles();
  return /*#__PURE__*/_react.default.createElement("table", {
    className: classes.table
  }, /*#__PURE__*/_react.default.createElement("thead", null, /*#__PURE__*/_react.default.createElement("tr", null, rows[0].map(function (col) {
    return /*#__PURE__*/_react.default.createElement(CalendarPickerWeekday, col);
  }))), /*#__PURE__*/_react.default.createElement("tbody", null, rows.map(function (row, r) {
    return /*#__PURE__*/_react.default.createElement("tr", {
      className: classes.tableRow,
      key: r
    }, row.map(function (col) {
      return /*#__PURE__*/_react.default.createElement(CalendarPickerDate, _extends({}, col, {
        today: today,
        startDate: startDate,
        endDate: endDate,
        onChange: onChange
      }));
    }));
  })));
};

var _default = CalendarPicker;
exports.default = _default;
module.exports = exports.default;
