// @kit-ui/admin/lib/Calendar/Timetable.stories.js
"use strict";

exports.__esModule = true;
exports.WithEvents = exports.Default = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _Timetable = _interopRequireDefault(require("./Timetable"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _default = {
  title: 'Calendar/Timetable',
  component: _Timetable.default,
  args: {},
  argTypes: {}
};
exports.default = _default;

var Template = function Template(args) {
  return /*#__PURE__*/_react.default.createElement(_Timetable.default, args);
};

var Default = Template.bind({});
exports.Default = Default;
Default.args = {};
var WithEvents = Template.bind({});
exports.WithEvents = WithEvents;
WithEvents.args = {
  date: "1999-05-26",
  days: 1,
  timeFrom: 18,
  timeTo: 23,
  items: [{
    dtStart: "1999-05-26 20:45",
    title: "Kick-off"
  }, {
    dtStart: "1999-05-26 21:51",
    title: "6. min: Basler 0-1"
  }, {
    dtStart: "1999-05-26 22:16",
    title: "91. min: Sheringham 1-1"
  }, {
    dtStart: "1999-05-26 22:18",
    title: "93. min: Sheringham 1-1"
  }]
};
