// @kit-ui/admin/lib/Calendar/TimetableModule.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _moment = _interopRequireDefault(require("moment"));

var _Module = require("../Module");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _inheritsLoose(subClass, superClass) { subClass.prototype = Object.create(superClass.prototype); subClass.prototype.constructor = subClass; subClass.__proto__ = superClass; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

// import './ModuleTimetable.scss';
var ModuleTimetable = /*#__PURE__*/function (_React$Component) {
  _inheritsLoose(ModuleTimetable, _React$Component);

  function ModuleTimetable() {
    var _this;

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = _React$Component.call.apply(_React$Component, [this].concat(args)) || this;

    _defineProperty(_assertThisInitialized(_this), "onToggle", function (item) {
      _this.props.action.onToggle(item);
    });

    _defineProperty(_assertThisInitialized(_this), "getStyle", function (start, duration) {
      var minutesHeight = 100 / 60;
      var top = Math.floor(start * minutesHeight);
      var height = Math.floor(duration * minutesHeight);
      var style = {
        top: top + '%',
        height: height + '%'
      };
      return style;
    });

    return _this;
  }

  var _proto = ModuleTimetable.prototype;

  _proto.render = function render() {
    var _this2 = this;

    var item = this.props.item;
    var title = item.title,
        _dtStart = item._dtStart,
        _dtEnd = item._dtEnd;
    var start = (0, _moment.default)(_dtStart).format('mm');
    var duration = (0, _moment.default)(_dtEnd).diff(_dtStart, 'minutes');
    var style = this.getStyle(start, duration);
    return /*#__PURE__*/_react.default.createElement("div", {
      className: "module  module--timetable",
      style: style,
      onClick: function onClick() {
        return _this2.onToggle(item);
      }
    }, /*#__PURE__*/_react.default.createElement(_Module.ModuleDate, {
      dt: _dtStart,
      format: "HH:mm"
    }), /*#__PURE__*/_react.default.createElement(_Module.ModuleTitle, {
      title: title
    }));
  };

  return ModuleTimetable;
}(_react.default.Component);

var _default = ModuleTimetable;
exports.default = _default;
module.exports = exports.default;
