// @kit-ui/admin/lib/Calendar/Timetable.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _moment = _interopRequireDefault(require("moment"));

var _utils = require("../utils/");

var _styles = require("@material-ui/core/styles");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var useStyles = (0, _styles.makeStyles)(function (theme) {
  return {
    timetable: {
      display: "flex",
      flexDirection: "row",
      alignItems: "flex-start",
      paddingLeft: theme.spacing(7),
      "& > *": {
        flexBasis: 0,
        flexGrow: 1
      }
    },
    date: {
      display: "flex",
      flexDirection: "column",
      alignItems: "flex-start",
      "& $hourLabel": {
        display: "none"
      },
      "&:first-child $hourLabel": {
        display: "block"
      }
    },
    dateLabel: {
      position: "sticky",
      zIndex: 2,
      top: 0,
      padding: theme.spacing(1, 0)
    },
    hour: {
      position: "relative",
      width: "100%",
      height: theme.spacing(7),
      borderTop: "1px solid",
      bordercolor: theme.palette.divider
    },
    hourLabel: {
      position: "absolute",
      width: theme.spacing(6),
      padding: theme.spacing(1, 0),
      display: "block",
      marginTop: "-1px",
      marginLeft: theme.spacing(-7),
      borderTop: "1px solid"
    }
  };
});

var TimetableDate = function TimetableDate(_ref) {
  var datetime = _ref.datetime,
      children = _ref.children;
  var date = (0, _moment.default)(datetime).format('LL');
  var classes = useStyles();
  return /*#__PURE__*/_react.default.createElement("div", {
    className: classes.date
  }, /*#__PURE__*/_react.default.createElement("date", {
    className: classes.dateLabel
  }, date), children);
};

var TimetableHour = function TimetableHour(_ref2) {
  var datetime = _ref2.datetime,
      _ref2$events = _ref2.events,
      events = _ref2$events === void 0 ? [] : _ref2$events;
  var hour = (0, _moment.default)(datetime).format('HH:mm');
  var classes = useStyles();
  return /*#__PURE__*/_react.default.createElement("div", {
    className: classes.hour
  }, /*#__PURE__*/_react.default.createElement("time", {
    className: classes.hourLabel
  }, hour), /*#__PURE__*/_react.default.createElement(TimetableEvents, {
    events: events
  }));
};

var TimetableEvents = function TimetableEvents(_ref3) {
  var _ref3$events = _ref3.events,
      events = _ref3$events === void 0 ? [] : _ref3$events;
  return /*#__PURE__*/_react.default.createElement("div", null, events && events.map(function (event) {
    return /*#__PURE__*/_react.default.createElement("div", null, event && event.title);
  }));
};

var Timetable = function Timetable(_ref4) {
  var date = _ref4.date,
      _ref4$items = _ref4.items,
      items = _ref4$items === void 0 ? [] : _ref4$items,
      days = _ref4.days,
      timeFrom = _ref4.timeFrom,
      timeTo = _ref4.timeTo;
  var timetable = (0, _utils.getTimetable)({
    date: date,
    items: items,
    days: days,
    timeFrom: timeFrom,
    timeTo: timeTo
  });
  var cols = timetable.cols;
  var classes = useStyles();
  return /*#__PURE__*/_react.default.createElement("div", {
    className: classes.timetable
  }, cols.map(function (rows, index) {
    return /*#__PURE__*/_react.default.createElement(TimetableDate, rows[0], rows.map(function (row) {
      return /*#__PURE__*/_react.default.createElement(TimetableHour, row);
    }));
  }));
};

Timetable.defaultProps = {
  days: 4,
  timeFrom: 0,
  timeTo: 23
};
Timetable.propTypes = process.env.NODE_ENV !== "production" ? {
  date: _propTypes.default.string,
  items: _propTypes.default.array,
  days: _propTypes.default.number,
  timeFrom: _propTypes.default.number,
  timeTo: _propTypes.default.number
} : {};
var _default = Timetable;
exports.default = _default;
module.exports = exports.default;
