// @kit-ui/admin/es/Section/SectionLayout.js
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import React from 'react';
import PropTypes from "prop-types";
import { EditableSection, CollapsibleSection, DefaultSection } from "./";

var SectionLayout = function SectionLayout(_ref) {
  var collapsible = _ref.collapsible,
      editable = _ref.editable,
      props = _objectWithoutPropertiesLoose(_ref, ["collapsible", "editable"]);

  if (editable) {
    return /*#__PURE__*/React.createElement(EditableSection, props);
  }

  if (collapsible) {
    return /*#__PURE__*/React.createElement(CollapsibleSection, props);
  }

  return /*#__PURE__*/React.createElement(DefaultSection, props);
};

SectionLayout.propTypes = process.env.NODE_ENV !== "production" ? {
  expanded: PropTypes.bool,
  onToggle: PropTypes.func
} : {};
SectionLayout.defaultProps = {
  position: "relative",
  editable: false,
  collapsible: false,
  expanded: true
};
export default SectionLayout;
