// @kit-ui/admin/lib/AppLoader/AppLoader.stories.js
"use strict";

exports.__esModule = true;
exports.LoadingCalendar = exports.LoadingHofEditor = exports.LoadingHof = exports.LoadingDmEditor = exports.LoadingDm = exports.Default = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _AppLoader = _interopRequireDefault(require("./AppLoader"));

var _AppLoader2 = _interopRequireDefault(require("./AppLoader.example"));

var _app = _interopRequireDefault(require("../data/apps/app"));

var _dm = _interopRequireDefault(require("../data/apps/dm"));

var _calendar = _interopRequireDefault(require("../data/apps/calendar"));

var _hof = _interopRequireDefault(require("../data/apps/hof"));

var _icons = require("@kit-ui/icons");

var _ = require("..");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var theme = (0, _.getRandomTheme)();
var _default = {
  title: 'App/AppLoader',
  component: _AppLoader.default,
  args: {
    title: "App title",
    description: "Loading text"
  },
  argTypes: {}
};
exports.default = _default;

var Template = function Template(args) {
  return /*#__PURE__*/_react.default.createElement("div", {
    style: {
      minHeight: '320px'
    }
  }, /*#__PURE__*/_react.default.createElement(_AppLoader2.default, args, JSON.stringify(args)));
};

var Default = Template.bind({});
exports.Default = Default;
Default.args = _extends({}, _app.default);
var LoadingDm = Template.bind({});
exports.LoadingDm = LoadingDm;
LoadingDm.args = _extends({}, _dm.default);
var LoadingDmEditor = Template.bind({});
exports.LoadingDmEditor = LoadingDmEditor;
LoadingDmEditor.args = _extends({}, _dm.default, {
  icon: /*#__PURE__*/_react.default.createElement(_icons.EditorIcon, {
    color: _dm.default.theme.palette.primary.main
  }),
  description: "Loading editor"
});
var LoadingHof = Template.bind({});
exports.LoadingHof = LoadingHof;
LoadingHof.args = _extends({}, _hof.default);
var LoadingHofEditor = Template.bind({});
exports.LoadingHofEditor = LoadingHofEditor;
LoadingHofEditor.args = _extends({}, _hof.default, {
  icon: /*#__PURE__*/_react.default.createElement(_icons.EditorIcon, {
    color: _hof.default.theme.palette.primary.main
  }),
  description: "Loading editor"
});
var LoadingCalendar = Template.bind({});
exports.LoadingCalendar = LoadingCalendar;
LoadingCalendar.args = _extends({}, _calendar.default);
