// @kit-ui/admin/lib/Layout/LayoutHeader.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _styles = require("@material-ui/core/styles");

var _Typography = _interopRequireDefault(require("@material-ui/core/Typography"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var useStyles = (0, _styles.makeStyles)(function (theme) {
  return {
    root: {
      display: "flex",
      alignItems: "center",
      marginTop: theme.spacing(2),
      marginBottom: theme.spacing(1)
    },
    title: {
      fontFamily: "Akkurat, sans-serif",
      fontSize: "14px",
      lineHeight: "24px",
      fontWeight: "bold"
    },
    description: {
      marginLeft: theme.spacing(1),
      fontFamily: "Akkurat, sans-serif",
      fontSize: "14px",
      lineHeight: "24px"
    }
  };
});

var LayoutHeader = function LayoutHeader(_ref) {
  var className = _ref.className,
      _ref$padding = _ref.padding,
      padding = _ref$padding === void 0 ? 1 : _ref$padding,
      title = _ref.title,
      description = _ref.description;
  var classes = useStyles({
    padding: padding
  });

  if (!title) {
    return false;
  }

  return /*#__PURE__*/_react.default.createElement("header", {
    className: className || classes.root
  }, /*#__PURE__*/_react.default.createElement(_Typography.default, {
    noWrap: true,
    component: "h2",
    className: classes.title
  }, title), /*#__PURE__*/_react.default.createElement(_Typography.default, {
    noWrap: true,
    component: "h3",
    className: classes.description
  }, description));
};

var _default = LayoutHeader;
exports.default = _default;
module.exports = exports.default;
