// @kit-ui/admin/lib/Layout/LayoutThumbnail.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _styles = require("@material-ui/core/styles");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var useStyles = (0, _styles.makeStyles)(function (theme) {
  return {
    root: {
      position: "relative",
      width: "100%",
      padding: function padding(props) {
        return theme.spacing(props.padding || 0);
      },
      display: "flex",
      "& > * + *": {
        flexGrow: 1,
        marginLeft: function marginLeft(props) {
          return theme.spacing(props.spacing || 0);
        }
      },
      "& > *:first-child": {
        flexGrow: 0,
        flexShrink: 0,
        flexBasis: function flexBasis(props) {
          return props.minWidth;
        },
        minHeight: function minHeight(props) {
          return props.minHeight;
        }
      }
    }
  };
});

var LayoutThumbnail = function LayoutThumbnail(_ref) {
  var _ref$minWidth = _ref.minWidth,
      minWidth = _ref$minWidth === void 0 ? 164 : _ref$minWidth,
      _ref$minHeight = _ref.minHeight,
      minHeight = _ref$minHeight === void 0 ? 164 : _ref$minHeight,
      _ref$padding = _ref.padding,
      padding = _ref$padding === void 0 ? 0 : _ref$padding,
      _ref$spacing = _ref.spacing,
      spacing = _ref$spacing === void 0 ? 2 : _ref$spacing,
      children = _ref.children,
      props = _objectWithoutPropertiesLoose(_ref, ["minWidth", "minHeight", "padding", "spacing", "children"]);

  var classes = useStyles({
    minWidth: minWidth,
    padding: padding,
    spacing: spacing
  });
  return /*#__PURE__*/_react.default.createElement("div", {
    className: classes.root
  }, children);
};

var _default = LayoutThumbnail;
exports.default = _default;
module.exports = exports.default;
