// @kit-ui/admin/lib/Layout/LayoutList.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _styles = require("@material-ui/core/styles");

var _LayoutHeader = _interopRequireDefault(require("./LayoutHeader"));

var _LayoutBody = _interopRequireDefault(require("./LayoutBody"));

var _LayoutFooter = _interopRequireDefault(require("./LayoutFooter"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var useStyles = (0, _styles.makeStyles)(function (theme) {
  return {
    root: {
      position: "relative",
      width: "100%",
      padding: function padding(props) {
        return theme.spacing(props.padding);
      },
      "& > header": {
        position: "sticky"
      }
    },
    list: {
      "& > * + *": {
        marginTop: function marginTop(props) {
          return theme.spacing(props.spacing);
        }
      }
    }
  };
});

var LayoutList = function LayoutList(_ref) {
  var _ref$spacing = _ref.spacing,
      spacing = _ref$spacing === void 0 ? 0 : _ref$spacing,
      _ref$padding = _ref.padding,
      padding = _ref$padding === void 0 ? 0 : _ref$padding,
      children = _ref.children,
      props = _objectWithoutPropertiesLoose(_ref, ["spacing", "padding", "children"]);

  var classes = useStyles({
    padding: padding,
    spacing: spacing
  });
  return /*#__PURE__*/_react.default.createElement("section", {
    className: classes.root,
    "data-layout": "list"
  }, /*#__PURE__*/_react.default.createElement(_LayoutHeader.default, props), /*#__PURE__*/_react.default.createElement("div", {
    className: classes.list
  }, children), /*#__PURE__*/_react.default.createElement(_LayoutFooter.default, props));
};

var _default = LayoutList;
exports.default = _default;
module.exports = exports.default;
