// @kit-ui/admin/lib/Layout/LayoutFormat.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _styles = require("@material-ui/core/styles");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var useStyles = (0, _styles.makeStyles)(function (theme) {
  return {
    format: {
      position: "relative",
      width: "100%",
      minHeight: function minHeight(props) {
        return props.minHeight;
      },
      paddingBottom: function paddingBottom(props) {
        return props.ratio * 100 + "%";
      }
    }
  };
});

var LayoutFormat = function LayoutFormat(_ref) {
  var format = _ref.format,
      minHeight = _ref.minHeight,
      children = _ref.children;
  var ratio;

  if (format) {
    var size = format.split(":");
    var w = size[0];
    var h = size[1];
    ratio = h / w;
  }

  var classes = useStyles({
    ratio: ratio,
    minHeight: minHeight
  });

  if (ratio || minHeight) {
    return /*#__PURE__*/_react.default.createElement("div", {
      className: classes.format,
      "data-minHeight": minHeight,
      "data-ratio": ratio,
      "data-format": format
    }, children);
  }

  return children;
};

var _default = LayoutFormat;
exports.default = _default;
module.exports = exports.default;
