// @kit-ui/admin/lib/Layout/LayoutFooter.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _styles = require("@material-ui/core/styles");

var _LayoutButtons = _interopRequireDefault(require("./LayoutButtons"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var useStyles = (0, _styles.makeStyles)(function (theme) {
  return {
    root: {
      marginTop: theme.spacing(2)
    }
  };
});

var LayoutFooter = function LayoutFooter(_ref) {
  var className = _ref.className,
      _ref$padding = _ref.padding,
      padding = _ref$padding === void 0 ? 1 : _ref$padding,
      _ref$buttons = _ref.buttons,
      buttons = _ref$buttons === void 0 ? [] : _ref$buttons,
      buttonGroupProps = _ref.buttonGroupProps,
      buttonProps = _ref.buttonProps;
  var classes = useStyles({
    padding: padding
  });

  if (!buttons.length) {
    return false;
  }

  return /*#__PURE__*/_react.default.createElement("footer", {
    className: className || classes.root
  }, buttons && /*#__PURE__*/_react.default.createElement(_LayoutButtons.default, {
    buttons: buttons,
    buttonGroupProps: buttonGroupProps,
    buttonProps: buttonProps
  }));
};

var _default = LayoutFooter;
exports.default = _default;
module.exports = exports.default;
