// @kit-ui/admin/lib/NavAction/NavActionMenuList.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _MenuList = _interopRequireDefault(require("@material-ui/core/MenuList"));

var _styles = require("@material-ui/core/styles");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var useStyles = (0, _styles.makeStyles)(function (theme) {
  return {
    actionList: {
      width: "100%",
      paddingTop: theme.spacing(1.5),
      paddingBottom: theme.spacing(1.5),
      "& $root": {
        paddingTop: 0,
        paddingBottom: 0
      }
    }
  };
});

var ActionMenuList = function ActionMenuList(_ref) {
  var className = _ref.className,
      children = _ref.children;
  var classes = useStyles();
  return /*#__PURE__*/_react.default.createElement(_MenuList.default, {
    className: className || classes.actionList
  }, children);
};

var _default = ActionMenuList;
exports.default = _default;
module.exports = exports.default;
