// @kit-ui/admin/lib/NavAction/NavAction.stories.js
"use strict";

exports.__esModule = true;
exports.Multiple = exports.SingleDisabled = exports.SingleCombo = exports.SingleWithChildren = exports.Single = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _NavAction = _interopRequireDefault(require("./NavAction"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _default = {
  title: 'Navigation/NavAction',
  component: _NavAction.default,
  argTypes: {}
};
exports.default = _default;

var Template = function Template(args) {
  return /*#__PURE__*/_react.default.createElement(_NavAction.default, args);
};

var Single = Template.bind({});
exports.Single = Single;
Single.args = {
  menu: [{
    title: "Create"
  }]
};
var SingleWithChildren = Template.bind({});
exports.SingleWithChildren = SingleWithChildren;
SingleWithChildren.args = {
  menu: [{
    role: "toggle",
    title: "Create",
    children: [{
      role: "group",
      children: [{
        icon: "business",
        title: "Business"
      }, {
        icon: "person",
        title: "User "
      }]
    }, {
      title: "Secondary 1"
    }, {
      title: "Secondary 2"
    }]
  }]
};
var SingleCombo = Template.bind({});
exports.SingleCombo = SingleCombo;
SingleCombo.args = {
  menu: [{
    title: "Create",
    url: "/create",
    children: [{
      role: "group",
      children: [{
        icon: "business",
        title: "Business"
      }, {
        icon: "person",
        title: "User "
      }]
    }, {
      title: "Secondary 1"
    }, {
      title: "Secondary 2"
    }]
  }]
};
var SingleDisabled = Template.bind({});
exports.SingleDisabled = SingleDisabled;
SingleDisabled.args = {
  menu: [{
    title: "Create",
    url: "/create",
    disabled: true,
    children: [{
      role: "group",
      children: [{
        icon: "business",
        title: "Business"
      }, {
        icon: "person",
        title: "User "
      }]
    }, {
      title: "Secondary 1"
    }, {
      title: "Secondary 2"
    }]
  }]
};
var Multiple = Template.bind({});
exports.Multiple = Multiple;
Multiple.args = {
  icons: {
    customIcon: "bug_report"
  },
  menu: [{
    role: "toggle",
    title: "Combo",
    url: "/combo",
    children: [{
      role: "group",
      children: [{
        icon: "business",
        title: "Business"
      }, {
        icon: "person",
        title: "User "
      }]
    }, {
      title: "Secondary 1"
    }, {
      title: "Secondary 2"
    }]
  }, {
    role: "toggle",
    title: "Toggle",
    children: [{
      role: "group",
      children: [{
        icon: "business",
        title: "Business"
      }, {
        icon: "person",
        title: "User "
      }]
    }, {
      title: "Click me!"
    }, {
      title: "Secondary 2"
    }]
  }, {
    title: "Click",
    onClick: function onClick() {
      return alert('Parent click');
    },
    children: [{
      icon: "customIcon",
      title: "Click child",
      onClick: function onClick() {
        return alert('Child click');
      }
    }]
  }, {
    title: "Custom icons",
    children: [{
      icon: "customIcon",
      title: "Custom icon"
    }]
  }, {
    title: "Action 2"
  }]
};
