// @kit-ui/admin/lib/NavAction/NavActionMenuItem.js
"use strict";

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _MenuItem = _interopRequireDefault(require("@material-ui/core/MenuItem"));

var _ListItem = _interopRequireDefault(require("@material-ui/core/ListItem"));

var _styles = require("@material-ui/core/styles");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var useStyles = (0, _styles.makeStyles)(function (theme) {
  return {
    actionListItem: {
      width: "100%",
      padding: 0,
      "&[role=group] + *": {
        borderTop: "1px solid",
        borderTopColor: theme.palette.divider,
        marginTop: theme.spacing(1.5),
        paddingTop: theme.spacing(1.5)
      },
      "& + &[role=group]": {
        borderTop: "1px solid",
        borderTopColor: theme.palette.divider,
        marginTop: theme.spacing(1.5),
        paddingTop: theme.spacing(1.5),
        "& + *": {
          borderTop: "1px solid",
          borderTopColor: theme.palette.divider,
          marginTop: theme.spacing(1.5),
          paddingTop: theme.spacing(1.5)
        }
      }
    }
  };
});

var ActionMenuItem = function ActionMenuItem(_ref) {
  var className = _ref.className,
      role = _ref.role,
      children = _ref.children;
  var classes = useStyles();
  return /*#__PURE__*/_react.default.createElement(_ListItem.default, {
    className: className || classes.actionListItem,
    role: role
  }, children);
};

var _default = ActionMenuItem;
exports.default = _default;
module.exports = exports.default;
